---
title: "CanaryVCF"
author: "Carson Stacy"
date: "27 February 2023"
output: html_document
---

Last updated: `r Sys.Date()`

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# VCF analysis 

## package
```{r}
if(!require(vcfR)){
    install.packages("vcfR")
}

if(!require(GenomicRanges)){
    BiocManager::install("GenomicRanges")
}
install.packages("rJava")
BiocManager::install("fastreeR")



install.packages("FactoMineR")
install.packages("vcd")
install.packages("factoextra")


library(rJava)
library(vcfR)
library(GenomicRanges, quietly = T)
library(tidyverse)
library(FactoMineR)
library(vcd)
library(factoextra)
```

# Data input
```{r}
vcf_file <- "/Users/clstacy/Downloads/MG_PASS_only.vcf.gz"
dna_file <- "~/Downloads/serCan2020.fna"
gtf_file <- "~/Downloads/serCan2020.gtf"
```

```{r}
vcf <- read.vcfR(vcf_file, verbose = FALSE)

vcf2 <- vcf[grep("NC_066317.1", vcf@fix[,1]),]

dna <- ape::read.dna(dna_file, format = "fasta")

dna2 <- dna[ grep( "NC_066317.1 ", names(dna) ) ]
names(dna2) <- "NC_066317.1"
dna2 <- as.matrix(dna2)

gtf <- read.table(gtf_file, sep="\t", quote="")

gtf2 <- gtf[grep("NC_066317.1", gtf[,1]),]

# remove(vcf, dna, gtf)
```


```{r}
chromosomes <- unique(vcf@fix[,1])

datalist = vector("list", length =0)


# gt_extracted <- data.frame(matrix(NA,nrow=1,ncol=15))
# colnames(gt_extracted) <- c("X625"  ,  "X209"   , "X601"   , "X615" ,   "X618"  ,  "Blue010" ,"X619" ,   "X48"   ,  "X101LS" , "AR1721","Blue013", "Blue015", "Blue031", "X106"  ,  "X610" )

for (chrm in chromosomes[1:32]) { #32
  #subset vcf
  vcf2 <- vcf[grep(chrm, vcf@fix[,1]),]
  #dna
  dna2 <- dna[ grep( paste0(chrm, " "), names(dna) ) ]
  names(dna2) <- chrm
  dna2 <- as.matrix(dna2)
  #gtf
  gtf2 <- gtf[grep(chrm, gtf[,1]),]
  
  #create chromR
  chrom <- create.chromR(name=chrm, vcf=vcf2, seq=dna2, ann=gtf2)
  chrom <- masker(chrom, min_DP = 200, max_DP = 700)
  
  cat("Chromosome", chrm)
  # print(chromoqc(chrom, dp.alpha=20))
  # print(plot(chrom))
  
  datalist[[chrm]] <- data.table::data.table(extract.gt(
   chrom,
   element = "GT",
   mask = FALSE,
   as.numeric = FALSE,
   return.alleles = TRUE,
   IDtoRowNames = TRUE,
   extract = TRUE,
   convertNA = TRUE),keep.rownames = TRUE
   )
  
  # gt_extracted <- do.call(rbind, list(gt_extracted,tmp))
  # gt_extracted=dplyr::bind_rows(gt_extracted, tmp)
  #   ))
}


remove(dna,dna2,gtf,gtf2,vcf,vcf2, chrom, tmp)

```

```{r}
# gt_extracted = data.table::data.table(do.call(rbind, datalist))

gt_extracted= data.table::rbindlist(datalist, use.names = TRUE)

remove(datalist)

gt_extracted %>%
  dplyr::filter(!is.na(`101LS`) & !is.na(`106`)) %>% 
  View()
```

```{r}
# Sys.setenv(JAVA_HOME="")
library(fastreeR)
myVcfIstats <- fastreeR::vcf2istats(inputFile = vcf_file)
plot(myVcfIstats[,7:9])

#crashes, doesn't work rn

# system.file("extdata", "samples.vcf.gz",
# package = "fastreeR"
# )
# myVcfDist <- fastreeR::vcf2dist(inputFile = system.file("extdata", "samples.vcf.gz",
# package = "fastreeR")
# )
vcf_file <- "/Users/clstacy/Downloads2/MG_PASS_only2.vcf"
myVcfDist <- fastreeR::vcf2dist(inputFile = vcf_file)

graphics::hist(myVcfDist, breaks = 100, main=NULL, 
                                xlab = "Distance", xlim = c(0,max(myVcfDist)))


```

```{r}
myVcfTree <- fastreeR::dist2tree(inputDist = myVcfDist)
plot(ape::read.tree(text = myVcfTree), direction = "down", cex = 0.9)
ape::add.scale.bar()
ape::axisPhylo(side = 2)
```


















<!-- ```{r} -->
<!-- factData <- gt_extracted %>% -->
<!--   mutate(rowsums = rowSums(is.na(.))) %>% -->
<!--   filter(rowsums >= 14) %>% -->
<!--   arrange(desc(rowsums)) %>% -->
<!--   select(-rowsums) %>% -->
<!--   replace(is.na(.), 0) %>% -->
<!--   # arrange(desc(`101LS`)) %>% -->
<!--   mutate_if(is.character, as.factor) %>% -->
<!--   mutate(rn = as.character(rn)) -->


<!-- famd <- FAMD(t(factData[1:2000,])[-1,],  -->
<!--                   graph=FALSE) -->
<!-- famd -->

<!-- fviz_famd_ind(famd)#,#col.ind = "rn",  -->
<!--              #gradient.cols = c("blue", "orange", "red"), -->
<!--             # repel = TRUE) -->
<!-- fviz_pca_ind(famd, col.ind = "cos2", repel = T) -->
<!-- # factCor <- cor(factData) -->
<!-- # factCov <- cov(factData) -->

<!-- ``` -->












<!-- ```{r} -->
<!-- chrom <- create.chromR(name='NC_066317.1', vcf=vcf2, seq=dna2, ann=gtf2) -->

<!-- chrom <- masker(chrom, min_DP = 300, max_DP = 700) -->
<!-- chrom <- proc.chromR(chrom, verbose = FALSE) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- plot(chrom) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- chromoqc(chrom, dp.alpha=20) -->
<!-- chromoqc(chrom, xlim=c(1e+05, 4e+05)) -->
<!-- chromoqc(chrom, xlim=c(16510000,17552729)) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- dp <- extract.gt(chrom, element="DP", as.numeric=TRUE) -->
<!-- rownames(dp) <- 1:nrow(dp) -->
<!-- is.na(dp[na.omit(dp == 0)]) <- TRUE -->

<!-- heatmap.bp(dp[1,100]) -->
<!-- ``` -->

