---
title: "Code Supplement Canary-Seq"
author: "Carson Stacy"
date: "2024-07-15"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#pak("eulerr")
library(eulerr)
library(cowplot)
library(ggplot2)
library(tidyverse)
library(ggpubr)
library(here)
library(kableExtra)
library(AnnotationHub)
library(gridtext)
library(graphics)
library(data.table)
library(edgeR)
library(ComplexUpset)
library(ComplexHeatmap)
library(tidyverse)
library(clusterProfiler)
library(flextable)
library(GGally)
library(biomartr)

options(ggrepel.max.overlaps = Inf)
```

# Data Preparation
```{r load annotations}
# get annotation hub (orgdb)
ah <- AnnotationHub()
orgs <- subset(ah, ah$rdataclass == "OrgDb")
orgdb <- query(orgs, "Serinus canaria")[[1]]
```

## load custom functions
```{r custom functions}
source(here::here("scripts/canary_data_processing_functions.R"))
```

```{r read-data}
# assign file path locaiton of rsem.merged.gene_counts.tsv file
filepath_rsem_gene_counts <- here::here("data", "rsem.merged.gene_counts.tsv")

# read in rsem.merged.gene_counts.tsv
g.counts.l <- data.table::fread(filepath_rsem_gene_counts, sep="\t") %>% as_tibble()
```


## Remove problematic samples

The Sample ES89 was removed due to failed sequencing.
Samples ES31,ES32, and ES65 were removed due to incomplete sample metadata.
Samples from bird SEM19-1481 had their metadata corrected due to an error in recording of diet received.
```{r remove-samples}
g.counts.l <- g.counts.l %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>% 
  # dplyr::select(-contains("Blue-050")) %>%
  # change diet group of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       # c(`ES64_SEM19-1481_MG-lipid_21`="ES64_SEM19-1481_MG-protein_21",
       #   `ES52_SEM19-1481_MG-lipid_14`="ES52_SEM19-1481_MG-protein_14",
       #   `ES83_SEM19-1481_MG-lipid_0`="ES83_SEM19-1481_MG-protein_0"))
       # )
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       )
```


## read metadata
```{r}
# see generate_metadata.R script for metadata generation code.
metadata_phenotypes <-
  read_tsv(here("sample_metadata/metadata_phenotypes.tsv"))

# annotation compiled in process_gene_annotation.R script.
result_BM_merged <- read_tsv(here::here("data/annotation/results_BM_merged.tsv"))
```

## Filter Hemoglobulin genes
```{r}
# Generate count matrix
gene_expression_matrix <- g.counts.l %>% #View()
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  # convert to matrix for edgeR
  as.matrix()
```


# Differential Expression Analysis with edgeR

```{r}
entrez_names_gene_expression_matrix <-
  gene_expression_matrix |>
  data.frame() |>
  rownames_to_column("external_gene_name") |> 
  left_join(#entrez_IDs |> dplyr::select(-go_id),
            result_BM_merged,
            by = "external_gene_name", multiple = "first") |> #View()
  mutate(entrezID = case_when(
    str_detect(external_gene_name, "^LOC") ~ str_replace(external_gene_name, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ external_gene_name
  )) |> #View()
  pull(entrezID)

rownames(gene_expression_matrix) <- entrez_names_gene_expression_matrix
```

```{r}
Diet <- as.factor(metadata_phenotypes$Diet)
MG.Active <- as.factor(metadata_phenotypes$MG.Status)


diet_inf <- interaction(Diet, MG.Active) %>% as.factor()
#%diet_inf <- interaction(Diet, MG.Active) #%diet_inf <- interaction(Diet, MG.Active) #%>% as.factor()
design <- model.matrix(~ 0 + diet_inf)
```

```{r}
y <- DGEList(
  counts = round(gene_expression_matrix,0),
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff <- as.numeric(round(cpm(10, mean(y$samples$lib.size)),1))

# apply cutoff
keep <- rowSums(cpm(y) > cpm_cutoff) >= 12
y <- y[keep,]
summary(keep)

y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=T)

y$samples$group <- #grouping_var
  diet_inf %>% as.factor()


contrast <- makeContrasts(
  #design, #metadata,
  contrasts = c(
    # Interaction contrast
    # infection response in protein - infection response in lipid
    "(diet_infprotein.MG - diet_infprotein.Control) - (diet_inflipid.MG - diet_inflipid.Control)",
    # Diet Contrast
    # Protein - lipid
    "(diet_infprotein.MG + diet_infprotein.Control)/2-(diet_inflipid.MG + diet_inflipid.Control)/2",
    # Infection contrast
    # Infected - Uninfected
    "(diet_infprotein.MG + diet_inflipid.MG)/2-(diet_infprotein.Control + diet_inflipid.Control)/2",
    # Infection in Protein Birds
    # infected protein - uninfected protein
    "diet_infprotein.MG - diet_infprotein.Control",
    # Infection in Lipid Birds
    # infected lipid - uninfected lipid
    "diet_inflipid.MG - diet_inflipid.Control",
    # Uninfected Diet Contrast
    # Protein - lipid
    "(diet_infprotein.Control)-(diet_inflipid.Control)"
  ),
  levels = design
)

fit <- glmQLFit(y, design)
```

## Infection x Diet interaction
```{r}
res_ixd_edgeR <- glmQLFTest(fit, contrast=contrast[,1])

topTags_ixd_edgeR <- topTags(res_ixd_edgeR, n=Inf) %>% 
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```

## Diet (Protein vs Lipid) (only uninfected samples)
```{r}
# res_diet_edgeR <- glmQLFTest(fit, contrast=contrast[,2])
# 
# 
# topTags_diet_edgeR <- topTags(res_diet_edgeR, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select(
#     c(
#       "gene",
#       "gene_name",
#       "logFC",
#       "logCPM",
#       "F",
#       "PValue",
#       "FDR",
#       "description",
#       "entrezgene_id"
#     )
#   ) %>%
#   distinct(gene, .keep_all = TRUE)

res_diet_uninf_edgeR <- glmQLFTest(fit, contrast=contrast[,6])

topTags_diet_uninf_edgeR <- topTags(res_diet_uninf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```


## Changes in expression following MG exposure
```{r}
res_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,3])

topTags_inf_edgeR <- topTags(res_inf_edgeR, n=Inf) %>% 
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)

# protein bird response to infection
res_protein_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,4])

topTags_protein_inf_edgeR <- topTags(res_protein_inf_edgeR, n=Inf) %>%
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)


# Lipid bird infection response
res_lipid_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,5])

topTags_lipid_inf_edgeR <- topTags(res_lipid_inf_edgeR, n=Inf) %>%
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)

```



## Phenotype

Using Pathogen load levels as regression covariate for gene expression.

```{r}
logLoad <- metadata_phenotypes$logLoad_alt
Diet <- metadata_phenotypes$Diet

design_logLoad <- model.matrix(~ 0 + logLoad * Diet)

y_logLoad <- DGEList(
  counts = gene_expression_matrix,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_logLoad <- as.numeric(round(cpm(10, mean(y_logLoad$samples$lib.size)),1))

# apply cutoff
keep_logLoad <- rowSums(cpm(y_logLoad) > cpm_cutoff_logLoad) >= 12
y_logLoad <- y_logLoad[keep_logLoad,]

y_logLoad <- calcNormFactors(y_logLoad)
y_logLoad <- estimateDisp(y_logLoad, design_logLoad)
fit_logLoad <- glmQLFit(y_logLoad, design_logLoad)
res_logLoad_inf <-  glmQLFTest(fit_logLoad, coef="logLoad")


# log pathogen Load
topTags_logLoad_inf <- topTags(res_logLoad_inf, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```

## merge results
```{r}
genes_higher_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_lower_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_protein_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_lipid_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>% 
  deframe() 

genes_bigger_protein_infection_difference <- topTags_ixd_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>% #View()
  deframe() 

genes_bigger_lipid_infection_difference <- topTags_ixd_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>% #View()
  dplyr::select(gene, logFC) %>%# View()
  deframe() 

genes_higher_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_lower_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

fdr_genes_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_ixd <- topTags_ixd_edgeR %>% 
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

DE_genes_edgeR_upset <- topTags_inf_edgeR %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixd_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixd_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),

         higher_in_logLoad = gene %in% names(genes_higher_in_logLoad),

         lower_in_logLoad = gene %in% names(genes_lower_in_logLoad),
        logFC_infection = 
          case_when(
            gene %in% names(genes_lower_in_infection) ~ genes_lower_in_infection[gene],
            gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
            TRUE ~ NA_real_
          ),
        logFC_diet = case_when(
          gene %in% names(genes_higher_protein_diet) ~ genes_higher_protein_diet[gene],
          gene %in% names(genes_higher_lipid_diet) ~ genes_higher_lipid_diet[gene],
          TRUE ~ NA_real_
        ),
        logFC_interaction = case_when(
          gene %in% names(genes_bigger_protein_infection_difference) ~ genes_bigger_protein_infection_difference[gene],
          gene %in% names(genes_bigger_lipid_infection_difference) ~ genes_bigger_lipid_infection_difference[gene],
          TRUE ~ NA_real_
        ),
        FDR_infection = case_when(
          gene %in% names(fdr_genes_infection) ~ fdr_genes_infection[gene],
          TRUE ~ NA_real_
        ),
        FDR_diet = case_when(
          gene %in% names(fdr_genes_diet) ~ fdr_genes_diet[gene],
          TRUE ~ NA_real_
        ),
        FDR_interaction = case_when(
          gene %in% names(fdr_genes_ixd) ~ fdr_genes_ixd[gene],
          TRUE ~ NA_real_
        ),
        logFC_logLoad = case_when(
          gene %in% names(genes_higher_in_logLoad) ~ genes_higher_in_logLoad[gene],
          gene %in% names(genes_lower_in_logLoad) ~ genes_lower_in_logLoad[gene],
          TRUE ~ NA_real_
        )
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixd = Protein_ixd_bigger_Inf | Lipid_ixd_bigger_Inf,
         DE_in_logLoad = higher_in_logLoad | lower_in_logLoad) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ))
```


# Gene Set Enrichment Analysis of DE genes

Grouped by comparison
```{r}
#gseGO

gseGO_ixd <- topTags_ixd_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

# gseGO_diet <- topTags_diet_edgeR %>%
#   arrange(desc(logFC)) %>%
#   pull(logFC, gene) %>%
#   gseGO(
#     geneList     = .,
#     OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "BP",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1
#   )

gseGO_diet_uninf <- topTags_diet_uninf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )



gseGO_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = ., 
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_logLoad <- topTags_logLoad_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    pvalueCutoff  = 1
  )

#gseKEGG


gseKEGG_ixd <- topTags_ixd_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(geneList     = .,
               organism = 'scan',
               pvalueCutoff = 1,
               verbose      = FALSE)

gseKEGG_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    by = "fgsea",
    exponent = 1,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

gseKEGG_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
               pvalueCutoff = 1,
               verbose      = FALSE)

gseKEGG_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
               pvalueCutoff = 1,
               verbose      = FALSE
  )
```

## Ordinal Logistic Regression of Eye Score
```{r}

```






# TODO: make sure we have all of these objects...
```{r}
#load figure code objects.
topTags_lipid_inf_edgeR
topTags_protein_inf_edgeR
topTags_inf_edgeR
topTags_ixd_edgeR
topTags_diet_uninf_edgeR
topTags_logLoad_inf
DE_genes_edgeR_upset

 tbl_tol_results

gseGO_infection
gseGO_infection_protein
gseGO_infection_lipid
gseKEGG_infection
gseKEGG_infection_protein
gseKEGG_infection_lipid
gseGO_diet_uninf

genes_lower_in_infection
genes_higher_in_infection
genes_higher_protein_diet
genes_higher_lipid_diet
genes_bigger_protein_infection_difference
genes_bigger_lipid_infection_difference
```
# Main Text Figures

## Figure 1

#TODO Add Erin's code for this figure here.
```{r}

```


## Figure 2

### Fig 2A
```{r}

source(here::here("scripts/canary_data_processing_functions.R"))

fig2A <- list("Infection" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
     "Diet" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet],
     "     Infection × Diet" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixd]
     ) %>%
  ggeulerr(shape="circle", text_size = 2.5) +
    theme_void(base_size = 2)+
    theme(legend.position = "none", 
          legend.title= element_blank(),
            text = element_text(size=6),
          plot.margin = margin(t = 5,  # Top margin
                             r = 0,  # Right margin
                             b = 5,  # Bottom margin
                             l = 5)) # Left margin

fig2A
  
# ggsave(plot = fig2A, filename= here("figures/venn_plot_interaction_fig.png"), width = 16, height = 16, units = "cm", dpi=1200)

```


### Fig 2B
```{r}
fig2b.ma.inf.L <- topTags_lipid_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Lipid)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  labs(x=expression("log"["2"]~"(counts per million)"),
       y=expression("log"["2"]~"(fold change)")) +
  ylim(-4,6) +
     guides(color = guide_legend(override.aes = list(size = 2) ) ) + 
      theme(
            text = element_text(size=8),
            axis.text = element_text(size=8),title = element_text(size=8))
  

fig2b.ma.inf.P <- topTags_protein_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Protein)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) + theme(legend.position = "right") + guides(color = guide_legend(reverse=TRUE)) +
  labs(x=expression("log"["2"]~"(counts per million)"),
       y=expression("log"["2"]~"(fold change)")) +
  ylim(-4,6) +
     guides(color = guide_legend(override.aes = list(size = 2) ) ) + 
      theme(
            text = element_text(size=8),
            axis.text = element_text(size=8),title = element_text(size=8))

fig2B <- ggpubr::ggarrange(fig2b.ma.inf.L, fig2b.ma.inf.P, ncol=1,common.legend = T, legend = "bottom") 

fig2B

# ggsave(plot = fig2B, filename = here("figures/logFC_density_DE_in_diet_fig.png"), width = 3, height = 4, units = "cm", dpi=1200)
```




### Fig 2C
```{r}
font_size <- 45#35

tbl_tol_results %>%
  left_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM"="logCPM", "description" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  left_join(topTags_lipid_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description")) %>%
  left_join(topTags_protein_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_inf_lipid", "_inf_prot")) %>%
  filter(#FDR_tol < 0.05 & 
           FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixn < 0.05 #& 
           & FDR_inf_lipid < 0.05
           # FDR_logLoad < 0.05 
           #|
           # gene_name_tol %in% c("JCHAIN", "EXOG",
           #                    "LOC103823677", "MZB1", "LOC103818060",
           #                    "MYD88", "LOC108961394",
           #                    "PTGDR", "PTGER2", "NEFM", "NEFL")
           ) %>%
  # filter(gene_name_tol %in% c("JCHAIN", "EXOG",
  #                             "LOC103823677", "MZB1", "LOC103818060", 
  #                             "MYD88", "LOC108961394",
  #                             "PTGDR", "PTGER2", "NEFM", "NEFL")) %>%
  dplyr::select(gene_name,#_tol, 
                description,#_tol, 
                logFC_inf, 
                logFC_inf_lipid,
                logFC_inf_prot,
                # logFC_logLoad, 
                logFC_diet, 
                logFC_ixn, 
                FDR_inf,
                #Estimate, #logCPM_inf,
                # FDR_tol, 
                FDR_diet,
                FDR_ixn,
                FDR_inf_lipid,
                FDR_inf_prot
                ) %>%
  arrange(desc(logFC_inf)) %>% 
  # mutate(across(is.numeric, round, 3)) %>% 
  mutate(gene_name = replace_na(gene_name, "TENT5C"),
         description = replace_na(description, "terminal nucleotidyltranferase 5C")) %>% 
  gt::gt() %>%
  gt::tab_options(table.font.size = font_size) %>%
  gt::cols_hide(c(FDR_inf, FDR_ixn, FDR_diet, FDR_inf_prot, FDR_inf_lipid)) %>%
  gt::tab_header(title = "31 Genes Differentially Expressed by Diet, Infection, and Diet×Infection") %>%
  gt::fmt_number(decimals = 2) %>%
  gt::cols_label(gene_name = "Gene Name",
                 description = "Description",
                 logFC_inf = "logFC Infection (all)",
                 logFC_inf_lipid = "logFC Infection (lipid)",
                 logFC_inf_prot = "logFC Infection (protein)",
                 logFC_diet = "logFC Diet (protein vs lipid)",
                 logFC_ixn = "logFC Interaction"
                 #Estimate = "Estimate (ES)",
                 #logCPM_inf = "logCPM"
                 ) %>%
  gt::cols_width(gene_name ~ "5em",
                 description ~ "20em",
                 logFC_inf ~ "5em",
                 logFC_inf_lipid ~ "5em",
                 logFC_inf_prot ~ "5em",
                 logFC_diet ~ "5em",
                 logFC_ixn ~ "5em",
                 # Estimate ~ "5em",
                 # logCPM_inf ~ "5em"
                 ) %>%
  gt::data_color(columns = c(logFC_inf, logFC_inf_lipid, logFC_inf_prot, logFC_diet, logFC_ixn),
             fn = scales::col_numeric(palette = c("blue","white", "red"), domain = c(-5,5))) %>%
  gt::cols_align(columns = c(logFC_inf, logFC_inf_lipid, logFC_inf_prot, logFC_diet, logFC_ixn), align="center") %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf,
                                   rows = FDR_inf < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_diet,
                                   rows = FDR_diet < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_ixn,
                                   rows = FDR_ixn < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf_lipid,
                                   rows = FDR_inf_lipid < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
            gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf_prot,
                                   rows = FDR_inf_prot < 0.05)
  ) %>%
  tab_options(data_row.padding = px(15))# %>%
  #gt::gtsave(file = here::here("tables/top_genes_ixndietinf_colors.png"), zoom=2, vwidth = 4000, vheight=4000, expand=20)
```


### Figure 2 combined
Figure 2 Panels
```{r, fig.width=8, fig.height=6}
fig2C <- ggdraw() + draw_image(here("tables/top_genes_ixndietinf_colors.png")) 

fig2AB <- cowplot::plot_grid(fig2A,NULL, fig2B, ncol=1, labels = c("A","B",""), rel_heights = c(1.1,0.12,2))
fig2 <- cowplot::plot_grid(fig2AB, NULL, fig2C, rel_widths = c(0.7, 0.025, 1.3), ncol=3,labels = c(" ","C", " "))

fig2

# ggsave(plot = fig2, filename = here("figures/fig2.png"), width = 8, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure 3



### Fig 3A

```{r}
cor_dietvInfL_DE <- left_join(topTags_lipid_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
    left_join(topTags_inf_edgeR,by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% #View()
  # filter(FDR < 0.05 & FDR.diet < 0.05) %>%
  # filter(FDR.inf < 0.05 & FDR.diet < 0.05) %>% #View()
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, color = "black", size=0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2.6, label.y = 6,  size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  #ggpubr::stat_regline_equation(label.x = -3, label.y = 2, size = 3) +
  # geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  geom_rug(col="#7CAE00", alpha = 0.2, sides="l",length = unit(0.05, "npc")) +
  geom_rug(col="#F8766D", alpha=0.2, sides = "b",length = unit(0.05, "npc")) +
  # ggrepel::geom_text_repel(aes(label=gene_name), box.padding = 0.5, size = 2) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ylab("") +
  xlab("log fold change\n infection (lipid)") +
  # xlab("logFC infection\n (lipid)")+
  #ggtitle("Cor(logFC) diet vs lipid infection (both FDR < 0.05)")+ 
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8)) #+
  # theme_classic()

#ggsave("figures/DEGenesBothCorrlipInfDietLogFC.png", width=16, height=16, units = "cm", dpi=1200)

cor_dietvInfP_DE <- left_join(topTags_protein_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  left_join(topTags_inf_edgeR,by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% #View()
  # filter(FDR < 0.05 & FDR.diet < 0.05) %>%
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, size=0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 6,  size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  #ggpubr::stat_regline_equation(label.x = -3, label.y = 2, size = 3) +
  geom_rug(col="#7CAE00", alpha = 0.2, sides="l",length = unit(0.05, "npc")) +
  geom_rug(col="#00BFC4", alpha=0.2, sides = "b",length = unit(0.05, "npc")) +
  # ggrepel::geom_text_repel(aes(label=gene_name), box.padding = 0.5, size = 2) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ylab("") +
  xlab("log fold change\n infection (protein)") +
  # xlab("logFC infection\n (protein)")+
  #ggtitle("Cor(logFC) diet vs lipid infection (both FDR < 0.05)")+ 
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8))  #+
  # theme_classic()

#ggsave("figures/DEGenesBothCorrprotInfDietLogFC.png", width=8, height=8, units = "cm", dpi=1200)

cor_dietvInf_DE <- left_join(topTags_inf_edgeR, topTags_diet_uninf_edgeR,
# left_join(topTags_lipid_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% #View()
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, size=0.5) +
  # ggrepel::geom_label_repel(aes(label=gene_name), box.padding = 0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 6, size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  #ggpubr::stat_regline_equation(label.x = -3, label.y = 2, size = 3) +
  geom_rug(col="#7CAE00", alpha = 0.2, sides="l",length = unit(0.05, "npc")) +
  geom_rug(col="#C77CFF", alpha = 0.2, sides = "b",length = unit(0.05, "npc")) +
  #ggrepel::geom_text_repel(aes(label=gene_name), box.padding = 0.5, size = 2) +
  
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  xlab("log fold change\n infection (all)") +
  ylab("log fold change\n diet (Protein vs Lipid)") +
  # xlab("logFC infection\n (all)")+
  #ggtitle("Cor(logFC) diet vs lipid infection (both FDR < 0.05)")+ 
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8)
        ) 

#ggsave("figures/DEGenesBothCorrInfDietLogFC.png", width=8, height=8, units = "cm", dpi=1200)

```
######

inf x diet.
```{r}
protvslipid_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR,
                                      topTags_lipid_inf_edgeR, 
                                      #topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  filter(FDR.protein < 0.05 | FDR.lipid < 0.05) %>% #View()
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC.protein, y=logFC.lipid)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm")+
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  # xlim(-2,2) +
  # ylim(-2,2) +
  theme_classic() +
  ggtitle("Cor(logFC) protein vs lipid infection")


allvslipid_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR,
                                      topTags_lipid_inf_edgeR, 
                                      #topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  filter(FDR < 0.05 | FDR.lipid < 0.05 | FDR.protein < 0.05 ) %>% #View()
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC, y=logFC.lipid)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm")+
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  # xlim(-2,2) +
  # ylim(-2,2) +
  theme_classic() +
  #
  ggtitle("Cor(logFC) combined vs lipid infection")

protvsall_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR,
                                      topTags_lipid_inf_edgeR, 
                                      #topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  #filter(FDR < 0.05 | FDR.protein < 0.05 ) %>% #View()
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC, y=logFC.protein)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm")+
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  # xlim(-2,2) +
  # ylim(-2,2) +
  theme_classic() +
  #ylab("") +
  ggtitle("Cor(logFC) combined vs lipid infection")
```


### Fig 3B

```{r}

fig4_plot_list <- list(
(gseGO_infection)@result %>%
  mutate(comparison = "infection (all)"),
(gseGO_infection_protein)@result %>%
  mutate(comparison = "infection (protein)"),
(gseGO_infection_lipid)@result %>%
  mutate(comparison = "infection (lipid)"),
gseKEGG_infection@result %>%
  mutate(comparison = "infection (all) KEGG"),
gseKEGG_infection_protein@result %>%
  mutate(comparison = "infection (protein) KEGG"),
gseKEGG_infection_lipid@result %>%
  mutate(comparison = "infection (lipid) KEGG"),
(gseGO_diet_uninf)@result %>%
  mutate(comparison = "diet (protein vs lipid)")
)
```
how about we scale the logFC values for each of those so they are on 0,1 scale? maybe, but that could be messy b/c of outliers.. z score fine but goes too far out.. maybe just leave it out? or rescale? we'll see.

```{r, fig.height=8}

DE_KEGG_terms_any_inf <- bind_rows(fig4_plot_list[[4]], 
          fig4_plot_list[[5]],
          fig4_plot_list[[6]]) %>%
  filter(p.adjust < 0.05) %>%
  pull(ID) %>% unique()

bind_rows(fig4_plot_list[[4]], 
          fig4_plot_list[[5]],
          fig4_plot_list[[6]]) %>%
  filter(ID %in% DE_KEGG_terms_any_inf) %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " (", ID, ") ")) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
    # Description == "monoatomic ion transport" & comparison == "infection (all)" ~ Term,
    # Description == "monoatomic ion transport" ~ NA,
    comparison == "infection (all) KEGG" ~ Term,
    TRUE ~ NA
  ),
  y_labels = case_when(
    Description == "Spliceosome" & comparison == "infection (protein) KEGG	" ~ 0.1,
    TRUE ~ 0
  )) %>%
  arrange(desc(enrichmentScore)) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`#-log(p.adjust)
                )) +
  geom_col(position = "dodge") +
  geom_text(aes(y = y_labels, label = Term_labels, hjust = as.numeric(enrichmentScore) > 0), alpha=1) + 
  xlab("GO Term") +
  scale_alpha_discrete(range=c(0.6, 1)) +
  scale_fill_manual(values = c("purple", "blue","red")) +
  coord_flip() + theme_classic() +
  theme(axis.title.y=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank())


barplot_enrichments <- bind_rows(fig4_plot_list[[1]], 
          fig4_plot_list[[2]],
          fig4_plot_list[[3]],
          fig4_plot_list[[7]]) %>%
  filter(ID %in% DE_GO_terms_any_inf_simple) %>%
  mutate(Description = str_replace_all(Description, "biological process involved in ", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " "#, "(", ID, ") "
                       )
         ) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
  #   Description == "monoatomic ion transport" & comparison == "infection (all)" ~ Term,
  #   Description == "monoatomic ion transport" ~ NA,
  #   comparison == "infection (all)" ~ Term,
     TRUE ~ NA
  ),
   y_labels = case_when(
  #   Description == "monoatomic ion transmembrane transport" & comparison == "infection (all)" ~ 0.42,
  #   Description == "monoatomic ion transport" & comparison == "infection (all)" ~ 0.36,
     TRUE ~ 0
   )) %>%
  group_by(Term) %>%
  mutate(Term_labels2 = case_when(
    n() == 4 & comparison == "infection (all)" ~ Term,
    n() == 3 & !is.na(Term_labels) & comparison == "infection (all)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (protein)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (lipid)" ~ Term,
    n() == 1 ~ Term,
    TRUE ~ NA) 
    )%>%
  ungroup() %>% 
  arrange(desc(enrichmentScore)) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`, color=`p.adj<0.05`#-log(p.adjust)
                )) +
    geom_col(#position = "dodge",
             position = position_dodge(preserve = 'single'), linewidth = 0.2) +
    geom_text(aes(y = y_labels, label = Term_labels2, hjust = as.numeric(enrichmentScore) > 0), alpha=1,
              size = 2.25, color = "black") + 
    xlab("GO Term") +
    scale_alpha_discrete(range=c(0.4, 1), guide="none") +
    scale_fill_manual(#values = c("yellow2","purple", "blue","red")
                      values = c("#7CAE00","#C77CFF", "#F8766D","#00BFC4") ) +
    scale_color_manual(values = c("#00000000","black"), guide="none") +
    coord_flip(clip="off") + theme_classic() +
    geom_hline(yintercept = 0) +
    theme(axis.title.y=element_blank(),
            axis.text.y=element_blank(),
            axis.ticks.y=element_blank(),
          axis.text = element_text(size = 10), axis.title=element_text(size=10)) +
      theme(legend.position = "top", legend.text = element_text(size=8,hjust=0.5),
            legend.title = element_text(size=8),
            #legend.box = "vertical",
            legend.key.size = unit(0.25, 'cm'),legend.justification="center") + 
    guides(fill = guide_legend(reverse=TRUE,nrow=2)) +
    annotate("text", y=0.75, x = 25, label = "upregulated", size=2.25) +
    annotate("text", y=-0.65, x = 25, label = "downregulated", size=2.25)

barplot_enrichments

# ggsave(plot = barplot_enrichments, "figures/GO_term_enrichment_comparison.png", width = 12, height = 10, units = "in", dpi = 1200)
```

### Fig 3C

Heatmap of DE genes in all contrasts
```{r, fig.height=10, fig.width=8}
logFC_all_contrasts <- topTags_inf_edgeR %>%
  arrange(gene) %>%
  #filter(FDR< 0.05) %>%
  #dplyr::select(gene) %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixd_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixd_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),
         # Here we use ifelse to assign logFC values from the named vector based on the gene presence.
        logFC_infection = topTags_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_interaction = topTags_ixn_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        FDR_infection = FDR, 
        FDR_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_interaction = topTags_ixn_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        #logFC_ES = topTags_ES_inf |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        # logFC_tolerance = topTags_tolerance |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        FDR_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        #FDR_ES = topTags_ES_inf |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(FDR)#,
        # FDR_tolerance = topTags_tolerance |> arrange(gene) |> filter(gene == gene) |> pull(FDR)
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixd = Protein_ixd_bigger_Inf | Lipid_ixd_bigger_Inf) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ),
  nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    TRUE ~ paste0("LOC",gene)
  ))

# Define significance level
sig.cutoff <- 0.05

df_heatmap <- logFC_all_contrasts %>%
  # filter(gene %in% genes_de_all_3) %>%
  filter(DE_in_infection | DE_in_diet | DE_in_ixd) %>%
  dplyr::select(
    gene,
    gene_name,
    nice_gene_name,
    description,
    logFC_infection,
    logFC_diet,
    logFC_interaction,
    FDR_infection,
    FDR_diet,
    FDR_interaction,
    logFC_inf_lipid,
    logFC_inf_protein,
    FDR_inf_lipid,
    FDR_inf_protein
  )

# Split into logFC and FDR matrices
logFC_matrix <- as.matrix(df_heatmap[, c("logFC_infection", "logFC_inf_lipid", "logFC_inf_protein", "logFC_diet", "logFC_interaction"
                                         )])
rownames(logFC_matrix) <-  paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)

FDR_matrix <- as.matrix(df_heatmap[, c("FDR_infection", "FDR_inf_lipid", "FDR_inf_protein", "FDR_diet", "FDR_interaction"
                                       )])
rownames(FDR_matrix) <- paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)



significance_matrix <- ifelse(FDR_matrix < sig.cutoff, "*", " ") %>%
  replace_na(" ")

colnames(logFC_matrix) <- c("Inf (all)", "Inf (L)", "Inf (P)", "Diet", "Interaction"
                            )

df <- scale(logFC_matrix)
factoextra::fviz_nbclust(k.max = 54, df, factoextra::hcut, method = "silhouette", nboot = 500) +
# geom_vline(xintercept = 4, linetype = 2)+
labs(subtitle = "silhouette method")

factoextra::fviz_nbclust(k.max = 54, df, factoextra::hcut, method = "wss", nboot = 500) +
# geom_vline(xintercept = 4, linetype = 2)+
labs(subtitle = "Elbow method")



heatmap_enrichment_terms <- list(
    text1 = list("Translation", "Translation Regulation", 
                 "Ribonucleoprotein Complex"),
    text2 = list("Nuclear Protein Complex", "Homologous Recombination"),
    text3 = list(expression(Transferase~Activity^"\u2020")),
    text4 = list("Lysosome", "Protein Kinase Activity", "Phosphorylation"),
    text5 = list("Innate Immune Response", "Defense Response", "Apoptosis"),
    text6 = list(expression(Lysosome^"\u2020"),
                 expression(Lytic~Vacuole^"\u2020")),
    text7 = list(expression(Apoptosis^"\u2020"),
                 expression(Channel~Activity^"\u2020")),
    text8 = list(expression(ECM-receptor~interaction^"\u2020"),
                 expression(Nucleosome^"\u2020"))
)

# note how we set the width of this empty annotation
ha = rowAnnotation(foo = anno_empty(border = FALSE, 
    width = max_text_width(unlist(heatmap_enrichment_terms)) + unit(4, "mm")))

# change padding 
ht_opt$DIMNAME_PADDING = unit(0.25, "cm")
#ht_opt$COLUMN_ANNO_PADDING = unit(0.25, "cm")

heatmap2 <- Heatmap(logFC_matrix, name = "logFC", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   width = unit(8, "cm"),
                   heatmap_legend_param = list(direction = "horizontal",
                                               legend_width = unit(3, "in"), 
                                               title = expression(
                                                 "log"["2"]~"(fold change)"
                                                 ),
                                               legend_position = "topleft",
                                               title_gp = gpar(fontsize = 12),
                                               labels_gp = gpar(fontsize=12)
                                               ),
                   cluster_columns = FALSE, 
                   split=8,
                   show_row_names = FALSE,
                   show_column_names = TRUE,
                   column_names_side = "top",
                   clustering_distance_rows = "pearson",
                   row_names_gp = gpar(fontsize = 12),
                   column_names_gp = gpar(fontsize = 12),
                   column_names_rot = 0,
                   column_names_centered = TRUE,
                   right_annotation = ha,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                       grid.rect(x, y, width, height, gp = gpar(fill = fill, col = NA))
                       if (significance_matrix[i, j] == "*") {
                           grid.text("*", x, y, gp = gpar(col = "black", alpha=0.5, fontsize = 6))
                       }
                   }
                   ) 
ht <- draw(heatmap2, heatmap_legend_side = "bottom")

ht
```

```{r, fig.height=9, fig.width=8, message=FALSE, warning=FALSE}
#png(file="figures/logFC_heatmap.png", width=8, height=16, res=1200,units = "in",bg = "transparent")

draw(heatmap2, heatmap_legend_side = "bot")

for (i in 1:8) {
    decorate_annotation("foo", slice = i, {
        grid.rect(x = 0.01, width = unit(0.75, "mm"), gp = gpar(fill = "black", col = NA), just = "left")
        text_lines <- heatmap_enrichment_terms[[i]]
        n_lines <- length(text_lines)
        # Calculate the center point: Half the number of lines from middle
        start_y <- unit(0.5, "npc")  # middle of the slice
        # Adjust starting y position up by half the number of lines
        adjust_y <- unit(n_lines / 2 - 0.5, "lines") * 1.5 # adjustment from center

        for (j in seq_along(text_lines)) {
            text_line <- text_lines[[j]]
            # Adjust position: start from center and move up or down
            line_y <- start_y - adjust_y + 1.5*unit(j - 1, "lines")
            grid.text(text_line, x = unit(4, "mm"), y = line_y, just = "left")
        }
    })
}

decorate_annotation("foo",  { 
    grid.text("           Functional Enrichments", 
              x = unit(0.4,"npc") - unit(0.2,"cm"),
              y = unit(1, "npc") + unit(0.55, "cm"),
                just = "top", 
              gp = gpar(fontface="bold")) 
})

#dev.off()
```


enrichment of each cluster
```{r}
list_GO_results <- list()
j=0
for (i in row_order(ht)) {
  # print(i)
  genes_i <- rownames(heatmap2@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_results[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.8,
                qvalueCutoff  = 0.8) %>%
    data.frame()
  
}

list_kegg_results <- list()
j=0
for (i in row_order(ht)) {
  # print(i)
  genes_i <- rownames(heatmap2@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_kegg_results[[j]] <- enrichKEGG(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
               organism = 'scan', keyType = "kegg",
               universe = topTags_diet_edgeR$gene,
               pvalueCutoff = 0.9,
                qvalueCutoff  = 0.9) %>%
    data.frame()
}
```


### Figure 3 combined

```{r, fig.width=7, fig.height=8}
fig3C <- ggdraw() + draw_image(here("figures/logFC_heatmap.png"), scale = 1.25, halign=0.4) +
  theme_void()#+ theme(plot.margin = unit(c(2, 0, 2, -5.0), "cm"))

fig3A <- cowplot::plot_grid(cor_dietvInf_DE+theme(rect = element_rect(fill = "transparent"),
                                                  plot.background = element_rect(fill = "transparent",
                                 colour = NA_character_)), NULL, 

                            cor_dietvInfL_DE+theme(rect = element_rect(fill = "transparent"),
                                                   plot.background = element_rect(fill = "transparent",
                                 colour = NA_character_)), NULL,
                            cor_dietvInfP_DE+theme(rect = element_rect(fill = "transparent"),
                                                   plot.background = element_rect(fill = "transparent",
                                 colour = NA_character_)), 
                            ncol=5, rel_widths = c(1,-0.15,1,-0.15,1))
# fig3AB <- cowplot::plot_grid(fig3A, fig3B, ncol=2, labels= c("A", "B"),
#                              rel_heights = c(1,1), align = "h"
#                              )



fig3AB <- cowplot::plot_grid(NULL, fig3A, NULL, barplot_enrichments ,  ncol=1, rel_heights = c(0.05,0.5, -0.01, 1.75), labels = c("A", " ", "", "B"), vjust=c(1.5, 0,0,4))
fig3AB


fig3 <- cowplot::plot_grid(fig3AB, NULL, fig3C, ncol=3, labels = c("", "", "C"),
                           # rel_widths = c(2.2,0.1,1.65)
                           rel_widths = c(2.3, 0, 1.8)
                           )
fig3


# ggsave(plot = fig3, filename = here("figures/fig3.png"), width = 7, height = 8, units = "in", dpi=1200, bg="white")
```


## Figure 4

### Fig 4A 

Venn Diagram of DE gene overlap between phenotype measures and infection

compile results.
```{r}
merged_results <- tbl_tol_results %>%
  left_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM"="logCPM", "description" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  left_join(topTags_lipid_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description")) %>%
  left_join(topTags_protein_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_inf_lipid", "_inf_prot")) %>% #View()
  mutate(
    Estimate = case_when(
      Estimate > 20 ~ 20,
      Estimate < -20 ~ -20,
      TRUE ~ Estimate
    ),
    est = as.numeric(scale(Estimate)), 
  DE_inf = case_when(
    FDR_inf < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_prot = case_when(
    FDR_inf_prot < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_diet = case_when(
    FDR_diet < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ixn = case_when(
    FDR_ixn < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_logLoad = case_when(
    FDR_logLoad < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ES = case_when(
    FDR_tol < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  goi = case_when(
    gene_name %in% c("JCHAIN","INSRR","NEFM", "ALDH1L2"
                         #"NEFL", "USP46"#,
                         #"LOC103823677", "MZB1", "LOC103818060", 
                         #"MYD88", "LOC108961394",
                         # "INSRR"
                         #"PTGDR", "PTGER2", "NEFM", "NEFL"
                         ) ~ gene_name,
    TRUE ~ NA
  ),
  goi_inf = case_when(
    FDR_inf < 0.05 & abs(logFC_inf)>1~ gene_name,
    TRUE ~ NA
  ),
  goi_inf_prot = case_when(
    FDR_inf_prot < 0.05 & abs(logFC_inf_prot)>1.5~ gene_name,
    TRUE ~ NA
  ),
  goi_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 & abs(logFC_inf_lipid)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_diet = case_when(
    FDR_diet < 0.05 & abs(logFC_diet)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_ixn = case_when(
    FDR_ixn < 0.05 & abs(logFC_ixn)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_logLoad = case_when(
    FDR_logLoad < 0.05 & abs(logFC_logLoad)>1~ gene_name,
    TRUE ~ NA
  ),
  goi_ES = case_when(
    FDR_tol < 0.05 & FDR_inf < 0.05 & abs(est)>4~ gene_name,
    abs(est) > 2 & FDR_tol < 0.05 & FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixn < 0.05 ~ gene_name,
    TRUE ~ NA
  )
  )  %>% unique()
```


```{r}
fig4a.1 <- list("Infection" = merged_results$gene.z[merged_results$DE_inf],
     "Pathogen\n Load" = merged_results$gene.z[merged_results$DE_logLoad],
     "Eye Score" = merged_results$gene.z[merged_results$DE_ES]
     ) %>%
  ggeulerr(shape="circle",, text_size = 3) +
  scale_fill_manual(#values = c("yellow2","purple", "blue","red")
                    values = c("darkblue","#C77CFF","orange") , name = " ") +
  # scale_fill_manual(#values = c("yellow2","purple", "blue","red")
                     #values = c("pink","#C77CFF", "#F8766D","#00BFC4") ) +
    theme_void(base_size = 18) +
    theme(#legend.position = "bottom", 
          legend.title= element_blank(),
          plot.margin = margin(t = 5,  # Top margin
                             r = 10,  # Right margin
                             b = 5,  # Bottom margin
                             l = 10)) %>% # Left margin
theme(legend.position = "none"#, #legend.text = element_text(size=8,hjust=8),
          #legend.title = element_text(size=12),
          #legend.box = "vertical",
          #legend.key.size = unit(0.25, 'cm'),
          #legend.justification="left"
      #text = element_text(size = 32)
      ) #+ 
  #guides(fill = guide_legend(reverse=TRUE,nrow=1))


fig4a.1
# ggsave(plot = fig4a.1, filename= here("figures/fig4a_venn_plot_phenotype_fig.png"), width = 16, height = 16, units = "cm", dpi=1200, bg="white")
```


part II: table Table of genes DE in infection, eye score, and pathogen load analyses
```{r}
merged_results %>%
  filter(FDR_inf < 0.05 & FDR_tol < 0.05) %>%
  filter(gene_name %in% c("JCHAIN", "EXOG",
                              "LOC103823677", "MZB1", "LOC103818060",
                              "MYD88", "LOC108961394",
                              "PTGDR", "PTGER2", #"NEFM", 
                          "NEFL")) %>%
  dplyr::select(gene_name, gene, description, 
                logFC_inf, Estimate, #logCPM_inf
                ) %>%
  arrange(desc(logFC_inf)) %>%
  mutate(across(where(is.numeric), round, 2)) %>%
  gt::gt() %>%
  gt::tab_header(title = "Notable Genes among 191 shared in Eye Score and Infection") %>%
  gt::cols_hide(gene) %>%
  gt::cols_label(gene_name = "Gene Name",
                 description = "Description",
                 logFC_inf = "logFC (Infection)",
                 Estimate = md("$\\hat\\beta$   (Eye Score)")#,
                 #logCPM_inf = "logCPM"
                 ) %>%
  gt::cols_width(gene_name ~ "5em",
                 description ~ "15em",
                  logFC_inf ~ "8em",
                 Estimate ~ "8em"#,
                 #logCPM_inf ~ "5em"
                 ) #%>%
  #gt::gtsave(file = here("tables/top_genes_all.png"))
```

```{r}
fig4a.2 <- ggdraw() + draw_image(here("tables/top_genes_all.png")) + theme_void()

fig4A <- cowplot::plot_grid(fig4a.1, fig4a.2, ncol=2, labels = c("A"," "),align = "h", rel_widths = c(1.4,2))
fig4A
```




### Fig 4B 

network plot of genes DE accoridng to Eye Scores.

```{r, fig.height=4, fig.width=8}
library(clusterProfiler)
# ah <- AnnotationHub()
# orgs <- subset(ah, ah$rdataclass == "OrgDb")
# orgdb <- query(orgs, "Serinus canaria")[[1]]

options(ggrepel.max.overlaps = Inf)
network_ES_plot <- enrichGO_tol_ord2 %>%
  #simplify() %>%
  setReadable(orgdb) %>%
#  enrichplot::pairwise_termsim() %>%
  filter(ID %in% c("GO:0006412","GO:0006511",
                   "GO:0031966","GO:0006955",# "GO:0009057",
                   "GO:0006952","GO:0006629")) %>%
clusterProfiler::cnetplot(., layout = "graphopt", 
           # foldChange = ., 
           showCategory = 6, nodel_label ="category",
           max.overlaps=Inf,
           species = "scan",#node_label="gene",
           cex.params = list(gene_label = 0, category_label = 0.55),
           color.params = list(foldChange = tbl_tol_results %>%
                                 mutate(est= -case_when(
    Estimate > 5 ~ 5,
    Estimate < -5 ~ -5,
    TRUE ~ Estimate
  )) %>%
  arrange(desc(est)) %>%
  pull(est, entrezgene_id))
  ) + theme(legend.position = "right") + scale_color_continuous( expression(hat(beta)~"Eye Score"),type = "viridis"#, values = c("gold","purple")
                                                                 )

network_ES_plot
```


### Figure 4 combined
```{r, fig.height=7, fig.width=7}

fig4 <- cowplot::plot_grid(fig4A, network_ES_plot, ncol=1, labels = c("", "B"), rel_heights = c(3,2.5))
fig4
# ggsave(plot = fig4, filename= here("figures/fig4_phenotype.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")
```



# Supplemental Figures

helper functions.
```{r plot-DE-cleanerCode}
sig.cutoff <- 0.05
logFC.cutoff <- 0

source(here::here("scripts/canary_data_processing_functions.R"))
```

## Figure S1: MA plots
```{r}
# Density plot of estimated gene expression changes due to infection, according to diet.
# this was moved to the supplement per feedback advice.
s1.density <- topTags_lipid_inf_edgeR %>%
  left_join(., topTags_protein_inf_edgeR, by = c("gene","gene_name","description"),suffix = c("_lipid", "_protein")) %>%
  left_join(., topTags_inf_edgeR, by = c("gene","gene_name","description")) %>%
  pivot_longer(c(logFC_lipid, logFC_protein), names_to = "diet", values_to = "logFC_plot") %>%
  mutate(diet =
           case_when(diet == "logFC_protein" ~ "protein",
                     diet == "logFC_lipid" ~ "lipid")) %>%
mutate(fdr = case_when(
  diet == "lipid" ~ FDR_lipid,
  diet == "protein" ~ FDR_protein
)) %>%
  filter(FDR_lipid < 0.05| FDR_protein < 0.05 | FDR < 0.05) %>%
  ggplot(aes(x=logFC_plot,  color = diet, fill = diet
             )) +
  geom_density(alpha=0.5#,bw=0.01,#,outline.type = "upper"
                  ) +
  # geom_histogram(aes(y=..density..), alpha=0.5, position="identity", bins=30) +
  xlim(-3, 3) +
  # facet_grid(rows=vars(diff_day), cols=vars(diet)) +
  # scale_y_sqrt() +
  #ggtitle("2B") +
  xlab("logFC") +
  theme_classic(base_size = 18) +
    theme(legend.position = c(0.8,0.6))


# ggsave(plot = figs1.density, filename = here("figures/logFC_density_DE_in_diet_fig.png"), width = 16, height = 16, units = "cm", dpi=1200)

s1.ma.inf <- topTags_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (all)",show.names = F, 
                   logFC.cutoff = logFC.cutoff)

s1.ma.diet <- topTags_diet_uninf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Diet",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  geom_text(inherit.aes = F,data= data.frame(label = c("higher on lipid diet", "higher on protein diet"),
                             x = c(10,9.5),
                             y= c(-4,5), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red"))


s1.ma.interaction <- topTags_ixn_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG-Diet Interaction",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  geom_text(inherit.aes = F,data= data.frame(label = c("Greater MG effect \non lipid diet", "Greater MG effect \non protein diet"),
                             x = c(10,9.5),
                             y= c(-4.25,4.25), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red"))

s1.ma.inf.L <- topTags_lipid_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Lipid)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) 

s1.ma.inf.P <- topTags_protein_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Protein)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) + theme(legend.position = "right") + guides(color = guide_legend(reverse=TRUE)) 

s1.legend <- get_legend(s1.ma.inf.P)

fig.s1.ma <- cowplot::plot_grid(s1.ma.inf, #s1.ma.inf.L, s1.ma.inf.P+ theme(legend.position = "none"), 
                             s1.ma.diet, s1.ma.interaction, 
                              s1.legend, labels = c(LETTERS[seq(1,3)], "") )

fig.s1.ma

fig.s1 <- cowplot::plot_grid(fig.s1.ma, s1.density, ncol=1, labels = c("", "D"))

fig.s1

# ggsave(plot = fig.s1, filename = here("figures/fig.s1.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")
```


## Figure S2: KEGG plots

```{r}
fig.s2a <- gseGO_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %in% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>% #str()
  arrange(pvalue) %>%
  #filter(p.adjust < 0.3) %>%
  slice_head(n=10) %>%
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","no")) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 45, side="left")) + # cut off long names
  #xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("GO Enrichments") +
  theme_bw() +
  facet_wrap(~comparison, nrow=5, scales = "free_y") +
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))


fig.s2b <- gseKEGG_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %in% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>%
  arrange(pvalue) %>%
  # arrange(desc(abs(enrichmentScore))) %>%
  #filter(p.adjust < 0.3) %>%
  slice_head(n=10) %>% 
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","yes")) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 42)) + # cut off long names
  #xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("KEGG Enrichments") +
  theme_bw() +
  facet_wrap(~comparison, nrow=5, scales = "free_y") + 
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))



# fig.s2 <- cowplot::plot_grid(fig.s2a, fig.s2b, nrow=1)
fig.s2 <- ggpubr::ggarrange(fig.s2a, fig.s2b, common.legend = T,legend = "right")
fig.s2

# ggsave(plot = fig.s2, filename = here("figures/fig.s2.png"), width = 7, height = 8, units = "in", dpi=1200, bg="white")
```


## Figure S3: Gene Expression Correlations by Contrast

```{r}
# merged_results %>%
#   dplyr::select(gene.z,entrezgene_id_inf, Estimate, logFC_inf, logFC_inf_lipid, logFC_inf_prot,
#                 logFC_diet, logFC_ixn, logFC_logLoad) %>%
#   mutate(rowid = paste0(gene.z, "_", entrezgene_id_inf)) %>%
#   group_by(rowid) %>%
#   slice_head(n=1) %>%
#   column_to_rownames("gene.z") %>%
#   dplyr::select(-c(rowid, entrezgene_id_inf)) %>%
#   cor(use="pairwise.complete.obs") %>%
#   corrplot::corrplot(diag = T, addCoef.col = 'black')

fig.s3 <- merged_results %>%
  dplyr::select(DE_inf, gene.z,entrezgene_id_inf, Estimate, logFC_inf, logFC_inf_lipid, logFC_inf_prot,
                logFC_diet, logFC_ixn, logFC_logLoad) %>%
  rename(all_of(c("logFC_inf" = "Infection \n(all)" ,
         "logFC_inf_lipid" = "Infection \n(lipid)",
         "logFC_inf_prot" = "Infection \n(protein)",
         "logFC_diet"="Diet" ,
         "logFC_ixn"="MG-Diet \nInteraction",
         "Estimate" = "Eye \nScore",
         "logFC_logLoad"="Pathogen\n Load" 
         ))) %>%
  mutate(`Differentially\nExpressed\nInfection` = ifelse(DE_inf,"yes","no")) %>%
  relocate(`Differentially\nExpressed\nInfection`, .before = DE_inf) %>%
  dplyr::select(-DE_inf) %>%
  mutate(rowid = paste0(gene.z, "_", entrezgene_id_inf)) %>%
  group_by(rowid) %>%
  slice_head(n=1) %>%
  column_to_rownames("gene.z") %>%
  dplyr::select(-c(rowid, entrezgene_id_inf)) %>%
  GGally::ggpairs(., aes(color=fct_rev(`Differentially\nExpressed\nInfection`), alpha=0.3),
    upper = list(continuous = GGally::wrap("cor", size = 2))) + 
  theme_classic() +
  theme(axis.text = element_text(size = 7), 
        axis.title=element_text(size=7), axis.text.x=element_text(angle=90, hjust=1))

fig.s3
# ggsave(plot = fig.s3, filename = here("figures/fig.s3.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")

```

## Figure S4: Upset All Contrasts

```{r}
library(ComplexUpset)
lab_size = 1.5
pt_size = 0.9

fig.s4 <- merged_results %>%
  upset(., c("DE_inf",#"DE_inf_prot", "DE_inf_lipid", 
             "DE_diet", "DE_ixn", "DE_ES"#, "DE_logLoad"
             ), name="Comparison",
        base_annotations=list(
        'Intersection size'=intersection_size(text = list(size=3, vjust=-0.15),
            text_colors=c(
                on_background='black', on_bar='white'
            )
        )
        + annotate(
            geom='text', x=Inf, y=Inf,
            label=paste('Total:', nrow(topTags_inf_edgeR)),
            vjust=1.2, hjust=1.2, size = 3
        )
        + ylab('# of genes')
    ),
        annotations = list(
        'Infection\n(all)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf,color=-log10(FDR_inf), label=goi#_inf
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",
                                                       size=lab_size,
nudge_x = -0.4,force=0.1,
                                                       max.overlaps = 10) + scale_color_continuous(name="-log10(adj.p.value)", limits = c(0,5)) + theme(legend.direction = "vertical")
        ),
        'Infection\n(lipid)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf_lipid,color=-log10(FDR_inf_lipid), label=goi#_inf_lipid
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE)+ ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR) ") + theme(legend.position='none')
        ),
        'Infection\n(protein)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf_prot,color=-log10(FDR_inf_prot), label=goi#_inf_lipid
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)  ") + theme(legend.position='none')
        ),
        'Diet\n(Protein-Lipid)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_diet,color=-log10(FDR_diet), label=goi#_diet
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)   ") + theme(legend.position='none')
        ),
        'Interaction\n(Protein-Lipid)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_ixn,color=-log10(FDR_ixn), label=goi#_ixn
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)    ") + theme(legend.position='none')
        ),
        'Eye Score'=(
            ggplot(mapping=aes(y=est,color=-log10(FDR_tol),label=goi#_ES
                               ))
            + ggbeeswarm::geom_quasirandom(size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)     ")+ guides(fill = guide_legend(override.aes = list(size = 0.5))) + theme(legend.position='none')
        )#,
        # 'logLoad'=(
        #     ggplot(mapping=aes(y=logFC_logLoad))
        #     + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.5,
        #         na.rm=TRUE)
        # )
        ),
        keep_empty_groups=TRUE, themes=upset_default_themes(text=element_text(size=8),axis.title=element_text(size=8,face="bold")),
        min_degree=2, min_size = 8,
    # moves legends over the set sizes
    guides='over') & theme(legend.margin=margin(t=0, r=0, b=1, l=0, 'cm'))



# ggsave(plot = fig.s4, filename = "figures/fig.s4.DE_genes_multinode_ES_combined_upsetplot.png", width = 6, height = 9, units = "in", dpi = 1200)
```


## Figure S5: MA plot for phenotype analysis
```{r}
s5.ma.logLoad <- topTags_logLoad_inf %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Pathogen Load Effect on Gene Expression",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  theme(plot.title = element_text(size=10), legend.position = "bottom")

s5.ma.eyeScore <- merged_results %>%
  filter(!is.na(Estimate)) %>%
    mutate(delabel = case_when(
      !is.na(gene_name) ~ gene_name,
      TRUE ~ paste0("LOC",gene)
    )) %>%
    mutate(delabel = case_when(
      FDR_tol < sig.cutoff ~ delabel,
      abs(Estimate) > logFC.cutoff ~ delabel,
      TRUE ~ NA
    )) %>%
    mutate(diffexpressed = case_when(
      FDR_tol > sig.cutoff ~ "Neither",
      Estimate < 0 ~ "Down",
      Estimate > 0 ~ "Up"
    )) %>%
    mutate(diffexpressed = factor(diffexpressed, levels = c("Neither", "Down", "Up"))) %>%
    arrange(diffexpressed) %>%
  mutate(logCPM = logCPM_inf) %>%
  ggplot(., aes(x = logCPM, 
                   y = Estimate, 
                   col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual("Differentially\n Expressed",values = c("black", "blue", "red")) +
    #geom_vline(xintercept = c(-logFC.cutoff, logFC.cutoff), col = "red") +
    #geom_hline(yintercept = -log10(sig.cutoff), col = "red") +
    #{if (show.names) ggrepel::geom_text_repel(size=2, na.rm=TRUE)} +
      scale_size_area() +
    # ggsci::scale_color_aaas() +
      ggtitle("Eye Score Effect on Gene Expression") + 
      theme(legend.position="none",
            text = element_text(size=10),
            axis.text = element_text(size=10)) +
    # add custom text to plot
  geom_text(inherit.aes = F,data= data.frame(label = c("Expression + correlates\n   with Eye Score", "Expression - correlates\n  with Eye Score"),
                             x = c(10,10),
                             y= c(-15,15), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red")) +
    #theme(legend.position = "right") + 
    guides(color = guide_legend(reverse=TRUE)) +
  theme(plot.title = element_text(size=10))

  
  
  
  
  
# s5.legend <- get_legend(s5.ma.eyeScore)

# fig.s5.no.legend <- cowplot::plot_grid(s5.ma.logLoad, s5.ma.eyeScore#+ theme(legend.position = "none")
#                                        , labels = c("AUTO"),nrow=2)

fig.s5 <- cowplot::plot_grid(s5.ma.logLoad, s5.ma.eyeScore, nrow=2, labels = c("AUTO"),
                             rel_heights = c(1.3,1))


fig.s5

# ggsave(plot = fig.s5, filename = here("figures/fig.s5.png"), width = 4, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure S6: GO and KEGG (Phenotype)

```{r}
fig.s6a <- gseGO_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %ni% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>% #str()
  arrange(pvalue) %>%
  #filter(p.adjust < 0.3) %>%
  slice_head(n=10) %>%
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","no")) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 45, side="right")) + # cut off long names
  #xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("GO Enrichments") +
  theme_bw() +
  facet_wrap(~fct_rev(comparison), nrow=5, scales = "free_y") +
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))


fig.s6b <- gseKEGG_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %ni% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>%
  arrange(pvalue) %>%
  # arrange(desc(abs(enrichmentScore))) %>%
  #filter(p.adjust < 0.3) %>%
  slice_head(n=10) %>% 
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","yes")) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 42)) + # cut off long names
  #xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("KEGG Enrichments") +
  theme_bw() +
  facet_wrap(~fct_rev(comparison), nrow=5, scales = "free_y") + 
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))

fig.s6 <- ggpubr::ggarrange(fig.s6a, fig.s6b, common.legend = T,legend = "right")
fig.s6

# ggsave(plot = fig.s6, filename = here("figures/fig.s6.png"), width = 7, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure S7: PCA plot
```{r}
PCA <- plotMDS(y_ixn, gene.selection="common",
               top = 300, pch = 2, cex = 0.5, main = "PCA plot of gene expression for all samples", dim.plot=c(1,2))

ggPCA <- tibble(
  SampID = colnames(PCA$distance.matrix.squared),
  x = PCA$x,
  y =PCA$y
) %>%
  mutate(SampID = str_replace_all(SampID, "\\.","-")) %>%
  left_join(metadata_phenotypes, by="SampID")

fig.s7 <- ggplot(ggPCA, aes(x=x,y=y, group=interaction(Diet,MG.active), #shape = Day,
                  color = interaction(MG.active,Diet), linetype = Diet)) +
  #geom_line(aes(alpha = MG.Status)) +
  geom_point() +
  theme_classic() +
  # theme(legend.position = "none") +
  labs(title="PCA plot of gene expression") +
  xlab(paste0("Principal Component 1 (",100*round(PCA$var.explained[1],2),"%)")) +
  ylab(paste0("Principal Component 2 (",100*round(PCA$var.explained[2],2),"%)")) +
  scale_color_discrete("MG.Diet") +
  stat_ellipse(type = "norm",level=0.8)
  # facet_wrap(~MG.active)

# ggsave(plot = fig.s7, filename = here("figures/fig.s7.png"), width = 7, height = 6, units = "in", dpi=1200, bg="white")
```



# Suppplemental Tables

## Table S2: DE genes in the interaction between diet and infection response

```{r}
# topTags_ixn_edgeR |>
#   filter(FDR < 0.05) |>
#   # arrange(desc(logFC)) %>%
#   arrange(logFC) |>
#   mutate(across(is.numeric, signif, 3)) |>
#   mutate(across(is.numeric, as.character, 3)) |>
#   dplyr::select(gene, gene_name, logFC, logCPM, `F`, PValue, FDR, description) |>
#   kbl(booktabs = T, longtable = T) |>
#     kable_styling("striped", latex_options = c("repeat_header")) #|>
#   save_kable(here("tables/table.s2.png"))

```
To include all comparisons in this table, let's instead show 20 largest magnitude DE (10 up and 10 down) genes for each contrast in a single table.
```{r}
table.s3.ixn <- topTags_ixn_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

table.s3.inf <- topTags_inf_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG Infection") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

table.s3.diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "Diet (Protein vs Lipid)") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

bind_rows(table.s3.ixn, table.s3.inf, table.s3.diet) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  dplyr::select(-comparison) |>
  rename(gene = "Top DE Genes",
         gene_name = "Gene Name") |>
  replace_na(list(rep("", 5))) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  pack_rows("MG-Diet Interaction", 1, 20) |>
  pack_rows("MG Infection", 21, 40) |>
  pack_rows("Diet (Protein-Lipid)", 41, 60) #|>
  # save_kable(here("tables/table.s2.png"))
```



## Table S3: Significant GO terms
```{r}
gseGO_all_df |>
    rownames_to_column("rowname") |>
  dplyr::select(comparison, ID, Description, setSize, enrichmentScore, NES, pvalue, p.adjust, qvalue, rank, ONTOLOGY) |>
  mutate(comparison = case_when(
    comparison == "infection_all" ~ "Infection (all)",
    comparison == "diet_uninf" ~ "Diet (Protein-Lipid)",
    comparison == "interaction_diet:infection" ~ "MG-Diet Interaction",
    comparison == "infection_protein" ~ "Infection (protein)",
    comparison == "infection_lipid" ~ "Infection (lipid)",
    comparison == "pathogen_load" ~ "Pathogen Load",
    comparison == "eye_score" ~ "Eye Score"
  )) |>
  mutate(across(is.numeric, signif, 3)) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  # pack_rows("MG-Diet Interaction", 1, 20) |>
  # pack_rows("MG Infection", 21, 40) |>
  # pack_rows("Diet (Protein-Lipid)", 41, 60) 
  collapse_rows(columns = 1, valign="top", latex_hline = "major") #|>
  #save_kable(here("tables/table.s3.png"))
```

## Table S4: Heatmap Cluster GO Enrichments
```{r}
table.s4.go <- list_GO_results |>
  bind_rows(.id= "cluster") |>
  rownames_to_column("rowname") |>
  group_by(cluster) |>
  arrange(pvalue) |>
  filter(Count > 3) |>
  slice_head(n=5) |>
  dplyr::select(cluster, ID, Description, GeneRatio, BgRatio,# pvalue,
                p.adjust#, qvalue, ONTOLOGY
                ) |>
  mutate(across(is.numeric, signif, 3)) #|>
  # kbl(booktabs = T, longtable = T) |>
  #   kable_styling("striped", latex_options = c("repeat_header")) |>
  # # pack_rows("MG-Diet Interaction", 1, 20) |>
  # # pack_rows("MG Infection", 21, 40) |>
  # # pack_rows("Diet (Protein-Lipid)", 41, 60) 
  # collapse_rows(columns = 1, valign="top", latex_hline = "major")

table.s4.kegg <- list_kegg_results |>
  bind_rows(.id= "cluster") |>
  rownames_to_column("rowname") |>
  group_by(cluster) |>
  arrange(pvalue) |>
  filter(Count > 3) |>
  slice_head(n=5) |>
  dplyr::select(cluster, ID, Description, GeneRatio, BgRatio,# pvalue,
                p.adjust#, qvalue, ONTOLOGY
                ) |>
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", "")) |>
  mutate(across(is.numeric, signif, 3)) #|>
  
  
bind_rows(table.s4.go, table.s4.kegg) |>
  group_by(cluster) |>
  arrange(cluster) |>
  mutate(across(is.numeric, as.character, 3)) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  # pack_rows("MG-Diet Interaction", 1, 20) |>
  # pack_rows("MG Infection", 21, 40) |>
  # pack_rows("Diet (Protein-Lipid)", 41, 60) 
  collapse_rows(columns = 1, valign="top", latex_hline = "major")


```

## Table S5: DE genes for pathogen load
```{r}
 topTags_logLoad_inf |>
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=20) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(gene, gene_name, logFC, FDR, description) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  rename(gene = "Top DE Genes",
         gene_name = "Gene Name") |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header"))
```

## Table S6: DE genes for Eye Score
```{r}
 tbl_tol_results |>
  filter(FDR_tol<0.05) %>%
  group_by(sign(Estimate)) %>%
  arrange(`Pr(>|z|)`) %>%
  slice_head(n=20) %>%
  arrange(Estimate) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(gene.z, gene_name, Estimate,`Pr(>|z|)`, description) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  rename(gene.z = "Top DE Genes",
         gene_name = "Gene Name"#,
         #FDR_tol = "FDR"
         ) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header"))
```


## Table S7: RNA-seq Mapping statistics
```{r}

```

