---
title: "Dream Differential Expression"
author: "Carson Stacy"
date: "`r Sys.Date()`"
output: html_document
---


```{r setup , include=FALSE}
knitr::opts_chunk$set(cache = TRUE,
                      echo = FALSE, 
                      dpi=300,
                      fig.align = "center",
                      fig.width = 8, 
                      fig.height = 5, 
                      dev = "png")
```

# Differential Expression Analysis with Random Effects in Dream

load packages
```{r library}
# library(tidyverse)
if (!require("pacman", quietly = TRUE)) install.packages("pacman")
pacman::p_load(edgeR, 
               tidyverse,
               limma, 
               variancePartition, #dream included
               topGO,
               biomaRt,
               GOplot,
               reactable,
               reactablefmtr,
               here,
               data.table
               # BiocParallel, 
               # pander
               )
```

define functions
```{r functions}
# A function to tidy gene expression matrix
get_tidy_expression <- function(data_path, meta, type = "tpm") {
  if (type == "tpm") {
    # tpm_df <- readr::read_tsv(data_path,
    #                           show_col_types = FALSE) %>%
    tpm_df <- data.table::fread(data_path, sep="\t") %>%
      tibble() %>%
      dplyr::select(
        -c(
          "ES89_Blue-049_Control-protein_0",
          "ES31_Blue-055_NA_0",
          "ES32_Blue-055_NA_14",
          "ES65_106_NA_0",
          'transcript_id(s)'
        )
      )
    
    meta_subset <- meta %>%
      dplyr::select(c(1,3,4,5,6,7))
    
    tpm_df <<- tpm_df %>%
      as_tibble() %>%
      pivot_longer(
        !gene_id,
        names_to = c("Lane", "Bird", "MG.Diet", "Day"),
        names_sep = "_",
        values_to = type
      ) %>%
      separate(col = MG.Diet, c("MG", "Diet"), remove=FALSE) %>%
      left_join(meta_subset, join_by(Lane==SampID,
                                     Bird==Bird.ID,
                                     Diet==Diet,
                                     Day==Day,
                                     MG.Diet==MG.Diet))
    
  } else if (type == "count") {
    # count_df <- readr::read_tsv(data_path,
    #                             show_col_types = FALSE) %>%
    count_df <- data.table::fread(data_path, sep="\t") %>%
      tibble() %>%
      dplyr::select(
        -c(
          "ES89_Blue-049_Control-protein_0",
          "ES31_Blue-055_NA_0",
          "ES32_Blue-055_NA_14",
          "ES65_106_NA_0"
        )
      )
    count_df <<-count_df %>%
      as_tibble(rownames = "gene") %>%
      pivot_longer(
        !gene,
        names_to = c("Lane", "Bird", "MG.Diet", "Day"),
        names_sep = "_",
        values_to = type
      ) %>%
      separate(col = MG.Diet, c("MG", "Diet"), remove = FALSE)
    
  } else if (length(type)==2){
    # tpm_df <- readr::read_tsv(filepath_rsem_gene_tpm,
    #                           show_col_types = FALSE) %>%
    tpm_df <- data.table::fread(filepath_rsem_gene_tpm, sep="\t") %>%
      tibble() %>%
      dplyr::select(
        -c(
          "ES89_Blue-049_Control-protein_0",
          "ES31_Blue-055_NA_0",
          "ES32_Blue-055_NA_14",
          "ES65_106_NA_0"
        )
      )
    tpm_df <<- tpm_df %>%
      as_tibble(rownames = "gene") %>%
      pivot_longer(
        !gene,
        names_to = c("Lane", "Bird", "MG.Diet", "Day"),
        names_sep = "_",
        values_to = "tpm"
      ) %>%
      separate(col = MG.Diet, c("MG", "Diet"))
    
    # count_df <- readr::read_tsv(filepath_rsem_gene_counts,
                                # show_col_types = FALSE) %>%
    count_df <- data.table::fread(filepath_rsem_gene_counts, sep="\t") %>%
      tibble() %>%
      dplyr::select(
        -c(
          "ES89_Blue-049_Control-protein_0",
          "ES31_Blue-055_NA_0",
          "ES32_Blue-055_NA_14",
          "ES65_106_NA_0"
        )
      )
    count_df <<-count_df %>%
      as_tibble(rownames = "gene") %>%
      pivot_longer(
        !gene,
        names_to = c("Lane", "Bird", "MG.Diet", "Day"),
        names_sep = "_",
        values_to = "count"
      ) %>%
      separate(col = MG.Diet, c("MG", "Diet"),remove=FALSE)
    count_df <<- count_df
  } else {
    stop("Error in inputs")
  }
}


# negate %in%
`%ni%` <- Negate(`%in%`)


# A function to take gene name and plot ggplot scatterboxplot
geneviz <- function(gene, tpm_df, subset) {
  ggdat <- 
    if (subset=="CTLvsINF") {
      tpm_df %>%
        filter(gene_id == gene) %>%
        dplyr::filter(MG == "Control" | Day != 0)
    } else if (subset=="DAY0") {
      tpm_df %>%
        filter(gene_id == gene) %>%
        dplyr::filter(Day == 0)
    } else if (subset=="ALL_UNINF") {
      tpm_df %>%
        filter(gene_id == gene) %>%
        dplyr::filter(MG == "Control" | Day == 0)
    } else {
      tpm_df %>%
        filter(gene_id == gene)
    }
  
  # identify where anova stat should go
  anova_y_low <- ggdat %>%
    dplyr::mutate(log2tpm = log2(tpm+1)) %>%
    dplyr::arrange(log2tpm) %>%
    head(1) %>%
    pull(log2tpm) %>% as.numeric()
  
  y_high <- ggdat %>%
    dplyr::mutate(log2tpm = log2(tpm+1)) %>%
    dplyr::arrange(desc(log2tpm)) %>%
    head(1) %>%
    pull(log2tpm) %>% as.numeric()
    
  
  ggdat %>%
    ggplot(aes(
      x = interaction(Diet, MG),
      y = log2(tpm+1),
      #color=Diet,
      shape = as.factor(Day),
      group = interaction(Diet, MG)
    )) +
    # ylim(0,3) +
    geom_boxplot(width = 0.3, outlier.shape = NA) +
    ggbeeswarm::geom_quasirandom(cex = 1.5) +
    ggpubr::stat_compare_means(
      comparisons =
        list(
          c("lipid.Control", "protein.Control"),
          c("lipid.MG", "protein.MG"),
          c("lipid.Control", "lipid.MG"),
          c("protein.Control", "protein.MG")
        ),
      method = "t.test",
      # label = "p.signif",
      label = "p.format",
      # Add pairwise comparisons p-value
      label.y = c(y_high+0.2,y_high+0.2,y_high+0.7,y_high+1.3)
    ) +
    # ylim(-1,8) +
    ylim(anova_y_low-1, y_high+2) +
    ggpubr::stat_compare_means(label.y = anova_y_low-0.5, method = "anova") +
    ggtitle(paste0(gene, " expression - ", subset)) +
    theme_classic()
}


# visualize gene expression over time
timevizMG <- function(gene, tpm_df, include_control=T, diff=T) {
  if (include_control==T) {
    if (diff==T) {
      tpm_df %>%
        filter(gene_id == gene) %>%
        group_by(Bird) %>%
        dplyr::select(c('gene_id', "Bird", "MG", "Diet", 
                        "Day", "tpm")) %>%
        dplyr::mutate(Day = paste0("Day", Day)) %>% 
        # dplyr::filter(tpm>1) %>% # remove samples where gene has very low expression
        pivot_wider(names_from=Day, values_from=tpm) %>%
        dplyr::mutate(diff14v0 = log(Day14)-log(Day0),
                  diff21v0 = log(Day21)-log(Day0)) %>%
        dplyr::select("gene_id", "Bird", "MG", "Diet", 
                        "diff14v0", "diff21v0") %>%
        pivot_longer(c("diff14v0", "diff21v0"),names_to="comparison", values_to="LogDiff") %>%
        ggplot(aes(
          x = interaction(Diet,MG),
          y = LogDiff,
          shape = Diet
        )) +
        # geom_line() +
        geom_boxplot(outlier.shape = NA) +
        geom_point() +
        ggtitle(paste0(gene, " expression changes")) +
        theme_classic() +
          facet_wrap(~comparison)
    } else {
    tpm_df %>%
      filter(gene_id == gene) %>%
      ggplot(aes(
        x = Day,
        y = log2(tpm+1),
        shape = Diet,
        group = Bird
      )) +
      geom_point() +
      geom_line() +
      # geom_boxplot() +
      ggtitle(paste0(gene, " expression over time")) +
      theme_classic() +
        facet_wrap(~MG)
    }
  } else {
  tpm_df %>%
    filter(gene_id == gene,
           MG == "MG") %>% # filter only the MG group
    ggplot(aes(
      x = Day,
      y = log2(tpm+1),
      shape = Diet,
      group = Bird
    )) +
    geom_point() +
    geom_line() +
    # geom_boxplot() +
    ggtitle(paste0(gene, " expression over time in MG infected birds")) +
    theme_classic()
  }
}
```

```{r read-data}
# assign file path locaiton of rsem.merged.gene_counts.tsv file
filepath_rsem_gene_counts <- here::here("SerCan2020", "rsem.merged.gene_counts.tsv")

#assign location of tpm file for later analysis
filepath_rsem_gene_tpm <- here::here("SerCan2020", "rsem.merged.gene_tpm.tsv")

# read in rsem.merged.gene_counts.tsv
g.counts.l <- data.table::fread(filepath_rsem_gene_counts, sep="\t") %>% as_tibble()
  # readr::read_tsv(filepath_rsem_gene_counts, show_col_types = FALSE, trim_ws = FALSE,
  #          name_repair = "minimal", progress = TRUE, skip=1)
# i don't know why read_tsv is not working, i'm using fread instead.
```

### Remove problematic samples

The Sample ES89 was removed due to failed sequencing.
Samples ES31,ES32, and ES65 were removed due to incomplete sample metadata.
Samples from bird SEM19-1481 had their metadata corrected due to an error in recording of diet received.
```{r remove-samples}
g.counts.l <- g.counts.l %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>% 
  # change diet group of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       # c(`ES64_SEM19-1481_MG-lipid_21`="ES64_SEM19-1481_MG-protein_21",
       #   `ES52_SEM19-1481_MG-lipid_14`="ES52_SEM19-1481_MG-protein_14",
       #   `ES83_SEM19-1481_MG-lipid_0`="ES83_SEM19-1481_MG-protein_0"))
       # )
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       )
```


### generate metadata

Using sample names, a tibble of metadata is generated.
```{r generate-metadata}
# create tibble metadata object, convert column names to sample ID column
metadata <-
  tibble(SampID = colnames(g.counts.l[, c(3:ncol(g.counts.l))])) %>%
  separate(
    SampID,
    c("Sample.ID", "Bird.ID",
      "MG.Diet", "Day"),
    "_",
    extra = "warn",
    fill = "warn",
    remove = FALSE
  ) %>%
  separate(
    MG.Diet,
    c("MG.Status", "Diet"),
    sep = "[-]",
    extra = "warn",
    fill = "warn",
    remove = FALSE
  ) %>%
  mutate(
    MG.Diet.Day = paste0(MG.Diet, ".", Day),
    MG.Day = paste0(MG.Status, ".", Day),
    Diet.Day = paste0(Diet, ".", Day)
  ) %>%
  mutate(MG.active = if_else(Day < 1, "NO",
                             if_else(MG.Status == "Control",
                                     "NO", "YES"))) %>%
  mutate(
    Exposure.MG = paste0(MG.active, ".", MG.Status),
    Exposure.Diet = paste0(MG.active, ".", Diet),
    Exposure.Diet.Day = paste0(Exposure.Diet, ".", Day)
  )


# read in phenotype data

## pathogen loads
canary_loads <-
  # only birds with both phenotype and genotype data
  # readr::read_csv(here("Geno.x.Pheno/canary_loads.csv")) %>% 
  data.table::fread(here("Geno.x.Pheno/canary_loads.csv"),sep=",") %>% 
  tibble() %>%
  # mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  mutate(Bird = stringr::str_replace(ID, " ", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  # mutate(Bird = stringr::str_replace(Bird, "-", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD.19 239", "LD19.239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue.045", "Blue.45")) %>%
  dplyr::filter(Bird %in% unique(metadata$Bird.ID)) %>%
  dplyr::select(-ID)

## semi-quantitative eyescore measures
eyescores <-
  # readr::read_csv(
  data.table::fread(
    here("Geno.x.Pheno/canary_eyescores.csv"),
    sep=","
    # col_types = cols(Day = col_integer(),
    #                  Date = col_character())
  ) %>%
  tibble() %>%
  mutate(Day = as.integer(Day),
         Date = as.character(Date)) %>%
  # correcting for errors in manual data entry
  mutate(Bird = stringr::str_replace(ID, " ", "-")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  mutate(Bird = stringr::str_replace(Bird, "\\.", "-")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Nlue", "Blue")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD-19 239", "LD19-239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue-045", "Blue-45")) %>%
  dplyr::filter(Bird %in% unique(metadata$Bird.ID)) %>%
  dplyr::mutate(day = replace(Day, which(Day<0), 0)) %>% # adjust day -1 to day 0
  dplyr::select(Bird, day, ES)

# inspired from https://support.bioconductor.org/p/p132992/

# generate columns of phenotypic metadata I'd like to have
# pathogen loads
day3_loads <- canary_loads %>%
  filter(day==3) %>%
  mutate(logLoad_day3 = logLoad, r_logLoad_day3=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day3, r_logLoad_day3))

day7_loads <- canary_loads %>%
  filter(day==7) %>%
  mutate(logLoad_day7 = logLoad, r_logLoad_day7=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day7, r_logLoad_day7))

day14_loads <- canary_loads %>%
  filter(day==14) %>%
  mutate(logLoad_day14 = logLoad, r_logLoad_day14=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day14, r_logLoad_day14))

day21_loads <- canary_loads %>%
  filter(day==21) %>%
  mutate(logLoad_day21 = logLoad, r_logLoad_day21=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day21, r_logLoad_day21))

#AUC estimates
day14_AUC_loads <- canary_loads %>%
  filter(day<=14) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,logLoad, type = "spline")
  ) %>%
  mutate(logLoad_AUC_14 = auc, r_logLoad_AUC_14=auc/max(auc)) %>%
  dplyr::select(c(Bird, logLoad_AUC_14, r_logLoad_AUC_14))

day21_AUC_loads <- canary_loads %>%
  filter(day<=21) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,logLoad, type = "spline")
  ) %>%
  mutate(logLoad_AUC_21 = auc, r_logLoad_AUC_21=auc/max(auc)) %>%
  dplyr::select(c(Bird, logLoad_AUC_21, r_logLoad_AUC_21))


load_phenotypes <- purrr::reduce(list(day3_loads,day7_loads,
                                      day14_loads, day21_loads,
                                      day14_AUC_loads, day21_AUC_loads), 
                                 dplyr::left_join, by = 'Bird')

## now do the same for eyescore data
day_eyes <- eyescores %>%
  filter(., day %in% c(0,14,21)) %>%
  mutate(day=as.character(day)) %>%
  dplyr::select(c(Bird, day, ES))

day3_eyes <- eyescores %>%
  filter(.,day==3) %>%
  mutate(ES_day3 = ES, r_ES_day3=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day3, r_ES_day3))

day7_eyes <- eyescores %>%
  filter(.,day==7) %>%
  mutate(ES_day7 = ES, r_ES_day7=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day7, r_ES_day7))

day14_eyes <- eyescores %>%
  filter(.,day==14) %>%
  mutate(ES_day14 = ES, r_ES_day14=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day14, r_ES_day14))

day21_eyes <- eyescores %>%
  filter(.,day==21) %>%
  mutate(ES_day21 = ES, r_ES_day21=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day21, r_ES_day21))

#AUC estimates
day7_AUC_eyes <- eyescores %>%
  filter(.,day<=7) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_7 = auc, r_ES_AUC_7=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_7, r_ES_AUC_7))

day14_AUC_eyes <- eyescores %>%
  filter(.,day<=14) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_14 = auc, r_ES_AUC_14=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_14, r_ES_AUC_14))

day21_AUC_eyes <- eyescores %>%
  filter(.,day<=21) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_21 = auc, r_ES_AUC_21=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_21, r_ES_AUC_21))



eye_phenotypes <- purrr::reduce(list(day3_eyes,day7_eyes,
                                      day14_eyes, day21_eyes,
                                      day7_AUC_eyes,
                                      day14_AUC_eyes, day21_AUC_eyes), 
                                 dplyr::left_join, by = 'Bird')

## add phenotype data to existing metadata df
metadata_phenotypes <- metadata %>%
  left_join(load_phenotypes, by = join_by('Bird.ID' == "Bird")) %>%
  left_join(eye_phenotypes, by = join_by('Bird.ID' == "Bird")) %>%
  left_join(day_eyes, by = join_by('Bird.ID' == "Bird", "Day" == "day")) %>%
  mutate_all(., ~replace_na(.,0)) %>%
  distinct() %>% # to get rid of duplicated row
  mutate(ES_minMax = ES/6) %>% # normalized to 0 to 1
  mutate(logLoad = case_when(
    Day == 0 ~ 1+logLoad_day3,
    Day == 14 ~ 1+logLoad_day14,
    Day == 21 ~ 1+logLoad_day21,
    TRUE ~ NA
  ),
  ES_day0_analysis = case_when(
    Day == 0 ~ ES_day3,
    TRUE ~ ES
  )) %>%
  mutate(tolerance = (logLoad/(ES_day0_analysis+1))-1) # tolerance index

# generate tidy data objects for downstream analysis
get_tidy_expression(data_path=filepath_rsem_gene_tpm, 
                    meta=metadata_phenotypes, 
                    type = "tpm")
```

```{r make-DGElist}
# Generate count matrix
gene_expression_matrix <- g.counts.l %>% #View()
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  #dplyr::filter(gene_id != "103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  
  # convert to matrix for edgeR
  as.matrix()

# group arrange samples by in DGEList
# group <- metadata$Exposure.Diet
group <- metadata$MG.active

# correct gene names
entrez_IDs <- getBM(attributes = c(#'ensembl_transcript_id',
                            #'ensembl_peptide_id',
                            # 'ensembl_gene_id', 
                            # 'uniprot_gn_symbol',
                            'wikigene_name',
                            'entrezgene_id',
                            'external_gene_name',
                            'go_id'
                            ),
             values = rownames(gene_expression_matrix),
             mart = mart) |>
  mutate(entrezgene_id = as.character(entrezgene_id))

entrez_names_gene_expression_matrix <-
  gene_expression_matrix |>
  data.frame() |>
  rownames_to_column() |> 
  left_join(entrez_IDs |> dplyr::select(-go_id),
            by = join_by("rowname" == "external_gene_name"), multiple = "first") |>
  mutate(entrezID = case_when(
    str_detect(rowname, "^LOC") ~ str_replace(rowname, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ rowname
  )) |> #View()
  pull(entrezID)

rownames(gene_expression_matrix) <- entrez_names_gene_expression_matrix



# make DGEList
y <- DGEList(
  counts = gene_expression_matrix,
  group = group,
  remove.zeros = TRUE
)
```


```{r assign-variables}
# Bird.ID <- metadata$Bird.ID
Bird <- factor(metadata$Bird.ID)
Disease <- factor(metadata$MG.active, levels=c("NO","YES"))
Diet <- factor(metadata$Diet, levels=c("protein","lipid"))
Day <- metadata$Day

Treatment <- factor(metadata$MG.Status, levels=c("Control","MG"))
# Trt.protein <- Treatment=="MG" & Diet == "protein"
# Trt.lipid <- Treatment=="MG" & Diet == "lipid"
# # MG <- Disease== "YES"
# protein <- Diet=="protein"
# control <- Disease=="NO"
```


```{r filter-genes}
# keep <- filterByExpr(y,min.total.count = 50)
# keep <- rowSums(cpm(y)>1) >= 15 # this was cpm>10 before, which missed some relevant genes.

#determine cpm cutoff
cpm_cutoff <- as.numeric(round(cpm(10, mean(y$samples$lib.size)),1))
# apply cutoff
keep <- rowSums(cpm(y) > cpm_cutoff) >= 4
#filter genes
y <- y[keep, , keep.lib.sizes=FALSE]
```

```{r normalize-expression}
y <- calcNormFactors(y)

# match sample names to metadata
colnames(y) <- str_replace_all(colnames(y), "\\.", "-")
```

```{r setup-dream}
param = SnowParam(5, "SOCK", progressbar=TRUE)
```

```{r fit-dream}
# design = model.matrix( ~ Disease + Diet, metadata_phenotypes)
# vobj_tmp = voom( y, design, plot=FALSE)


# The variable to be tested must be a fixed effect
form <- ~ MG.active + Diet + (1|Bird.ID)

# form <- ~ 0 + Exposure.Diet + (1|Bird.ID)

# estimate weights using linear mixed model of dream
vobjDream = voomWithDreamWeights(y, form, metadata, BPPARAM=param )

# Fit the dream model on each gene
# By default, uses the Satterthwaite approximation for the hypothesis test
fitmm = dream( vobjDream, form, metadata )
fitmm = eBayes(fitmm)
```

```{r save-dream}
dir_path <- here("results/dream")

if (!dir.exists(dir_path)) {
  dir.create(dir_path, recursive = TRUE)
  cat("Directory created:", dir_path, "\n")
} else {
  cat("Directory already exists:", dir_path, "\n")
}

saveRDS(fitmm, here("results/dream/fitmm.rds"))

fitmm <- readRDS(here("results/dream/fitmm.rds"))
```

```{r fit-deseq2}
```

```{r fit-voom}
```

```{r fit-limma}
```

Get gene information from BioMart
```{r generate-geneInfo}
mart <- if(exists('mart') == TRUE){
  mart
} else {
  biomaRt::useMart(biomart = "ensembl", dataset = "scanaria_gene_ensembl")
}


gene_info <- biomaRt::getBM(attributes = c(#'ensembl_transcript_id',
                            # 'ensembl_peptide_id',
                            'ensembl_gene_id', 
                            # 'uniprot_gn_symbol',
                            'wikigene_name',
                            'entrezgene_id',
                            'entrezgene_description',
                            # '',
                            'external_gene_name'
                            # 'go_id',
                            # 'description'
                            ),
             values = g.counts.l$gene_id,
             mart = mart) %>%
  dplyr::mutate(entrezgene_id = paste0("LOC",entrezgene_id)) %>%
  dplyr::mutate(gene = ifelse(entrezgene_id %in% rownames(fitmm@.Data[[1]]), entrezgene_id, external_gene_name))
  # dplyr::select(c("entrezgene_id", "wikigene_name", "entrezgene_description")) %>%
  # distinct()

```

# fitmm: MG.active + Diet + (1|Bird.ID)

```{r identify-DEgenes}
head(fitmm$design, 3)

dream_DE_MG_genes <- variancePartition::topTable( fitmm, coef='MG.activeYES',number = Inf, p.value = 0.1
                             ) %>% rownames_to_column("gene")# %>%
   # filter(abs(logFC)>0.2)
dream_DE_Diet_genes <- variancePartition::topTable( fitmm, coef='Dietprotein',number = Inf, p.value = 0.1
                             ) %>% rownames_to_column("gene")
# dream_DE_MG_genes <- variancePartition::topTable( fitmm, coef='MG.activeYES',number = Inf, p.value = 0.1
#                              ) %>% rownames_to_column("gene") %>%
#   filter(abs(logFC)>log2(1.2))

# dream_MG_genes <- variancePartition::topTable( fitmm, coef='MG.activeYES',number = Inf#, p.value = 0.05
#                              ) %>% rownames_to_column("gene")# %>% gt()

#MG effect
dream_MG_genes <-  variancePartition::topTable( fitmm, coef='MG.activeYES',number = Inf#, p.value = 0.2
                             ) %>% rownames_to_column("gene") %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
    dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) #%>%
  # View()

#Protein effect
dream_Diet_genes <- variancePartition::topTable( fitmm, coef='Dietprotein',number = Inf,adjust.method	= "fdr", confint=T,#, p.value = 0.2
                             ) %>% rownames_to_column("gene") %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
    dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) # %>% View()

# #Dayz effect
# variancePartition::topTable( fitmm, coef='Day21',number = Inf,adjust.method	= "fdr", confint=T,#, p.value = 0.2
#                              ) %>% rownames_to_column("gene") %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
#     dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) %>% View()

# interaction between protein and MG
# variancePartition::topTable( fitmm, coef='MG.activeYES:Dietprotein',number = Inf#, p.value = 0.2
#                              ) %>% rownames_to_column("gene") %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
#     dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) #%>%
  # View()


dream_MG_genes %>% 
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fitmm: MG.activeYES coefficient")

dream_Diet_genes %>% 
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fitmm: Dietprotein coefficient")


# variancePartition::topTable( fitmm, coef='Exposure.DietYES.lipid',number = Inf) %>% rownames_to_column("gene") #%>% View()
```
This design did not include interactions, so we can't break down the analysis further. Instead, let's follow an approach used in the edgeR manual to make dream contrasts.

# fit: 0 + Exposure.Diet + (1|Bird.ID)

```{r estimate-blockedDream, fig.height=10, fig.width=14}
# estimate weights using linear mixed model of dream

form <- ~ 0 + Exposure.Diet + (1|Bird.ID) #+ Day

vobjDream = voomWithDreamWeights( y, form, metadata_phenotypes, BPPARAM=param )

L = makeContrastsDream( form, metadata, 
  contrasts = c("Exposure.DietYES.lipid - Exposure.DietNO.lipid", 
                "Exposure.DietYES.protein - Exposure.DietNO.protein",
                "(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2",
                "Exposure.DietYES.protein - Exposure.DietYES.lipid",
                "Exposure.DietNO.protein - Exposure.DietNO.lipid",
                "(Exposure.DietYES.protein - Exposure.DietNO.protein) - (Exposure.DietYES.lipid - Exposure.DietNO.lipid)",
                "(Exposure.DietYES.protein + Exposure.DietNO.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietYES.lipid)/2" # this used to break it...
                # "Day21 - Day14"
                ))

fit = dream( vobjDream, form, metadata_phenotypes, L)
fit = eBayes(fit)

# get names of available coefficients and contrasts for testing
colnames(fit)
head(fit$design, 3)



# Visualize contrast matrix
plotContrasts(L)
```

```{r save-fit-dream}
dir_path <- here("results/dream")

if (!dir.exists(dir_path)) {
  dir.create(dir_path, recursive = TRUE)
  cat("Directory created:", dir_path, "\n")
} else {
  cat("Directory already exists:", dir_path, "\n")
}

saveRDS(fit, here("results/dream/fit.rds"))

fit <- readRDS(here("results/dream/fit.rds"))
```

```{r identify-blockedDEgenes}
# look at DE genes across contrasts

# difference in response to infection by diet.
interaction_topTags_fit <- variancePartition::topTable( fit, coef="(Exposure.DietYES.protein - Exposure.DietNO.protein) - (Exposure.DietYES.lipid - Exposure.DietNO.lipid)",number = Inf, adjust.method = "BH") %>% rownames_to_column("gene")  %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  distinct(gene, .keep_all = TRUE) %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(adj.P.Val)

# no DE genes for interaction

# effect of infection across all samples.
infVSuninf_both_topTags <- variancePartition::topTable( fit, coef="(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2",number = Inf) %>% rownames_to_column("gene")  %>%
left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  # filter(adj.P.Val < 0.1) %>%
  distinct(gene, .keep_all = TRUE) %>%
  arrange(adj.P.Val)
# overall inf vs uninf DE genes

infVSuninf_both_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of infection across all birds")

# effect of infection on just lipid birds
infected_lipid_topTags <- variancePartition::topTable( fit, coef="Exposure.DietYES.lipid - Exposure.DietNO.lipid", number = Inf) %>% rownames_to_column("gene") %>%
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) %>%
  # filter(adj.P.Val < 0.1) %>%
  distinct(gene, .keep_all = TRUE) %>%
  arrange(adj.P.Val) #%>%

  # View()

infected_lipid_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of infection on lipid diet birds")
# wow awesome, these look crazy.

# effect of infection on just protein diet birds.
infected_protein_topTags <- variancePartition::topTable( fit, coef="Exposure.DietYES.protein - Exposure.DietNO.protein", number = Inf) %>% 
  rownames_to_column("gene")  %>% left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
    dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) %>%
  # filter(adj.P.Val < 0.1) %>%
  distinct(gene, .keep_all = TRUE) %>%
  arrange(adj.P.Val)

infected_protein_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of infection on protein diet birds")


# effect of diet on expression among infected birds
infected_protVSlipid_topTags <- variancePartition::topTable( fit, coef="Exposure.DietYES.protein - Exposure.DietYES.lipid", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  # filter(adj.P.Val < 0.1) %>%
  distinct(gene, .keep_all = TRUE) %>%
  arrange(adj.P.Val)

infected_protVSlipid_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of diet on infected birds (protein-lipid)")

# effect of diet in samples that aren't infected
uninfected_both_topTags <- variancePartition::topTable( fit, coef="Exposure.DietNO.protein - Exposure.DietNO.lipid", number = Inf) %>% rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>% #left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  distinct(gene, .keep_all = TRUE) %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(adj.P.Val)

uninfected_both_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of diet on uninfected birds (protein-lipid)")

# infected_both_topTags <- variancePartition::topTable( fit, coef="Exposure.DietYES.protein - Exposure.DietYES.lipid", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
#   dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
#   # filter(adj.P.Val < 0.1) %>%
#   arrange(logFC)


# effect of diet across all birds
diet_topTags <- variancePartition::topTable( fit, coef="(Exposure.DietYES.protein + Exposure.DietNO.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietYES.lipid)/2", number = Inf) %>% rownames_to_column("gene")  %>%
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  # left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  distinct(gene, .keep_all = TRUE) %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(adj.P.Val)

diet_topTags %>%
  reactable(defaultColDef = colDef(cell = function(value) format(value, digits = 2))) %>%
  add_title("fit: Effect of diet across all birds (protein-lipid)")

# variancePartition::topTable( fit, coef="Day21 - Day 14", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
#   dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description")) %>%
#   View()
```

Save for reproducibility.
```{r}
# save estimates for all genes
write_csv(diet_topTags, here("supplemental_data","AllSamples_ProteinVsLipid_AllGenes.csv"))
write_csv(uninfected_both_topTags, here("supplemental_data","Uninfected_ProteinVsLipid_AllGenes.csv"))
write_csv(infVSuninf_both_topTags, here("supplemental_data","AllBirds_InfectionEffect_AllGenes.csv"))
write_csv(infected_protein_topTags, here("supplemental_data","ProteinBirds_InfectionEffect_AllGenes.csv"))
write_csv(infected_lipid_topTags, here("supplemental_data","LipidBirds_InfectionEffect_AllGenes.csv"))
write_csv(infected_protVSlipid_topTags, here("supplemental_data","Infected_DietEffect_AllGenes.csv"))

# save for just DE genes
AllSamples_ProteinVsLipid_DEGenes <- diet_topTags[diet_topTags$adj.P.Val < 0.05,]
Uninfected_ProteinVsLipid_DEGenes <- uninfected_both_topTags[uninfected_both_topTags$adj.P.Val < 0.05,]
AllBirds_InfectionEffect_DEGenes <- infVSuninf_both_topTags[infVSuninf_both_topTags$adj.P.Val < 0.05,]
ProteinBirds_InfectionEffect_DEGenes <- infected_protein_topTags[infected_protein_topTags$adj.P.Val < 0.05,]
LipidBirds_InfectionEffect_DEGenes <- infected_lipid_topTags[infected_lipid_topTags$adj.P.Val < 0.05,]
Infected_DietEffect_DEGenes <- infected_protVSlipid_topTags[infected_protVSlipid_topTags$adj.P.Val < 0.05,]

# write the DE genes to files as well
write_csv(AllSamples_ProteinVsLipid_DEGenes, here("supplemental_data","AllSamples_ProteinVsLipid_DEGenes.csv"))
write_csv(Uninfected_ProteinVsLipid_DEGenes, here("supplemental_data","Uninfected_ProteinVsLipid_DEGenes.csv"))
write_csv(AllBirds_InfectionEffect_DEGenes, here("supplemental_data","AllBirds_InfectionEffect_DEGenes.csv"))
write_csv(ProteinBirds_InfectionEffect_DEGenes, here("supplemental_data","ProteinBirds_InfectionEffect_DEGenes.csv"))
write_csv(LipidBirds_InfectionEffect_DEGenes, here("supplemental_data","LipidBirds_InfectionEffect_DEGenes.csv"))
write_csv(Infected_DietEffect_DEGenes, here("supplemental_data","Infected_DietEffect_DEGenes.csv"))
```


## KEGG enrichments by contrast

### Infection
```{r}
# Filter genes based on p-value
significant_infection_genes <- #diet_topTags[diet_topTags$P.Value < 0.05,]
  infVSuninf_both_topTags[infVSuninf_both_topTags$adj.P.Val < 0.05,]

# Prepare the list of gene identifiers
gene_list_inf_up <- significant_infection_genes %>%
  # dplyr::filter(logFC > 0.2) %>%
  pull(gene)

gene_list_inf_down <- significant_infection_genes %>%
  # dplyr::filter(logFC < -0.2) %>%
  pull(gene)

dream_inf_kegga <- kegga(fit, coef="(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2", 
          species.KEGG = "scan", FDR=0.05, restrict.universe	= T)

library(clusterProfiler)
dream_inf_kegga_up <- enrichKEGG(gene = gene_list_inf_up,
                 organism = 'scan'#,
                 #universe = infVSuninf_both_topTags$gene,
                 #restrict.universe	= T#,
                 #use_internal_data = F
                 #pvalueCutoff = 0.05
                 )

dream_inf_kegga_down <- enrichKEGG(gene = gene_list_inf_down,
                 organism = 'scan', #readable=TRUE
                 universe = infVSuninf_both_topTags$gene
                 #pvalueCutoff = 0.05
                 )

dream_inf_kegga_up_df <- dream_inf_kegga_up@result
dream_inf_kegga_down_df <- dream_inf_kegga_down@result

```


### Diet
```{r}
# Filter genes based on p-value
significant_diet_genes <- #diet_topTags[diet_topTags$P.Value < 0.05,]
  uninfected_both_topTags[uninfected_both_topTags$P.Value < 0.05,]

# Prepare the list of gene identifiers
gene_list_diet_up <- significant_diet_genes %>%
  dplyr::filter(logFC > 0.4) %>%
  pull(gene)

gene_list_diet_down <- significant_diet_genes %>%
  dplyr::filter(logFC < -0.4) %>%
  pull(gene)

dream_diet_kegga <- kegga(fit, coef="(Exposure.DietYES.protein + Exposure.DietNO.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietYES.lipid)/2", 
          species.KEGG = "scan", FDR=0.9997, restrict.universe	= T)

dream_diet_uninf_kegga <- kegga(fit, coef="Exposure.DietNO.protein - Exposure.DietNO.lipid", 
          species.KEGG = "scan", FDR=0.9549757, restrict.universe	= T)

#dream_diet_kegga_up <- kegga(gene_list_diet_up, species.KEGG = "scan", restrict.universe	= T, universe = diet_topTags$gene)
#dream_diet_kegga_down <- kegga(gene_list_diet_down, species.KEGG = "scan", restrict.universe	= T, universe = diet_topTags$gene)
```

Try KEGG with clusterProfiler so I can view

```{r}
library(clusterProfiler)

dream_diet_kegga_up <- enrichKEGG(gene = gene_list_diet_up,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 )

dream_diet_kegga_down <- enrichKEGG(gene = gene_list_diet_down,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 )

dream_diet_kegga_cP<- enrichKEGG(gene = #significant_diet_genes$gene,
                                   c(gene_list_diet_down,gene_list_diet_up),
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 )

browseKEGG(dream_diet_kegga_down, 'scan04080')

browseKEGG(dream_diet_kegga_up, "scan04672")
browseKEGG(dream_diet_kegga_up, "scan04060")


mkk <- enrichMKEGG(gene = gene_list_diet_up,
                   organism = 'scan',
                   universe = diet_topTags$gene,
                   #pvalueCutoff = 1,
                   #qvalueCutoff = 1
                   )
```


The above analysis looks really nice. note that we see higher expression of Cytokine-cytokine receptor interaction related genes in the protein diet. On the other hand, Neuroactive ligand-receptor interaction related genes are higher in the lipid diet. This is consistent with the idea that the lipid diet may be more neuroactive.

Next, I want to do gseKEGG to see if I can get a better idea of the pathways that are differentially expressed between the diets.

```{r}
library(clusterProfiler)

# create ordered gene list
ordered_list_diet <- significant_diet_genes %>%
  # dplyr::filter(P.Value < 0.1) %>%
  #dplyr::filter(logFC > 0.1) %>%
  pull(logFC, name = gene) %>%
  sort(decreasing = T)



dream_diet_gseKEGG <- gseKEGG(gene = ordered_list_diet,
                 organism = 'scan',
                 #universe = diet_topTags$gene
                 pvalueCutoff = 1,
                 minGSSize = 10,
                 maxGSSize = 50,
                 #scoreType = 'pos'
                 #by
                 )

enrichplot::gseaplot2(dream_diet_gseKEGG, geneSetID = 6, title = dream_diet_gseKEGG$Description[6])
```


## plot individual genes
```{r}
# plot individual genes
y$counts %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = "-", too_many = "merge") %>%
  # filter(gene == "103826733") %>%
  # filter(gene == "103823426") %>%
  filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  #filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,SampID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  mutate(diff_21 = (day21) - (day0),
         diff_14 = (day14) - (day0)) %>%
  pivot_longer(c("diff_14", diff_21), names_to = "diff_day", values_to = "counts_diff") %>% #View()
  #filter(day==0) %>%
  ggplot(aes(x = diet, y = counts_diff, color = condition, group = "BirdID")) +
    geom_point(position = "jitter") +
    # geom_line() +
    facet_wrap(~diff_day) +
    theme_bw()
```


## Phenotype model (MM)
we don't use this I think.
```{r fit-dreamPheno}
form_pheno <- ~ ES + (1|Bird.ID) # this one worked well, trying to add path load scores as well.
                                 # can't do path load scores easily b/c they don't correspond to same days.
form_pheno <- ~ ES + Diet + (1|Bird.ID)

vobjDream_pheno = voomWithDreamWeights( y, form_pheno, metadata_phenotypes, BPPARAM=param)

L_pheno = makeContrastsDream( form, metadata, 
  contrasts = c("Exposure.DietYES.lipid - Exposure.DietNO.lipid", 
                "Exposure.DietYES.protein - Exposure.DietNO.protein",
                "(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2",
                "Exposure.DietYES.protein - Exposure.DietYES.lipid",
                "Exposure.DietNO.protein - Exposure.DietNO.lipid"#,
                # "Day21 - Day14"
                ))

fit_pheno = dream( vobjDream_pheno, form_pheno, metadata_phenotypes)#, L_pheno)
fit_pheno = eBayes(fit_pheno)

# get names of available coefficients and contrasts for testing
colnames(fit_pheno)
head(fit_pheno$design, 3)

pheno_topTable_ES <- variancePartition::topTable( fit_pheno, coef="ES", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(logFC)

pheno_topTable_Diet <- variancePartition::topTable( fit_pheno, coef="Dietprotein", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(logFC)
```

## pheno model with ES normalized to 0,1
```{r fit-dreamEyescoreMinmax}
form_pheno_MM <- ~ ES_minMax + Diet + (1|Bird.ID)

vobjDream_pheno_MM = voomWithDreamWeights( y, form_pheno_MM, metadata_phenotypes, BPPARAM=param )

L_pheno_MM = makeContrastsDream( form, metadata_phenotypes, 
  contrasts = c("Exposure.DietYES.lipid - Exposure.DietNO.lipid", 
                "Exposure.DietYES.protein - Exposure.DietNO.protein",
                "(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2",
                "Exposure.DietYES.protein - Exposure.DietYES.lipid",
                "Exposure.DietNO.protein - Exposure.DietNO.lipid"#,
                # "Day21 - Day14"
                ))

fit_pheno_MM = dream( vobjDream_pheno_MM, form_pheno_MM, metadata_phenotypes)#, L_pheno_MM)
fit_pheno_MM = eBayes(fit_pheno_MM)


pheno_topTable_ES_MM <- variancePartition::topTable( fit_pheno_MM, coef="ES_minMax", number = Inf) %>% rownames_to_column("gene")  %>% left_join(gene_info, by=c(gene = 'entrezgene_id')) %>%
  dplyr::select(c("gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"))  %>%
  # filter(adj.P.Val < 0.1) %>%
  arrange(logFC)

```

Create a dream model with tolerance scores
```{r}
tolerance_dream_model <- ~ 0 + diet.tolerance + Day + (1|Bird.ID)
  # ES_minMax + Diet 

colnames(y_tolerance) <- str_replace_all(colnames(y_tolerance), "\\.", "-")

vobjDream_tolerance = voomWithDreamWeights( y_tolerance, tolerance_dream_model, metadata_phenotypes_day0)#, BPPARAM=param )

dream_contrasts_tolerance = makeContrastsDream( tolerance_dream_model, metadata_phenotypes_day0, 
  contrasts = c("(diet.toleranceprotein.3 - diet.toleranceprotein.2) - (diet.tolerancelipid.3 - diet.tolerancelipid.2)",
                # "Diet_1protein",
                "(diet.toleranceprotein.3 + diet.toleranceprotein.2 + diet.toleranceprotein.1)-(diet.tolerancelipid.3 + diet.tolerancelipid.2 + diet.tolerancelipid.1)",
                "(diet.toleranceprotein.3 + diet.tolerancelipid.3)-(diet.toleranceprotein.2 + diet.tolerancelipid.2)"
                #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
                #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
                ))

fit_tolerance_dream = dream( vobjDream_tolerance, tolerance_dream_model, metadata_phenotypes_day0, dream_contrasts_tolerance)
fit_tolerance_dream = eBayes(fit_tolerance_dream)


# tolerance_topTable_dream <- 
  variancePartition::topTable( fit_tolerance_dream, 
                                                         coef=
                                                           #"(diet.toleranceprotein.3 - diet.toleranceprotein.1) - (diet.tolerancelipid.3 - diet.tolerancelipid.1)",
                                                           #"(diet.toleranceprotein.3 + diet.tolerancelipid.3) - (diet.toleranceprotein.1 + diet.tolerancelipid.1)",
                                                           "(diet.toleranceprotein.3 + diet.toleranceprotein.2 + diet.toleranceprotein.1) - (diet.tolerancelipid.3 + diet.tolerancelipid.2 + diet.tolerancelipid.1)",
                                                           number = Inf) %>% 
  rownames_to_column("gene")  %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      #"gene", "external_gene_name", "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B", "z.std", "entrezgene_description"
      "gene",
      "external_gene_name",
      "logFC",
      "AveExpr",
      "t",
      "P.Value",
      "adj.P.Val",
      "B",
      "entrezgene_description"
    )
  ) %>%
  distinct() 
```


## Tolerance scores

Tolerance: get the regression. Get the individual level tolerance. Correlate the uninfected day0 fold changes with index. How do conditions of host on day 0 impact later tolerance after infection by diet. What Erin wants to know is: LogFC is predicting the future response. Tolerance. Question is how does expression prior to infection predict tolerance after infection. We could do day 7.      tl;dr: letâ€™s see how expression on day 0 correlates with (ES/log10load) onTl;dr7. or day 14, or merge them all.

Since we aren't included random effects here, we analyze using `edgeR` pipeline instead

```{r determine-tolerance}
# generate tolerance score(s)
metadata_phenotypes_day0 <- metadata_phenotypes |>
  # filter(Day==0) |>
  # add 1 to avoid undefined
  mutate(tolerance_1 = ES_AUC_7/(1+logLoad_AUC_14), 
         tolerance_2 = ES_AUC_7/(1+logLoad_day14),
         tolerance_3 = ES_day7/(1+logLoad_day7),
         tolerance_4 = r_ES_day3/(1+r_logLoad_day3), # maybe best?
         tolerance_5 = (1+r_logLoad_day3)/(1+r_ES_day3)-1 # this format works better for "tolerance"
         #tolerance_6 = (1+logLoad_day14)/(1+r_ES_AUC_14)-1,
         # tolerance_0 = ES/logLoad
         ) %>%
  filter(MG.Status != "Control") %>%
  # filter(!Sample.ID %in% c("ES12", "ES25", "ES36", "ES58", "ES84", "ES53")) %>%
  mutate(tolerance_5_cat = dplyr::ntile(tolerance_5,3),
         tolerance_cat = dplyr::ntile(tolerance,3)) %>%
  mutate(diet.tolerance = paste0(Diet, ".",tolerance_cat)) %>%
  # mutate(SampID = str_replace_all(SampID,"\\.","-")) %>%
  column_to_rownames("SampID")
  
  
  
  

# get count matrix of just day 0 birdies
gene_expression_matrix_day0 <- gene_expression_matrix |>
  data.frame() |>
  # dplyr::select(ends_with("_0")) |>
  dplyr::select(contains("MG")) |>
  # these are birds who PCR shows no pathogen
  # dplyr::select(-contains("ES12")) |>
  # dplyr::select(-contains("ES25")) |>
  # dplyr::select(-contains("ES36")) |>
  # dplyr::select(-contains("ES58")) |>
  # dplyr::select(-contains("ES84")) |>
  # dplyr::select(-contains("ES53")) |>
  as.matrix()

# design matrix
tolerance_1 <- metadata_phenotypes_day0$tolerance_5 # careful need to update this...
tolerance_1 <- metadata_phenotypes_day0$tolerance # careful need to update this...
#Diet_1 <- 1+as.numeric(as.factor(metadata_phenotypes_day0$Diet)) #%>% gsub(1,4, .) %>% gsub(2,1, .)
Diet_1 <- as.factor(metadata_phenotypes_day0$Diet)
Day_tolerance <- (metadata_phenotypes_day0$Day)

grouping_var <- interaction(Diet_1, metadata_phenotypes_day0$MG.Status) %>% as.factor()

# this one is not appropriate to use
#design <- model.matrix(~ Diet_1:tolerance_1)
#design

# try this one 12/18/23
# is appropriate to use, doesn't give significant FDR
design <- model.matrix(~ Diet_1+tolerance_1+Diet_1:tolerance_1)#+ Day_tolerance)

# above design works well, but hard to make contrasts. use pseudogroupings instead
# generate groupings
diet_tolerance <- interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) #%>% as.factor()
design <- model.matrix(~ 0 + diet_tolerance + Day_tolerance)

entrez_IDs <- getBM(attributes = c(#'ensembl_transcript_id',
                            #'ensembl_peptide_id',
                            # 'ensembl_gene_id', 
                            # 'uniprot_gn_symbol',
                            'wikigene_name',
                            'entrezgene_id',
                            'external_gene_name',
                            'go_id'
                            ),
             values = rownames(gene_expression_matrix_day0),
             mart = mart) |>
  mutate(entrezgene_id = as.character(entrezgene_id))

entrez_names_gene_expression_matrix_day0 <-
  gene_expression_matrix_day0 |>
  data.frame() |>
  rownames_to_column() |> 
  left_join(entrez_IDs |> dplyr::select(-go_id),
            by = join_by("rowname" == "external_gene_name"), multiple = "first") |>
  mutate(entrezID = case_when(
    str_detect(rowname, "^LOC") ~ str_replace(rowname, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ rowname
  )) |> #View()
  pull(entrezID)

rownames(gene_expression_matrix_day0) <- entrez_names_gene_expression_matrix_day0

y_tolerance <- DGEList(
  counts = gene_expression_matrix_day0,
  # names= entrez_names_gene_expression_matrix_day0,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff <- as.numeric(round(cpm(10, mean(y_tolerance$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep <- rowSums(cpm(y_tolerance) > cpm_cutoff) >= 4
y_tolerance <- y_tolerance[keep,]
summary(keep)

y_tolerance <- calcNormFactors(y_tolerance)
y_tolerance <- estimateDisp(y_tolerance, design, robust=TRUE)

y_tolerance$samples$group <- #grouping_var
  interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) %>% as.factor()

y_tolerance$samples$group <- #grouping_var
  interaction(Diet_1, metadata_phenotypes_day0$tolerance_cat) %>% as.factor()

contrast_ixn_tolerance <- makeContrasts(#design, #metadata, 
  contrasts = c("(diet_toleranceprotein.3 - diet_toleranceprotein.1) - (diet_tolerancelipid.3 - diet_tolerancelipid.1)",
                # "Diet_1protein",
                "(diet_toleranceprotein.3 + diet_toleranceprotein.2 + diet_toleranceprotein.1)-(diet_tolerancelipid.3 + diet_tolerancelipid.2 + diet_tolerancelipid.1)",
                "(diet_toleranceprotein.3 + diet_tolerancelipid.3)-(diet_toleranceprotein.1 + diet_tolerancelipid.1)"
                #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
                #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
                ), levels=design)

fit <- glmQLFit(y_tolerance, design, robust=TRUE)



# res1 <-  glmQLFTest(fit, coef = "Diet_1protein:tolerance_1")
#res2 <-  glmQLFTest(fit, coef = "Diet_1lipid:tolerance_1")
# 
# topTags_prot_tolerance_ixn <- topTags(res1, n=Inf) %>% data.frame() %>% 
#   arrange(FDR) %>%
#   mutate(logFC=round(logFC,2)) %>%
#   # mutate(across(where(is.numeric), signif, 3)) %>%
#   mutate_if(is.numeric, signif, 3) %>%
#   rownames_to_column("gene") %>% 
#   left_join(gene_info |> 
#               mutate(gene = str_replace(gene, "^LOC", "")),
#             by = "gene") %>%
#   dplyr::select(
#     c(
#       "gene",
#       "external_gene_name",
#       "logFC",
#       "logCPM",
#       "F",
#       "PValue",
#       "FDR",
#       "entrezgene_description"
#     )
#   ) %>%
#   distinct()# %>% View()
# 
# View(topTags_prot_tolerance_ixn)
# 
# topTags_lipid_tolerance_ixn <- topTags(res2, n=Inf) %>% data.frame() %>% 
#   arrange(FDR) %>%
#   mutate(logFC=round(logFC,2)) %>%
#   # mutate(across(where(is.numeric), signif, 3)) %>%
#   mutate_if(is.numeric, signif, 3)  %>%
#   rownames_to_column("gene") %>% 
#   left_join(gene_info |> 
#               mutate(gene = str_replace(gene, "^LOC", "")),
#             by = "gene") %>%
#   dplyr::select(
#     c(
#       "gene",
#       "external_gene_name",
#       "logFC",
#       "logCPM",
#       "F",
#       "PValue",
#       "FDR",
#       "entrezgene_description"
#     )
#   ) %>%
#   distinct()
# 
# # Summary of the first test
# decideTests(res1, threshold = 0.05) %>% summary()

# Summary of the second test
#decideTests(res2, threshold = 0.05) %>% summary()

```

# venn diagram of genes DE in diet vs infection

Note that we don't have any DE genes in diet, so let's look at genes whose avg exp greater than 1 and logFC > 0.5

```{r}
# up in diet
uninfected_both_topTags %>%
  filter(logFC > 0.3) %>%
  filter(AveExpr > 0) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()
  #intersect(uninfected_prot_topTags %>% pull(gene)) %>%
  #intersect(uninfected_lipid_topTags %>% pull(gene)) %>%
  #length()
uninfected_both_topTags %>%
  filter(logFC < -0.3) %>%
  filter(AveExpr > 0) %>%
  pull(gene) %>%clipr::write_clip()

#inf up
infVSuninf_both_topTags  %>%
  filter(adj.P.Val < 0.05) %>%
  filter(logFC > 0) %>%
  filter(AveExpr > 0) %>%
  pull(gene) %>% clipr::write_clip()

infVSuninf_both_topTags  %>%
  filter(adj.P.Val < 0.05) %>%
  filter(logFC < -0) %>%
  filter(AveExpr > 0) %>%
  pull(gene) %>% clipr::write_clip()

# get top 1k genes by abs(lfc) for each
# up in protein
uninfected_both_topTags %>%
  arrange(desc(abs(logFC))) %>%
  head(1000) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()

uninfected_both_topTags %>%
  arrange(desc(abs(logFC))) %>%
  head(1000) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()

### diet (not just uninfected birds)
# get top 1k genes by abs(lfc) for each
# up in protein
diet_topTags %>%
  arrange(desc(abs(logFC))) %>%
  head(1000) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()






#inf up
infVSuninf_both_topTags  %>%
  arrange(desc(abs(logFC))) %>%
  head(1000) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()

infVSuninf_both_topTags  %>%
  arrange(desc(abs(logFC))) %>%
  head(1000) %>%
  pull(gene) %>% #cat() %>%
  clipr::write_clip()


# KEGG enrichment of genes that are higher in protein birds & repressed when infected
KEGG_higher_in_lipid_and_infection <- intersect(
  infVSuninf_both_topTags  %>%
  arrange(logFC) %>%
  head(2000) %>%# View()
  pull(gene),
  uninfected_both_topTags %>%
  arrange(desc((logFC))) %>%
  head(2000) %>%
  pull(gene) 
) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 ) 
```

```{r}
# see the KEGG enrichment of the opposite set
KEGG_higher_in_protein_and_uninfected <- intersect(
  infVSuninf_both_topTags  %>%
  arrange(desc(logFC)) %>%
  head(1000) %>%# View()
  pull(gene),
  uninfected_both_topTags %>%
  arrange(((logFC))) %>%
  head(1000) %>%
  pull(gene) 
) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 ) 
```

```{r}
# how much do the genes most different by diet in uninfected birds compare to those in all birds?
# how much do they overlap?
intersect(
  diet_topTags %>%
    arrange(desc(abs(logFC))) %>%
    head(500) %>%
    pull(gene),
  uninfected_both_topTags %>%
  arrange(desc(abs(logFC))) %>%
  head(500) %>%
  pull(gene)
  ) %>%
  length()

# are those genes enriched for anything?
# specifically, genes that are higher in protein birds than lipid birds
KEGG_higher_in_protein_diet_inf_and_uninf <- intersect(
  diet_topTags %>%
    arrange(desc((logFC))) %>%
    # arrange((abs(P.Value))) %>%
    head(200) %>%
    pull(gene),
  uninfected_both_topTags %>%
  arrange(desc((logFC))) %>%
  # arrange((abs(P.Value))) %>%
  head(200) %>%
  pull(gene)
  ) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 #pvalueCutoff = 0.05
                 )

KEGG_higher_in_protein_diet_inf_and_uninf@result %>% View()
```





# Correlate the logFC estimates of genes who are DE in protein vs lipid infection

Generate a scatter plot with x axis being lipid bird's and y axis being protein bird's. include all genes DE in either. color corresponding to DE in one, other, or both.
```{r}
DE_during_infection_comparing_protein_lipid_topTags <-
  infected_lipid_topTags%>%
  # filter(adj.P.Val < 0.1) %>%
  mutate(diet = 'lipid') %>%
  left_join(
    infected_protein_topTags %>%
      # filter(adj.P.Val < 0.1), #%>%
      mutate(diet = 'protein'),
    by = c('gene', 'external_gene_name', 'entrezgene_description'),
    suffix = c('.lipid',".protein")
  ) %>%
  dplyr::select(-c("AveExpr.lipid", "t.lipid", "B.lipid", "z.std.lipid",
                   "AveExpr.protein", "t.protein", "B.protein", "z.std.protein")) %>% 
  filter(adj.P.Val.lipid < 0.1 | adj.P.Val.protein < 0.1) %>%
  mutate(DE_diet = case_when(
    adj.P.Val.lipid < 0.1 & adj.P.Val.protein < 0.1 ~ 'both',
    adj.P.Val.lipid < 0.1 ~ 'lipid',
    adj.P.Val.protein < 0.1 ~ 'protein'
  )) %>%
  mutate(label = case_when(
    is.na(external_gene_name) ~ paste0("LOC",gene),
    external_gene_name == "" ~ paste0("LOC",gene),
    TRUE ~ external_gene_name
  )) %>%
  mutate(direction =case_when(
    logFC.protein/logFC.lipid > 0 ~ "Same",
    logFC.protein/logFC.lipid < 0 ~ "Opposite",
    TRUE ~ "NA"
  ))

DE_during_infection_comparing_protein_lipid_topTags %>%
  ggplot(., aes(x = logFC.lipid, y = logFC.protein, color = DE_diet)) +
  geom_abline(intercept = 0, slope = 1, color = 'grey', linetype = 'dashed') +
  geom_smooth(method="lm") +
  geom_point() +
  geom_label(aes(label = label), size = 2, label.padding = unit(0.1, "lines")
             ) +
  theme_minimal()
```

What enrichments are there for the genes that are DE in both?
```{r}
KEGG_DE_during_infection_both_protein_lipid_topTags <- DE_during_infection_comparing_protein_lipid_topTags %>%
  dplyr::filter(DE_diet == 'both') %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 )

KEGG_DE_during_infection_both_protein_lipid_topTags@result %>% View()
```

What enrichments for genes only DE infection in protein?
```{r}
KEGG_DE_during_infection_protein_topTags <- DE_during_infection_comparing_protein_lipid_topTags %>%
  dplyr::filter(DE_diet == 'protein') %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 )

KEGG_DE_during_infection_protein_topTags@result %>% View()
```

What enrichments for genes only DE infection in lipid?
```{r}
KEGG_DE_during_infection_lipid_topTags <- DE_during_infection_comparing_protein_lipid_topTags %>%
  dplyr::filter(DE_diet == 'lipid') %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
                 organism = 'scan',
                 universe = diet_topTags$gene
                 )

KEGG_DE_during_infection_lipid_topTags@result %>% View()
```


```{r}
#makeContrasts()
res <- glmQLFTest(fit, contrast = contrast_ixn_tolerance)
decideTests(res, threshold = 0.05) %>% summary()

res_tolerance3v1 <-  glmQLFTest(fit, contrast = contrast_ixn_tolerance[,3])
res_diet_prot_v_lipid <-  glmQLFTest(fit, contrast = contrast_ixn_tolerance[,2])
res_diet_tol_ixn <-  glmQLFTest(fit, contrast = contrast_ixn_tolerance[,1])

topTags(res, n=Inf) %>% data.frame() %>% 
  arrange(FDR) %>%
  # mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  mutate_if(is.numeric, signif, 3) %>%
  # remove_rownames() |> 
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  # dplyr::select(
  #   c(
  #     "gene",
  #     "external_gene_name",
  #     "logFC",
  #     "logCPM",
  #     "F",
  #     "PValue",
  #     "FDR",
  #     "entrezgene_description"
  #   )
  # ) %>%
  distinct() %>%
  View()

topTags(res_tolerance3v1, n=Inf) %>% data.frame() %>% 
  arrange(FDR) %>%
  mutate(logFC=round(logFC,2)) %>%
  mutate(across(where(is.numeric), signif, 3)) %>%
  mutate_if(is.numeric, signif, 3) %>%
  # remove_rownames() |> 
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct() %>%
  View()


topTags(res_diet_tol_ixn, n=Inf) %>% data.frame() %>% 
  arrange(FDR) %>%
  mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  mutate_if(is.numeric, signif, 3) %>%
  # remove_rownames() |> 
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct() %>%
  View()

top_tolerance_genes <-
  topTags(res_diet_tol_ixn, n = Inf) %>% 
  data.frame() %>%
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct()



# res_diet <- glmQLFTest(fit, coef = "Diet_1protein")

topTags(res_diet_prot_v_lipid, n=Inf) %>% data.frame() %>% 
  arrange(FDR) %>%
  mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  mutate_if(is.numeric, signif, 3) %>%
  # remove_rownames() |> 
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct() %>%
  View()
```


## Diet interaction in tolerance model
```{r}
res_diet_ixn <- glmQLFTest(fit, coef = "Diet_1protein:tolerance_1")

top_tolerance_ixn_genes <-
  topTags(res_diet_ixn, n = Inf) %>% 
  data.frame() %>%
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct()

top_tolerance_ixn_genes %>% View()


res_diet_genes <- glmQLFTest(fit, coef = "Diet_1protein")

top_diet_genes <-
  topTags(res_diet_genes, n = Inf) %>% 
  data.frame() %>%
  rownames_to_column("gene") %>% 
  left_join(gene_info |> 
              mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "external_gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "entrezgene_description"
    )
  ) %>%
  distinct()
```






## kegga on tolerance genes
```{r}
# clusterProfiler::search_kegg_organism("canary", by="common_name")
ixn_kegga <- kegga(res_diet_tol_ixn, #coef="logFC..diet_toleranceprotein.3...diet_toleranceprotein.1.....diet_tolerancelipid.3...diet_tolerancelipid.1.", 
          species.KEGG = "scan",FDR = 0.05,restrict.universe	= T )

tolerance_kegga <- kegga(res_tolerance3v1, 
          species.KEGG = "scan", FDR = 0.05, restrict.universe	= T )

diet_kegga <- kegga(res_diet_prot_v_lipid, #coef="diet_tolerancelipid.3 - diet_tolerancelipid.1", 
          species.KEGG = "scan", FDR = 0.05,restrict.universe	= T )

topKEGG(ixn_kegga, p.value=0.05)
topKEGG(tolerance_kegga,p.value = 0.05)
topKEGG(diet_kegga,p.value = 0.05)

# topKEGG(k,sort = "Down")
```

## clusterProfiler KEGG analysis

```{r}
library(pathview)
library(clusterProfiler)
# save the logFC values from the res object to a new variable
gene_data_logFC <- res$table %>% dplyr::select(logFC)

# put the ORF names as names for each entry
fold_change_geneList <- setNames(object = data.frame(res)$logFC,
                                 nm = rownames(res))

# create fc genelist
fold_change_geneList_ixn <- setNames(object = data.frame(res_diet_ixn)$logFC,
                                 nm = rownames(res_diet_ixn))

# we can see what this look like with head()
head(fold_change_geneList)

DE_tol_genes <-  res %>% 
  data.frame() %>%
  filter(PValue < 0.05 & abs(logFC)>0.5
         ) %>%
  rownames_to_column() |>
  pull(rowname)

kegg_results <- enrichKEGG(gene = DE_tol_genes, #entrez_ids$ENTREZID, 
                           organism = 'scan' #options: https://www.genome.jp/kegg/catalog/org_list.html
                           )

head(kegg_results)

kegg_results@result$Description <- kegg_results@result$Description %>% print() %>% str_replace_all(., fixed(" - Serinus canaria"), "")

```
### visualize those results
```{r}
enrichplot::upsetplot(kegg_results)
heatplot(kegg_results, showCategory=10, foldChange = fold_change_geneList) +
  # the below code hides the messy overlapping gene names.
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

cnetplot(kegg_results,  
         showCategory=6,
         # circular=TRUE, # this changes the output graphing
         node_label="category",
         color.params=list(foldChange = fold_change_geneList),
         cex.params=list(gene_label = 0.5,
                         category_label=0.5), 
         max.overlaps=100) +
  # change the color scale
  scale_colour_gradientn(colours = rev(RColorBrewer::brewer.pal(n = 10, name = "RdYlBu")), limits=c(-6, 6)) +
  labs(colour="logFC")
```
### GSEA
```{r}
# For gseKEGG, we need a gene list with na removed and sorted. let's do that now.
# omit any NA values 
kegg_gene_list = na.omit(fold_change_geneList_ixn)

# sort the list in decreasing order by FOLD CHANGE (required for this analysis)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

gse_kegg <- gseKEGG(geneList = kegg_gene_list,
               organism      = 'scan',
               # minGSSize   = 120,
               pvalueCutoff  = 0.05,
               verbose       = FALSE)

head(gse_kegg)

# shorten the gse_kegg descriptions that are too long
gse_kegg@result$Description <- gse_kegg@result$Description %>% 
  str_replace_all(., fixed(" - Serinus canaria"), "")

```

```{r}
enrichplot::upsetplot(gse_kegg)
```
```{r gse-multipleGseaplot2, fig.height=6, fig.width=12, dpi=300}
enrichplot::gseaplot2(gse_kegg, geneSetID = 1:4, # trying to get variety of outputs.
                      pvalue_table = TRUE,
                      color = c("#E495A5", "#86B875", "#7DB0DD", "purple"), 
                      ES_geom = "dot"
                      )
```



# Compare logFC patterns across all these contrasts
```{r}
mergedl2fc <- top_tolerance_genes |>
  left_join(top_diet_genes, by = c("gene", "external_gene_name", "entrezgene_description"), suffix=c("_tolerance_score_infection","_tolerance_score_diet")) |>
  left_join(top_tolerance_ixn_genes, by = c("gene", "external_gene_name", "entrezgene_description")) |>
  left_join(dream_MG_genes|> mutate(gene = str_replace(gene, "^LOC", "")), by = c("gene", "external_gene_name", "entrezgene_description"), suffix=c("_tolerance_score_interaction","_dream_MG")) |>
  left_join(dream_Diet_genes|> mutate(gene = str_replace(gene, "^LOC", ""), logFC_dream_Diet = logFC) |> select(-logFC), by = c("gene", "external_gene_name", "entrezgene_description"), suffix=c("_dream_MG","_dream_Diet"))

col <- colorRampPalette(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA"))

mergedl2fc |>
  select(starts_with("logFC")) |>
  cor(use = "pairwise.complete.obs", method = "spearman") |> #View()
corrplot::corrplot(method="color", col=col(200),  
         type="upper", 
         # order="hclust", 
         addCoef.col = "black", # Add coefficient of correlation
         tl.col="black", tl.srt=45, #Text label color and rotation
         # Combine with significance
         # p.mat = p.mat, sig.level = 0.01, insig = "blank", 
         # hide correlation coefficient on the principal diagonal
         diag=FALSE 
         )

mergedl2fc |>
  select(c(starts_with("logFC"),"gene","external_gene_name", "entrezgene_description")) |>
  # filter(abs(logFC_tolerance_score_infection)>1) |># pull(gene) |> clipr::write_clip()
  ggplot(aes(x=logFC_dream_Diet, y = logFC_tolerance_score_diet, label=external_gene_name)) +
  geom_point()
```


# use the contrasts so we can get just the diet ones
```{r}

# form <- ~ 0 + Exposure.Diet + (1|Bird.ID) #+ Day
# vobjDream = voomWithDreamWeights( y, form, metadata_phenotypes, BPPARAM=param )
# L = makeContrastsDream( form, metadata, 
#   contrasts = c("Exposure.DietYES.lipid - Exposure.DietNO.lipid", 
#                 "Exposure.DietYES.protein - Exposure.DietNO.protein",
#                 "(Exposure.DietYES.lipid + Exposure.DietYES.protein)/2 - (Exposure.DietNO.lipid + Exposure.DietNO.protein)/2",
#                 "Exposure.DietYES.protein - Exposure.DietYES.lipid",
#                 "Exposure.DietNO.protein - Exposure.DietNO.lipid"#,
#                 # "Day21 - Day14"
#                 ))


# fit = dream( vobjDream, form, metadata_phenotypes, L)
# fit = eBayes(fit)
# # get names of available coefficients and contrasts for testing
# colnames(fit)
# head(fit$design, 3)
# # Visualize contrast matrix
# plotContrasts(L)


```




# Volcano Plots






### needs to be checked
```{r plot-DE-cleanerCode}
# Function to clean and mutate the data
clean_and_mutate_data <- function(data) {
  data %>%
    mutate(delabel = case_when(
      stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
      TRUE ~ gene
    )) %>%
    mutate(delabel = case_when(
      adj.P.Val < 0.05 ~ delabel,
      abs(logFC) > 0.8 ~ delabel,
      TRUE ~ NA
    )) %>%
    mutate(diffexpressed = case_when(
      adj.P.Val > 0.05 ~ "NO",
      logFC < 0 ~ "DOWN",
      logFC > 0 ~ "UP"
    )) %>%
    mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP")))
}

# Function to create the plot
create_gene_plot <- function(data, title) {
  ggplot(data, aes(x = logFC, y = -log10(adj.P.Val), col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point() +
    theme_minimal() +
    scale_color_manual(values = c("blue", "black", "red")) +
    geom_vline(xintercept = c(-0.585, 0.585), col = "red") +
    geom_hline(yintercept = -log10(0.05), col = "red") +
    ggrepel::geom_text_repel(size = 2) +
    scale_size_area() +
    ggtitle(title)
}


## dream infection contrast
infVSuninf_both_topTags %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Genes Differentially Expressed (infected / uninfected contrast)")

# infected lipid birds
infected_lipid_topTags %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Genes Differentially Expressed in infected lipid birds")

# infected protein birds
infected_protein_topTags %>%
  clean_and_mutate_data() %>%
  create_gene_plot("Genes Differentially Expressed in infected protein birds")

## dream diet contrast
uninfected_both_topTags %>%
  clean_and_mutate_data() %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("N.S.", "Higher in Lipid", "Higher in Protein"))) %>%
  create_gene_plot("Genes Differentially Expressed by Diet (in uninfected birds)")

## phenotype (Eyescore)
pheno_topTable_ES %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene
  )) %>%
  mutate(delabel = case_when(
    adj.P.Val > 0.05 ~ NA,
    TRUE ~ delabel
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  create_gene_plot("Genes Differentially Expressed According to Eyescore Phenotype")
```


```{r is-replaced1}
## dream infection contrast
infVSuninf_both_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val > 0.05 ~ NA,
    TRUE ~ delabel
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("black", "blue", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="grey") +
        geom_hline(yintercept=-log10(0.00726), col="grey") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed (infected / uninfected contrast)")
# 
# infVSuninf_both_topTags2 %>%
#   mutate(delabel = case_when(
#     stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
#     TRUE ~ gene)
#     ) %>%
#   mutate(delabel = case_when(
#     adj.P.Val > 0.05 ~ NA,
#     TRUE ~ delabel
#   )) %>%
#   mutate(diffexpressed = case_when(
#     adj.P.Val > 0.05 ~ "NO",
#     logFC < 0 ~ "DOWN",
#     logFC > 0 ~ "UP"
#   )) %>%
#   mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
#   ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
#         geom_point() + 
#         theme_minimal() +
#         scale_color_manual(values=c("black", "blue", "red")) +
#         geom_vline(xintercept=c(-0.585, 0.585), col="grey") +
#         geom_hline(yintercept=-log10(0.00726), col="grey") +
#         ggrepel::geom_text_repel(size=2) +
#   ggtitle("Genes Differentially Expressed (infected / uninfected contrast)")
```

```{r is-replaced2}
# infected lipid birds
infected_lipid_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ delabel,
    abs(logFC) > 0.8 ~ delabel,
    TRUE ~ NA
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  ggplot(aes(x=logFC, y=-log10(adj.P.Val), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="red") +
        geom_hline(yintercept=-log10(0.05), col="red") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed in infected lipid birds")

# infected protein birds
infected_protein_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ delabel,
    abs(logFC) > 0.8 ~ delabel,
    TRUE ~ NA
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  ggplot(aes(x=logFC, y=-log10(adj.P.Val), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="red") +
        geom_hline(yintercept=-log10(0.05), col="red") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed in infected protein birds")


## dream diet contrast
uninfected_both_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ delabel,
    abs(logFC) > 0.8 ~ delabel,
    TRUE ~ NA
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "N.S.",
    logFC < 0 ~ "Higher in Lipid",
    logFC > 0 ~ "Higher in Protein"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("N.S.", 
                                                          "Higher in Lipid", "
                                                          Higher in Protein"))) %>%
  ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="red") +
        geom_hline(yintercept=-log10(0.05), col="red") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed in uninfected birds") +
  annotate("text", label = "Higher in Lipid", x = -1.5, y = 0) +
  annotate("text", label = "Higher in Protein", x = 1, y = 0) 
                             

## phenotype (Eyescore)
pheno_topTable_ES %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val > 0.05 ~ NA,
    TRUE ~ delabel
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="red") +
        geom_hline(yintercept=-log10(0.05), col="red") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed According to Eyescore Phenotype")

# phenotype (Diet factor)
pheno_topTable_Diet %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ NA,
    abs(logFC) < 0.8 ~ NA,
    TRUE ~ delabel
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "N.S.",
    logFC < 0 ~ "Higher in Lipid",
    logFC > 0 ~ "Higher in Protein"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        scale_color_manual(values=c("black", "blue", "red")) +
        geom_vline(xintercept=c(-0.585, 0.585), col="grey") +
        geom_hline(yintercept=-log10(0.05), col="grey") +
        ggrepel::geom_text_repel(size=2) +
  ggtitle("Genes Differentially Expressed According to Diet factor in Phenotype model (not good to use prb)")
```

## generate a boxplot that shows FC values in response to MG on left, another boxplot for FC in response to diet on the right, and lines connecting the genes.
```{r generate-boxplots, warning=FALSE}
# MG
tmp_MG_tidy <- infVSuninf_both_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val > 0.05 ~ NA,
    TRUE ~ delabel
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "NO",
    logFC < 0 ~ "DOWN",
    logFC > 0 ~ "UP"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>%
  mutate(contrast = "MG")
  # ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
  #       geom_point() + 
  #       theme_minimal() +
  #       scale_color_manual(values=c("black", "blue", "red")) +
  #       geom_vline(xintercept=c(-0.585, 0.585), col="grey") +
  #       geom_hline(yintercept=-log10(0.00726), col="grey") +
  #       ggrepel::geom_text_repel(size=2) +
  # ggtitle("Genes Differentially Expressed (infected / uninfected contrast)")

# Diet
## dream diet contrast
tmp_Diet_tidy <- uninfected_both_topTags %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ delabel,
    abs(logFC) > 0.8 ~ delabel,
    TRUE ~ NA
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "N.S.",
    logFC < 0 ~ "Higher in Lipid",
    logFC > 0 ~ "Higher in Protein"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("N.S.", 
                                                          "Higher in Lipid", "
                                                          Higher in Protein"))) %>%
  mutate(contrast = "Diet")
  # ggplot(aes(x=logFC, y=-log10(P.Value), col=diffexpressed, label=delabel)) +
  #       geom_point() + 
  #       theme_minimal() +
  #       scale_color_manual(values=c("blue", "black", "red")) +
  #       geom_vline(xintercept=c(-0.585, 0.585), col="red") +
  #       geom_hline(yintercept=-log10(0.05), col="red") +
  #       ggrepel::geom_text_repel(size=2) +
  # ggtitle("Genes Differentially Expressed in uninfected birds") +
  # annotate("text", label = "Higher in Lipid", x = -1.5, y = 0) +
  # annotate("text", label = "Higher in Protein", x = 1, y = 0) 

# boxplot - paired
tmp_MG_tidy %>%
  bind_rows(tmp_Diet_tidy) %>%
  filter(AveExpr > 1.5) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene
  )) %>%
  mutate(diffexpressed = case_when(
    diffexpressed == "N.S." ~ "NO",
    TRUE ~ diffexpressed) %>% fct_relevel("NO", "UP", "DOWN")) %>%
  mutate(delabel = case_when(abs(logFC) < 0.4 ~ NA,
                             TRUE ~ delabel)) %>%
  ggplot(aes(contrast, logFC, fill = contrast, label = delabel)) +
  geom_line(aes(group = gene), alpha = 0.1, position = position_dodge(0.2), show.legend=FALSE) +
  geom_point(aes(fill = contrast,  group = gene, shape = diffexpressed), size = 0.7, position = position_dodge(0.2), show.legend = c(shape=TRUE, fill=FALSE, group = FALSE)) +
  # geom_line(aes(group = gene), alpha = 0.1) +
  # ggbeeswarm::geom_quasirandom() +
  geom_boxplot(width = 0.25, outlier.shape = NA,show.legend=FALSE) +
  ggrepel::geom_text_repel(size = 2) +
  theme_classic() +
  # theme(legend.position = "none") +
  theme(legend.position="right") +
  annotate("text",
           label = "Higher in Protein",
           x = 0.9,
           y = 1.35) +
  annotate("text",
           label = "Higher in Lipid",
           x = 0.9,
           y = -1.15) +
  annotate("text",
           label = "Higher in Infection",
           x = 2.2,
           y = 1.35) +
  annotate("text",
           label = "Lower in Infection",
           x = 2.2,
           y = -1.2) +
  ggtitle("Same genes top lists of Expression Fold Changes in Diet and MG treatments")

```
# alternative visualization (scatter correlation plot):

```{r}
tmp_MG_tidy %>%
  left_join(tmp_Diet_tidy, by = join_by(gene, external_gene_name, entrezgene_description)) %>%
  filter(AveExpr.x > 1.5) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene
  )) %>%
  mutate(delabel = case_when(abs(logFC.x) < 0.4 & abs(logFC.y) < 0.4 ~ NA,
                             TRUE ~ delabel)) %>%
  ggplot(aes(logFC.x, logFC.y, label = delabel)) +
  # geom_line(aes(group = gene), alpha = 0.1, position = position_dodge(0.2)) +
  geom_point(aes(group = gene),  alpha=0.2) +
  # geom_boxplot(width = 0.25, outlier.shape = NA) +
  # geom_smooth(method="lm") +
  geom_abline(alpha=0.5) +
  ggrepel::geom_text_repel(size = 2) +
  theme_classic() +
  theme(legend.position = "none") +
  annotate("text",
           label = "Higher in Protein",
           x = 1,
           y = 1.35) +
  annotate("text",
           label = "Higher in Lipid",
           x = -0.9,
           y = -0.85) +
  ggtitle("Correlation of Expression Fold Changes in effects of Diet and MG") +
  xlab("logFC (MG infection)") +
  ylab("logFC (Diet)")
```




## correlation plot beetween uninfected_both_toptags and pheno_topTable_Diet(not _uninfected)
```{r}

tmp_pheno_Diet_tidy <- pheno_topTable_Diet %>% # was pheno_topTable_Diet_uninfected 
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    adj.P.Val < 0.05 ~ delabel,
    abs(logFC) > 0.8 ~ delabel,
    TRUE ~ NA
  )) %>%
  mutate(diffexpressed = case_when(
    adj.P.Val > 0.05 ~ "N.S.",
    logFC < 0 ~ "Higher in Lipid",
    logFC > 0 ~ "Higher in Protein"
  )) %>%
  mutate(diffexpressed = factor(diffexpressed, levels = c("N.S.", 
                                                          "Higher in Lipid", "
                                                          Higher in Protein"))) %>%
  mutate(contrast = "Diet_pheno")

# alternative visualization:
tmp_pheno_Diet_tidy %>%
  left_join(tmp_Diet_tidy, by = join_by(gene, external_gene_name, entrezgene_description)) %>%
  filter(AveExpr.x > 1.5) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene
  )) %>%
  mutate(delabel = case_when(abs(logFC.x) < 0.4 & abs(logFC.y) < 0.4 ~ NA,
                             TRUE ~ delabel)) %>%
  ggplot(aes(logFC.x, logFC.y, label = delabel)) +
  # geom_line(aes(group = gene), alpha = 0.1, position = position_dodge(0.2)) +
  geom_point(aes(group = gene), position = position_dodge(0.2), alpha=0.2) +
  # geom_boxplot(width = 0.25, outlier.shape = NA) +
  # geom_smooth(method="lm") +
  geom_abline(alpha=0.5) +
  ggrepel::geom_text_repel(size = 2) +
  theme_classic() +
  theme(legend.position = "none") +
  annotate("text",
           label = "Higher in Protein",
           x = 2,
           y = 1.35) +
  annotate("text",
           label = "Higher in Lipid",
           x = -1,
           y = -1) +
  ggtitle("Correlation of Expression Fold Changes in effects of Diet (contrast vs pheno)") +
  xlab("logFC (Diet Pheno)") +
  ylab("logFC (Diet)")
```



## correlation plot between logFC MG and logFC EyeScores.
```{r}
full_join(pheno_topTable_ES, infVSuninf_both_topTags, by=c("gene", "external_gene_name", "entrezgene_description")) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  dplyr::select(logFC.x, logFC.y) %>%
  corrr::correlate()

full_join(pheno_topTable_ES, infVSuninf_both_topTags, by=c("gene", "external_gene_name", "entrezgene_description")) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    abs(logFC.x) < 0.3 ~ NA,
    TRUE ~ delabel
  )) %>%
  ggplot(aes(x=logFC.x, y = logFC.y, label = delabel)) +
  geom_point(alpha=0.5) +
  ggrepel::geom_text_repel(size=2) +
  geom_smooth(method="lm") +
  # ggpubr::stat_cor(aes(label = ..rr.label..)) +
  ggpubr::stat_regline_equation(
    aes(label =  paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),
    formula = y ~ x
  ) +
  theme_classic() +
  xlim(-1.2,1.5) +
  ylim(-1.2,1.5) +
  labs(
    title = "Correlation of logFC estimates between EyeScore and MG group regression",
    x = "logFC.EyeScore",
    y = "logFC.MG"
  )
```
# comparing eyescores normalized to 0,1 with MG
```{r, warning=FALSE}
full_join(pheno_topTable_ES_MM, infVSuninf_both_topTags, by=c("gene", "external_gene_name", "entrezgene_description")) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  dplyr::select(logFC.x, logFC.y) %>%
  corrr::correlate()

full_join(pheno_topTable_ES_MM, infVSuninf_both_topTags, by=c("gene", "external_gene_name", "entrezgene_description")) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene)
    ) %>%
  mutate(delabel = case_when(
    abs(logFC.x) < 0.3 ~ NA,
    TRUE ~ delabel
  )) %>%
  ggplot(aes(x=logFC.x, y = logFC.y, label = delabel)) +
  geom_point(alpha=0.5) +
  geom_smooth(method="lm") +
  ggrepel::geom_text_repel(size=2) +
  # ggpubr::stat_cor(aes(label = ..rr.label..)) +
  ggpubr::stat_regline_equation(
    aes(label =  paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),
    formula = y ~ x
  ) +
  theme_classic() +
  # xlim(-1.2,1.2) +
  # ylim(-1.2,1.2) +
  labs(
    title = "Correlation of logFC estimates between EyeScore and MG group regression",
    x = "logFC.EyeScore (minMax)",
    y = "logFC.MG"
  )
```





# GO enrichment of MG genes.
```{r prep-GO}
# get biomart ensembl dataset for scanaria (has most GO terms)
mart <- if(exists('mart') == TRUE){
  mart
} else {
  biomaRt::useMart(biomart = "ensembl", dataset = "scanaria_gene_ensembl")
}

# get biomart gene information of all genes in expression matrix
res <- getBM(attributes = c('ensembl_transcript_id',
                            'ensembl_peptide_id',
                            'ensembl_gene_id', 
                            'uniprot_gn_symbol',
                            'wikigene_name',
                            'entrezgene_id',
                            'external_gene_name',
                            'go_id',
                            'description'
                            ),
             values = dream_MG_genes$gene,
             mart = mart)


# getting GO terms grouped to gene names
res_min <- res %>% 
    mutate(wikigene_name = ifelse(wikigene_name %in% "", 
                                  external_gene_name, 
                                  wikigene_name)) %>% # get gene names
    group_by(ensembl_gene_id) %>% # I think I need to change this...
    mutate(GOterms = paste0(go_id, collapse = ",")) %>%
    slice_head(n=1) %>% # get only one row per gene
    ungroup()

# tidy up data formatting
res_min <- res_min %>%
    mutate(GOterms = gsub("^,+|,+$|(,)+", "\\1", GOterms, perl=TRUE)) %>%  # tidy up the extra ;;; in Go terms
    mutate(across(everything(), ~ifelse(.=="", NA, as.character(.))))

  # combine different nomenclature types
geneID2GO_combine_names <- res_min[,c(5,6,10)] %>%
    mutate(gene = ifelse(is.na(res_min$wikigene_name), res_min$entrezgene_id, res_min$wikigene_name)) %>%
    dplyr::select(-c(wikigene_name, entrezgene_id))

geneID2GO_combine_names <- res_min[,c(5,6,10)] %>%
    filter(!is.na(GOterms)) %>%
    mutate(gene = paste0("LOC",entrezgene_id)) #%>%
    # dplyr::select(-c(wikigene_name, entrezgene_id))
  
  # merge the data from BioMart and our input genes
geneID_names <- left_join(dream_MG_genes, geneID2GO_combine_names, by = "gene") %>%
  left_join(geneID2GO_combine_names[,c(1,3)], by = join_by("gene" == "wikigene_name")) %>%
  mutate(GOterms = ifelse(is.na(GOterms.x), GOterms.y, GOterms.x))

colnames(geneID_names)[1] <- "seq_name"


# build the table of original GO terms with gene IDs (weird format needed)
geneID2GO_build_BM <- tibble(geneID_names$seq_name,
                             "\t",
                             geneID_names$GOterms)
# give appropriate column names
colnames(geneID2GO_build_BM) <- c("seq_name", "\t", "go_i_ds")


# load in Blast2GO results
Blast2GO <-
  read_delim(
    "~/Desktop/canary/newGOterms_nomatchDEgenes.tsv",
    delim = "\t",
    escape_double = FALSE,
    trim_ws = TRUE
  ) %>%
  janitor::clean_names() %>%
  dplyr::select(seq_name, go_i_ds) %>%
  #change formatting of the GO term seperators to ,
  dplyr::mutate(go_i_ds = str_replace_all(go_i_ds, "; ", ",")) %>%
  dplyr::mutate(go_i_ds = str_replace_all(go_i_ds, "[PFC]:", ""))

combinedGOterms <-
  left_join(geneID2GO_build_BM, Blast2GO, by = "seq_name")

# now condense the two go_i_ds columns into single column of GO terms.
combinedGOterms <-
  tidyr::unite(
    combinedGOterms,
    GOterms,
    go_i_ds.x:go_i_ds.y,
    na.rm = TRUE,
    remove = FALSE,
    sep = ","
  )

geneID2GO_build_BM <- tibble(# canary_annotations$geneID_ncbi,
  seq_name = combinedGOterms$seq_name,
  "\t",
  GOterms = combinedGOterms$GOterms)


  # write out the geneID2GO_BM.map file
  write.table(geneID2GO_build_BM, file=here("test_geneID2GO_BM.map"),
              quote = F, row.names = F, col.names = F)
  
  # save(geneID2GO_build_BM, 
  #      file = paste( "~/Documents/GitHub/CanarySeq/geneAnnotations/geneID2GO_", names(contrasts)[i], ".RData", sep= ""))
  
  # read in the mappings with readMappings from topGO
  canary_geneID2GO_BM <- topGO::readMappings(file = here("test_geneid2GO_BM.map"))

matches <- which(toupper(names(canary_geneID2GO_BM)) %in% dream_MG_genes$gene)

# used unique because there are some duplicates in output. (per isoform)
genesMatched <- unique(toupper(names(canary_geneID2GO_BM))[matches])
```

```{r enrich-GO}
  # assign desired fold change cutoff value (will be log2 tranformed in script)
  j = 1.3

  ## which gene list we are looking at
  DEgenes_up <- filter(as.data.frame(dream_DE_MG_genes), logFC > log2(j)) %>% pull(gene)
  DEgenes_down <- filter(dream_DE_MG_genes, logFC < -log2(j)) %>% pull(gene)
    
  # genes names with GO terms and DE.
  DEgeneswithGO_up <- which(DEgenes_up %in% genesMatched)
  DEgeneswithGO_down <- which(DEgenes_down %in% genesMatched)
  
  # select only genes that are DE and have GO terms
  DEgenesGO_up <- DEgenes_up[DEgeneswithGO_up]
  DEgenesGO_down <- DEgenes_down[DEgeneswithGO_down]
   
  #first, let's subset our canary geneIDtoGO to just these genes.
  canary_geneID2GO_BM_DE_up <-
    canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% DEgenesGO_up)]
    
  names(canary_geneID2GO_BM_DE_up)[!(DEgenesGO_up %in% names(canary_geneID2GO_BM_DE_up))]
  
  canary_geneID2GO_BM_DE_down <-
    canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% DEgenesGO_down)]
    
  names(canary_geneID2GO_BM_DE_down)[!(DEgenesGO_down %in% names(canary_geneID2GO_BM_DE_down))]
    
    
  #Let me also get my full edgeR gene list.
  canary_geneID2GO_BM_edgeR <- canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% genesMatched)]

    
  # weird formatting creating the geneList for topGO. this code seemed to work
  all_genes_up <- factor(as.integer(toupper(filter(dream_MG_genes,AveExpr>1)$gene) %in% DEgenesGO_up),labels=c(0,1)) # create 0 1 factors.
    names(all_genes_up) <- toupper(filter(dream_MG_genes,AveExpr>1)$gene) # add gene names
    
  all_genes_down <- factor(as.integer(toupper(filter(dream_MG_genes,AveExpr>1)$gene) %in% DEgenesGO_down)) # create 0 1 factors.
  names(all_genes_down) <- toupper(filter(dream_MG_genes,AveExpr>1)$gene) # add gene names
    
    
  GOdata_up=new('topGOdata', 
         # ontology='CC', # 'BP' = Biological Process. Also tried 'ALL' but doesn't work.
         ontology='BP',
         allGenes = all_genes_up, # all 10188 genes that are in MGpost0 data from edgeR. 
                                  # factor=1 for those that are DE.
         annot = annFUN.gene2GO, # annotation function for custom annotation
# I am thinking about switching this for one filtered to just edgeR genes:
         gene2GO = canary_geneID2GO_BM_edgeR # annotation db from BiomaRt.
           )
    
  GOdata_down=new('topGOdata', 
         # ontology='CC', # 'BP' = Biological Process. Also tried 'ALL' but doesn't work.
         ontology='BP',
         allGenes = all_genes_down, # all 10188 genes that are in MGpost0 data from edgeR. 
                                    # factor=1 for those that are DE.
         annot = annFUN.gene2GO, # annotation function for custom annotation
# I am thinking about switching this for one filtered to just edgeR genes:
         gene2GO = canary_geneID2GO_BM_edgeR # annotation db from BiomaRt.

           )
    
  #analysis
  # define test using the weight01 algorithm (default) with fisher
  # weight_fisher_result_up=runTest(GOdata_up, algorithm='weight01', statistic='fisher')
  # weight_fisher_result_down=runTest(GOdata_down, algorithm='weight01', statistic='fisher')
  weight_fisher_result_up=runTest(GOdata_up, algorithm = "elim", statistic = "fisher")
  weight_fisher_result_down=runTest(GOdata_down, algorithm = "elim", statistic = "fisher")
  
     
  # generate a table of results: we can use the GenTable function to generate a summary table with the results from tests applied to the topGOdata object.
  allGO_up=usedGO(GOdata_up)
  all_res_up=GenTable(GOdata_up, 
                      weightFisher=weight_fisher_result_up, 
                      orderBy='weightFisher', 
                      topNodes=length(allGO_up), 
                      numChar = 200)
    
    allGO_down=usedGO(GOdata_down)
    all_res_down=GenTable(GOdata_down, 
                          weightFisher=weight_fisher_result_down, 
                          orderBy='weightFisher', 
                          topNodes=length(allGO_down), 
                          numChar = 200)
    
    #performing BH correction on our p values
    p.adj_up=round(p.adjust(all_res_up$weightFisher,method="BH"),digits = 4)
    p.adj_down=round(p.adjust(all_res_down$weightFisher,method="BH"),digits = 4)
     
    # create the file with all the statistics from GO analysis
    all_res_final_up=cbind(all_res_up,p.adj_up)
    all_res_final_up=all_res_final_up[order(all_res_final_up$p.adj),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";"))
    
    all_res_final_down=cbind(all_res_down,p.adj_down)
    all_res_final_down=all_res_final_down[order(all_res_final_down$p.adj),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";"))
     
    #get list of significant GO before multiple testing correction
    results.table.p_up= all_res_final_up[which(as.numeric(all_res_final_up$weightFisher)<=0.1),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";")) %>%
      type_convert() %>%
      filter(Annotated > 1) %>%
      mutate(Sig_Anno = Significant/Expected)
    
    results.table.p_down= all_res_final_down[which(as.numeric(all_res_final_down$weightFisher)<=0.1),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";")) %>%
      type_convert() %>%
      filter(Annotated > 1) %>%
      mutate(Sig_Exp = Significant/Expected)
    
    #get list of significant GO after multiple testing correction
    results.table.bh_up=all_res_final_up[which(all_res_final_up$p.adj<=0.1),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";"))
    
    results.table.bh_down=all_res_final_down[which(all_res_final_down$p.adj<=0.1),] %>%
      mutate(across(everything(), gsub, pattern = ",", replacement = ";"))
    
# PLOT the GO hierarchy plot: the enriched GO terms are colored in yellow/red according to significance level
    
# pdf(file='~/Desktop/canary/test_topGO_MG_DAG_DOWN.pdf', height=12, width=12, paper='special', pointsize=18)
showSigOfNodes(GOdata_down, score(weight_fisher_result_down), useInfo = "none",
               sigForAll=TRUE, firstSigNodes=2)
# dev.off()
```

```{r}
showSigOfNodes(GOdata_up, score(weight_fisher_result_up), useInfo = "all", sigForAll=TRUE, firstSigNodes=5)
```





# functions for GO enrichment
```{r functional-GO}
get_DE_genes <- function(dream_DE_MG_genes, j) {
  DEgenes_up <- filter(as.data.frame(dream_DE_MG_genes), logFC > log2(j)) %>% pull(gene)
  DEgenes_down <- filter(dream_DE_MG_genes, logFC < -log2(j)) %>% pull(gene)
  
  return(list(up = DEgenes_up, down = DEgenes_down))
}

get_DE_genes_with_GO <- function(DEgenes, genesMatched) {
  DEgenes_with_GO <- which(DEgenes %in% genesMatched)
  DEgenes_GO <- DEgenes[DEgenes_with_GO]
  
  return(DEgenes_GO)
}

# get_canary_geneID2GO_BM_DE <- function(canary_geneID2GO_BM, DEgenes_GO) {
#   canary_geneID2GO_BM_DE <- canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% DEgenes_GO)]
#   names(canary_geneID2GO_BM_DE)[!(DEgenes_GO %in% names(canary_geneID2GO_BM_DE))]
#   
#   return(canary_geneID2GO_BM_DE)
# }

get_canary_geneID2GO_BM_edgeR <- function(canary_geneID2GO_BM, genesMatched) {
  canary_geneID2GO_BM_edgeR <- canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% genesMatched)]
  
  return(canary_geneID2GO_BM_edgeR)
}

create_topGOdata <- function(all_genes, canary_geneID2GO_BM_edgeR, ontology) {
  GOdata <- new('topGOdata',
                ontology = paste0(ontology),
                allGenes = all_genes,
                annot = annFUN.gene2GO,
                gene2GO = canary_geneID2GO_BM_edgeR)
  
  return(GOdata)
}

perform_enrichment_analysis <- function(GOdata) {
  weight_fisher_result <- runTest(GOdata, algorithm = 'elim', statistic = 'fisher')
  allGO <- usedGO(GOdata)
  all_res <- GenTable(GOdata, weightFisher = weight_fisher_result, orderBy = 'weightFisher', topNodes = length(allGO), numChar = 200)
  
  return(all_res)
}

correct_p_values <- function(all_res_final) {
  p.adj <- round(p.adjust(all_res_final$weightFisher, method = 'BH'), digits = 4)
  all_res_final_adj <- cbind(all_res_final, p.adj)
  all_res_final_adj <- all_res_final_adj[order(all_res_final_adj$p.adj),] %>%
    mutate(across(everything(), gsub, pattern = ',', replacement = ';'))
  
  return(all_res_final_adj)
}

get_significant_GO <- function(all_res_final_adj, cutoff) {
  results.table_p <- all_res_final_adj[which(as.numeric(all_res_final_adj$weightFisher) <= cutoff),] %>%
    mutate(across(everything(), gsub, pattern = ',', replacement = ';')) %>%
    type_convert() %>%
    filter(Annotated > 1)
  results.table_p <- results.table_p %>% mutate(Sig_Anno = Significant / Expected)
  
  return(results.table_p)
}

get_significant_GO_bh <- function(all_res_final_adj, cutoff) {
  results.table_bh <- all_res_final_adj[which(all_res_final_adj$p.adj <= cutoff),]
  results.table_bh <- results.table_bh %>% mutate(across(everything(), gsub, pattern = ',', replacement = ';'))
  
  return(results.table_bh)
}
```                         
                                        
```{r}                                        
# Perform enrichment analysis for upregulated & downregulated genes
  MG.DEgenes_up <- get_DE_genes(dream_MG_genes, 1.2)$up
  MG.DEgenes_down <- get_DE_genes(dream_MG_genes, 1.2)$down

  MG.DEgenes_with_GO_up <- get_DE_genes_with_GO(MG.DEgenes_up, genesMatched)
  MG.DEgenes_with_GO_down <- get_DE_genes_with_GO(MG.DEgenes_down, genesMatched)

  # canary_geneID2GO_BM_DE_up <- get_canary_geneID2GO_BM_DE(canary_geneID2GO_BM, MG.DEgenes_with_GO_up)
  # canary_geneID2GO_BM_DE_down <- get_canary_geneID2GO_BM_DE(canary_geneID2GO_BM, DEgenes_with_GO_down)

  canary_geneID2GO_BM_edgeR <- get_canary_geneID2GO_BM_edgeR(canary_geneID2GO_BM, genesMatched)

  MG.all_genes_up <- factor(as.integer(toupper(filter(dream_MG_genes, AveExpr > 1)$gene) %in% MG.DEgenes_with_GO_up),
                         labels = c(0, 1))
  names(MG.all_genes_up) <- toupper(filter(dream_MG_genes, AveExpr > 1)$gene)

  MG.all_genes_down <- factor(as.integer(toupper(filter(dream_MG_genes, AveExpr > 1)$gene) %in% MG.DEgenes_with_GO_down))
  names(MG.all_genes_down) <- toupper(filter(dream_MG_genes, AveExpr > 1)$gene)

  
  #CC
  ccMG.GOdata_up <- create_topGOdata(MG.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "CC")
  ccMG.GOdata_down <- create_topGOdata(MG.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "CC")

  ccMG.all_res_up <- perform_enrichment_analysis(ccMG.GOdata_up)
  ccMG.all_res_down <- perform_enrichment_analysis(ccMG.GOdata_down)

  ccMG.all_res_final_up <- correct_p_values(ccMG.all_res_up)
  ccMG.all_res_final_down <- correct_p_values(ccMG.all_res_down)

  ccMG.results.table_p_up <- get_significant_GO(ccMG.all_res_final_up,0.1)
  ccMG.results.table_p_down <- get_significant_GO(ccMG.all_res_final_down,0.1)

  ccMG.results.table_bh_up <- get_significant_GO_bh(ccMG.all_res_final_up,0.1)
  ccMG.results.table_bh_down <- get_significant_GO_bh(ccMG.all_res_final_down,0.1)
  
  #BP
  bpMG.GOdata_up <- create_topGOdata(MG.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "BP")
  bpMG.GOdata_down <- create_topGOdata(MG.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "BP")

  bpMG.all_res_up <- perform_enrichment_analysis(bpMG.GOdata_up)
  bpMG.all_res_down <- perform_enrichment_analysis(bpMG.GOdata_down)

  bpMG.all_res_final_up <- correct_p_values(bpMG.all_res_up)
  bpMG.all_res_final_down <- correct_p_values(bpMG.all_res_down)

  bpMG.results.table_p_up <- get_significant_GO(bpMG.all_res_final_up,0.1)
  bpMG.results.table_p_down <- get_significant_GO(bpMG.all_res_final_down,0.1)

  bpMG.results.table_bh_up <- get_significant_GO_bh(bpMG.all_res_final_up,0.1)
  bpMG.results.table_bh_down <- get_significant_GO_bh(bpMG.all_res_final_down,0.1)
  
  #MF
  mfMG.GOdata_up <- create_topGOdata(MG.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "MF")
  mfMG.GOdata_down <- create_topGOdata(MG.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "MF")

  mfMG.all_res_up <- perform_enrichment_analysis(mfMG.GOdata_up)
  mfMG.all_res_down <- perform_enrichment_analysis(mfMG.GOdata_down)

  mfMG.all_res_final_up <- correct_p_values(mfMG.all_res_up)
  mfMG.all_res_final_down <- correct_p_values(mfMG.all_res_down)

  mfMG.results.table_p_up <- get_significant_GO(mfMG.all_res_final_up,0.1)
  mfMG.results.table_p_down <- get_significant_GO(mfMG.all_res_final_down,0.1)

  mfMG.results.table_bh_up <- get_significant_GO_bh(mfMG.all_res_final_up,0.1)
  mfMG.results.table_bh_down <- get_significant_GO_bh(mfMG.all_res_final_down,0.1)
```


```{r}
#diet GO enrichments
  Diet.DEgenes_up <- get_DE_genes(dream_Diet_genes, 1.2)$up
  Diet.DEgenes_down <- get_DE_genes(dream_Diet_genes, 1.2)$down

  Diet.DEgenes_with_GO_up <- get_DE_genes_with_GO(Diet.DEgenes_up, genesMatched)
  Diet.DEgenes_with_GO_down <- get_DE_genes_with_GO(Diet.DEgenes_down, genesMatched)
  
  
  Diet.all_genes_up <- factor(as.integer(toupper(filter(dream_Diet_genes, AveExpr > 1)$gene) %in% Diet.DEgenes_with_GO_up),
                         labels = c(0, 1))
  names(Diet.all_genes_up) <- toupper(filter(dream_Diet_genes, AveExpr > 1)$gene)

  Diet.all_genes_down <- factor(as.integer(toupper(filter(dream_Diet_genes, AveExpr > 1)$gene) %in% Diet.DEgenes_with_GO_down))
  names(Diet.all_genes_down) <- toupper(filter(dream_Diet_genes, AveExpr > 1)$gene)

 
  #CC
  ccDiet.GOdata_up <- create_topGOdata(Diet.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "CC")
  ccDiet.GOdata_down <- create_topGOdata(Diet.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "CC")

  ccDiet.all_res_up <- perform_enrichment_analysis(ccDiet.GOdata_up)
  ccDiet.all_res_down <- perform_enrichment_analysis(ccDiet.GOdata_down)

  ccDiet.all_res_final_up <- correct_p_values(ccDiet.all_res_up)
  ccDiet.all_res_final_down <- correct_p_values(ccDiet.all_res_down)

  ccDiet.results.table_p_up <- get_significant_GO(ccDiet.all_res_final_up,0.1)
  ccDiet.results.table_p_down <- get_significant_GO(ccDiet.all_res_final_down,0.1)

  ccDiet.results.table_bh_up <- get_significant_GO_bh(ccDiet.all_res_final_up,0.1)
  ccDiet.results.table_bh_down <- get_significant_GO_bh(ccDiet.all_res_final_down,
                                                        0.1)
  
  #BP
  bpDiet.GOdata_up <- create_topGOdata(Diet.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "BP")
  bpDiet.GOdata_down <- create_topGOdata(Diet.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "BP")

  bpDiet.all_res_up <- perform_enrichment_analysis(bpDiet.GOdata_up)
  bpDiet.all_res_down <- perform_enrichment_analysis(bpDiet.GOdata_down)

  bpDiet.all_res_final_up <- correct_p_values(bpDiet.all_res_up)
  bpDiet.all_res_final_down <- correct_p_values(bpDiet.all_res_down)

  bpDiet.results.table_p_up <- get_significant_GO(bpDiet.all_res_final_up,0.1)
  bpDiet.results.table_p_down <- get_significant_GO(bpDiet.all_res_final_down,0.1)

  bpDiet.results.table_bh_up <- get_significant_GO_bh(bpDiet.all_res_final_up,
                                                      0.1)
  bpDiet.results.table_bh_down <- get_significant_GO_bh(bpDiet.all_res_final_down,
                                                        0.1)
  
  #MF
  mfDiet.GOdata_up <- create_topGOdata(Diet.all_genes_up, 
                                     canary_geneID2GO_BM_edgeR, "MF")
  mfDiet.GOdata_down <- create_topGOdata(Diet.all_genes_down, 
                                       canary_geneID2GO_BM_edgeR, "MF")

  mfDiet.all_res_up <- perform_enrichment_analysis(mfDiet.GOdata_up)
  mfDiet.all_res_down <- perform_enrichment_analysis(mfDiet.GOdata_down)

  mfDiet.all_res_final_up <- correct_p_values(mfDiet.all_res_up)
  mfDiet.all_res_final_down <- correct_p_values(mfDiet.all_res_down)

  mfDiet.results.table_p_up <- get_significant_GO(mfDiet.all_res_final_up,0.1)
  mfDiet.results.table_p_down <- get_significant_GO(mfDiet.all_res_final_down,0.1)

  mfDiet.results.table_bh_up <- get_significant_GO_bh(mfDiet.all_res_final_up,
                                                      0.1)
  mfDiet.results.table_bh_down <- get_significant_GO_bh(mfDiet.all_res_final_down,
                                                        0.1)
```


# temporary, from the heatmap.Rmd file is clusDesignation. want to see if a cluster of de genes are enriched in GO.
```{r}
# clusDesignation21[clusDesignation21%in%c(2)] %>% names()
# 
# MG.heat_genes <- factor(as.integer(toupper(filter(dream_MG_genes, AveExpr > 0)$gene) %in% names(clusDesignation[clusDesignation21==2])),
#                          labels = c(0, 1))
# 
# names(MG.heat_genes) <- toupper(filter(dream_MG_genes, AveExpr > 0)$gene)
# 
#   #CC
#   ccMG.GOdata_heat <- create_topGOdata(MG.heat_genes, 
#                                      canary_geneID2GO_BM_edgeR, "CC")
# 
#   ccMG.all_res_heat <- perform_enrichment_analysis(ccMG.GOdata_heat)
# 
#   ccMG.all_res_final_heat <- correct_p_values(ccMG.all_res_heat)
# 
#   ccMG.results.table_p_heat <- get_significant_GO(ccMG.all_res_final_heat,0.1)
# 
#   ccMG.results.table_bh_heat <- get_significant_GO_bh(ccMG.all_res_final_heat,0.1)
#   
#   #BP
#   bpMG.GOdata_heat <- create_topGOdata(MG.heat_genes, 
#                                      canary_geneID2GO_BM_edgeR, "BP")
# 
#   bpMG.all_res_heat <- perform_enrichment_analysis(bpMG.GOdata_heat)
# 
#   bpMG.all_res_final_heat <- correct_p_values(bpMG.all_res_heat)
# 
#   bpMG.results.table_p_heat <- get_significant_GO(bpMG.all_res_final_heat,0.1)
# 
#   bpMG.results.table_bh_heat <- get_significant_GO_bh(bpMG.all_res_final_heat,0.1)
#   
#   #MF
#   mfMG.GOdata_heat <- create_topGOdata(MG.heat_genes, 
#                                      canary_geneID2GO_BM_edgeR, "MF")
# 
# 
#   mfMG.all_res_heat <- perform_enrichment_analysis(mfMG.GOdata_heat)
#   mfMG.all_res_final_heat <- correct_p_values(mfMG.all_res_heat)
#   mfMG.results.table_p_heat <- get_significant_GO(mfMG.all_res_final_heat,0.1)
#   mfMG.results.table_bh_heat <- get_significant_GO_bh(mfMG.all_res_final_heat,0.1)
```



# ?

```{r viz-GO.MG}
#add genes to terms for GOplot (from https://github.com/W-L/ADA_coursework)
add_genes_to_terms <- function(GO2genes, results){
  for (i in seq(1, nrow(results))){
    GOterm <- results$GO.ID[i]
    results[i, "genes"] <- paste(GO2genes[GOterm][[1]], collapse=', ')
  }
  return(results)
}

GO2genes <- inverseList(canary_geneID2GO_BM_edgeR)

all_res_down_GOplot <- add_genes_to_terms(GO2genes=GO2genes, results=all_res_down) %>%
  mutate(Category="CC",
         ID=GO.ID,
         adj_pval=weightFisher) %>%
  dplyr::select(Category, ID, Term, adj_pval, genes) %>%
  readr::type_convert()

DE_down_GOplot_genes <- infVSuninf_both_topTags %>%
  # filter(gene %in% names(all_genes_down[all_genes_down == 1])) %>%
  mutate(ID=gene) %>%
  readr::type_convert()

circ <- circle_dat(all_res_down_GOplot, DE_down_GOplot_genes) 
```


#redo classic GO term viz
```{r fig.height=8, fig.width=10}
visualize_results <- function(all_res_down, all_res_up, gg_title) {
  ntop <- 15
  
  # MG down
  ggdata_down <- all_res_down[1:ntop,] %>%
    mutate(enrichment = paste("MG","_DOWN", sep = ""),
           `% overrepresented` = (Significant-Expected)/Annotated)
  
  ggdata_down$Term <- factor(ggdata_down$Term, levels = rev(ggdata_down$Term))
  
  gg_down <- ggplot(ggdata_down,
                    aes(x = Term, y = -log10(as.numeric(weightFisher)), size = log10(Annotated), fill = `% overrepresented`)) +
    expand_limits(y = 1) +
    geom_hline(yintercept = c(-log10(0.05), -log10(0.01), -log10(0.001)),
               linetype = c("dotted", "longdash", "solid"),
               colour = "black",
               linewidth = c(0.5, 1.5, 3)) +
    geom_point(shape = 21) +
    scale_size(range = c(2.5,12.5)) +
    scale_fill_continuous(low = 'royalblue', high = 'red4') +
    xlab('') + 
    ylab('Enrichment score') +
    labs(
      # title = "Downregulated",
      title = paste0("Downregulated ", gg_title, sep=" "),
      caption = 'Cut-off lines drawn at equivalents of p=0.05, p=0.01, p=0.001') +
    theme_bw(base_size = 20) +
    theme(
      legend.position = 'right',
      legend.background = element_rect(),
      plot.title = element_text(angle = 0, size = 16, face = 'bold', vjust = 1, hjust = 0.5),
      plot.subtitle = element_text(angle = 0, size = 14, face = 'bold', vjust = 1),
      plot.caption = element_text(angle = 0, size = 12, face = 'bold', vjust = 1),
      axis.text.x = element_text(angle = 0, size = 10, face = 'bold', hjust = 1.10),  # Set the maximum width and overflow behavior),
      axis.text.y = element_text(angle = 0, size = 12, face = 'bold', vjust = 0.5),
      axis.title = element_text(size = 12, face = 'bold'),
      axis.title.x = element_text(size = 12, face = 'bold'),
      axis.title.y = element_text(size = 12, face = 'bold'),
      axis.line = element_line(colour = 'black'),
      legend.key = element_blank(),
      legend.key.size = unit(1, "cm"),
      legend.text = element_text(size = 14, face = "bold"),
      title = element_text(size = 14, face = "bold")) +
    scale_x_discrete(label = function(x) stringr::str_trunc(x, 50)) +
    coord_flip() 
  
  # MG up
  ggdata_up<- all_res_up[1:ntop,] %>%
    mutate(enrichment = paste("MG","_UP", sep = ""),
           `% overrepresented` = (Significant-Expected)/Annotated)
  
  ggdata_up$Term <- factor(ggdata_up$Term, levels = rev(ggdata_up$Term)) 
  
  gg_up <- ggdata_up %>%
    ggplot(
                  aes(x = Term, y = -log10(as.numeric(weightFisher)), size = log10(Annotated), fill = `% overrepresented`)) +
       expand_limits(y = 1) +
    geom_hline(yintercept = c(-log10(0.05), -log10(0.01), -log10(0.001)),
               linetype = c("dotted", "longdash", "solid"),
               colour = "black",
               linewidth = c(0.5, 1.5, 3)) +
    geom_point(shape = 21) +
    scale_size(range = c(2.5,12.5)) +
    scale_fill_continuous(low = 'royalblue', high = 'red4') +
    xlab('') + 
    ylab('Enrichment score') +
    labs(
      # title = "Upregulated",
      title = paste0("Upregulated ", gg_title, sep=" "),
      caption = 'Cut-off lines drawn at equivalents of p=0.05, p=0.01, p=0.001') +
    theme_bw(base_size = 20) +
    theme(
      legend.position = 'right',
      legend.background = element_rect(),
      plot.title = element_text(angle = 0, size = 16, face = 'bold', vjust = 1, hjust = 0.5),
      plot.subtitle = element_text(angle = 0, size = 14, face = 'bold', vjust = 1),
      plot.caption = element_text(angle = 0, size = 12, face = 'bold', vjust = 1),
      axis.text.x = element_text(angle = 0, size = 10, face = 'bold', hjust = 1.10),
      axis.text.y = element_text(angle = 0, size = 12, face = 'bold', vjust = 0.5),
      axis.title = element_text(size = 12, face = 'bold'),
      axis.title.x = element_text(size = 12, face = 'bold'),
      axis.title.y = element_text(size = 12, face = 'bold'),
      axis.line = element_line(colour = 'black'),
      legend.key = element_blank(),
      legend.key.size = unit(1, "cm"),
      legend.text = element_text(size = 14, face = "bold"),
      title = element_text(size = 14, face = "bold")) +
    scale_x_discrete(label = function(x) stringr::str_trunc(x, 50)) +
    coord_flip()
  
  return(list(gg_down = gg_down, gg_up = gg_up))
}

visualize_results(bpMG.all_res_down, bpMG.all_res_up,
                  # "Biological Processes: MG infection",
                  "Biological Processes: MG infection")
visualize_results(ccMG.all_res_down, ccMG.all_res_up,
                  "Cellular Compartment: MG infection")
visualize_results(mfMG.all_res_down, mfMG.all_res_up,
                  "Molecular Function: MG infection")
```

# Diet GO term plots
```{r fig.height=8, fig.width=10}
visualize_results(bpDiet.all_res_down, bpDiet.all_res_up,
                  # "Biological Processes: MG infection",
                  "Biological Processes: Response to Protein Diet")
visualize_results(ccDiet.all_res_down, ccDiet.all_res_up,
                  "Cellular Compartment: Response to Protein Diet")
visualize_results(mfDiet.all_res_down, mfDiet.all_res_up,
                  "Molecular Function: Response to Protein Diet")
```


# classic GO term visualization

```{r}

# visualize results
ntop <- 15

ggdata_down <- all_res_down[1:ntop,] %>%
    mutate(enrichment = paste("MG","_DOWN", sep = ""),
           `% overrepresented` = (Significant-Expected)/Annotated)

ggdata_down$Term <- factor(ggdata_down$Term, levels = rev(ggdata_down$Term))
  

gg_down <- ggplot(ggdata_down,
    aes(x = Term, y = -log10(as.numeric(weightFisher)), size = log10(Annotated), fill = `% overrepresented`)) +
  
    expand_limits(y = 1) +
    geom_hline(yintercept = c(-log10(0.05), -log10(0.01), -log10(0.001)),
      linetype = c("dotted", "longdash", "solid"),
      colour = "black",
      linewidth = c(0.5, 1.5, 3)) +
    geom_point(shape = 21) +
    scale_size(range = c(2.5,12.5)) +
    scale_fill_continuous(low = 'royalblue', high = 'red4') +
  
    xlab('') + 
    ylab('Enrichment score') +
    labs(
      title = "Downregulated",
      # subtitle = paste('Top ',ntop,' terms ordered by weighted Fisher p-value', sep=""),
      caption = 'Cut-off lines drawn at equivalents of p=0.05, p=0.01, p=0.001') +
  
    theme_bw(base_size = 24) +
    theme(
      legend.position = 'right',
      legend.background = element_rect(),
      plot.title = element_text(angle = 0, size = 16, face = 'bold', vjust = 1, hjust = 0.5),
      plot.subtitle = element_text(angle = 0, size = 14, face = 'bold', vjust = 1),
      plot.caption = element_text(angle = 0, size = 12, face = 'bold', vjust = 1),
  
      axis.text.x = element_text(angle = 0, size = 12, face = 'bold', hjust = 1.10),
      axis.text.y = element_text(angle = 0, size = 12, face = 'bold', vjust = 0.5),
      axis.title = element_text(size = 12, face = 'bold'),
      axis.title.x = element_text(size = 12, face = 'bold'),
      axis.title.y = element_text(size = 12, face = 'bold'),
      axis.line = element_line(colour = 'black'),
  
      #Legend
      legend.key = element_blank(), # removes the border
      legend.key.size = unit(1, "cm"), # Sets overall area/size of the legend
      legend.text = element_text(size = 14, face = "bold"), # Text size
      title = element_text(size = 14, face = "bold")) +
  
    # facet_grid(rows = vars(enrichment), scales = "free") +
    coord_flip()

#MG up
ggdata_up<- all_res_up[1:ntop,] %>%
    mutate(enrichment = paste("MG","_UP", sep = ""),
           `% overrepresented` = (Significant-Expected)/Annotated)

ggdata_up$Term <- factor(ggdata_up$Term, levels = rev(ggdata_up$Term))
  

gg_up <- ggplot(ggdata_up,
    aes(x = Term, y = -log10(as.numeric(weightFisher)), size = log10(Annotated), fill = `% overrepresented`)) +
    expand_limits(y = 1) +
    geom_hline(yintercept = c(-log10(0.05), -log10(0.01), -log10(0.001)),
      linetype = c("dotted", "longdash", "solid"),
      colour = "black",
      linewidth = c(0.5, 1.5, 3)) +
    geom_point(shape = 21) +
    scale_size(range = c(2.5,12.5)) +
    scale_fill_continuous(low = 'royalblue', high = 'red4') +
  
    xlab('') + 
    ylab('Enrichment score') +
    labs(
      title = "Upregulated",
      # subtitle = paste('Top ',ntop,' terms ordered by weighted Fisher p-value', sep=""),
      caption = 'Cut-off lines drawn at equivalents of p=0.05, p=0.01, p=0.001') +
  
    theme_bw(base_size = 24) +
    theme(
      legend.position = 'right',
      legend.background = element_rect(),
      plot.title = element_text(angle = 0, size = 16, face = 'bold', vjust = 1, hjust = 0.5),
      plot.subtitle = element_text(angle = 0, size = 14, face = 'bold', vjust = 1),
      plot.caption = element_text(angle = 0, size = 12, face = 'bold', vjust = 1),
  
      axis.text.x = element_text(angle = 0, size = 12, face = 'bold', hjust = 1.10),
      axis.text.y = element_text(angle = 0, size = 12, face = 'bold', vjust = 0.5),
      axis.title = element_text(size = 12, face = 'bold'),
      axis.title.x = element_text(size = 12, face = 'bold'),
      axis.title.y = element_text(size = 12, face = 'bold'),
      axis.line = element_line(colour = 'black'),
  
      #Legend
      legend.key = element_blank(), # removes the border
      legend.key.size = unit(1, "cm"), # Sets overall area/size of the legend
      legend.text = element_text(size = 14, face = "bold"), # Text size
      title = element_text(size = 14, face = "bold")) +
  
    # facet_grid(rows = vars(enrichment), scales = "free") +
    coord_flip()

gg_down
gg_up


  title <- paste('GO Biological processes enriched in contrast: ', 
                 "MG")

  # Create a text grob
  tgrob <- ggpubr::text_grob(label = title, size = 20)
  # Draw the text
  plot_0 <- ggpubr::as_ggplot(tgrob) + theme(plot.margin = margin(0,2,0,0, "cm"))
  
  gg <- ggpubr::ggarrange(plot_0,
                          gg_up, gg_down,
                          nrow=3,
                          align = "v",
                          common.legend = TRUE,
                          legend = "right",
                          heights = c(0.5,5,5)
                          ) +
        ggpubr::bgcolor("White")
  
gg
```

I like the above figure, but need it for CC, BP, MF for MG & Diet, up & Down. Need to do both of those asap. I have the output data for all of those now, just need to make the graphs.




```{r}
tmp_MG_tidy %>%
  left_join(tmp_Diet_tidy, by = join_by(gene, external_gene_name, entrezgene_description)) %>%
  filter(AveExpr.x > 1.5) %>%
  mutate(delabel = case_when(
    stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
    TRUE ~ gene
  )) %>%
  mutate(delabel = case_when(abs(logFC.x) < 0.4 & abs(logFC.y) < 0.4 ~ NA,
                             TRUE ~ delabel)) %>%
  filter(gene %in% GO2genes$`GO:0006955`) %>% # this subsets to only genes "immune response"
  ggplot(aes(logFC.x, logFC.y, label = delabel)) +
  # geom_line(aes(group = gene), alpha = 0.1, position = position_dodge(0.2)) +
  geom_point(aes(group = gene),  alpha=0.2) +
  # geom_boxplot(width = 0.25, outlier.shape = NA) +
  # geom_smooth(method="lm") +
  geom_abline(alpha=0.5) +
  ggrepel::geom_text_repel(size = 2) +
  theme_classic() +
  theme(legend.position = "none") +
  annotate("text",
           label = "Higher in Protein",
           x = 1,
           y = 1.35) +
  annotate("text",
           label = "Higher in Lipid",
           x = -0.9,
           y = -0.85) +
  ggtitle("Correlation of Expression Fold Changes in effects of Diet and MG (immune genes)") +
  xlab("logFC (MG infection)") +
  ylab("logFC (Diet)")
```

# newer GO term alternative visualizations

```{r fig.height=12}
# circ %>%
#   dplyr::filter(adj_pval < 0.05) %>%
#   dplyr::filter(category == 'CC') %>%
#   dplyr::arrange(adj_pval) %>%
#   GOBar(display='multiple') +
#   coord_flip()
```

```{r}
# Generate the bubble plot with a label threshold of 3
# NEED TO FIX, has 
GOBubble(circ, ID = T, table.legend = T, labels=3)
```

```{r}
#Generate a circular visualization of the results of gene- annotation enrichment analysis
GOCircle(circ)
```

```{r}

DE_down_GOplot_genes_p05 <- DE_down_GOplot_genes %>%
  filter(adj.P.Val < 0.1) %>%
  dplyr::select(ID, logFC)

process_enrich_down_CC <- all_res_down_GOplot %>%
  filter(adj_pval < 0.05) %>%
  pull(Term) %>% head(6)

chord <- chord_dat(circ, DE_down_GOplot_genes_p05, process_enrich_down_CC)

# chord <- circ %>%
#   drop_na() #%>%
#   filter(adj_pval < 0.2 | ID %in% c("GO:0032449", "GO:0042105", "GO:0044297", "GO:0016020", "GO:0043203", 
# "GO:0005604", "GO:0072686", "GO:0000139", "GO:0000242", "GO:0045121", 
# "GO:0098857"))




GOCluster(
  circ,
  head(process_enrich_down_CC,6),
  metric='minkowski',
  clust.by = 'logFC',
  term.width = 0.5,
  lfc.width = 0.4,
  lfc.space = 0.05,
  term.space = 0.1,
  clust = 'centroid'
)

```

```{r}
process_enrich_down_CC_chord <- all_res_down_GOplot %>%
  filter(adj_pval < 0.05) %>%
  pull(Term)# %>% head(10)

# chord_GO <- chord_dat(circ, DE_down_GOplot_genes_p05)

# GOChord(chord_GO, space = 0.03, gene.order = 'logFC', gene.space = 0.25, gene.size = 2)
```

```{r}
# GOHeat(chord[,-9], nlfc = 0)

# GOHeat(chord, nlfc=1, fill.col = c('darkgoldenrod1', 'black', 'cyan1'))
```

I see that JCAHIN (LOC103823426) has the largest logFC when looking at eyescores vs expression. Let's visualize just that gene to see how those correlate.
```{r}



# compare eyescore of the given day to FC of JCHAIN gene
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID",
                                        "Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=ES_minMax, y=log2FC_normToDay0, color = Diet)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Eyescore") +
  facet_wrap(~Day) +
  theme_classic()

# comparing day7 eyescore to 
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=r_ES_day7, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Day7 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()

# comparing day3 eyescore to 
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=r_ES_day3, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Day3 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()

# logLoads
# comparing day3 load to JCHAIN FC
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=r_logLoad_day3, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Day3 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()

# comparing day7 load to JCHAIN FC
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=r_logLoad_day7, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Day7 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()

# comparing day14 logload to JCHAIN FC
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID", "Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=r_logLoad_day14, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs Day14 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()


## with AUC instead
# comparing day14 auclogload to JCHAIN FC
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>%
  ggplot(aes(x=logLoad_AUC_14, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs aucDay14 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()


# comparing day21 aucload to JCHAIN FC
tpm_df %>%
  dplyr::filter(gene_id == "LOC103823426") %>%
  left_join(metadata_phenotypes, by = c("Bird" = "Bird.ID","Day",
                                        "Diet"#, "MG.Diet.Day", "MG.Day",
                                        # "Diet.Day", "MG.active", "Exposure.MG",
                                        # "Exposure.Diet", "Exposure.Diet.Day"
                                        )) %>%
  mutate(gene="JCHAIN") %>%
  # normalize tpm to day 0 tpm.
  group_by(Bird) %>%
  mutate(RN = row_number(), log2FC_normToDay0 = log2(tpm/tpm[RN ==1])) %>%
  filter(Day > 0, MG.active == "YES") %>% 
  ggplot(aes(x=logLoad_AUC_21, y = log2FC_normToDay0, color = Day)) +
  geom_point() +
  geom_smooth(method="lm") +
  ggtitle("JCHAIN log2FC vs aucDay21 Eyescore") +
  facet_wrap(~Diet) +
  theme_classic()

```





# variance partition visualization
```{r}
varpart.form <- ~ (1|Diet) + (1|MG.active) + (1|Bird.ID)
varPart <- fitExtractVarPartModel( vobjDream, varpart.form, metadata )

vp <- sortCols( varPart )


vp %>%
  as.data.frame() %>% 
  rownames_to_column("gene") %>%
  dplyr::filter(gene %in% DEgenes_up | gene %in% DEgenes_down) %>%
  column_to_rownames("gene") %>%
  slice_head(n=15) %>%
  plotPercentBars()
# plotPercentBars( vp[1:20,] )
plotVarPart( vp )

# vp %>% as.data.frame() %>% rownames_to_column("gene") %>% head()
```