---
title: "edgeR Canary DE Analysis"
author: "Carson Stacy"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(edgeR)
library(ComplexUpset)
library(ComplexHeatmap)
library("AnnotationHub")
library(tidyverse)
library(here)
library(clusterProfiler)
library(flextable)
library(GGally)
```

# Data Preparation
```{r load annotations}
# get annotation hub (orgdb)

ah <- AnnotationHub()
orgs <- subset(ah, ah$rdataclass == "OrgDb")
orgdb <- query(orgs, "Serinus canaria")[[1]]
```

## load custom functions
```{r custom functions}
source(here::here("scripts/canary_data_processing_functions.R"))
```

```{r read-data}
# assign file path locaiton of rsem.merged.gene_counts.tsv file
filepath_rsem_gene_counts <- here::here("data", "rsem.merged.gene_counts.tsv")

#assign location of tpm file for later analysis
# filepath_rsem_gene_tpm <- here::here("SerCan2020", "rsem.merged.gene_tpm.tsv")

# read in rsem.merged.gene_counts.tsv
g.counts.l <- data.table::fread(filepath_rsem_gene_counts, sep="\t") %>% as_tibble()
  # readr::read_tsv(filepath_rsem_gene_counts, show_col_types = FALSE, trim_ws = FALSE,
  #          name_repair = "minimal", progress = TRUE, skip=1)
# i don't know why read_tsv is not working, i'm using fread instead.
```

## Remove problematic samples

The Sample ES89 was removed due to failed sequencing.
Samples ES31,ES32, and ES65 were removed due to incomplete sample metadata.
Samples from bird SEM19-1481 had their metadata corrected due to an error in recording of diet received.
```{r remove-samples}
g.counts.l <- g.counts.l %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>% 
  # dplyr::select(-contains("Blue-050")) %>%
  # change diet group of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       # c(`ES64_SEM19-1481_MG-lipid_21`="ES64_SEM19-1481_MG-protein_21",
       #   `ES52_SEM19-1481_MG-lipid_14`="ES52_SEM19-1481_MG-protein_14",
       #   `ES83_SEM19-1481_MG-lipid_0`="ES83_SEM19-1481_MG-protein_0"))
       # )
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       )
```



## generate metadata

Using sample names, a tibble of metadata is generated.
```{r generate-metadata}
# create tibble metadata object, convert column names to sample ID column
metadata <-
  tibble(SampID = colnames(g.counts.l[, c(3:ncol(g.counts.l))])) %>%
  separate(
    SampID,
    c("Sample.ID", "Bird.ID",
      "MG.Diet", "Day"),
    "_",
    extra = "warn",
    fill = "warn",
    remove = FALSE
  ) %>%
  separate(
    MG.Diet,
    c("MG.Status", "Diet"),
    sep = "[-]",
    extra = "warn",
    fill = "warn",
    remove = FALSE
  ) %>%
  mutate(
    MG.Diet.Day = paste0(MG.Diet, ".", Day),
    MG.Day = paste0(MG.Status, ".", Day),
    Diet.Day = paste0(Diet, ".", Day)
  ) %>%
  mutate(MG.active = if_else(Day < 1, "NO",
                             if_else(MG.Status == "Control",
                                     "NO", "YES"))) %>%
  mutate(
    Exposure.MG = paste0(MG.active, ".", MG.Status),
    Exposure.Diet = paste0(MG.active, ".", Diet),
    Exposure.Diet.Day = paste0(Exposure.Diet, ".", Day)
  )


# read in phenotype data

## pathogen loads
canary_loads <-
  # only birds with both phenotype and genotype data
  # readr::read_csv(here("Geno.x.Pheno/canary_loads.csv")) %>% 
  data.table::fread(here("sample_metadata/canary_loads.csv"),sep=",") %>% 
  tibble() %>%
  # mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  mutate(Bird = stringr::str_replace(ID, " ", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  # mutate(Bird = stringr::str_replace(Bird, "-", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD.19 239", "LD19.239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue.045", "Blue.45")) %>%
  dplyr::filter(Bird %in% unique(metadata$Bird.ID)) %>%
  dplyr::select(-ID)

## semi-quantitative eyescore measures
eyescores <-
  # readr::read_csv(
  data.table::fread(
    here("sample_metadata/canary_eyescores.csv"),
    sep=","
    # col_types = cols(Day = col_integer(),
    #                  Date = col_character())
  ) %>%
  tibble() %>%
  mutate(Day = as.integer(Day),
         Date = as.character(Date)) %>%
  # correcting for errors in manual data entry
  mutate(Bird = stringr::str_replace(ID, " ", "-")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  mutate(Bird = stringr::str_replace(Bird, "\\.", "-")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Nlue", "Blue")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD-19 239", "LD19-239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue-045", "Blue-45")) %>%
  dplyr::filter(Bird %in% unique(metadata$Bird.ID)) %>%
  dplyr::mutate(day = replace(Day, which(Day<0), 0)) %>% # adjust day -1 to day 0
  dplyr::select(Bird, day, ES)

# inspired from https://support.bioconductor.org/p/p132992/

# generate columns of phenotypic metadata I'd like to have
# pathogen loads
day3_loads <- canary_loads %>%
  filter(day==3) %>%
  mutate(logLoad_day3 = logLoad, r_logLoad_day3=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day3, r_logLoad_day3))

day7_loads <- canary_loads %>%
  filter(day==7) %>%
  mutate(logLoad_day7 = logLoad, r_logLoad_day7=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day7, r_logLoad_day7))

day14_loads <- canary_loads %>%
  filter(day==14) %>%
  mutate(logLoad_day14 = logLoad, r_logLoad_day14=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day14, r_logLoad_day14))

day21_loads <- canary_loads %>%
  filter(day==21) %>%
  mutate(logLoad_day21 = logLoad, r_logLoad_day21=logLoad/max(logLoad)) %>%
  dplyr::select(c(Bird, logLoad_day21, r_logLoad_day21))

#AUC estimates
day14_AUC_loads <- canary_loads %>%
  filter(day<=14) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,logLoad, type = "spline")
  ) %>%
  mutate(logLoad_AUC_14 = auc, r_logLoad_AUC_14=auc/max(auc)) %>%
  dplyr::select(c(Bird, logLoad_AUC_14, r_logLoad_AUC_14))

day21_AUC_loads <- canary_loads %>%
  filter(day<=21) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,logLoad, type = "spline")
  ) %>%
  mutate(logLoad_AUC_21 = auc, r_logLoad_AUC_21=auc/max(auc)) %>%
  dplyr::select(c(Bird, logLoad_AUC_21, r_logLoad_AUC_21))


load_phenotypes <- purrr::reduce(list(day3_loads,day7_loads,
                                      day14_loads, day21_loads,
                                      day14_AUC_loads, day21_AUC_loads), 
                                 dplyr::left_join, by = 'Bird')

## now do the same for eyescore data
day_eyes <- eyescores %>%
  filter(., day %in% c(0,14,21)) %>%
  mutate(day=as.character(day)) %>%
  dplyr::select(c(Bird, day, ES))

day3_eyes <- eyescores %>%
  filter(.,day==3) %>%
  mutate(ES_day3 = ES, r_ES_day3=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day3, r_ES_day3))

day7_eyes <- eyescores %>%
  filter(.,day==7) %>%
  mutate(ES_day7 = ES, r_ES_day7=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day7, r_ES_day7))

day14_eyes <- eyescores %>%
  filter(.,day==14) %>%
  mutate(ES_day14 = ES, r_ES_day14=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day14, r_ES_day14))

day21_eyes <- eyescores %>%
  filter(.,day==21) %>%
  mutate(ES_day21 = ES, r_ES_day21=ES/max(ES)) %>%
  dplyr::select(c(Bird, ES_day21, r_ES_day21))

#AUC estimates
day7_AUC_eyes <- eyescores %>%
  filter(.,day<=7) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_7 = auc, r_ES_AUC_7=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_7, r_ES_AUC_7))

day14_AUC_eyes <- eyescores %>%
  filter(.,day<=14) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_14 = auc, r_ES_AUC_14=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_14, r_ES_AUC_14))

day21_AUC_eyes <- eyescores %>%
  filter(.,day<=21) %>%
  group_by(Bird) %>%
  summarise(
  auc = MESS::auc(day,ES, type = "spline")
  ) %>%
  mutate(ES_AUC_21 = auc, r_ES_AUC_21=auc/max(auc)) %>%
  dplyr::select(c(Bird, ES_AUC_21, r_ES_AUC_21))



eye_phenotypes <- purrr::reduce(list(day3_eyes,day7_eyes,
                                      day14_eyes, day21_eyes,
                                      day7_AUC_eyes,
                                      day14_AUC_eyes, day21_AUC_eyes), 
                                 dplyr::left_join, by = 'Bird')

## add phenotype data to existing metadata df
metadata_phenotypes <- metadata %>%
  left_join(load_phenotypes, by = join_by('Bird.ID' == "Bird")) %>%
  left_join(eye_phenotypes, by = join_by('Bird.ID' == "Bird")) %>%
  left_join(day_eyes, by = join_by('Bird.ID' == "Bird", "Day" == "day")) %>%
  mutate_all(., ~replace_na(.,0)) %>%
  distinct() %>% # to get rid of duplicated row
  mutate(ES_minMax = ES/6) %>% # normalized to 0 to 1
  # View()
  mutate(logLoad = case_when(
    Day == 0 ~ logLoad_day3,
    Day == 14 ~ logLoad_AUC_14/14,
    Day == 21 ~ logLoad_AUC_21/21,
    TRUE ~ NA
  ),
  logLoad_alt = case_when(
    Day == 0 ~ 0,
    Day == 14 ~ logLoad_day14,
    Day == 21 ~ logLoad_day21,
    TRUE ~ NA
  ),
  ES_day0_analysis = case_when(
    Day == 0 ~ ES_day3/6,
    TRUE ~ ES_minMax
  )
  ) %>%
  mutate(tolerance_loadoverES = (1+logLoad_alt)/(ES_day0_analysis+1)-1, 
         tolerance_erin = 1-((ES)/(logLoad_alt)),
         tolerance = 1-(ES/logLoad_alt))
```



```{r}
metadata_phenotypes_ixn <- metadata_phenotypes |>
  group_by(Bird.ID) |>
  mutate(tolerance_median = median(tolerance, na.rm = TRUE)) |>
  ungroup()|>
  # metadata_phenotypes |>
  # # filter(Day==0) |>
  # ## filter(MG.active=="YES") |>
  # # add 1 to avoid undefined
  # mutate(tolerance_1 = ES_AUC_7/(1+logLoad_AUC_14), 
  #        tolerance_2 = ES_AUC_7/(1+logLoad_day14),
  #        tolerance_3 = ES_day7/(1+logLoad_day7),
  #        tolerance_4 = r_ES_day3/(1+r_logLoad_day3), # maybe best?
  #        tolerance_5 = (1+r_logLoad_day3)/(1+r_ES_day3)-1, # this format works better for "tolerance"
  #        #tolerance_6 = (1+logLoad_day14)/(1+r_ES_AUC_14)-1,
  #        tolerance_0 = 1+logLoad/(1+ES)-1
  #        ) %>%
  # ## filter(MG.Status != "Control") %>%
  # # filter(!Sample.ID %in% c("ES12", "ES25", "ES36", "ES58", "ES84", "ES53")) %>%
  #mutate(#tolerance_5_cat = dplyr::ntile(tolerance_5,3),
  #        tolerance_5_cat = dplyr::ntile(tolerance_5,3),
         # tolerance_cat = dplyr::ntile(tolerance,4)) %>%
  # group_by(Bird.ID) |>
  # mutate(tolerance_median = median(tolerance, na.rm = TRUE)) |>
  # ungroup() |>#)
  mutate(
         tolerance_cat = case_when(
           tolerance_median > 0 ~ "tolerant",
           tolerance_median < 0 ~ "susceptible",
           TRUE ~ "intermediate"
         )) %>%
  mutate(tolerance_cat = factor(tolerance_cat, levels = c("susceptible", "intermediate",#"intermediate1",
                                                          "tolerant"))) %>%
  mutate(diet.tolerance = paste0(Diet, ".",tolerance_cat)) %>%
  # mutate(SampID = str_replace_all(SampID,"\\.","-")) %>%
  column_to_rownames("SampID")

metadata_phenotypes_ixn  %>%
  rownames_to_column("Sample.Name") %>%
  dplyr::select("Sample.Name", "Sample.ID", "Bird.ID", "MG.Status", "MG.active", "Diet", "Day", "ES_day0_analysis", "logLoad", "tolerance", "tolerance_median", "tolerance_cat") %>%
  rename("ES_day0_analysis"= "ES" ,  "tolerance"="tolerance.day" ,  "tolerance_median" = "tolerance.bird",
         "tolerance_cat" = "tolerance.category" 
         ) %>%
  mutate(Day = as.numeric(Day)) %>% 
  # left_join(eyescores[,-3], by = c("Bird.ID"="Bird", "Day" = "day"))
  write_tsv(., "metadata_tolerance_scores.tsv")
```

```{r}

```

# Pull gene annotations
```{r}
# mart2 <- if(exists('mart2') == TRUE){
#   mart2
# } else {
#   biomaRt::useMart(biomart = "ensembl", dataset = "scanaria_gene_ensembl")
# }
# 
# listDatasets(mart2)
# 
# 
# gene_info2 <- biomaRt::getBM(attributes = c(#'ensembl_transcript_id',
#                             # 'ensembl_peptide_id',
#                             'ensembl_gene_id', 
#                             # 'uniprot_gn_symbol',
#                             'wikigene_name',
#                             'entrezgene_id',
#                             'entrezgene_description',
#                             # '',
#                             'external_gene_name'
#                             # 'go_id',
#                             # 'description'
#                             ),
#              values = g.counts.l$gene_id,
#              mart = mart) %>%
#   dplyr::mutate(entrezgene_id = paste0("LOC",entrezgene_id)) %>%
#   dplyr::mutate(gene = ifelse(entrezgene_id %in% rownames(fitmm@.Data[[1]]), entrezgene_id, external_gene_name))

library(biomartr)
# more complete annotation for genes.
# biomartr::getCollection( db = "refseq", organism = "Serinus canaria")

scan_gff <- biomartr::read_gff("./_db_downloads/collections/refseq/Serinus_canaria/Serinus_canaria_genomic_refseq.gff.gz") %>%
  filter(type=="gene", source == "Gnomon") %>%
  # create new column pulling out gene name
  mutate(gene_common_name = str_extract(attribute, "ID=gene-([^;]+)"),
         gene_ID = str_extract(attribute, "Dbxref=GeneID:([^;]+)")) %>%
  mutate(gene_common_name = str_remove(gene_common_name, "ID=gene-") ,
         gene_ID = str_remove(gene_ID, "Dbxref=GeneID:"))

# get list of gene IDs
gene_set <- unique(scan_gff$gene_ID)

# get result BM
result_BM <- biomartr::biomart( genes      = gene_set, # genes were retrieved using biomartr::getGenome()
                                mart       = "ENSEMBL_MART_ENSEMBL", # marts were selected with biomartr::getMarts()
                                dataset    = "scanaria_gene_ensembl", # datasets were selected with biomartr::getDatasets()
                                attributes = c(#'ensembl_transcript_id',
                            # 'ensembl_peptide_id',
                            'ensembl_gene_id',
                            # 'uniprot_gn_symbol',
                            'wikigene_name',
                            #'entrezgene_id',
                            'entrezgene_description',
                            # '',
                            'external_gene_name'
                            # 'go_id',
                            # 'description'
                            ), # attributes were selected with biomartr::getAttributes()
                              filters    = "entrezgene_id"
                            ) # specify what ID type was stored in the fasta file retrieved with biomartr::getGenome()

head(result_BM)

# GO_tbl <- biomartr::getGO(organism = "Serinus canaria",
#                           genes    = gene_set,
#                           filters  = "entrezgene_id"
#                           )

# head(GO_tbl)


# gff anno
gene_anno <- scan_gff %>%
  dplyr::select(gene_ID, gene_common_name, attribute) %>%
  dplyr::rename(entrezgene_id = gene_ID,
                gene_gff = gene_common_name) %>%
  # split attribute column based on ; separator, column name comes before =
  separate_wider_delim(attribute, delim = ";", names= c("ID", "LOC_id", "common_name", "description",
                                                        "endrange", "genbank_key", "gene_name", "gene_type"),
                       too_many = "merge",
                       too_few = "align_start") %>%
  dplyr::select(entrezgene_id, gene_gff, description) %>%
  mutate(description = str_remove(description, "description="))

# gene_GO_tbl <- merge(result_BM, GO_tbl, by = "entrezgene_id", all = TRUE)


# Generate count matrix
gene_expression_matrix <- g.counts.l %>% #View()
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  # convert to matrix for edgeR
  as.matrix()

result_BM_merged <- result_BM %>%
  mutate(entrezgene_id = as.character(entrezgene_id)) %>%
  full_join(gene_anno, by = "entrezgene_id") %>%
  mutate(gene_name = case_when(
    str_detect(gene_gff, "LOC1") & !str_detect(external_gene_name, "LOC1") & !is.na(external_gene_name) & external_gene_name != "" ~ external_gene_name,
    str_detect(gene_gff, "LOC1") & !str_detect(wikigene_name, "LOC1") & !is.na(wikigene_name) ~ wikigene_name,
    #str_detect(gene_gff, "LOC10") ~ gene_gff,
    TRUE ~ gene_gff#paste0("LOC",entrezgene_id)
  ),
  gene_tag = case_when(
    paste0("LOC",entrezgene_id) %in% rownames(gene_expression_matrix) ~ paste0("LOC",entrezgene_id),
    gene_name %in% rownames(gene_expression_matrix) ~ gene_name,
    TRUE ~ NA),
  gene = case_when(
    paste0("LOC",entrezgene_id) %in% rownames(gene_expression_matrix) ~ entrezgene_id,
    gene_name %in% rownames(gene_expression_matrix) ~ gene_name,
    TRUE ~ NA)
  ) %>%
  # remove formatting issue for commas in description.
  mutate(description = str_replace_all(description, "%2", ",")) %>%
  mutate(description = str_replace_all(description, ",C", ",")) 

```

```{r make-DGElist}
# generate tidy data objects for downstream analysis
get_tidy_expression(
  #data_path=filepath_rsem_gene_tpm,
  data =
    g.counts.l %>% #View()
    dplyr::select(-`transcript_id(s)`) %>% 
    # remove globin genes
    dplyr::filter(gene_id != "LOC103824466") %>%
    dplyr::filter(gene_id != "LOC103817748") %>%
    dplyr::filter(gene_id != "LOC103817749") %>%
    dplyr::filter(gene_id != "LOC115485052") %>%
    dplyr::filter(gene_id != "LOC103817748")%>%
    # Add rownames
    column_to_rownames("gene_id") %>%
    # convert to matrix for edgeR
    as.matrix(),
  from_file = FALSE,
  meta = metadata_phenotypes,
  type = "tpm"
)


```

```{r}
gene_expression_matrix_ixn <- gene_expression_matrix |>
  data.frame() |>
  ## dplyr::select(!ends_with("_0")) |>
  ## dplyr::select(contains("MG")) |>
  # these are birds who PCR shows no pathogen
  # dplyr::select(-contains("ES12")) |>
  # dplyr::select(-contains("ES25")) |>
  # dplyr::select(-contains("ES36")) |>
  # dplyr::select(-contains("ES58")) |>
  # dplyr::select(-contains("ES84")) |>
  # dplyr::select(-contains("ES53")) |>
  as.matrix()

# design matrix
#tolerance_1 <- metadata_phenotypes_ixn$tolerance_5 # careful need to update this...
#tolerance_1 <- metadata_phenotypes_ixn$tolerance # careful need to update this...
#Diet_ixn <- 1+as.numeric(as.factor(metadata_phenotypes_ixn$Diet)) #%>% gsub(1,4, .) %>% gsub(2,1, .)
Diet_ixn <- as.factor(metadata_phenotypes_ixn$Diet)

MG.Active <- as.factor(metadata_phenotypes_ixn$MG.Status)
#grouping_var <- interaction(Diet_ixn, metadata_phenotypes_ixn$MG.Status) %>% as.factor()

# Bird.ID <- as.factor(metadata_phenotypes_ixn$Bird.ID)
# this one is not appropriate to use
#design <- model.matrix(~ Diet_ixn:tolerance_1)
#design


# above design works well, but hard to make contrasts. use pseudogroupings instead
# generate groupings
diet_inf <- interaction(Diet_ixn, MG.Active) #%>% as.factor()
design_ixn <- model.matrix(~ 0 + diet_inf) #+ Day_tolerance)

# entrez_IDs <- getBM(attributes = c(#'ensembl_transcript_id',
#                             #'ensembl_peptide_id',
#                             # 'ensembl_gene_id', 
#                             # 'uniprot_gn_symbol',
#                             'wikigene_name',
#                             'entrezgene_id',
#                             'external_gene_name',
#                             'go_id'
#                             ),
#              values = rownames(gene_expression_matrix_ixn),
#              mart = mart) |>f
#   mutate(entrezgene_id = as.character(entrezgene_id))

entrez_names_gene_expression_matrix_ixn <-
  gene_expression_matrix_ixn |>
  data.frame() |>
  rownames_to_column("external_gene_name") |> 
  left_join(#entrez_IDs |> dplyr::select(-go_id),
            result_BM_merged,
            by = "external_gene_name", multiple = "first") |> #View()
  mutate(entrezID = case_when(
    str_detect(external_gene_name, "^LOC") ~ str_replace(external_gene_name, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ external_gene_name
  )) |> #View()
  pull(entrezID)

rownames(gene_expression_matrix_ixn) <- entrez_names_gene_expression_matrix_ixn



# get tidy expression data for later
# returns tpm_df
# get_tidy_expression(
#   #data_path=filepath_rsem_gene_tpm,
#   data = gene_expression_matrix_ixn,
#   from_file = FALSE,
#   meta = metadata_phenotypes,
#   type = "tpm"
# )
```

# Differential expression analysis

```{r}
y_ixn <- DGEList(
  counts = gene_expression_matrix_ixn,
  # names= entrez_names_gene_expression_matrix_ixn,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff <- as.numeric(round(cpm(10, mean(y_ixn$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep <- rowSums(cpm(y_ixn) > cpm_cutoff) >= 12
y_ixn <- y_ixn[keep,]
summary(keep)

y_ixn <- calcNormFactors(y_ixn)
y_ixn <- estimateDisp(y_ixn, design_ixn)#, robust=TRUE) 

y_ixn$samples$group <- #grouping_var
  diet_inf %>% as.factor()
  # interaction(Diet_ixn, metadata_phenotypes_ixn$tolerance_5_cat) %>% as.factor()

# y_ixn$samples$group <- #grouping_var
#   interaction(Diet_ixn, metadata_phenotypes_ixn$tolerance_cat) %>% as.factor()


contrast_ixn <- makeContrasts(
  #design, #metadata,
  contrasts = c(
    # Interaction contrast
    # infection response in protein - infection response in lipid
    "(diet_infprotein.MG - diet_infprotein.Control) - (diet_inflipid.MG - diet_inflipid.Control)",
    # Diet Contrast
    # Protein - lipid
    "(diet_infprotein.MG + diet_infprotein.Control)/2-(diet_inflipid.MG + diet_inflipid.Control)/2",
    # Infection contrast
    # Infected - Uninfected
    "(diet_infprotein.MG + diet_inflipid.MG)/2-(diet_infprotein.Control + diet_inflipid.Control)/2",
    # Infection in Protein Birds
    # infected protein - uninfected protein
    "diet_infprotein.MG - diet_infprotein.Control",
    # Infection in Lipid Birds
    # infected lipid - uninfected lipid
    "diet_inflipid.MG - diet_inflipid.Control",
    # Uninfected Diet Contrast
    # Protein - lipid
    "(diet_infprotein.Control)-(diet_inflipid.Control)"
  ),
  levels = design_ixn
)

fit <- glmQLFit(y_ixn, design_ixn)#, robust=TRUE)

# res1 <-  glmQLFTest(fit, contrast=contrast_ixn[,1])

res_ixn_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,1])
# res1 <-  glmQLFTest(fit, coef = "Diet_ixnprotein:tolerance_1")
#res2 <-  glmQLFTest(fit, coef = "Diet_ixnlipid:tolerance_1")
# 
topTags_ixn_edgeR <- topTags(res_ixn_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_GO_tbl |>
              #mutate(gene = as.character(entrezgene_id)),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```


# Functional enrichments of genes Differentally Expressed in Interaction

## KEGG 
```{r}
gseKEGG_ixn <- topTags_ixn_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(geneList     = .,
               organism = 'scan',
               #universe = topTags_ixn$gene,
               #minGSSize    = 120,
               pvalueCutoff = 1,
               verbose      = FALSE)

# enrichplot::upsetplot(gseKEGG_ixn)

enrichKEGG_ixn_lipid_DE <- clusterProfiler::enrichKEGG(gene = filter(topTags_ixn_edgeR, logFC < 0 & FDR < 0.1) |> pull(gene), organism = 'scan', pvalueCutoff = 1)

enrichKEGG_ixn_prot_DE <- clusterProfiler::enrichKEGG(gene = filter(topTags_ixn_edgeR, logFC > 0 & FDR < 0.1) |> pull(gene), organism = 'scan', pvalueCutoff = 1)

# enrichplot::upsetplot(enrichKEGG_ixn_lipid_DE)
# enrichplot::upsetplot(enrichKEGG_ixn_prot_DE)
```


```{r}
enrichGO(gene         = 
  topTags_ixn_edgeR  %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  filter(FDR < 0.05) %>%
  # filter(logFC > 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene)
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() #%>%
  #View()
```




# lipid vs protein
```{r}

res_diet_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,2])



topTags_diet_edgeR <- topTags(res_diet_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_info |>
              #mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)


res_diet_uninf_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,6])


topTags_diet_uninf_edgeR <- topTags(res_diet_uninf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_info |>
              #mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)

```



# infection
```{r}
res_inf_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,3])


topTags_inf_edgeR <- topTags(res_inf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_info |>
              #mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```




See infection response, one contrast per diet.
```{r}
# protein bird response to infection
res_protein_inf_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,4])


topTags_protein_inf_edgeR <- topTags(res_protein_inf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_info |>
              #mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)


# Lipid bird infection response
# protein bird response to infection
res_lipid_inf_edgeR <- glmQLFTest(fit, contrast=contrast_ixn[,5])


topTags_lipid_inf_edgeR <- topTags(res_lipid_inf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  #mutate(logFC=round(logFC,2)) %>%
  # mutate(across(where(is.numeric), signif, 3)) %>%
  # mutate_if(is.numeric, signif, 3) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
    #gene_info |>
              #mutate(gene = str_replace(gene, "^LOC", "")),
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)



gseGO_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = ., 
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseKEGG_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
               #universe = topTags_ixn$gene,
               #minGSSize    = 120,
               pvalueCutoff = 1,
               verbose      = FALSE)

gseKEGG_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
               #universe = topTags_ixn$gene,
               #minGSSize    = 120,
               pvalueCutoff = 1,
               verbose      = FALSE
  )
```



## Supplement: Correlation of logFC between protein and lipid
```{r}
left_join(topTags_protein_inf_edgeR,topTags_lipid_inf_edgeR, by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join( topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  left_join(topTags_diet_edgeR, by=c("gene","gene_name", "description"), suffix = c(".inf", ".diet")) %>%
  mutate(logFC_diff = logFC.protein - logFC.lipid) %>%# View()
  #arrange(desc(logFC_diff)) %>%  #View()
  filter(FDR.protein < 0.05 | FDR.lipid < 0.05) %>%
  dplyr::select(#gene_name, description, gene,
                logFC.protein, logFC.lipid, #logFC_diff, 
                logFC.inf, logFC.diet) %>%
  cor() %>%
  # corrr::correlate() %>%
  corrplot::corrplot()
  

protvslipid_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR, topTags_lipid_inf_edgeR, topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  filter(FDR.protein < 0.05 | FDR.lipid < 0.05) %>% #View()
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC.protein, y=logFC.lipid)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm")+
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  # xlim(-2,2) +
  # ylim(-2,2) +
  theme_classic() +
  ggtitle("Cor(logFC) protein vs lipid infection")

protvsall_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR, topTags_lipid_inf_edgeR, topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  filter(FDR.protein < 0.05 | FDR.lipid < 0.05) %>% 
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC.protein, y=logFC)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm")+
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ggtitle("Cor(logFC) protein vs combined infection")

lipidvsall_inf_logfc_gg <- left_join(topTags_protein_inf_edgeR, topTags_lipid_inf_edgeR, topTags_inf_edgeR, 
          by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
  filter(FDR.protein < 0.05 | FDR.lipid < 0.05) %>%
  # pivot_longer(cols = c(logFC.protein, logFC.lipid), names_to = "Diet", values_to = "logFC") %>% View()
  ggplot(aes(x=logFC.lipid, y=logFC)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 1.5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2) +
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ggtitle("Cor(logFC) lipid vs combined infection")


ggpubr::ggarrange(protvslipid_inf_logfc_gg, protvsall_inf_logfc_gg, 
          lipidvsall_inf_logfc_gg, ncol = 2, nrow = 2)

ggsave("figures/inf_logfc_correlation.png", width = 10, height = 10, dpi = 600)
```

## Correlation between FC estimates for diet and infection

```{r}
left_join(topTags_inf_edgeR, topTags_diet_uninf_edgeR,
# left_join(topTags_lipid_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  filter(FDR.inf < 0.05 & FDR.diet < 0.05) %>% #View()
  ggplot(aes(x=logFC.diet, y=logFC.inf)) +
  geom_point(alpha = 0.5) +
  # ggrepel::geom_label_repel(aes(label=gene_name), box.padding = 0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 1.5, size = 5) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2, size = 5) +
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ggtitle("Cor(logFC) diet vs combined infection (both FDR < 0.05)")+ 
  theme(axis.text = element_text(size = 14), axis.title=element_text(size=16)) #+
  # theme_classic()

ggsave("figures/DEGenesBothCorrInfDietLogFC.png", width=16, height=16, units = "cm", dpi=1200)

left_join(topTags_lipid_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
    left_join(topTags_inf_edgeR,by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 & FDR.diet < 0.05) %>% #View()
  # filter(FDR < 0.05 & FDR.diet < 0.05) %>%
  # filter(FDR.inf < 0.05 & FDR.diet < 0.05) %>% #View()
  ggplot(aes(x=logFC.diet, y=logFC.inf)) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -3, label.y = 1.5, size = 5) +
  ggpubr::stat_regline_equation(label.x = -3, label.y = 2, size = 5) +
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  ggrepel::geom_text_repel(aes(label=gene_name), box.padding = 0.5) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ggtitle("Cor(logFC) diet vs lipid infection (both FDR < 0.05)")+ 
  theme(axis.text = element_text(size = 14), axis.title=element_text(size=16)) #+
  # theme_classic()

ggsave("figures/DEGenesBothCorrlipInfDietLogFC.png", width=16, height=16, units = "cm", dpi=1200)

left_join(topTags_protein_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  left_join(topTags_inf_edgeR,by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 & FDR.diet < 0.05) %>% #View()
  # filter(FDR < 0.05 & FDR.diet < 0.05) %>%
  ggplot(aes(x=logFC.diet, y=logFC.inf)) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = 0, label.y = 2.5, size = 5) +
  ggpubr::stat_regline_equation(label.x = 0, label.y = 2, size = 5) +
  ggrepel::geom_text_repel(aes(label=gene_name), box.padding = 0.5) +
  geom_rug(col=rgb(.5,0,0,alpha=.2)) +
  theme_classic() +
  # xlim(-2,2) +
  # ylim(-2,2) +
  ggtitle("Cor(logFC) diet vs protein infection (both FDR < 0.05)")+ 
  theme(axis.text = element_text(size = 14), axis.title=element_text(size=16)) #+
  # theme_classic()

ggsave("figures/DEGenesBothCorrprotInfDietLogFC.png", width=16, height=16, units = "cm", dpi=1200)


```

```{r}
left_join(topTags_inf_edgeR, topTags_diet_uninf_edgeR, #topTags_inf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  left_join(topTags_ixn_edgeR) %>% 
  filter(FDR.inf > 0.05 & FDR.diet < 0.05 & FDR < 0.05) %>%
  View()
```

```{r}
# gseGO_compareDietFits_inf <- left_join(topTags_protein_inf_edgeR,topTags_lipid_inf_edgeR, by=c("gene","gene_name", "description"),suffix = c(".protein", ".lipid")) %>%
#   left_join( topTags_inf_edgeR, by=c("gene","gene_name", "description")) %>%
#   mutate(logFC_diff = logFC.protein - logFC.lipid) %>%
#   arrange(desc(logFC_diff)) %>%
#   pull(logFC_diff, gene) %>%
#   gseGO(
#     geneList     = .,
#     OrgDb         = orgdb,
#     keyType       = 'ENTREZID',
#     ont           = "BP",
#     pAdjustMethod = "BH",
#     pvalueCutoff  = 1
#   )
```


# Phenotype Correlation

Here, we will look at the correlation between the logFC of the genes in the infection and diet comparisons and the tolerance score we have calculated based on eyescores and log loads of pathogens detected.




## Tolerance
```{r}
# get just the infected birds
tolerance_inf <- metadata_phenotypes_ixn %>%
  # filter(MG.active == "YES") %>%
  filter(logLoad > 0) %>%
  # pull(tolerance_median)
  pull(tolerance_loadoverES)

diet_inf <- metadata_phenotypes_ixn %>%
  # filter(MG.active == "YES") %>%
  filter(logLoad > 0) %>%
  pull(Diet)
#
# diet.tolerance.inf_ <- metadata_phenotypes_ixn %>%
#   filter(MG.active == "YES") %>%
#   pull(diet.tolerance) %>%
#   as.factor()
design_tolerance <- model.matrix(~ 0 + diet_inf * tolerance_inf)
# design_tolerance_inf <- model.matrix(~ 0 + diet.tolerance.inf_)
#
gene_expression_matrix_inf <- gene_expression_matrix_ixn[, metadata_phenotypes_ixn$logLoad > 0]

y_tolerance <- DGEList(
  counts = gene_expression_matrix_inf,
            # gene_expression_matrix_inf, # to remove control birds
  # names= entrez_names_gene_expression_matrix_day0,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_tolerance <- as.numeric(round(cpm(10, mean(y_tolerance$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep_tolerance <- rowSums(cpm(y_tolerance) > cpm_cutoff_tolerance) >= 4
y_tolerance <- y_tolerance[keep_tolerance,]
summary(keep_tolerance)

y_tolerance <- calcNormFactors(y_tolerance)
y_tolerance <- estimateDisp(y_tolerance, design_tolerance)#, robust=TRUE)

fit_tolerance <- glmQLFit(y_tolerance, design_tolerance)#, robust=TRUE)

res_tolerance <-  glmQLFTest(fit_tolerance, coef = "tolerance_inf")#, contrast=contrast_ixn_tolerance[,1])

# res_tolerance_ixn <-  glmQLFTest(fit_tolerance, coef="Dietprotein:tolerance")
# res_tolerance_diet <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,2])
# res_tolerance_inf <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,3])


# tolerance ixn contrast
# topTags_tolerance_ixn <- topTags(res_tolerance_ixn, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()

# # tolerance diet contrast
# topTags_tolerance_diet <- topTags(res_tolerance_diet, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()

# tolerance inf contrast
topTags_tolerance <- topTags(res_tolerance, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```

```{r}
# diet_inf <- interaction(Diet_ixn, MG.Active) #%>% as.factor()
# design_ixn <- model.matrix(~ 0 + diet_inf) #+ Day_tolerance)
# design matrix
# Diet_1 <- as.factor(metadata_phenotypes_day0$Diet)
# Day_tolerance <- (metadata_phenotypes_day0$Day)

# above design works well, but hard to make contrasts. use pseudogroupings instead
# generate groupings
tolerance <- metadata_phenotypes_ixn$tolerance_loadoverES#_median #%>% as.factor()
# tolerance <- metadata_phenotypes_ixn$tolerance
# tolerance <- metadata_phenotypes_ixn$tolerance_cat #%>% as.factor()

# tolerance <- metadata_phenotypes_ixn$tolerance/max(
#   metadata_phenotypes_ixn$tolerance) 

# tolerance_cat <- metadata_phenotypes_ixn$tolerance_cat
# diet.tolerance_ <- as.factor(metadata_phenotypes_ixn$diet.tolerance)
# # design_tolerance <- model.matrix(~ 0 + Diet_ixn + tolerance)
# # diet.tolerance_ <- diet.tolerance.inf_
# 
# design_tolerance <- model.matrix(~ 0 + diet.tolerance_)
# 
# # replace with the one above
# # design_tolerance <- design_tolerance_inf
# 
# #rownames(gene_expression_matrix_day0) <- entrez_names_gene_expression_matrix_day0
# 
# y_tolerance <- DGEList(
#   counts = gene_expression_matrix_ixn,
#             # gene_expression_matrix_inf, # to remove control birds
#   # names= entrez_names_gene_expression_matrix_day0,
#   # group=Active_1,
#   remove.zeros = TRUE
# )
# 
# #determine cpm cutoff
# cpm_cutoff_tolerance <- as.numeric(round(cpm(10, mean(y_tolerance$samples$lib.size)),1))
# # cpm_cutoff <- 1
# # apply cutoff
# keep_tolerance <- rowSums(cpm(y_tolerance) > cpm_cutoff_tolerance) >= 2#4
# y_tolerance <- y_tolerance[keep_tolerance,]
# summary(keep_tolerance)
# 
# y_tolerance <- calcNormFactors(y_tolerance)
# y_tolerance <- estimateDisp(y_tolerance, design_tolerance)#, robust=TRUE)
# 
# # if using all birds
# y_tolerance$samples$group <- diet.tolerance_#grouping_var
#   # interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) %>% as.factor()
# 
# # y_tolerance$samples$group <- #grouping_var
# #   interaction(Diet_1, metadata_phenotypes_day0$tolerance_cat) %>% as.factor()
# 
# contrast_ixn_tolerance <- makeContrasts(#design, #metadata,
#   contrasts = c(
#     # protein's resistant vs susceptible / lipids's resistant vs susceptible
#     "(diet.tolerance_protein.tolerant - diet.tolerance_protein.susceptible) - (diet.tolerance_lipid.tolerant - diet.tolerance_lipid.susceptible)",
#     # protein vs lipid (diet contrast)
#     "(diet.tolerance_protein.tolerant + diet.tolerance_protein.intermediate +  diet.tolerance_protein.susceptible)/3-(diet.tolerance_lipid.tolerant + diet.tolerance_lipid.intermediate + diet.tolerance_lipid.susceptible)/3",
#     #  tolerant vs susceptible (infection contrast)
#     "(diet.tolerance_protein.tolerant + diet.tolerance_lipid.tolerant)/2 - (diet.tolerance_protein.susceptible + diet.tolerance_lipid.susceptible)/2"
#                 #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
#                 #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
#                 ), levels=design_tolerance)
# 
# # contrast_ixn_tolerance <- makeContrasts(#design, #metadata,
# #   contrasts = c(
# #     # protein's resistant vs susceptible / lipids's resistant vs susceptible
# #     "(diet.tolerance.inf_protein.tolerant - diet.tolerance.inf_protein.susceptible) - (diet.tolerance.inf_lipid.tolerant - diet.tolerance.inf_lipid.susceptible)",
# #     # protein vs lipid (diet contrast)
# #     "(diet.tolerance.inf_protein.tolerant + diet.tolerance.inf_protein.intermediate +  diet.tolerance.inf_protein.susceptible)/3-(diet.tolerance.inf_lipid.tolerant + diet.tolerance.inf_lipid.intermediate + diet.tolerance.inf_lipid.susceptible)/3",
# #     #  tolerant vs susceptible (infection contrast)
# #     "(diet.tolerance.inf_protein.tolerant + diet.tolerance.inf_lipid.tolerant)/2 - (diet.tolerance.inf_protein.susceptible + diet.tolerance.inf_lipid.susceptible)/2"
# #                 #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
# #                 #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
# #                 ), levels=design_tolerance)
# 
# fit_tolerance <- glmQLFit(y_tolerance, design_tolerance)#, robust=TRUE)
# 
# # res_tolerance <-  glmQLFTest(fit_tolerance,coef = "tolerance")#, contrast=contrast_ixn_tolerance[,1])
# 
# res_tolerance_ixn <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,1])
# res_tolerance_diet <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,2])
# res_tolerance_inf <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,3])
# 
# 
# # tolerance ixn contrast
# topTags_tolerance_ixn <- topTags(res_tolerance_ixn, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()
# 
# # tolerance diet contrast
# topTags_tolerance_diet <- topTags(res_tolerance_diet, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()
# 
# # tolerance inf contrast
# topTags_tolerance_inf <- topTags(res_tolerance_inf, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()
```

TESTING: This worked out really well.
```{r}
# tolerance <- metadata_phenotypes_ixn$tolerance #%>% as.factor()
Diet <- metadata_phenotypes_ixn$Diet
# tolerance_01 <- scale(tolerance,center=min(tolerance),scale=diff(range(tolerance)))
############################
## TRY WITHOUT CATEGORIES
# design_tolerance_continuous <- model.matrix(~ 0 + Diet + tolerance_01)
design_tolerance_continuous <- model.matrix(~ 0 + Diet * tolerance)

y_tolerance <- DGEList(
  counts = gene_expression_matrix_ixn,
            # gene_expression_matrix_inf, # to remove control birds
  # names= entrez_names_gene_expression_matrix_day0,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_tolerance <- as.numeric(round(cpm(10, mean(y_tolerance$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep_tolerance <- rowSums(cpm(y_tolerance) > cpm_cutoff_tolerance) >= 12
y_tolerance <- y_tolerance[keep_tolerance,]
summary(keep_tolerance)

y_tolerance <- calcNormFactors(y_tolerance)
y_tolerance <- estimateDisp(y_tolerance, design_tolerance_continuous)#, robust=TRUE)

# if using all birds
# y_tolerance$samples$group <- Diet#grouping_var
  # interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) %>% as.factor()

# y_tolerance$samples$group <- #grouping_var
#   interaction(Diet_1, metadata_phenotypes_day0$tolerance_cat) %>% as.factor()

# contrast_ixn_tolerance <- makeContrasts(#design, #metadata,
#   contrasts = c(
#     # protein's resistant vs susceptible / lipids's resistant vs susceptible
#     "(diet.tolerance_protein.tolerant - diet.tolerance_protein.susceptible) - (diet.tolerance_lipid.tolerant - diet.tolerance_lipid.susceptible)",
#     # protein vs lipid (diet contrast)
#     "(diet.tolerance_protein.tolerant + diet.tolerance_protein.intermediate +  diet.tolerance_protein.susceptible)/3-(diet.tolerance_lipid.tolerant + diet.tolerance_lipid.intermediate + diet.tolerance_lipid.susceptible)/3",
#     #  tolerant vs susceptible (infection contrast)
#     "(diet.tolerance_protein.tolerant + diet.tolerance_lipid.tolerant)/2 - (diet.tolerance_protein.susceptible + diet.tolerance_lipid.susceptible)/2"
#                 #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
#                 #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
#                 ), levels=design_tolerance)

# contrast_ixn_tolerance <- makeContrasts(#design, #metadata,
#   contrasts = c(
#     # protein's resistant vs susceptible / lipids's resistant vs susceptible
#     "(diet.tolerance.inf_protein.tolerant - diet.tolerance.inf_protein.susceptible) - (diet.tolerance.inf_lipid.tolerant - diet.tolerance.inf_lipid.susceptible)",
#     # protein vs lipid (diet contrast)
#     "(diet.tolerance.inf_protein.tolerant + diet.tolerance.inf_protein.intermediate +  diet.tolerance.inf_protein.susceptible)/3-(diet.tolerance.inf_lipid.tolerant + diet.tolerance.inf_lipid.intermediate + diet.tolerance.inf_lipid.susceptible)/3",
#     #  tolerant vs susceptible (infection contrast)
#     "(diet.tolerance.inf_protein.tolerant + diet.tolerance.inf_lipid.tolerant)/2 - (diet.tolerance.inf_protein.susceptible + diet.tolerance.inf_lipid.susceptible)/2"
#                 #"diet_tolerancelipid.3 - diet_tolerancelipid.1",
#                 #"diet_toleranceprotein.3 - diet_tolerancelipid.3"
#                 ), levels=design_tolerance)

fit_tolerance <- glmQLFit(y_tolerance, design_tolerance_continuous)#, robust=TRUE)

res_tolerance <-  glmQLFTest(fit_tolerance, coef = "tolerance")#, contrast=contrast_ixn_tolerance[,1])

res_tolerance_ixn <-  glmQLFTest(fit_tolerance, coef="Dietprotein:tolerance")
# res_tolerance_diet <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,2])
# res_tolerance_inf <-  glmQLFTest(fit_tolerance, contrast=contrast_ixn_tolerance[,3])


# tolerance ixn contrast
topTags_tolerance_ixn <- topTags(res_tolerance_ixn, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct()

# # tolerance diet contrast
# topTags_tolerance_diet <- topTags(res_tolerance_diet, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()

# tolerance inf contrast
topTags_tolerance <- topTags(res_tolerance, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```

check correlation of logFC estimates between two analyses
```{r}
left_join(topTags_inf_edgeR, topTags_tolerance, by = c("gene", "gene_name", "description"),suffix = c("_edgeR", "_tolerance")) %>% #View()
  filter(FDR_edgeR < 0.05 | FDR_tolerance < 0.05) %>% #View()
  ggplot(aes(x = logFC_edgeR, y = logFC_tolerance)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_cor(method = "pearson",) +
  ggpubr::stat_regline_equation(label.x = -1, label.y = 0.1)+
  # xlim(-8,8) +
  # ylim(-8,8) +
  #geom_label() +
  theme_minimal()

left_join(topTags_ixn_edgeR, topTags_tolerance_ixn, by = c("gene", "gene_name", "description"),suffix = c("_edgeR_ixn", "_tolerance_ixn")) %>% #View()
  filter(FDR_edgeR_ixn < 0.05 | FDR_tolerance_ixn < 0.05) %>%
  ggplot(aes(x = logFC_edgeR_ixn, y = logFC_tolerance_ixn)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_regline_equation(label.x = -5, label.y = 2)+
  ggpubr::stat_cor(method = "pearson") +
  #geom_label() +
  theme_minimal()


# left_join(topTags_inf_edgeR, topTags_ES_inf, by = c("gene", "gene_name", "description"),suffix = c("_edgeR_inf", "_ES_inf")) %>% #View()
#   filter(FDR_edgeR_inf < 0.05 | FDR_ES_inf < 0.05) %>%
#   ggplot(aes(x = logFC_edgeR_inf, y = logFC_ES_inf)) +
#   geom_point() +
#   ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
#   ggpubr::stat_regline_equation(label.x = -5, label.y = 2)+
#   ggpubr::stat_cor(method = "pearson") +
#   #geom_label() +
#   theme_minimal()
# 
# 
# 
# left_join(topTags_inf_edgeR, topTags_ES0_inf, by = c("gene", "gene_name", "description"),suffix = c("_edgeR_inf", "_ES0_inf")) %>% #View()
#   filter(FDR_edgeR_inf < 0.05 | FDR_ES0_inf < 0.05) %>%
#   ggplot(aes(x = logFC_edgeR_inf, y = logFC_ES0_inf)) +
#   geom_point() +
#   ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
#   ggpubr::stat_regline_equation(label.x = -5, label.y = 2)+
#   ggpubr::stat_cor(method = "pearson") +
#   #geom_label() +
#   theme_minimal()\
```
## Eye Score
```{r}
# ES_mm <- metadata_phenotypes_ixn$ES_minMax #%>% as.factor()
# tolerance_cat <- metadata_phenotypes_ixn$tolerance_cat
# diet.tolerance_ <- as.factor(metadata_phenotypes_ixn$diet.tolerance)
# design_tolerance <- model.matrix(~ 0 + Diet_ixn + tolerance)
# ES_minMax <- metadata_phenotypes_ixn$ES_day0_analysis
# ES_minMax <- metadata_phenotypes_ixn$r_ES_AUC_21

ES_minMax <- metadata_phenotypes_ixn$ES_minMax


design_ES <- model.matrix(~ 0 + ES_minMax * Diet)



#rownames(gene_expression_matrix_day0) <- entrez_names_gene_expression_matrix_day0

y_ES <- DGEList(
  counts = gene_expression_matrix_ixn,
  # names= entrez_names_gene_expression_matrix_day0,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_ES <- as.numeric(round(cpm(10, mean(y_ES$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep_ES <- rowSums(cpm(y_ES) > cpm_cutoff_ES) >= 12
y_ES <- y_ES[keep_ES,]
summary(keep_ES)

y_ES <- calcNormFactors(y_ES)
y_ES <- estimateDisp(y_ES, design_ES)#, robust=TRUE)

# y_ES$samples$group <- diet.tolerance_#grouping_var
  # interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) %>% as.factor()

# y_ES$samples$group <- #grouping_var
#   interaction(Diet_1, metadata_phenotypes_day0$tolerance_cat) %>% as.factor()

# contrast_ixn_ES <- makeContrasts(#design, #metadata, 
#   contrasts = c(
#     # protein's resistant vs susceptible / lipids's resistant vs susceptible
#     "(diet.ES_protein.tolerant - diet.ES_protein.susceptible) - (diet.ES_lipid.tolerant - diet.ES_lipid.susceptible)", 
#     # protein vs lipid (diet contrast)
#     "(diet.ES_protein.tolerant + diet.ES_protein.intermediate +  diet.ES_protein.susceptible)/3-(diet.ES_lipid.tolerant + diet.ES_lipid.intermediate + diet.ES_lipid.susceptible)/3",
#     #  tolerant vs susceptible (infection contrast)
#     "(diet.ES_protein.tolerant + diet.ES_lipid.tolerant)/2 - (diet.ES_protein.susceptible + diet.ES_lipid.susceptible)/2"
#                 #"diet_ESlipid.3 - diet_ESlipid.1",
#                 #"diet_ESprotein.3 - diet_ESlipid.3"
#                 ), levels=design_ES)

fit_ES <- glmQLFit(y_ES, design_ES)#, robust=TRUE)

# res_ES <-  glmQLFTest(fit_ES,coef = "ES")#, contrast=contrast_ixn_ES[,1])

res_ES_ixn <-  glmQLFTest(fit_ES, coef="ES_minMax:Dietprotein")
# res_ES_diet <-  glmQLFTest(fit_ES, contrast=contrast_ixn_ES[,2])
res_ES_inf <-  glmQLFTest(fit_ES, coef="ES_minMax")


# ES ixn contrast
topTags_ES_ixn <- topTags(res_ES_ixn, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct()
# 
# # ES diet contrast
# topTags_ES_diet <- topTags(res_ES_diet, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()

# ES inf contrast
topTags_ES_inf <- topTags(res_ES_inf, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```

```{r}
left_join(topTags_inf_edgeR, topTags_ES_inf, by = c("gene", "gene_name", "description"),suffix = c("_edgeR", "_ES")) %>% #View()
  filter(FDR_edgeR < 0.05 & FDR_ES < 0.05) %>% #View()
  ggplot(aes(x = logFC_edgeR, y = logFC_ES)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_cor(method = "pearson",) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  # xlim(-8,8) +
  # ylim(-8,8) +
  #geom_label() +
  theme_minimal()

left_join(topTags_ixn_edgeR, topTags_ES_ixn, by = c("gene", "gene_name", "description"),suffix = c("_edgeR_ixn", "_ES_ixn")) %>% #View()
  filter(FDR_edgeR_ixn < 0.05 | FDR_ES_ixn < 0.05) %>%
  ggplot(aes(x = logFC_edgeR_ixn, y = logFC_ES_ixn)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_regline_equation(label.x = -5, label.y = 2)+
  ggpubr::stat_cor(method = "pearson") +
  #geom_label() +
  theme_minimal()
```

## logLoad
```{r}
logLoad <- metadata_phenotypes_ixn$logLoad_alt
# logLoad <- metadata_phenotypes_ixn$logLoad_AUC_21/max(metadata_phenotypes_ixn$logLoad_AUC_21)
Diet <- metadata_phenotypes_ixn$Diet

design_logLoad <- model.matrix(~ 0 + logLoad * Diet)

#rownames(gene_expression_matrix_day0) <- entrez_names_gene_expression_matrix_day0

y_logLoad <- DGEList(
  counts = gene_expression_matrix_ixn,
  # names= entrez_names_gene_expression_matrix_day0,
  # group=Active_1,
  remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_logLoad <- as.numeric(round(cpm(10, mean(y_logLoad$samples$lib.size)),1))
# cpm_cutoff <- 1
# apply cutoff
keep_logLoad <- rowSums(cpm(y_logLoad) > cpm_cutoff_logLoad) >= 12
y_logLoad <- y_logLoad[keep_logLoad,]
summary(keep_logLoad)

y_logLoad <- calcNormFactors(y_logLoad)
y_logLoad <- estimateDisp(y_logLoad, design_logLoad)#, robust=TRUE)

# y_logLoad$samples$group <- diet.tolerance_#grouping_var
  # interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) %>% as.factor()

# y_logLoad$samples$group <- #grouping_var
#   interaction(Diet_1, metadata_phenotypes_day0$tolerance_cat) %>% as.factor()

# contrast_ixn_logLoad <- makeContrasts(#design, #metadata, 
#   contrasts = c(
#     # protein's resistant vs susceptible / lipids's resistant vs susceptible
#     "(diet.logLoad_protein.tolerant - diet.logLoad_protein.susceptible) - (diet.logLoad_lipid.tolerant - diet.logLoad_lipid.susceptible)", 
#     # protein vs lipid (diet contrast)
#     "(diet.logLoad_protein.tolerant + diet.logLoad_protein.intermediate +  diet.logLoad_protein.susceptible)/3-(diet.logLoad_lipid.tolerant + diet.logLoad_lipid.intermediate + diet.logLoad_lipid.susceptible)/3",
#     #  tolerant vs susceptible (infection contrast)
#     "(diet.logLoad_protein.tolerant + diet.logLoad_lipid.tolerant)/2 - (diet.logLoad_protein.susceptible + diet.logLoad_lipid.susceptible)/2"
#                 #"diet_logLoadlipid.3 - diet_logLoadlipid.1",
#                 #"diet_logLoadprotein.3 - diet_logLoadlipid.3"
#                 ), levels=design_logLoad)

fit_logLoad <- glmQLFit(y_logLoad, design_logLoad)#, robust=TRUE)

# res_logLoad <-  glmQLFTest(fit_logLoad,coef = "logLoad")#, contrast=contrast_ixn_logLoad[,1])

res_logLoad_ixn <-  glmQLFTest(fit_logLoad, coef="logLoad:Dietprotein")
# res_logLoad_ixn <-  glmQLFTest(fit_logLoad, contrast=contrast_ixn_logLoad[,1])
# res_logLoad_diet <-  glmQLFTest(fit_logLoad, contrast=contrast_ixn_logLoad[,2])
res_logLoad_inf <-  glmQLFTest(fit_logLoad, coef="logLoad")


# logLoad ixn contrast
topTags_logLoad_ixn <- topTags(res_logLoad_ixn, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct()
# 
# # logLoad diet contrast
# topTags_logLoad_diet <- topTags(res_logLoad_diet, n=Inf) %>% data.frame() %>%
#   arrange(FDR) %>%
#   rownames_to_column("gene") %>%
#   left_join(result_BM_merged,
#             by = "gene") %>%
#   dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
#       "PValue", "FDR", "description", "entrezgene_id") ) %>%
#   distinct()

# logLoad inf contrast
topTags_logLoad_inf2 <- topTags(res_logLoad_inf, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```


```{r}
left_join(topTags_inf_edgeR, topTags_logLoad_inf, by = c("gene", "gene_name", "description"),suffix = c("_edgeR", "_logLoad")) %>% #View()
  filter(FDR_edgeR < 0.05 | FDR_logLoad < 0.05) %>% #View()
  ggplot(aes(x = logFC_edgeR, y = logFC_logLoad)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_cor(method = "pearson",) +
  ggpubr::stat_regline_equation(label.x = -2, label.y = 2)+
  # xlim(-8,8) +
  # ylim(-8,8) +
  #geom_label() +
  theme_minimal()

left_join(topTags_ixn_edgeR, topTags_logLoad_ixn, by = c("gene", "gene_name", "description"),suffix = c("_edgeR_ixn", "_logLoad_ixn")) %>% #View()
  filter(FDR_edgeR_ixn < 0.05 | FDR_logLoad_ixn < 0.05) %>%
  ggplot(aes(x = logFC_edgeR_ixn, y = logFC_logLoad_ixn)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name), max.overlaps = 20) +
  ggpubr::stat_regline_equation(label.x = -5, label.y = 2)+
  ggpubr::stat_cor(method = "pearson") +
  #geom_label() +
  theme_minimal()
```


# Save DE text files
```{r}
### CONTINUE HERE>
write_tsv(topTags)
```



## summary stats DE
```{r}
decideTests(res_inf_edgeR, p.value = 0.05) %>% table()
decideTests(res_diet_edgeR, p.value = 0.05) %>% table()
decideTests(res_diet_uninf_edgeR, p.value = 0.05) %>% table()
decideTests(res_ixn_edgeR, p.value = 0.05) %>% table()

decideTests(res_protein_inf_edgeR, p.value = 0.05) %>% table()
decideTests(res_lipid_inf_edgeR, p.value = 0.05) %>% table()

decideTests(res_logLoad_inf, p.value = 0.05) %>% table()
decideTests(res_logLoad_ixn, p.value = 0.05) %>% table()
decideTests(res_ES_inf, p.value = 0.05) %>% table()
decideTests(res_ES_ixn, p.value = 0.05) %>% table()
decideTests(res_tolerance, p.value = 0.05) %>% table()
decideTests(res_tolerance_ixn, p.value = 0.05) %>% table()




```


## gseGO
```{r}
gseGO_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_diet <- topTags_diet_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
                keyType       = 'ENTREZID',
    # by="DOSE", nPerm=100,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_diet_uninf <- topTags_diet_uninf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
                keyType       = 'ENTREZID',
    # by="DOSE", nPerm=100,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )


gseGO_ixn <- topTags_ixn_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb, #minGSSize = 10, maxGSSize = 50,
                keyType       = 'ENTREZID',
    # by="DOSE", nPerm=100,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1#,
    # scoreType = "pos"
  )

gseGO_tolerance_inf <- topTags_tolerance %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    pvalueCutoff  = 1
  )

gseGO_ES <- topTags_ES_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    pvalueCutoff  = 1
  )

gseGO_logLoad <- topTags_logLoad_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    pvalueCutoff  = 1
  )


# simplify(gseGO_infection) %>%
#   ridgeplot(showCategory = 10)
# 
# simplify(gseGO_diet) %>%
#   ridgeplot(showCategory = 10)
# 
# simplify(gseGO_ixn) %>%
#   ridgeplot(showCategory = 10)
# 
# simplify(gseGO_tolerance_inf) %>%
#   ridgeplot(showCategory = 10)
# 
# simplify(gseGO_ES) %>%
#   ridgeplot(showCategory = 10)
# 
# simplify(gseGO_logLoad) %>%
#   ridgeplot(showCategory = 10)

```


```{r, fig.height=5}
gseaplot(gseGO_infection_protein, by = "all", 
         title = gseGO_infection_protein$Description[37], geneSetID = 37)
gseaplot(gseGO_infection_lipid, by = "all", 
         title = gseGO_infection_lipid$Description[280], geneSetID = 280)
gseaplot(gseGO_infection, by = "all", 
         title = gseGO_infection$Description[364], geneSetID = 364)


infection_ribosome_kegg_gene_list <- data.frame(
  #arrange(topTags_diet_edgeR,gene)$logFC,
  arrange(topTags_protein_inf_edgeR,gene)$logFC,
 arrange(topTags_lipid_inf_edgeR,gene)$logFC,                                               arrange(topTags_inf_edgeR,gene)$logFC
 ) %>% as.matrix(byrow = F)

rownames(infection_ribosome_kegg_gene_list) <- arrange(topTags_inf_edgeR,gene)$gene

# infection_ribosome_kegg_gene_list <- sort(infection_ribosome_kegg_gene_list,
#                                           decreasing = TRUE)

library(pathview)
ribo_inf <- pathview::pathview(gene.data = infection_ribosome_kegg_gene_list,
                pathway.id="scan03010",
                species = "scan")

oxphos_inf <- pathview::pathview(gene.data = infection_ribosome_kegg_gene_list,
                pathway.id="scan00190",
                species = "scan")

```



# Figure 0
An Upset plot comparing DE genes across all of the different comparisons we are looking at.
```{r}
genes_higher_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_lower_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_protein_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_lipid_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>% 
  deframe() 

genes_bigger_protein_infection_difference <- topTags_ixn_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>% #View()
  deframe() 

genes_bigger_lipid_infection_difference <- topTags_ixn_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>% #View()
  dplyr::select(gene, logFC) %>%# View()
  deframe() 

genes_higher_in_tolerance <- topTags_tolerance %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_higher_in_ES <- topTags_ES_inf %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_higher_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_lower_in_tolerance <- topTags_tolerance %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_lower_in_ES <- topTags_ES_inf %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_lower_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

fdr_genes_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_diet <- topTags_diet_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_ixn <- topTags_ixn_edgeR %>% 
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()
```

```{r, fig.height=9, fig.width=8, message=FALSE, warning=FALSE}
DE_genes_edgeR_upset <- topTags_inf_edgeR %>%
  #filter(FDR< 0.05) %>%
  #dplyr::select(gene) %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixn_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixn_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),
         higher_in_tolerance = gene %in% names(genes_higher_in_tolerance),
         higher_in_ES = gene %in% names(genes_higher_in_ES),
         higher_in_logLoad = gene %in% names(genes_higher_in_logLoad),
         lower_in_tolerance = gene %in% names(genes_lower_in_tolerance),
         lower_in_ES = gene %in% names(genes_lower_in_ES),
         lower_in_logLoad = gene %in% names(genes_lower_in_logLoad),
         # Here we use ifelse to assign logFC values from the named vector based on the gene presence.
        # logFC_repressed_in_infection = 
        #   ifelse(gene %in% names(genes_lower_in_infection), genes_lower_in_infection[gene], NA),
        # logFC_induced_in_infection = ifelse(gene %in% names(genes_higher_in_infection), genes_higher_in_infection[gene], NA),
        # 
        # logFC_higher_in_Protein = ifelse(gene %in% names(genes_higher_protein_diet), genes_higher_protein_diet[gene], NA),
        # logFC_higher_in_Lipid = ifelse(gene %in% names(genes_higher_lipid_diet), genes_higher_lipid_diet[gene], NA),
        # logFC_Protein_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_protein_infection_difference), genes_bigger_protein_infection_difference[gene], NA),
        # logFC_Lipid_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_lipid_infection_difference), genes_bigger_lipid_infection_difference[gene], NA)
        logFC_infection = 
          case_when(
            gene %in% names(genes_lower_in_infection) ~ genes_lower_in_infection[gene],
            gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
            TRUE ~ NA_real_
          ),
          # )
        # logFC_induced_in_infection = ifelse(gene %in% names(genes_higher_in_infection), genes_higher_in_infection[gene], NA),
        
        logFC_diet = case_when(
          gene %in% names(genes_higher_protein_diet) ~ genes_higher_protein_diet[gene],
          gene %in% names(genes_higher_lipid_diet) ~ genes_higher_lipid_diet[gene],
          TRUE ~ NA_real_
        ),
        # )
          # ifelse(gene %in% names(genes_higher_protein_diet), genes_higher_protein_diet[gene], NA),
        # logFC_higher_in_Lipid = ifelse(gene %in% names(genes_higher_lipid_diet), genes_higher_lipid_diet[gene], NA),
        logFC_interaction = case_when(
          gene %in% names(genes_bigger_protein_infection_difference) ~ genes_bigger_protein_infection_difference[gene],
          gene %in% names(genes_bigger_lipid_infection_difference) ~ genes_bigger_lipid_infection_difference[gene],
          TRUE ~ NA_real_
        ),
        FDR_infection = case_when(
          gene %in% names(fdr_genes_infection) ~ fdr_genes_infection[gene],
          #gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
          TRUE ~ NA_real_
        ),
        FDR_diet = case_when(
          gene %in% names(fdr_genes_diet) ~ fdr_genes_diet[gene],
          # gene %in% names(genes_higher_protein_diet) ~ filter(topTags_diet_edgeR, paste0(gene) == paste0(gene))$FDR,
          # gene %in% names(genes_higher_lipid_diet) ~ filter(topTags_diet_edgeR, paste0(gene) == paste0(gene))$FDR,
          TRUE ~ NA_real_
        ),
        FDR_interaction = case_when(
          # gene %in% ((topTags_ixn_edgeR |>
  # filter(FDR < 0.05) |> pull(gene))) ~ (filter(topTags_ixn_edgeR, paste0(gene) == paste0(gene))$FDR),
          gene %in% names(fdr_genes_ixn) ~ fdr_genes_ixn[gene],
          # gene %in% names(genes_bigger_protein_infection_difference) ~ filter(topTags_ixn_edgeR, gene == gene)$FDR,
          # gene %in% names(genes_bigger_lipid_infection_difference) ~ filter(topTags_ixn_edgeR, gene == gene)$FDR,
          TRUE ~ NA_real_
        ),
        logFC_tolerance = case_when(
          gene %in% names(genes_higher_in_tolerance) ~ genes_higher_in_tolerance[gene],
          gene %in% names(genes_lower_in_tolerance) ~ genes_lower_in_tolerance[gene],
          TRUE ~ NA_real_
        ),
        logFC_ES = case_when(
          gene %in% names(genes_higher_in_ES) ~ genes_higher_in_ES[gene],
          gene %in% names(genes_lower_in_ES) ~ genes_lower_in_ES[gene],
          TRUE ~ NA_real_
        ),
        logFC_logLoad = case_when(
          gene %in% names(genes_higher_in_logLoad) ~ genes_higher_in_logLoad[gene],
          gene %in% names(genes_lower_in_logLoad) ~ genes_lower_in_logLoad[gene],
          TRUE ~ NA_real_
        )
        # )
          # ifelse(gene %in% names(genes_bigger_protein_infection_difference), genes_bigger_protein_infection_difference[gene], NA),
        # logFC_Lipid_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_lipid_infection_difference), genes_bigger_lipid_infection_difference[gene], NA)
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixn = Protein_ixn_bigger_Inf | Lipid_ixn_bigger_Inf,
         DE_in_tolerance = higher_in_tolerance | lower_in_tolerance,
         DE_in_ES = higher_in_ES | lower_in_ES,
         DE_in_logLoad = higher_in_logLoad | lower_in_logLoad) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ))
  #dplyr::select(-"gene")




DE_genes_edgeR_upsetplot <- DE_genes_edgeR_upset %>%
  upset(., colnames(DE_genes_edgeR_upset[,c(9:14)]), name= "Comparison", width_ratio = 0.2,
        #mode="union",
        #intersections="all",
        annotations = list(
        # 1st method - passing list:
        # 'logFC'=list(
        #     aes=aes(x=intersection, y=logFC),
        #     # provide a list if you wish to add several geoms
        #     geom=geom_boxplot(na.rm=TRUE)
        # ),
        # 2nd method - using ggplot
        'Infection'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_infection))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(#aes(color=FDR),
                                           na.rm=TRUE)
            + geom_text(aes(label=gene_name, y = logFC_infection + 0.5 * sign(logFC_infection), 
                            label = logFC_infection), size=2, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            + geom_violin(alpha=0.5, na.rm=TRUE) #+
    # ggplot2::theme_bw()
            #+ geom_label(aes(label=gene_label),size=2, na.rm=TRUE)
        ),
        'Diet'=(
            ggplot(mapping=aes(y=logFC_diet))
            + ggbeeswarm::geom_quasirandom(#aes(color=FDR),
                                           na.rm=TRUE)
            + geom_text(aes(label=gene_name, y = logFC_diet + 0.5 * sign(logFC_diet), 
                            label = logFC_diet), size=2, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            + geom_violin(alpha=0.5, na.rm=TRUE) #+
        ),
        'Interaction'=(
            ggplot(mapping=aes(y=logFC_interaction))
            + ggbeeswarm::geom_quasirandom(#aes(color=FDR),
                                           na.rm=TRUE)
            + geom_text(aes(label=gene_name, y = logFC_interaction + 0.5 * sign(logFC_interaction), 
                            label = logFC_interaction), size=2, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            + geom_violin(alpha=0.5, na.rm=TRUE) #+
        )
    ),
        keep_empty_groups=FALSE, min_degree=1,# drop=TRUE,
    min_size=0#, #mode="inclusive_union"
    )# +
    # ggplot2::theme_bw()

# DE_genes_edgeR_upsetplot$theme$legend.text.align <- "left"
DE_genes_edgeR_upsetplot

ggsave("figures/DE_genes_edgeR_upsetplot.png", DE_genes_edgeR_upsetplot, width = 8, height = 8, units = "in", dpi = 300)










DE_genes_edgeR_upset %>%
  upset(., colnames(DE_genes_edgeR_upset[,c(30:32)]), name= "Comparison", width_ratio = 0.2,
        #mode="union",
        #intersections="all",
        base_annotations=list(
        'Intersection size'=intersection_size(text = list(size=5, vjust=-0.15),
            text_colors=c(
                on_background='black', on_bar='white'
            )
        )
        + annotate(
            geom='text', x=Inf, y=Inf,
            label=paste('Total:', nrow(DE_genes_edgeR_upset)),
            vjust=1.2, hjust=1.2
        )
        + ylab('# of genes')
    ),
        annotations = list(
        # 1st method - passing list:
        # 'Length'=list(
        #     aes=aes(x=intersection, y=logFC),
        #     # provide a list if you wish to add several geoms
        #     geom=geom_boxplot(na.rm=TRUE)
        # ),
        # 2nd method - using ggplot
        # 'logFC'=(
        #     # note that aes(x=intersection) is supplied by default and can be skipped
        #     ggplot(mapping=aes(y=logFC))
        #     # checkout ggbeeswarm::geom_quasirandom for better results!
        #     + ggbeeswarm::geom_quasirandom(#aes(color=FDR),
        #                                    na.rm=TRUE)
        #     # + geom_jitter(aes(color=log10(FDR)), na.rm=TRUE)
        #     #+ geom_line()
        #     + geom_violin(alpha=0.5, na.rm=TRUE)# +
    #ggplot2::theme_bw()()
            #+ geom_text(aes(label=gene_label),size=3, na.rm=TRUE)
        'Infection'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_infection))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.8,
                na.rm=TRUE)
            # + geom_text(aes(label=gene_name, y = logFC_infection + 0.5 * sign(logFC_infection), 
            #                 label = logFC_infection), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
    # ggplot2::theme_bw()
            #+ geom_label(aes(label=gene_label),size=2, na.rm=TRUE)
        ),
        'Diet'=(
            ggplot(mapping=aes(y=logFC_diet))
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.8,
                na.rm=TRUE)
            # + geom_text(aes(label=gene_name, y = logFC_diet + 0.5 * sign(logFC_diet), 
            #                 label = logFC_diet), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
        ),
        'Interaction'=(
            ggplot(mapping=aes(y=logFC_interaction))
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.8,
                na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
            # + geom_text(aes(label=gene_name, y = logFC_interaction + 0.5 * sign(logFC_interaction), 
            #                 label = logFC_interaction), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            
        )
        
    ),
    set_sizes	=F,
    themes=upset_default_themes(text=element_text(size=20),axis.title=element_text(size=20,face="bold")),
    # themes=upset_default_themes(text=element_text(size=rel(2)),axis.title=element_text(size=rel(2),face="bold")),
        keep_empty_groups=TRUE, min_degree=1) #+ theme(axis.text = element_text(size = 14))

ggsave("figures/DE_genes_edgeR_combined_upsetplot.png", width = 8, height = 9, units = "in", dpi = 1200)
```

Upset plot for phenotypes
```{r}
DE_genes_edgeR_upset %>%
  upset(., colnames(DE_genes_edgeR_upset[,c(30,33,34,35)]), name= "Comparison", width_ratio = 0.2,
        #mode="union",
        #intersections="all",
        annotations = list(
        'Infection'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_infection))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.2,
                na.rm=TRUE)
            # + geom_text(aes(label=gene_name, y = logFC_infection + 0.5 * sign(logFC_infection), 
            #                 label = logFC_infection), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
    # ggplot2::theme_bw()
            #+ geom_label(aes(label=gene_label),size=2, na.rm=TRUE)
        ),
    'Tolerance'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_tolerance))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.2,
                na.rm=TRUE)
            # + geom_text(aes(label=gene_name, y = logFC_infection + 0.5 * sign(logFC_infection), 
            #                 label = logFC_infection), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
    # ggplot2::theme_bw()
            #+ geom_label(aes(label=gene_label),size=2, na.rm=TRUE)
        ),
        'ES'=(
            ggplot(mapping=aes(y=logFC_ES))
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.2,
                na.rm=TRUE)
            # + geom_text(aes(label=gene_name, y = logFC_diet + 0.5 * sign(logFC_diet), 
            #                 label = logFC_diet), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
        ),
        'logLoad'=(
            ggplot(mapping=aes(y=logFC_logLoad))
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.5,
                na.rm=TRUE)
            # + geom_violin(alpha=0.5, na.rm=TRUE) #+
            # + geom_text(aes(label=gene_name, y = logFC_interaction + 0.5 * sign(logFC_interaction), 
            #                 label = logFC_interaction), size=0.8, na.rm=TRUE, check_overlap = TRUE)
            # + geom_jitter(#aes(color=FDR),
            #               na.rm=TRUE)
            
        )
        
    ),
        keep_empty_groups=TRUE, min_degree=1)#+
    # ggplot2::theme_bw()

ggsave("figures/DE_genes_edgeR_PHENO_combined_upsetplot.png", width = 8, height = 8, units = "in", dpi = 300)
```


Venn diagram
```{r}

# beautiful version made with eulerr.co
ggVennDiagram::ggVennDiagram(
  x = list(DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
           DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet],
           DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixn],
           tbl_tol_results %>% filter(FDR_tol < 0.05) %>% pull(gene.y)
           ),
  category.names = c("Infection", "Diet", "Interaction","ES")#,
  # category.col=c("red", "magenta", "blue"),
  # cat.col=c("red", "magenta", "blue"),
  # cat.fontface=2,
  # margin=0.05,
  # main="Venn Diagram of DE genes"
)
```

Genes that are DE in all 3 contrasts
```{r}
genes_de_all_3 <- DE_genes_edgeR_upset %>%
  filter(DE_in_infection & DE_in_diet & DE_in_ixn) %>%
  pull(gene)

DE_genes_edgeR_upset %>%
  filter(repressed_in_infection & higher_in_Protein) #%>% View()
```



## Genes that are DE (table)
```{r}
DE_gene_table <- DE_genes_edgeR_upset %>%
  filter(DE_in_infection | DE_in_diet | DE_in_ixn) %>%
  dplyr::select(-c(PValue,F,FDR,gene_label,logFC)) %>%
  # dplyr::select(c("gene","gene_name", "description", contains("logFC_"))) %>%
  mutate(number_DE_contrasts = rowSums(!is.na(dplyr::select(., starts_with("logFC"))))) %>%
  arrange(desc(number_DE_contrasts), desc(max(abs(logFC_infection), abs(logFC_diet), abs(logFC_interaction)))
  # desc((abs(logFC_infection)+abs(logFC_diet)+abs(logFC_interaction))*logCPM/(mean(abs(logFC_infection),abs(logFC_diet),abs(logFC_interaction), na.rm=T,trim=0)))
          ) %>%
  dplyr::relocate(c("gene_name", "description", "logFC_infection", "logFC_diet", "logFC_interaction", "FDR_infection", "FDR_diet", "FDR_interaction","gene","number_DE_contrasts")) #%>%

write_tsv(DE_gene_table, "DE_gene_table_all_contrasts.tsv")

```



# Figure 2 - Infection

## 2A: Volcano Plot for Infection

helper functions.
```{r plot-DE-cleanerCode}
sig.cutoff <- 0.05
logFC.cutoff <- 0

# Function to clean and mutate the data
clean_and_mutate_data <- function(data, sig.cutoff=0.05, logFC.cutoff=0.575) {
  data %>%
    mutate(delabel = case_when(
      !is.na(gene_name) ~ gene_name,
      TRUE ~ paste0("LOC",gene)
    )) %>%
    mutate(delabel = case_when(
      FDR < sig.cutoff ~ delabel,
      abs(logFC) > logFC.cutoff ~ delabel,
      TRUE ~ NA
    )) %>%
    mutate(diffexpressed = case_when(
      FDR > sig.cutoff ~ "NO",
      logFC < 0 ~ "DOWN",
      logFC > 0 ~ "UP"
    )) %>%
    mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP")))
}

# Function to create the plot
create_gene_plot <- function(data, title, sig.cutoff=0.05, logFC.cutoff=0.575, show.names=T) {
  ggplot(data, aes(x = logFC, y = -log10(FDR), col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual(values = c("black", "blue", "red")) +
    #geom_vline(xintercept = c(-logFC.cutoff, logFC.cutoff), col = "red") +
    #geom_hline(yintercept = -log10(sig.cutoff), col = "red") +
    {if (show.names) ggrepel::geom_text_repel(size=2, na.rm=TRUE)} +
      scale_size_area() +
    # ggsci::scale_color_aaas() +
      ggtitle(title) + 
      theme(legend.position="none")

      # scale_size_area() +
      # # ggsci::scale_color_aaas() +
      # ggtitle(title) + 
      # theme(legend.position="none")
    # }
    #ggrepel::geom_label_repel(size = 2) +
    # scale_size_area() +
    # # ggsci::scale_color_aaas() +
    # ggtitle(title) + 
    # theme(legend.position="none")
}
```

Create volcano plot
```{r}
## infection contrast
topTags_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Genes DE following MG Infection",show.names = F, 
                   logFC.cutoff = logFC.cutoff) + theme(text = element_text(size=18),
                                                        axis.text = element_text(size=18))
                    #theme(text = element_text(size=rel(20)))

ggsave(here("figures/volcano_topTags_inf_edgeR.png"), width = 10, height = 10, units = "cm", dpi=600)
# 
# # infected lipid birds
# infected_lipid_topTags %>%
#   clean_and_mutate_data() %>% 
#   create_gene_plot("Genes Differentially Expressed in infected lipid birds")
# 
# # infected protein birds
# infected_protein_topTags %>%
#   clean_and_mutate_data() %>%
#   create_gene_plot("Genes Differentially Expressed in infected protein birds")
# 
# ## dream diet contrast
# uninfected_both_topTags %>%
#   clean_and_mutate_data() %>%
#   mutate(diffexpressed = factor(diffexpressed, levels = c("N.S.", "Higher in Lipid", "Higher in Protein"))) %>%
#   create_gene_plot("Genes Differentially Expressed by Diet (in uninfected birds)")
# 
# ## phenotype (Eyescore)
# # pheno_topTable_ES %>%
# topTags_tolerance_ixn %>%
#   mutate(delabel = case_when(
#     stringr::str_length(external_gene_name) > 2 ~ external_gene_name,
#     TRUE ~ gene
#   )) %>%
#   mutate(adj.P.Val=FDR) %>%
#   mutate(delabel = case_when(
#     adj.P.Val > 0.05 ~ NA,
#     TRUE ~ delabel
#   )) %>%
#   mutate(diffexpressed = case_when(
#     adj.P.Val > 0.05 ~ "NO",
#     logFC < 0 ~ "DOWN",
#     logFC > 0 ~ "UP"
#   )) %>%
#   mutate(diffexpressed = factor(diffexpressed, levels = c("NO", "DOWN", "UP"))) %>% #View()
#   create_gene_plot("Genes Differentially Expressed According to Eyescore Tolerance Phenotype")
```


## 2B: Heatmap of INF DE genes.

Heatmap of inf DE genes d14vd0 (by sample)

```{r}
genes_DE_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  pull(gene) %>%
  unique()

tpm_df <- gene_expression_matrix %>%
  as_tibble(rownames="external_gene_name") %>%
  # head() %>%
      pivot_longer(
        !external_gene_name,
        names_to = c("Lane", "Bird", "MG.Diet", "Day"),
        names_sep = "_",
        values_to = "tpm"
      ) %>%
      separate(col = MG.Diet, c("MG", "Diet"), remove=FALSE) %>%
      left_join(#entrez_IDs |> dplyr::select(-go_id),
            result_BM_merged,
            by = "external_gene_name", multiple = "first") %>%
  mutate(entrezID = case_when(
    str_detect(external_gene_name, "^LOC") ~ str_replace(external_gene_name, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ external_gene_name
  )) %>%
      mutate(MG.Status = case_when(
        Day == 0 ~ "Uninfected",
        MG == "MG" ~ "Infected",
        MG == "Control" ~ "Uninfected",
        TRUE ~ "NA"
      ))
```




```{r}
df_heatmap_14_tpm <- tpm_df %>%
  filter(entrezID %in% c(genes_DE_infection)) %>%
  filter(Day %in% c(0,14)) %>%
  mutate(nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    !is.na(gene) ~ paste0("LOC",gene),
    TRUE ~ external_gene_name)) %>%
  dplyr::select(Bird, MG, Diet, Day, nice_gene_name,#external_gene_name, 
                tpm) %>%
  group_by(nice_gene_name, Bird) %>% #pull(entrezID) %>% unique() %>% length()
  mutate(log_tpm = log2(tpm+1)) %>%
  mutate(log_tpm = (tpm)) %>%
  # mutate(z = scale(log_tpm)) %>%
  filter(!is.na(tpm)) %>%
  dplyr::select(-tpm) %>%
  # pivot_wider(names_from = Day, values_from = tpm) %>% #View()
  pivot_wider(names_from = Day, values_from = log_tpm) %>% #View()
  group_by(nice_gene_name) %>%
  # mutate(tpm_change = `14` - median(`14`,na.rm=T)) %>%
  mutate(tpm_change = scale(`14`) - scale(`0`)) %>%
  # mutate(tpm_change = scale(tpm_change)) %>%
  # mutate(tpm_change_14v0 = (`14` - `0`)/ (`14`+ `0`)) %>%
  dplyr::select(-c(`0`,`14`, MG, Diet)) %>%
  # dplyr::select(-c(`0`, MG, Diet)) %>%
  # pivot_wider(., names_from = Bird, values_from = tpm_change_14v0) %>%
  pivot_wider(., names_from = Bird, values_from = tpm_change) %>%
  purrr::discard(~all(is.na(.))) %>%
  # replace any 0/0 with 0
  mutate(across(where(anyNA), ~ replace_na(., 0))) %>%
  column_to_rownames("nice_gene_name")
```


```{r, fig.width=6, fig.height=10}
cpm14_matrix <- as.matrix(df_heatmap_14_tpm)
# add MG_group
MG_group <- tpm_df %>%
  dplyr::select(MG,Bird) %>%
  unique() %>%
  filter(Bird %in% colnames(cpm14_matrix)) %>%
  arrange(factor(Bird,levels=colnames(cpm14_matrix))) %>%
  pull(MG, Bird)

pheno_tolerance <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  #filter(Day == 14) %>%
  dplyr::select(tolerance,Bird.ID) %>%
  group_by(Bird.ID) %>%
  summarise(tolerance = median(tolerance,na.rm=T)) %>%
  unique() %>%
  dplyr::filter(Bird.ID %in% colnames(cpm14_matrix)) %>%
  arrange(factor(Bird.ID,levels=colnames(cpm14_matrix))) %>%
  pull(tolerance, Bird.ID)

ES_pheno_heatmap_14 <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  filter(Day == 14) %>%
  dplyr::select(ES_day14,Bird.ID) %>%
  ungroup() %>%
  unique() %>%
  dplyr::filter(Bird.ID %in% colnames(cpm14_matrix)) %>%
  arrange(factor(Bird.ID,levels=colnames(cpm14_matrix))) %>%
  pull(ES_day14, Bird.ID)

logLoad_pheno_heatmap_14 <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  filter(Day == 14) %>%
  dplyr::select(logLoad_day14,Bird.ID) %>%
  ungroup() %>%
  unique() %>%
  dplyr::filter(Bird.ID %in% colnames(cpm14_matrix)) %>%
  arrange(factor(Bird.ID,levels=colnames(cpm14_matrix))) %>%
  pull(logLoad_day14, Bird.ID)

diet_heatmap_14 <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  filter(Day == 14) %>%
  dplyr::select(Diet,Bird.ID) %>%
  ungroup() %>%
  unique() %>%
  dplyr::filter(Bird.ID %in% colnames(cpm14_matrix)) %>%
  arrange(factor(Bird.ID,levels=colnames(cpm14_matrix))) %>%
  pull(Diet, Bird.ID)

# add metadata to top
column_ha = HeatmapAnnotation(Infection = MG_group, Diet = diet_heatmap_14, #tolerance = anno_barplot(pheno_tolerance),
                              EyeScore = anno_barplot(ES_pheno_heatmap_14), logLoad = anno_barplot(logLoad_pheno_heatmap_14),col = list(Infection = c("Control" = "grey", "MG" = "red"), tolerance = c("Resistant" = "cyan", "Susceptible" = "yellow")), annotation_legend_param = list(
        Infection = list(direction = "horizontal",nrow=1),
        Diet = list(direction = "horizontal",nrow=1)#,
        # EyeScore = list(direction = "horizontal",nrow=1),
        # logLoad = list(direction = "horizontal",nrow=1)
                              )
)

# Generate the heatmap with internal annotations
heatmap_cpm14 <- Heatmap(cpm14_matrix, 
                         name = "log2 change\nin cpm (day 14vs0)", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   heatmap_legend_param = list(direction = "horizontal"),
                   
                   cluster_columns = TRUE,
                   # column_split = MG_group,#
                   column_split = interaction(MG_group,diet_heatmap_14),
                   top_annotation = column_ha,
                   split=3, 
                   show_row_names = FALSE,
                   show_column_names = TRUE, 
                   # clustering_distance_rows = "pearson",
                   clustering_distance_rows = "pearson",
                   # clustering_distance_rows = "spearman",
                   clustering_method_rows = "ward.D2",
                   row_names_gp = gpar(fontsize = 6),
                   column_gap = unit(0.8, "mm"),
                   column_names_gp = gpar(fontsize = 6),
                   # column_names_rot = 0,
                   column_title_rot	=45
                   # column_names_centered = TRUE
                   )
# )

# Draw the heatmap
ht_cpm14 <- draw(heatmap_cpm14, heatmap_legend_side = "bottom", annotation_legend_side = "bottom",merge_legend = TRUE
                 )
ht_cpm14

#row_order(ht)
```

```{r}
png(file="figures/day14_cpm_change_heatmap.png", width=6, height=10, res=1200,units = "in")
ht_cpm14 <- draw(heatmap_cpm14, heatmap_legend_side = "bottom")
dev.off()
```

enrichment on clusters of genes
```{r}
list_GO_heatmap_inf <- list()
j=0
for (i in row_order(ht_cpm14)) {
  # print(i)
  genes_i <- rownames(heatmap_cpm14@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_heatmap_inf[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = genes_DE_infection,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) %>%
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  # mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% 
    data.frame()
  
  k <- j+length(row_order(ht_cpm14))
  
  list_GO_heatmap_inf[[k]] <- 
    logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_inf_edgeR$gene,
               pvalueCutoff = 0.5) %>% as.data.frame()
}
```



## Ontology enrichment plot on Inf DE genes

### GO enirchments

#### DE genes only

Only the genes higher in infection exhibited a significant enrichment in GO terms.
```{r}
# up in infection (DE GENES)
enrichGO_higher_in_inf_DE <- enrichGO(gene         = 
  topTags_inf_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # top_n(2000, logFC) %>%
  filter(FDR < 0.05) %>%
  filter(logFC > 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene) ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_inf_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Infected vs Uninfected",
         Direction = "Up",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

enrichGO_higher_in_inf_DE

# down in infection (DE GENES)
enrichGO_lower_in_inf_DE <- enrichGO(gene         = 
  topTags_inf_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # top_n(2000, logFC) %>%
  filter(FDR < 0.05) %>%
  filter(logFC < 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene),
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_inf_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Infected vs Uninfected",
         Direction = "Up",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

enrichGO_lower_in_inf_DE
```

#### Top 2000 genes by logFC

```{r}
# up in infection
enrichGO_higher_in_inf <- enrichGO(gene         = 
  topTags_inf_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
    
  # top_n(2000, logFC) %>% #View()
  
  filter(logFC>0) %>%
  top_n(2000, -FDR) %>%
  # filter(FDR < 0.1) %>%
  # filter(logFC > 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene) ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_inf_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Infected vs Uninfected",
         Direction = "Up",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

# down in infection
enrichGO_lower_in_inf <- enrichGO(gene         = 
  topTags_inf_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  filter(logFC<0) %>%
  # top_n(2000, -logFC) %>%
    top_n(2000, -FDR) %>%
  # filter(FDR < 0.05) %>%
  # filter(logFC < -0.1) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene)
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_inf_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # simplify(by="ID") |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Infected vs Uninfected",
         Direction = "Down",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()
  #View()
```
 
Create GO plots
```{r}
# up in infection
enrichGO_higher_in_inf_fig <- ggplot(enrichGO_higher_in_inf,
       showCategory = 15,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Enriched for genes induced following infection") +
  theme_classic()

# down in infection
enrichGO_lower_in_inf_fig <- ggplot(enrichGO_lower_in_inf,
       showCategory = 15,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Enriched for genes repressed following infection") +
  theme_classic()

ggpubr::ggarrange(enrichGO_higher_in_inf_fig, enrichGO_lower_in_inf_fig, ncol = 1, nrow = 2)


bind_rows(enrichGO_lower_in_inf, enrichGO_higher_in_inf) %>%
  arrange(desc(Description)) %>%
  group_by(geneID) %>%
  slice_head(n=1) %>%
  ungroup() %>%
  group_by(Direction) %>%
  #arrange(desc(richFactor)) %>%
  slice_max(richFactor, n=8) %>%
  ggplot(.,
       showCategory = 8,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  # facet_grid(rows=vars(Direction), scales = "free") +
  facet_wrap(~Direction, scales = "free", ncol=1) +
  ggtitle("Processes enriched for genes with large Fold Change following infection") +
  theme_classic() +
  theme(text = element_text(size = 20))

ggsave(here("figures/enrichGO_DE_in_inf_fig.png"), width = 20, height = 20, units = "cm", dpi=300)
  
```

## find KEGG enrichments
```{r}
# up in infection
enrichKEGG_higher_in_inf <- topTags_inf_edgeR  %>%
  # top_n(2000, logFC) %>%
  filter(logFC > 0) %>%
  filter(FDR < 0.05) %>%
  # slice_min(FDR, n=1000) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_inf_edgeR$gene) #%>% data.frame() %>% View()

# down in infection
enrichKEGG_lower_in_inf <- topTags_inf_edgeR  %>%
  # top_n(2000, -logFC) %>%
  filter(logFC < 0) %>%
  filter(FDR<0.05) %>%
  # slice_min(FDR, n=1000) %>%
    pull(gene) %>%
    enrichKEGG(gene = .,
               organism = 'scan', minGSSize = 5, maxGSSize=1000,
               universe = topTags_inf_edgeR$gene)

# either way
enrichKEGG_in_inf <- topTags_inf_edgeR  %>%
  top_n(4000, abs(logFC)) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_inf_edgeR$gene)

# gse enrichment
gseKEGG_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    by = "fgsea",
    exponent = 1,
    #minGSSize    = 120,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

# library(pathview)
topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  cnetplot(gseGO_infection, foldChange = ., 
           species = "scan",#node_label="gene", 
        cex_label_gene = 0.3,
        cex_label_pathway = 0.4) 
# pathview(pathway.id="scan03010", gene.data=., species="scan"#,
#          # list(gene=2.7))
# )

cnetplot(enrichGO_tol_ord, 
           # foldChange = ., 
           showCategory = 15,
           species = "scan",#node_label="gene",
           cex.params = list(gene_label = 0.3, category_label = 0.3),
           color.params = list(foldChange = tbl_tol_results %>%
                                 mutate(est=Estimate) %>%
  arrange(desc(est)) %>%
  pull(est, entrezgene_id))
  ) 


upsetplot_gseKEGG_infection <- enrichplot::upsetplot(gseKEGG_infection)

upsetplot_gseKEGG_infection

ggsave(plot = upsetplot_gseKEGG_infection, here("figures/upsetplot_gseKEGG_infection.png"), width = 30, height = 20, units = "cm", dpi=600)

# gseaplot(
#   gseKEGG_infection,
#   geneSetID = 1,
#   by = "preranked",
#   title = gseKEGG_infection$Description[1]
# )

# gse GO enrichment
gseGO_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
  OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )



dotplot(gseGO_infection, showCategory = 8, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign)

ridgeplot(gseKEGG_infection) + labs(x = "enrichment distribution")
```
remove me.
```{r}
par(cex = 0.5)
plotGOgraph(
  gseGO_infection, useFullNames = T, firstSigNodes = 8,
)
```


## 2C: Functional Enrichments of DE INF genes
```{r, fig.height=20, fig.width=20}
## add KEGG terms to GO plot
enrichGOKEGG_higher_in_inf_fig <-
  enrichKEGG_higher_in_inf %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Infected vs Uninfected",
    Direction = "Up",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_higher_in_inf %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Infected vs Uninfected",
        Direction = "Up",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("red", "#FFD6D7"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50)
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes upregulated in infection") +
  theme_classic() +
  theme(text = element_text(size = 18),
        legend.position='right'#,
        # legend.justification='top'#,
        # legend.direction='horizontal'
        )

# lower in inf GO and KEGG
enrichGOKEGG_lower_in_inf_fig <-
  enrichKEGG_lower_in_inf %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Infected vs Uninfected",
    Direction = "Down",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_lower_in_inf %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Infected vs Uninfected",
        Direction = "Down",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    # colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    colors = c("blue", "lightblue"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50),
    position = "right"
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes downregulated in infection") +
  scale_x_reverse() +
  theme_classic() +
  theme(text = element_text(size = 18),
        legend.position='left'#, 
        # legend.justification='top',
        # legend.direction='horizontal'
        )



enrichGOKEGG_higher_in_inf_fig
enrichGOKEGG_lower_in_inf_fig


ggpubr::ggarrange(enrichGOKEGG_lower_in_inf_fig, enrichGOKEGG_higher_in_inf_fig, ncol = 2, nrow = 1)

ggsave(here("figures/enrichGOKEGG_DE_in_inf_fig.png"), width = 40, height = 20, units = "cm", dpi=1200)


```

```{r}
cnetplot(enrichKEGG_in_inf,  
         showCategory=10,
         node_label="category",
         color.params=list(foldChange = #fold_change_geneList
                           setNames(topTags_inf_edgeR$logFC, topTags_inf_edgeR$gene)),
         cex.params=list(gene_label = 0.1,
                         category_label=0.5), 
         max.overlaps=100) +
  # change the color scale
  # scale_colour_gradientn(colours = rev(RColorBrewer::brewer.pal(n = 10, name = "RdYlBu"))#, limits=c(-6, 6)
                         # ) +
  labs(colour="logFC")
```

## optional volcano of immune genes

I think I will opt not to include this figure, unless desired.
```{r}
go_id <- "GO:0006955" # Immune Response GO term ID for biological_process
go_id <- "GO:0006629" # lipid metabolism

go_ids <- c("GO:0006955",# immune response
            "GO:0006952", # defense response
            "GO:0002376", # immune system process
            "GO:0006954" # inflammatory response
            )

immune_genes <- select(orgdb, keys = go_ids, columns = c("SYMBOL", "GENENAME"), keytype = "GO")

# create volcano plot
topTags_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  filter(gene_name %in% immune_genes$SYMBOL) %>%
  create_gene_plot("Genes Differentially Expressed (infected / uninfected contrast)", 
                   logFC.cutoff = logFC.cutoff)

ggsave(here("figures/volcano_immunegenes_inf_edgeR.png"), width = 10, height = 10, units = "cm", dpi=600)
```

## ribosome genes
```{r}
ribo_genes <- AnnotationDbi::select(orgdb, keys = "GO:0005840", columns = c("SYMBOL", "GENENAME"), keytype = "GO")
```


# Figure 3: Diet

## 3A: Volcano plot of diet DE genes

```{r}
topTags_diet_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Effect of Diet on Gene Expression (P/L)", show.names = FALSE,
                   logFC.cutoff = logFC.cutoff)

ggsave(here("figures/volcano_topTags_diet_edgeR.png"), width = 10, height = 10, units = "cm", dpi=600)
```


## 3B: Heatmap of DIET DE genes.

Heatmap of Diet DE genes (by sample)

```{r}
genes_DE_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05) %>%
  pull(gene) %>%
  unique()

# tpm_df <- gene_expression_matrix %>%
#   as_tibble(rownames="external_gene_name") %>%
#   # head() %>%
#       pivot_longer(
#         !external_gene_name,
#         names_to = c("Lane", "Bird", "MG.Diet", "Day"),
#         names_sep = "_",
#         values_to = "tpm"
#       ) %>%
#       separate(col = MG.Diet, c("MG", "Diet"), remove=FALSE) %>%
#       left_join(#entrez_IDs |> dplyr::select(-go_id),
#             result_BM_merged,
#             by = "external_gene_name", multiple = "first") %>%
#   mutate(entrezID = case_when(
#     str_detect(external_gene_name, "^LOC") ~ str_replace(external_gene_name, "^LOC", ""),
#     !is.na(entrezgene_id) ~ as.character(entrezgene_id),
#     TRUE ~ external_gene_name
#   )) %>%
#       mutate(MG.Status = case_when(
#         Day == 0 ~ "Uninfected",
#         MG == "MG" ~ "Infected",
#         MG == "Control" ~ "Uninfected",
#         TRUE ~ "NA"
#       ))
```




```{r}
df_heatmap_diet_tpm <- tpm_df %>%
  filter(#entrezID 
         entrezgene_id %in% c(genes_DE_diet)) %>%
  # filter(Day %in% c(0,14)) %>%
  mutate(nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    !is.na(gene) ~ paste0("LOC",gene),
    TRUE ~ external_gene_name)) %>%
  dplyr::select(Bird, MG, Diet, Day, nice_gene_name,#external_gene_name, 
                tpm) %>%
  group_by(nice_gene_name, Bird, Day) %>% #pull(entrezID) %>% unique() %>% length()
  mutate(log_tpm = log2(tpm+1)) %>%
  # mutate(log_tpm = (tpm)) %>%
  filter(!is.na(tpm)) %>%
  dplyr::select(-tpm) %>%
  # pivot_wider(names_from = Day, values_from = tpm) %>% #View()
  # pivot_wider(names_from = Day, values_from = log_tpm) %>% #View()
  group_by(nice_gene_name) %>%
  mutate(tpm_change = log_tpm - median(log_tpm,na.rm=T)) %>%
  # mutate(tpm_change = `14` - `0`) %>%
  # mutate(tpm_change_14v0 = (`14` - `0`)/ (`14`+ `0`)) %>%
  
  
  
  dplyr::select(-c(log_tpm,
                   MG, Diet)) %>%
  # dplyr::select(-c(`0`, MG, Diet)) %>%
  # pivot_wider(., names_from = Bird, values_from = tpm_change_14v0) %>%
  pivot_wider(., names_from = c(Bird,Day), values_from = tpm_change) %>%
  purrr::discard(~all(is.na(.))) %>%
  # replace any 0/0 with 0
  mutate(across(where(anyNA), ~ replace_na(., 0))) %>%
  column_to_rownames("nice_gene_name")
```


```{r fig.height=8, fig.width=8}
diet_heatmap_matrix <- as.matrix(df_heatmap_diet_tpm)

# add MG_group
MG_group_diet <- #tpm_df %>%
  # dplyr::select(MG,Bird) %>%
  metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(MG.Status,Bird.ID,Day) %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(diet_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(diet_heatmap_matrix))) %>%
  pull(MG.Status, Bird.ID)

pheno_tolerance_diet <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  #filter(Day == 14) %>%
  dplyr::select(tolerance,Bird.ID,Day) %>%
  group_by(Bird.ID, Day) %>%
  summarise(tolerance = median(tolerance,na.rm=T)) %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(diet_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(diet_heatmap_matrix))) %>%
  pull(tolerance, Bird.ID)

ES_pheno_heatmap_diet <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(ES_day14,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(diet_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(diet_heatmap_matrix))) %>%
  pull(ES_day14, Bird.ID)

logLoad_pheno_heatmap_diet <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(logLoad_day14,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(diet_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(diet_heatmap_matrix))) %>%
  pull(logLoad_day14, Bird.ID)

diet_heatmap_anno_diet <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(Diet,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(diet_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(diet_heatmap_matrix))) %>%
  pull(Diet, Bird.ID)

# add metadata to top
column_ha = HeatmapAnnotation(Infection = MG_group_diet, Diet = diet_heatmap_anno_diet, #tolerance = anno_barplot(pheno_tolerance),
                              EyeScore = anno_barplot(ES_pheno_heatmap_diet), logLoad = anno_barplot(logLoad_pheno_heatmap_diet),col = list(Infection = c("Control" = "grey", "MG" = "red"), tolerance = c("Resistant" = "cyan", "Susceptible" = "yellow")), annotation_legend_param = list(
        Infection = list(direction = "horizontal",nrow=1),
        Diet = list(direction = "horizontal",nrow=1)#,
        # EyeScore = list(direction = "horizontal",nrow=1),
        # logLoad = list(direction = "horizontal",nrow=1)
                              )
)

# Generate the heatmap with internal annotations
heatmap_diet <- Heatmap(diet_heatmap_matrix, 
                         name = "gene expression of DE genes in diet)", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   heatmap_legend_param = list(direction = "horizontal"),
                   cluster_columns = TRUE,
                   # column_split = MG_group,#
                   column_split = interaction(MG_group_diet,diet_heatmap_anno_diet),
                   top_annotation = column_ha,
                   split=3, 
                   show_row_names = FALSE,
                   show_column_names = TRUE, 
                   # clustering_distance_rows = "pearson",
                   clustering_distance_rows = "pearson",
                   # clustering_distance_rows = "spearman",
                   clustering_method_rows = "ward.D2",
                   row_names_gp = gpar(fontsize = 6),
                   column_gap = unit(0.8, "mm"),
                   column_names_gp = gpar(fontsize = 6),
                   # column_names_rot = 0,
                   column_title_rot	=45
                   # column_names_centered = TRUE
                   )
# )

# Draw the heatmap
ht_diet <- draw(heatmap_diet, heatmap_legend_side = "bottom", annotation_legend_side = "bottom",merge_legend = TRUE
                 )
ht_diet

#row_order(ht)
```

```{r}
png(file="figures/diet_heatmap.png", width=6, height=10, res=1200,units = "in")
ht_diet <- draw(heatmap_diet, heatmap_legend_side = "bottom")
dev.off()
```

enrichment on clusters of genes
```{r}
list_GO_heatmap_diet <- list()
j=0
for (i in row_order(ht_diet)) {
  # print(i)
  genes_i <- rownames(heatmap_diet@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_heatmap_diet[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = genes_DE_infection,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) %>%
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  # mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% 
    data.frame()
  
  k <- j+length(row_order(ht_diet))
  
  list_GO_heatmap_diet[[k]] <- 
    logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene,
               pvalueCutoff = 0.5) %>% as.data.frame()
  
}
```






## Ontology enrichment plot on diet DE genes

### GO enirchments
```{r}
# up in protein
enrichGO_higher_in_prot <- enrichGO(gene         = 
  topTags_diet_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  top_n(2000, logFC) %>%
  # filter(FDR < 0.5) %>%
  # filter(logFC < -0.1) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene) ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_diet_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Protein vs Lipid",
         Direction = "Up",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

# up in lipid
enrichGO_higher_in_lipid <- enrichGO(gene         = 
  topTags_diet_edgeR  %>%
  # arrange(desc(logFC)) %>%
  
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  top_n(2000, -logFC) %>%
  # filter(FDR < 0.5) %>%
  # filter(logFC < -0.1) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene)
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_diet_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # simplify(by="ID") |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "Protein vs lipid",
         Direction = "Down") %>% data.frame() #%>%
  #View()
```
 
Create GO plots
```{r}
# up in protein
enrichGO_higher_in_prot_fig <- ggplot(enrichGO_higher_in_prot,
       showCategory = 15,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Enriched for genes higher in Protein Diet") +
  theme_classic()

# up in lipid
enrichGO_higher_in_lipid_fig <- ggplot(enrichGO_higher_in_lipid,
       showCategory = 15,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Enriched for genes higher in Lipid Diet") +
  theme_classic()

ggpubr::ggarrange(enrichGO_higher_in_prot_fig, enrichGO_higher_in_lipid_fig, ncol = 1, nrow = 2)


bind_rows(enrichGO_higher_in_prot, enrichGO_higher_in_lipid) %>%
  arrange(desc(Description)) %>%
  group_by(geneID) %>%
  slice_head(n=1) %>%
  ungroup() %>%
  group_by(Direction) %>%
  #arrange(desc(richFactor)) %>%
  slice_max(richFactor, n=8) %>%
  ggplot(.,
       showCategory = 8,
       aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description)) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 50)) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  # facet_grid(rows=vars(Direction), scales = "free") +
  facet_wrap(~Direction, scales = "free", ncol=1) +
  ggtitle("Processes enriched for genes DE by Diet") +
  theme_classic()

ggsave(here("figures/enrichGO_DE_in_diet_fig.png"), width = 20, height = 20, units = "cm", dpi=300)
  
```

### KEGG enrichments
```{r}
# up in high protein diet
enrichKEGG_higher_in_prot <- topTags_diet_edgeR  %>%
  top_n(2000, logFC) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene)

# up in high fat diet
enrichKEGG_higher_in_lipid <- topTags_diet_edgeR  %>%
  top_n(2000, -logFC) %>%
    pull(gene) %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene)

# either way
enrichKEGG_in_diet <- topTags_diet_edgeR  %>%
  top_n(4000, abs(logFC)) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene)

# gse enrichment
gseKEGG_diet <- topTags_diet_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    pAdjustMethod = "BH",
    by = "fgsea",
    exponent = 1,
    #minGSSize    = 120,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

upsetplot_gseKEGG_diet <- enrichplot::upsetplot(gseKEGG_diet)

upsetplot_gseKEGG_diet

ggsave(plot = upsetplot_gseKEGG_diet, here("figures/upsetplot_gseKEGG_diet.png"), width = 30, height = 20, units = "cm", dpi=600)


# gse enrichment
gseKEGG_diet_uninf <- topTags_diet_uninf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    pAdjustMethod = "BH",
    by = "fgsea",
    exponent = 1,
    #minGSSize    = 120,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

# gse GO enrichment
gseGO_diet <- topTags_diet_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
                keyType       = 'ENTREZID',
    # by="DOSE", nPerm=100,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
  )

gseGO_diet %>%
  simplify() %>%
dotplot(., showCategory = 8, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign)

simplify(gseGO_diet, by = "p.adjust") %>%
  ridgeplot() + labs(x = "enrichment distribution")
# ridgeplot(gseGO_diet) + labs(x = "enrichment distribution")
```

### Enrichments for uninfected only diet
```{r}
# up in high protein diet
enrichKEGG_higher_in_prot_uninf <- topTags_diet_uninf_edgeR  %>%
  top_n(2000, logFC) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene)

# up in high fat diet
enrichKEGG_higher_in_lipid_uninf <- topTags_diet_uninf_edgeR  %>%
  top_n(2000, -logFC) %>%
    pull(gene) %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_diet_edgeR$gene)
```


##3C: Functional Enrichments
```{r}
## add KEGG terms to GO plot
enrichGOKEGG_higher_in_prot_fig <-
  enrichKEGG_higher_in_prot %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Protein vs Lipid",
    Direction = "Protein",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_higher_in_prot %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Protein vs Lipid",
        Direction = "Protein",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("red", "#FFD6D7"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50)
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes higher protein") +
  theme_classic()

# higher in lipid GO and KEGG
enrichGOKEGG_higher_in_lipid_fig <-
  enrichKEGG_higher_in_lipid %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Protein vs Lipid",
    Direction = "Lipid",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_higher_in_lipid %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Protein vs Lipid",
        Direction = "Lipid",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    # colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    colors = c("blue", "lightblue"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50),
    position = "right"
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes up in lipid") +
  scale_x_reverse() +
  theme_classic()



enrichGOKEGG_higher_in_prot_fig
enrichGOKEGG_higher_in_lipid_fig


ggpubr::ggarrange(enrichGOKEGG_higher_in_lipid_fig, enrichGOKEGG_higher_in_prot_fig, ncol = 2, nrow = 1)

ggsave(here("figures/enrichGOKEGG_DE_in_diet_fig.png"), width = 30, height = 15, units = "cm", dpi=1200)
```


## raw count diff by diet.
I'm skeptical of the results below, lets use tpm data instead and see what we get.
```{r read-data}
#assign location of tpm file for later analysis
filepath_rsem_gene_tpm <- here::here("SerCan2020", "rsem.merged.gene_tpm.tsv")

# read in rsem.merged.gene_counts.tsv
tpm_data <- data.table::fread(filepath_rsem_gene_tpm, sep="\t") %>% as_tibble()

tpm_data <- tpm_data %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>% 
  # dplyr::select(-contains("Blue-050")) %>%
  # change diet group of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       # c(`ES64_SEM19-1481_MG-lipid_21`="ES64_SEM19-1481_MG-protein_21",
       #   `ES52_SEM19-1481_MG-lipid_14`="ES52_SEM19-1481_MG-protein_14",
       #   `ES83_SEM19-1481_MG-lipid_0`="ES83_SEM19-1481_MG-protein_0"))
       # )
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       ) %>% #View()
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  # convert to matrix for edgeR
  as.matrix()
```

```{r}
topTags_lipid_inf_edgeR %>%
  left_join(., topTags_protein_inf_edgeR, by = c("gene","gene_name","description"),suffix = c("_lipid", "_protein")) %>%
  left_join(., topTags_inf_edgeR, by = c("gene","gene_name","description")) %>%
  pivot_longer(c(logFC_lipid, logFC_protein), names_to = "diet", values_to = "logFC_plot") %>%
  mutate(diet = 
           case_when(diet == "logFC_protein" ~ "protein",
                     diet == "logFC_lipid" ~ "lipid")) %>%
mutate(fdr = case_when(
  diet == "lipid" ~ FDR_lipid,
  diet == "protein" ~ FDR_protein
)) %>%
  filter(FDR_lipid < 0.05| FDR_protein < 0.05 | FDR < 0.05) %>%
  ggplot(aes(x=logFC_plot,  color = diet, fill = diet
             )) +
  geom_density(alpha=0.5#,bw=0.01,#,outline.type = "upper"
                  ) +
  # geom_histogram(aes(y=..density..), alpha=0.5, position="identity", bins=30) +
  xlim(-3, 3) +
  # facet_grid(rows=vars(diff_day), cols=vars(diet)) +
  # scale_y_sqrt() +
  ggtitle("2B") +
  xlab("logFC") +
  theme_classic(base_size = 24) +
    theme(legend.position = c(0.8,0.6))


ggsave(here("figures/logFC_density_DE_in_diet_fig.png"), width = 16, height = 16, units = "cm", dpi=1200)
```


## Supplemental figure: Difference in count_diffs by Diet
```{r}
# curios how densities are in y, in y$tolerance very different between diets.
y_ixn %>%
  cpm(.,log=FALSE) %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  #filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,SampID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  mutate(diff_21 = (day21) - (day0),
          diff_14 = (day14) - (day0)) %>%
  #pivot_longer(c("day14", "day21", "day0"), names_to = "day", values_to = "counts") %>% #View()
  pivot_longer(c("diff_14", diff_21), names_to = "diff_day", values_to = "counts_diff") %>% #View()
  mutate(counts_diff_norm = counts_diff/max(c(day0,day14,day21), na.rm=T)) %>%
  dplyr::select(gene,counts_diff,counts_diff_norm,diff_day, condition, diet) %>% 
  filter(diff_day == "diff_14") %>%
  distinct() %>%# View()
  ggplot(aes(x=counts_diff, color = condition, fill = condition
             )) +
  geom_density(alpha=0.5#,outline.type = "upper"
               ) +
  # xlim(-250,250) +
  xlim(-10,10) +
  facet_grid(rows=vars(diff_day), cols=vars(diet)) +
  # scale_y_sqrt() +
  theme_classic()



```


Functional enrichment of the fat tails on MG diff day 14 (from density plots)
```{r}
gene_count_diff_df <- y_ixn$counts %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%

  left_join(metadata, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,SampID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  mutate(diff_21 = (day21) - (day0),
          diff_14 = (day14) - (day0)) %>%
  pivot_longer(c("diff_14", diff_21), names_to = "diff_day", values_to = "counts_diff") %>% #View()
  mutate(counts_diff_norm = counts_diff/max(c(day0,day14,day21), na.rm=T)) %>%
  ungroup() %>%
  dplyr::select(gene,counts_diff,counts_diff_norm,diff_day, condition, diet) %>% 
  filter(diff_day == "diff_14") %>%
  distinct()
  
# enrichments for genes higher in lipid control in left side of tail
gene_count_diff_df %>% #View()
  filter(diet == "lipid") %>%
  filter(counts_diff < -100 & counts_diff > -250) %>%
  group_by(gene) %>%
  # summarise number of samples that a gene is found in MG vs control
  summarise(MG = sum(condition == "MG"),
            Control = sum(condition == "Control")) %>% #View()
  filter(Control - MG > 0) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 

# enrichments for genes higher in lipid MG in right side of tail
gene_count_diff_df %>% #View()
  filter(diet == "lipid") %>%
  filter(counts_diff < -100 & counts_diff > -250) %>%
  group_by(gene) %>%
  # summarise number of samples that a gene is found in MG vs control
  summarise(MG = sum(condition == "MG"),
            Control = sum(condition == "Control")) %>% #View()
  filter(Control - MG < 0) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 

# gene_count_diff_df %>%
#   filter(diet == "lipid" & condition == "MG") %>%
#   filter(as.numeric(counts_diff) > 100 & as.numeric(counts_diff) < 250) %>%
#   ungroup() %>% #View()
#   dplyr::select(gene) %>% 
#   distinct() %>%
#   pull(gene) %>%
#   enrichGO(gene = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 #universe      = unfiltered_genes_with_GO,
#                 ont           = "BP",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 0.01,
#                 qvalueCutoff  = 0.01) |>
#   # filter(ONTOLOGY %in% c("MF","BP")) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 


## Does this same pattern appear for protein birds?

# enrichments for genes higher in protein control in left side of tail
gene_count_diff_df %>%
  filter(diet == "protein" & condition == "Control") %>%
  filter(as.numeric(counts_diff) < -100 & as.numeric(counts_diff) > -250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 
  
# enrichments for genes higher in protein MG in right side of tail
gene_count_diff_df %>%
  filter(diet == "protein" & condition == "MG") %>%
  filter(as.numeric(counts_diff) > 100 & as.numeric(counts_diff) < 250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 
```

The pattern of enrichments appears to be conserved between protein and lipid infection responses, what if we look at the opposite?
```{r}

# enrichments for genes higher in lipid MG in left side of tail
gene_count_diff_df %>%
  filter(diet == "lipid" & condition == "MG") %>%
  filter(as.numeric(counts_diff) < -100 & as.numeric(counts_diff) > -250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 
  
# enrichments for genes higher in lipid Control in right side of tail
gene_count_diff_df %>%
  filter(diet == "lipid" & condition == "Control") %>%
  filter(as.numeric(counts_diff) > 100 & as.numeric(counts_diff) < 250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 


## Does this same pattern appear for protein birds?

# enrichments for genes higher in protein MG in left side of tail
gene_count_diff_df %>%
  filter(diet == "protein" & condition == "MG") %>%
  filter(as.numeric(counts_diff) < -100 & as.numeric(counts_diff) > -250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 
  
# enrichments for genes higher in protein control in right side of tail
gene_count_diff_df %>%
  filter(diet == "protein" & condition == "Control") %>%
  filter(as.numeric(counts_diff) > 100 & as.numeric(counts_diff) < 250) %>%
  ungroup() %>% #View()
  dplyr::select(gene) %>% 
  distinct() %>%
  pull(gene) %>%
  enrichGO(gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.01) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() 
```




# Figure 4: Interaction


```{r}
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  # mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  dplyr::select(logFC.tolerance, logFC.ES, logFC.logLoad, logFC) %>%
  cor()

# Corr Phenotype measures correlation ES vs logload vs tolerance
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  # mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  dplyr::select(logFC.tolerance, logFC.ES, logFC.logLoad, logFC) %>%
  ggpairs(mapping = aes(alpha=0.5)) +
  theme_classic() +
  ggtitle("ALL Genes: Correlation of Phenotypic LogFC estimates")

ggsave("figures/AllGenesCorrPhenotypeLogFC.png", width=16, height=16, units = "cm", dpi=1200)
```


## 4A: volcano ixn
```{r}
topTags_ixn_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Genes Differentially Expressed (protein inf response / lipid inf response)", show.names=FALSE, 
                   logFC.cutoff = logFC.cutoff
                   )


ggsave(here("figures/volcano_topTags_interaction_edgeR.png"), width = 10, height = 10, units = "cm", dpi=600)
```

```{r}
topTags_ixn_edgeR %>%
  clean_and_mutate_data() %>% 
  ggplot(., aes(x = logCPM, y = logFC, col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual(values = c("black", "blue", "red")) +
    #geom_vline(xintercept = c(-logFC.cutoff, logFC.cutoff), col = "red") +
    #geom_hline(yintercept = -log10(sig.cutoff), col = "red") +
    # ggrepel::geom_text_repel(size=2, na.rm=TRUE) +
      scale_size_area() +
    # ggsci::scale_color_aaas() +
      ggtitle(title) + 
      theme(legend.position="none")

topTags_lipid_inf_edgeR %>%
  clean_and_mutate_data() %>%
  ggplot(., aes(x = logCPM, y = logFC, col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual(values = c("black", "blue", "red")) +
    #geom_vline(xintercept = c(-logFC.cutoff, logFC.cutoff), col = "red") +
    #geom_hline(yintercept = -log10(sig.cutoff), col = "red") +
    # ggrepel::geom_text_repel(size=2, na.rm=TRUE) +
      # scale_size_area() +
    # ggsci::scale_color_aaas() +
      ggtitle("lipid only MG gene response") + 
      theme(legend.position="none")

topTags_protein_inf_edgeR %>%
  clean_and_mutate_data() %>%
  ggplot(., aes(x = logCPM, y = logFC, col = diffexpressed, 
                   label = delabel#, size=AveExpr
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual(values = c("black", "blue", "red")) +
    #geom_vline(xintercept = c(-logFC.cutoff, logFC.cutoff), col = "red") +
    #geom_hline(yintercept = -log10(sig.cutoff), col = "red") +
    # ggrepel::geom_text_repel(size=2, na.rm=TRUE) +
      # scale_size_area() +
    # ggsci::scale_color_aaas() +
      ggtitle("protein only MG gene response") + 
      theme(legend.position="none")
```


## 4B: Heatmap of Interaction DE genes.

Heatmap of Interaction DE genes (by sample)

```{r}
genes_DE_ixn <- topTags_ixn_edgeR %>%
  filter(FDR < 0.05) %>%
  pull(gene) %>%
  unique()
```




```{r}
df_heatmap_ixn_tpm <- tpm_df %>%
  filter(entrezID %in% c(genes_DE_ixn)) %>%
  # filter(Day %in% c(0,14)) %>%
  mutate(nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    !is.na(gene) ~ paste0("LOC",gene),
    TRUE ~ external_gene_name)) %>%
  dplyr::select(Bird, MG, Diet, Day, nice_gene_name,#external_gene_name, 
                tpm) %>%
  group_by(nice_gene_name, Bird, Day) %>% #pull(entrezID) %>% unique() %>% length()
  mutate(log_tpm = log2(tpm+1)) %>%
  # mutate(log_tpm = (tpm)) %>%
  filter(!is.na(tpm)) %>%
  dplyr::select(-tpm) %>%
  # pivot_wider(names_from = Day, values_from = tpm) %>% #View()
  # pivot_wider(names_from = Day, values_from = log_tpm) %>% #View()
  group_by(nice_gene_name) %>%
  mutate(tpm_change = log_tpm - median(log_tpm,na.rm=T)) %>%
  # mutate(tpm_change = `14` - `0`) %>%
  # mutate(tpm_change_14v0 = (`14` - `0`)/ (`14`+ `0`)) %>%
  dplyr::select(-c(log_tpm,
                   MG, Diet)) %>%
  # dplyr::select(-c(`0`, MG, Diet)) %>%
  # pivot_wider(., names_from = Bird, values_from = tpm_change_14v0) %>%
  pivot_wider(., names_from = c(Bird,Day), values_from = tpm_change) %>%
  purrr::discard(~all(is.na(.))) %>%
  # replace any 0/0 with 0
  mutate(across(where(anyNA), ~ replace_na(., 0))) %>%
  column_to_rownames("nice_gene_name")
```


```{r, fig.width=6, fig.height=10}
ixn_heatmap_matrix <- as.matrix(df_heatmap_ixn_tpm)

# add MG_group
MG_group_ixn <- #tpm_df %>%
  # dplyr::select(MG,Bird) %>%
  metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(MG.Status,Bird.ID,Day) %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(ixn_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(ixn_heatmap_matrix))) %>%
  pull(MG.Status, Bird.ID)

pheno_tolerance_ixn <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  #filter(Day == 14) %>%
  dplyr::select(tolerance,Bird.ID,Day) %>%
  group_by(Bird.ID, Day) %>%
  # summarise(tolerance = mean(tolerance_median,na.rm=T)) %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(ixn_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(ixn_heatmap_matrix))) %>%
  pull(tolerance, Bird.ID)

ES_pheno_heatmap_ixn <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(ES,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(ixn_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(ixn_heatmap_matrix))) %>%
  pull(ES, Bird.ID)

logLoad_pheno_heatmap_ixn <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(logLoad,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(ixn_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(ixn_heatmap_matrix))) %>%
  pull(logLoad, Bird.ID)

diet_heatmap_anno_ixn <- metadata_phenotypes_ixn %>%
  rownames_to_column("sample") %>%
  # filter(Day == 14) %>%
  dplyr::select(Diet,Bird.ID,Day) %>%
  ungroup() %>%
  unique() %>%
  mutate(Bird.ID_Day = paste0(Bird.ID,"_",Day)) %>%
  dplyr::filter(Bird.ID_Day %in% colnames(ixn_heatmap_matrix)) %>%
  arrange(factor(Bird.ID_Day,levels=colnames(ixn_heatmap_matrix))) %>%
  pull(Diet, Bird.ID)

# add metadata to top
column_ha_ixn = HeatmapAnnotation(Infection = MG_group_ixn, Diet = diet_heatmap_anno_ixn, #tolerance = anno_barplot(pheno_tolerance_ixn),
                              EyeScore = anno_barplot(ES_pheno_heatmap_ixn), logLoad = anno_barplot(logLoad_pheno_heatmap_ixn),col = list(Infection = c("Control" = "grey", "MG" = "red"), tolerance = c("Resistant" = "cyan", "Susceptible" = "yellow")), annotation_legend_param = list(
        Infection = list(direction = "horizontal",nrow=1),
        Diet = list(direction = "horizontal",nrow=1)#,
        # EyeScore = list(direction = "horizontal",nrow=1),
        # logLoad = list(direction = "horizontal",nrow=1)
                              )
)

# Generate the heatmap with internal annotations
heatmap_ixn <- Heatmap(ixn_heatmap_matrix, 
                         name = "gene expression of DE genes in interaction)", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   heatmap_legend_param = list(direction = "horizontal"),
                   cluster_columns = TRUE,
                   # column_split = MG_group,#
                   column_split = interaction(MG_group_ixn,diet_heatmap_anno_ixn),
                   top_annotation = column_ha_ixn,
                   split=3, 
                   show_row_names = FALSE,
                   show_column_names = TRUE, 
                   # clustering_distance_rows = "pearson",
                   clustering_distance_rows = "pearson",
                   # clustering_distance_rows = "spearman",
                   clustering_method_rows = "ward.D2",
                   row_names_gp = gpar(fontsize = 6),
                   column_gap = unit(0.8, "mm"),
                   column_names_gp = gpar(fontsize = 6),
                   # column_names_rot = 0,
                   column_title_rot	=45
                   # column_names_centered = TRUE
                   )
# )

# Draw the heatmap
ht_ixn <- draw(heatmap_ixn, heatmap_legend_side = "bottom", annotation_legend_side = "bottom", merge_legend = TRUE
                 )
ht_ixn

#row_order(ht)
```


```{r}
png(file="figures/interaction_heatmap.png", width=6, height=10, res=1200,units = "in")
ht_diet <- draw(heatmap_ixn, heatmap_legend_side = "bottom", annotation_legend_side = "bottom",)
dev.off()
```

enrichment on clusters of genes
```{r}
list_GO_heatmap_ixn <- list()
j=0
for (i in row_order(ht_ixn)) {
  # print(i)
  genes_i <- rownames(heatmap_ixn@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_heatmap_ixn[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = genes_DE_infection,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) %>%
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  # mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% 
    data.frame()
  
  k <- j+length(row_order(ht_ixn))
  
  list_GO_heatmap_ixn[[k]] <- 
    logFC_all_contrasts %>%
    filter(gene_name %in% genes_i | paste0("LOC",gene) %in% genes_i) %>%
    pull(gene)  %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_ixn_edgeR$gene,
               pvalueCutoff = 0.5) %>% as.data.frame()
  
}
```


## 4C: Functional Enrichments

Run enrichments:
```{r}
# GO enrichment
enrichGO_bigger_delta_in_prot <- enrichGO(gene         = 
  topTags_ixn_edgeR  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  top_n(2000, logFC) %>%
  # filter(FDR < 0.5) %>%
  # filter(logFC < -0.1) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene) ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_ixn_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "ixn",
         Direction = "Prot",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

# up in lipid
enrichGO_bigger_delta_in_lipid <- enrichGO(gene         = 
  topTags_ixn_edgeR  %>%
  # arrange(desc(logFC)) %>%
  
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  top_n(2000, -logFC) %>%
  # filter(FDR < 0.05) %>%
  # filter(logFC < 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene)
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_ixn_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) |>
  # simplify(by="ID") |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "ixn",
         Direction = "Lipid") %>% data.frame()



##################
# KEGG enrichments
##################


# up in high protein ixn
enrichKEGG_bigger_delta_in_prot <- topTags_ixn_edgeR  %>%
  top_n(2000, logFC) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_ixn_edgeR$gene)

# up in high fat ixn
enrichKEGG_bigger_delta_in_lipid <- topTags_ixn_edgeR  %>%
  top_n(2000, -logFC) %>%
    pull(gene) %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_ixn_edgeR$gene)

# either way
# enrichKEGG_in_diet <- topTags_diet_edgeR  %>%
#   top_n(4000, abs(logFC)) %>%
#   pull(gene) %>%
#   enrichKEGG(gene = .,
#                organism = 'scan',
#                universe = topTags_diet_edgeR$gene)

# gse enrichment
gseKEGG_ixn <- topTags_ixn_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    by = "fgsea",
    exponent = 1,
    #minGSSize    = 120,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

# gse GO enrichment
gseGO_ixn <- topTags_ixn_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb, #minGSSize = 10,
                keyType       = 'ENTREZID',
    # by="DOSE", nPerm=100,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
  )

dotplot(gseGO_ixn, showCategory = 8, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign)

ridgeplot(gseGO_ixn) + labs(x = "enrichment distribution")


# simplify(gseGO_ixn) %>% data.frame() %>% View()
```


```{r}
## add KEGG terms to GO plot
enrichGOKEGG_bigger_delta_in_prot_fig <-
  enrichKEGG_bigger_delta_in_prot %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Interaction",
    Direction = "bigger_delta_protein",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_bigger_delta_in_prot %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Interaction",
        Direction = "bigger_delta_protein",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("red", "#FFD6D7"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50)
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes bigger change protein") +
  theme_classic()

# higher in lipid GO and KEGG
enrichGOKEGG_bigger_delta_in_lipid_fig <-
  enrichKEGG_bigger_delta_in_lipid %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Interaction",
    Direction = "bigger_delta_lipid",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_bigger_delta_in_lipid %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Interaction",
        Direction = "bigger_delta_lipid",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    # colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    colors = c("blue", "lightblue"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50),
    position = "right"
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched for genes bigger change in lipid") +
  scale_x_reverse() +
  theme_classic()



enrichGOKEGG_bigger_delta_in_prot_fig
enrichGOKEGG_bigger_delta_in_lipid_fig


ggpubr::ggarrange(enrichGOKEGG_bigger_delta_in_lipid_fig, enrichGOKEGG_bigger_delta_in_prot_fig, ncol = 2, nrow = 1)

ggsave(here("figures/enrichGOKEGG_DE_in_ixn_fig.png"), width = 30, height = 15, units = "cm", dpi=1200)
```

 Table of Genes DE in all Contrasts
```{r}
# set_flextable_defaults(digits = 2)

# make pretty table of genes that are DE in all 3 contrasts
DE_all3_table <- DE_genes_edgeR_upset %>%
  filter(gene %in% genes_de_all_3) %>%
  mutate_if(is.numeric, round, 2) %>%
  dplyr::select(
    gene_name,
    description,
    logFC_infection,
    logFC_diet,
    logFC_interaction,
    FDR_infection,
    FDR_diet,
    FDR_interaction
  ) %>%
  flextable(
    col_keys = c(
      "gene_name",
      "description",
      "logFC_infection",
      "logFC_diet",
      "logFC_interaction"
    )
  ) %>%
  add_header_row(
    x = .,
    colwidths = c(2, 3),
    values = c(" ", "log2 fold change")
  ) %>%
  theme_vanilla(.) %>%
  align(.,
        i = 1,
        part = "header",
        align = "center") %>%
  add_footer_lines(
    .,
    "Positive values indicate higher expression in protein birds. Negative values indicate higher expression in lipid birds. FDR < 0.01 is bolded."
  ) %>%
  set_table_properties(align = "left", layout = "autofit") %>%
  italic(., j = 1) %>%
  bold(., ~ FDR_infection < 0.01, ~ logFC_infection, bold = TRUE) %>%
  labelizor(
    x = .,
    part = "header",
    labels = c(
      "gene_name" = "gene name",
      "logFC_infection" = "Infection",
      "logFC_diet" = "Diet",
      "logFC_interaction" = "Interaction"
    )
  ) %>%
  labelizor(x = .,
            part = "header",
            labels = stringr::str_to_title)


DE_all3_table

save_as_pptx(
  "DE_all3_table" = DE_all3_table,
  path = "tables/Genes_DE_all3_contrasts.pptx")
```









# Figure 5: Heatmap of DE genes in all contrasts
```{r, fig.height=10, fig.width=8}
logFC_all_contrasts <- topTags_inf_edgeR %>%
  arrange(gene) %>%
  #filter(FDR< 0.05) %>%
  #dplyr::select(gene) %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixn_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixn_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),
         # Here we use ifelse to assign logFC values from the named vector based on the gene presence.
        # logFC_repressed_in_infection = 
        #   ifelse(gene %in% names(genes_lower_in_infection), genes_lower_in_infection[gene], NA),
        # logFC_induced_in_infection = ifelse(gene %in% names(genes_higher_in_infection), genes_higher_in_infection[gene], NA),
        # 
        # logFC_higher_in_Protein = ifelse(gene %in% names(genes_higher_protein_diet), genes_higher_protein_diet[gene], NA),
        # logFC_higher_in_Lipid = ifelse(gene %in% names(genes_higher_lipid_diet), genes_higher_lipid_diet[gene], NA),
        # logFC_Protein_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_protein_infection_difference), genes_bigger_protein_infection_difference[gene], NA),
        # logFC_Lipid_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_lipid_infection_difference), genes_bigger_lipid_infection_difference[gene], NA)
        logFC_infection = topTags_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
          # case_when(
          #   gene %in% names(genes_lower_in_infection) ~ genes_lower_in_infection[gene],
          #   gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
          #   TRUE ~ NA_real_
          # ),
          # )
        # logFC_induced_in_infection = ifelse(gene %in% names(genes_higher_in_infection), genes_higher_in_infection[gene], NA),
        
        logFC_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        #case_when(
          # gene %in% names(genes_higher_protein_diet) ~ genes_higher_protein_diet[gene],
          # gene %in% names(genes_higher_lipid_diet) ~ genes_higher_lipid_diet[gene],
          # TRUE ~ NA_real_
        #),
        # )
          # ifelse(gene %in% names(genes_higher_protein_diet), genes_higher_protein_diet[gene], NA),
        # logFC_higher_in_Lipid = ifelse(gene %in% names(genes_higher_lipid_diet), genes_higher_lipid_diet[gene], NA),
        logFC_interaction = topTags_ixn_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
  #case_when(
        #   gene %in% names(genes_bigger_protein_infection_difference) ~ genes_bigger_protein_infection_difference[gene],
        #   gene %in% names(genes_bigger_lipid_infection_difference) ~ genes_bigger_lipid_infection_difference[gene],
        #   TRUE ~ NA_real_
        # ),
        FDR_infection = FDR, #case_when(
          # gene %in% names(fdr_genes_infection) ~ fdr_genes_infection[gene],
          #gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
          # TRUE ~ NA_real_
       # ),
        FDR_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
  # case_when(
  #         gene %in% names(fdr_genes_diet) ~ fdr_genes_diet[gene],
  #         # gene %in% names(genes_higher_protein_diet) ~ filter(topTags_diet_edgeR, paste0(gene) == paste0(gene))$FDR,
  #         # gene %in% names(genes_higher_lipid_diet) ~ filter(topTags_diet_edgeR, paste0(gene) == paste0(gene))$FDR,
  #         TRUE ~ NA_real_
  #       ),
        FDR_interaction = topTags_ixn_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
  # case_when(
  #         # gene %in% ((topTags_ixn_edgeR |>
  # # filter(FDR < 0.05) |> pull(gene))) ~ (filter(topTags_ixn_edgeR, paste0(gene) == paste0(gene))$FDR),
  #         gene %in% names(fdr_genes_ixn) ~ fdr_genes_ixn[gene],
  #         # gene %in% names(genes_bigger_protein_infection_difference) ~ filter(topTags_ixn_edgeR, gene == gene)$FDR,
  #         # gene %in% names(genes_bigger_lipid_infection_difference) ~ filter(topTags_ixn_edgeR, gene == gene)$FDR,
  #         TRUE ~ NA_real_
  #       )
        # )
          # ifelse(gene %in% names(genes_bigger_protein_infection_difference), genes_bigger_protein_infection_difference[gene], NA),
        # logFC_Lipid_ixn_bigger_Inf = ifelse(gene %in% names(genes_bigger_lipid_infection_difference), genes_bigger_lipid_infection_difference[gene], NA)
        logFC_ES = topTags_ES_inf |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        # logFC_tolerance = topTags_tolerance |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        FDR_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_ES = topTags_ES_inf |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(FDR)#,
        # FDR_tolerance = topTags_tolerance |> arrange(gene) |> filter(gene == gene) |> pull(FDR)
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixn = Protein_ixn_bigger_Inf | Lipid_ixn_bigger_Inf) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ),
  nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    TRUE ~ paste0("LOC",gene)
  ))

# Define significance level
sig.cutoff <- 0.05

df_heatmap <- logFC_all_contrasts %>%
  # filter(gene %in% genes_de_all_3) %>%
  filter(DE_in_infection | DE_in_diet | DE_in_ixn) %>%
  # filter(FDR_interaction < 0.05 & abs(logFC_interaction)>2 | FDR_infection < 0.05 & abs(logFC_infection)>1 | FDR_diet < 0.05 & abs(logFC_diet)>1 #| FDR_interaction < 0.05 & FDR_infection < 0.05  & FDR_diet < 0.05 
         # ) %>% #View()
  
  # filter(FDR_interaction < sig.cutoff & FDR_diet < sig.cutoff & (abs(logFC_interaction) > 1 | abs(logFC_diet) > 1)) %>%
  
  # filter(str_detect(gene_name, '^RPL|^RPS')) %>%
  # filter(gene_name %in% immune_genes$SYMBOL) %>%
  # filter(FDR_interaction < 0.05 &  FDR_infection < 0.05 & FDR_diet < 0.05) %>% #View()
  # slice_head(n=20) %>%
  # mutate_if(is.numeric, round, 2) %>%
  dplyr::select(
    gene,
    gene_name,
    nice_gene_name,
    description,
    logFC_infection,
    logFC_diet,
    logFC_interaction,
    FDR_infection,
    FDR_diet,
    FDR_interaction,
    logFC_inf_lipid,
    logFC_inf_protein,
    FDR_inf_lipid,
    FDR_inf_protein
    # logFC_ES,
    # logFC_logLoad,
    # FDR_ES,
    # FDR_logLoad
  )

# Split into logFC and FDR matrices
logFC_matrix <- as.matrix(df_heatmap[, c("logFC_infection", "logFC_inf_lipid", "logFC_inf_protein", "logFC_diet", "logFC_interaction"
                                         )])
rownames(logFC_matrix) <-  paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)

FDR_matrix <- as.matrix(df_heatmap[, c("FDR_infection", "FDR_inf_lipid", "FDR_inf_protein", "FDR_diet", "FDR_interaction"
                                       )])
rownames(FDR_matrix) <- paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)





significance_matrix <- ifelse(FDR_matrix < sig.cutoff, "*", " ") %>%
  replace_na(" ")

colnames(logFC_matrix) <- c("Inf (all)", "Inf (L)", "Inf (P)", "Diet", "Interaction"
                            )

df <- scale(logFC_matrix)
factoextra::fviz_nbclust(k.max = 54, df, factoextra::hcut, method = "silhouette", nboot = 500) +
# geom_vline(xintercept = 4, linetype = 2)+
labs(subtitle = "Elbow method")

# Generate the heatmap with internal annotations
heatmap <- Heatmap(logFC_matrix, name = "logFC", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   width = unit(8, "cm"),
                   heatmap_legend_param = list(direction = "horizontal"),
                   cluster_columns = FALSE, 
                   split=8,
                   show_row_names = FALSE,
                   show_column_names = TRUE, 
                   clustering_distance_rows = "pearson",
                   row_names_gp = gpar(fontsize = 10),
                   column_names_gp = gpar(fontsize = 10),
                   column_names_rot = 0,
                   column_names_centered = TRUE,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                       grid.rect(x, y, width, height, gp = gpar(fill = fill, col = NA))
                       if (significance_matrix[i, j] == "*") {
                           grid.text("*", x, y, gp = gpar(col = "black", alpha=0.5, fontsize = 6))
                       }
                   }
                   )

# Draw the heatmap
ht <- draw(heatmap)

ht

#row_order(ht)

png(file="figures/logFC_heatmap.png", width=10, height=9, res=1200,units = "in")
ht <- draw(heatmap, heatmap_legend_side = "bottom")
dev.off()
```

enrichment of some of those
```{r}
list_GO_results <- list()
j=0
for (i in row_order(ht)) {
  # print(i)
  genes_i <- rownames(heatmap@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_results[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe = topTags_diet_edgeR$gene,
                # universe      = df_heatmap$gene,#unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05) %>%
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  # mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% 
    data.frame()
  
}

list_kegg_results <- list()
j=0
for (i in row_order(ht)) {
  # print(i)
  genes_i <- rownames(heatmap@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_kegg_results[[j]] <- enrichKEGG(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
               organism = 'scan', keyType = "kegg",
               universe = topTags_diet_edgeR$gene,
               pvalueCutoff = 0.05) %>%
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  # mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% 
    data.frame()
}
```

```{r}
```



```{r}
# overlap of top and bottom 2000 genes for each:
top_2000_genes_ist <- list(
    # c(
      inf.u = topTags_inf_edgeR  %>%
        top_n(2000, logFC) %>%
        pull(gene),
      inf.d = topTags_inf_edgeR  %>%
        top_n(2000,-logFC) %>%
        pull(gene),
    # ), 
    # c(
      diet.p = topTags_diet_edgeR  %>%
        top_n(2000, logFC) %>%
        pull(gene),
      diet.l = topTags_diet_edgeR  %>%
        top_n(2000,-logFC) %>%
        pull(gene),
    # ), 
    # c(
      ixn.p = topTags_ixn_edgeR  %>%
        top_n(2000, logFC) %>%
        pull(gene),
      ixn.l = topTags_ixn_edgeR  %>%
        top_n(2000,-logFC) %>%
        pull(gene)
    # )
    ) 

upset_top2000_each <- UpSetR::upset(UpSetR::fromList(top_2000_genes_ist),nsets=6, order.by="freq")


upset_top2000_each

# ggsave("", width=50, height=50, units="cm", dpi=300)

png(file="figures/upset_top2000_each.png", width=8, height=8, res=1200,units = "in")
upset_top2000_each
dev.off()
```



# Figure 6: Phenotype Correlation


first thing, let's show how the tolerance metric corresponds would look across the possible range of values:
```{r}
possible_ES = seq(0.5,6)
possible_logLoad = seq(0,10, by = 0.1)

df_possible <- expand.grid(possible_ES, possible_logLoad) %>%
  setNames(c("ES", "logLoads")) %>%
  mutate(tolerance = (1-(ES/logLoads)))#(1+logLoads) / (1+ES)-1)

df_possible %>%
  ggplot(aes(x=tolerance)) + geom_density()

df_possible %>%
    # filter(abs(tolerance)<1) %>%
  ggplot(aes(x=logLoads, y=ES, fill=(tolerance))) +
  geom_tile() +
  scale_fill_viridis_c() +
  theme_minimal() +
  labs(title="Tolerance Metric", x="log Load", y="ES") +
  theme(legend.position = "bottom")

metadata_phenotypes_ixn %>%
  filter(logLoad > 0) %>%
  ggplot(aes(x=tolerance) ) +
  geom_histogram() +
  theme_minimal() +
  labs(title="Tolerance Score Distribution", x="Tolerance Score", y="Density") +
  
  theme(legend.position = "none") + facet_wrap(~Bird.ID)
```

```{r}
```


```{r}
# delete this code chunk
# plot phenotype correlation (interaction)
topTags_tolerance_ixn %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Tolerance Score Interaction (protein inf response / lipid inf response)", 
                   logFC.cutoff = logFC.cutoff,
                   show.names=FALSE)

# plot phenotype correlation (diet)
topTags_tolerance_diet %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Tolerance Score Diet (protein / lipid)", 
                   logFC.cutoff = logFC.cutoff,
                   show.names=FALSE)

# plot phenotype correlation (infection)
topTags_tolerance_inf %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Tolerance Score Infection (inf/uninf)", 
                   logFC.cutoff = logFC.cutoff,
                   show.names=FALSE)
```

## 6A: Volcano tolerance
```{r}
topTags_tolerance %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Tolerance Score", 
                   logFC.cutoff = logFC.cutoff,
                   show.names=FALSE)

ggsave(here("figures/volcano_topTags_tolerance_edgeR.png"), width = 10, height = 10, units = "cm", dpi=1200)

topTags_ES_inf %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Eye Score",show.names = F, 
                   logFC.cutoff = logFC.cutoff)

ggsave(here("figures/volcano_topTags_EyeScore_edgeR.png"), width = 10, height = 10, units = "cm", dpi=1200)


topTags_logLoad_inf %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("logLoad Pathogen",show.names = F, 
                   logFC.cutoff = logFC.cutoff)

ggsave(here("figures/volcano_topTags_logLoad_edgeR.png"), width = 10, height = 10, units = "cm", dpi=1200)

```


Final Venn diagram for table 4.
```{r}

ggVennDiagram::ggVennDiagram(
  x = list(#"Tolerance" = topTags_tolerance %>% filter(FDR < 0.05) %>% pull(gene),
           # "Diet_tol" = topTags_tolerance_diet %>% filter(FDR < 0.05) %>% pull(gene),
           # "Interaction_tol"  = topTags_tolerance_ixn%>% filter(FDR < 0.05) %>% pull(gene),
           "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.05) %>% pull(gene),
           "infection" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
           "EyeScore" = tbl_tol_results %>% filter(FDR_tol < 0.05) %>% pull(gene.y)#, 
           # "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.01) %>% pull(gene)#,
           )
  )


```
## 6B: Vennn of these
```{r}
ggVennDiagram::ggVennDiagram(
  x = list(#"Tolerance" = topTags_tolerance %>% filter(FDR < 0.05) %>% pull(gene),
           # "Diet_tol" = topTags_tolerance_diet %>% filter(FDR < 0.05) %>% pull(gene),
           # "Interaction_tol"  = topTags_tolerance_ixn%>% filter(FDR < 0.05) %>% pull(gene),
           "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.05) %>% pull(gene),
           "infection" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
           "EyeScore" =topTags_ES_inf %>% filter(FDR < 0.05) %>% pull(gene)#, 
           # "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.01) %>% pull(gene)#,
           )
)

# and for their interactions
ggVennDiagram::ggVennDiagram(
  x = list("Tol_ixn" = topTags_tolerance_ixn %>% filter(FDR < 0.05) %>% pull(gene),
           # "Diet_tol" = topTags_tolerance_diet %>% filter(FDR < 0.05) %>% pull(gene),
           # "Interaction_tol"  = topTags_tolerance_ixn%>% filter(FDR < 0.05) %>% pull(gene),
           "inf_ixn" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixn],
           "ES_ixn" =topTags_ES_ixn %>% filter(FDR < 0.05) %>% pull(gene)#, 
           # "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.01) %>% pull(gene)#,
           )
)

(list("Tol_ixn" = topTags_tolerance_ixn %>% filter(FDR < 0.05) %>% pull(gene),
           # "Diet_tol" = topTags_tolerance_diet %>% filter(FDR < 0.05) %>% pull(gene),
           # "Interaction_tol"  = topTags_tolerance_ixn%>% filter(FDR < 0.05) %>% pull(gene),
           "inf_ixn" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixn],
           "ES_ixn" =topTags_ES_ixn %>% filter(FDR < 0.05) %>% pull(gene)#, 
           # "logLoad" = topTags_logLoad_inf %>% filter(FDR < 0.01) %>% pull(gene)#,
           ))
```



## 6B: Heatmap of pheno DE genes

```{r}

```

## 6C: Ontology of pheno DE genes.

Run TOLERANCE enrichments:
```{r}
# GO enrichment
enrichGO_higher_in_tolerance <- enrichGO(gene         = 
  topTags_tolerance  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  top_n(2000, logFC) %>%
  filter(FDR < 0.05) %>%
  filter(logFC > 0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene) ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_ixn_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "ixn",
         Direction = "Prot",
         Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")) %>% data.frame()

# up in lipid
enrichGO_lower_in_tolerance <- enrichGO(gene         = 
  topTags_tolerance  %>%
  # arrange(desc(logFC)) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # top_n(2000, -logFC) %>%
  filter(FDR < 0.05) %>%
  filter(logFC < -0) %>%
  # filter(gene %in% unfiltered_genes_with_GO) %>%
  # arrange(desc(logFC)) %>%
  # slice_max(logFC, prop=0.1) %>%
  pull(gene)
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                universe      = topTags_ixn_edgeR$gene,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # simplify(by="ID") |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
         comparison = "ixn",
         Direction = "Lipid") %>% data.frame()



##################
# KEGG enrichments
##################


# up in high protein ixn
enrichKEGG_higher_in_tolerance <- topTags_tolerance  %>%
  top_n(2000, logFC) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_tolerance$gene)

# up in high fat ixn
enrichKEGG_lower_in_tolerance <- topTags_tolerance  %>%
  top_n(2000, -logFC) %>%
    pull(gene) %>%
    enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_tolerance$gene)

# either way
# enrichKEGG_in_diet <- topTags_diet_edgeR  %>%
#   top_n(4000, abs(logFC)) %>%
#   pull(gene) %>%
#   enrichKEGG(gene = .,
#                organism = 'scan',
#                universe = topTags_diet_edgeR$gene)

# gse enrichment
gseKEGG_tolerance <- topTags_tolerance %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    by = "fgsea",
    exponent = 1,
    #minGSSize    = 120,
    pvalueCutoff = 0.05,
    verbose      = FALSE
  )
```


```{r}
## add KEGG terms to GO plot
enrichGOKEGG_higher_in_tolerance_fig <-
  enrichKEGG_higher_in_tolerance %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "Interaction",
    Direction = "bigger_delta_protein",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_higher_in_tolerance %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "Interaction",
        Direction = "bigger_delta_protein",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    colours = c("red", "#FFD6D7"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50)
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched: up with tolerance") +
  theme_classic()

# higher in lipid GO and KEGG
enrichGOKEGG_lower_in_tolerance_fig <-
  enrichKEGG_lower_in_tolerance %>%
  as.data.frame() |>
  mutate(
    richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
    comparison = "tolerance",
    Direction = "up_tolerance",
    Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
  ) %>%
  bind_rows(
    enrichGO_lower_in_tolerance %>%
      as.data.frame() |>
      mutate(
        richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)),
        comparison = "tolerance",
        Direction = "up_tolerance",
        Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ")
      )
  ) %>%
  ggplot(.,
         showCategory = 8,
         aes(richFactor, fct_reorder(Description, richFactor))) +
  geom_segment(aes(xend = 0, yend = Description),
               color = "black",
               linewidth = 1) +
  geom_point(aes(color = p.adjust, size = Count)) +
  scale_color_gradientn(
    # colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    colors = c("blue", "lightblue"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(3, 10)) +
  scale_y_discrete(
    label = function(x)
      stringr::str_trunc(x, 50),
    position = "right"
  ) + # cut off long names
  xlab("Rich Factor") +
  ylab(NULL) +
  ggtitle("Functions Enriched: down with tolerance") +
  scale_x_reverse() +
  theme_classic()



enrichGOKEGG_higher_in_tolerance_fig
enrichGOKEGG_lower_in_tolerance_fig


ggpubr::ggarrange(enrichGOKEGG_lower_in_tolerance_fig, enrichGOKEGG_higher_in_tolerance_fig, ncol = 2, nrow = 1)

ggsave(here("figures/enrichGOKEGG_DE_in_tolerance_fig.png"), width = 30, height = 15, units = "cm", dpi=1200)
```


....SIDEBAR: 
Plot venn diagram of inf genes in general analysis vs tolerance phenotype
```{r}
ggVennDiagram::ggVennDiagram(
  x = list("Infection_tol" = topTags_tolerance %>% filter(FDR < 0.05) %>% pull(gene),
           # "Diet_tol" = topTags_tolerance_diet %>% filter(FDR < 0.05) %>% pull(gene),
           # "Interaction_tol"  = topTags_tolerance_ixn%>% filter(FDR < 0.05) %>% pull(gene),
           "infection" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
           "Diet" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet], 
           "Interaction" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixn]
           )#,
  #category.names = c("Infection", "Diet", "Interaction")
  #category.names = c("Infection", "Diet", "Interaction")#,
  # category.col=c("red", "magenta", "blue"),
  # cat.col=c("red", "magenta", "blue"),
  # cat.fontface=2,
  # margin=0.05,
  # main="Venn Diagram of DE genes"
)
```
<!-- # Phenotype of Day 0 tolerance -->
<!-- this is ugly and doesn't provide any insights into the data, I think we should exclude it. -->
<!-- ```{r} -->
<!-- metadata_phenotypes_day0 <- metadata_phenotypes |> -->
<!--   filter(Day==0) |> -->
<!--   ## filter(MG.active=="YES") |> -->
<!--   # add 1 to avoid undefined -->
<!--   mutate(tolerance_1 = ES_AUC_7/(1+logLoad_AUC_14),  -->
<!--          tolerance_2 = ES_AUC_7/(1+logLoad_day14), -->
<!--          tolerance_3 = ES_day7/(1+logLoad_day7), -->
<!--          tolerance_4 = r_ES_day3/(1+r_logLoad_day3), # maybe best? -->
<!--          tolerance_5 = (1+r_logLoad_day3)/(1+r_ES_day3)-1, # this format works better for "tolerance" -->
<!--          #tolerance_6 = (1+logLoad_day14)/(1+r_ES_AUC_14)-1, -->
<!--          tolerance_0 = 1+logLoad/(1+ES)-1 -->
<!--          ) %>% -->
<!--   filter(MG.Status != "Control") %>% -->
<!--   # filter(!Sample.ID %in% c("ES12", "ES25", "ES36", "ES58", "ES84", "ES53")) %>% -->
<!--   mutate(tolerance_5_cat = dplyr::ntile(tolerance_5,3), -->
<!--          #tolerance_5_cat = dplyr::ntile(tolerance_5,3), -->
<!--          # tolerance_5_cat = dplyr::ntile(tolerance_1,3), -->
<!--          tolerance_cat = dplyr::ntile(tolerance,3)) %>% -->
<!--   mutate(diet.tolerance = paste0(Diet, ".",tolerance_cat)) %>% -->
<!--   # mutate(SampID = str_replace_all(SampID,"\\.","-")) %>% -->
<!--   column_to_rownames("SampID") -->

<!-- gene_expression_matrix_day0 <- gene_expression_matrix |> -->
<!--   data.frame() |> -->
<!--   dplyr::select(ends_with("_0")) |> -->
<!--   dplyr::select(contains("MG")) |> -->

<!--   # these are birds who PCR shows no pathogen -->
<!--   # dplyr::select(-contains("ES12")) |> -->
<!--   # dplyr::select(-contains("ES25")) |> -->
<!--   # dplyr::select(-contains("ES36")) |> -->
<!--   # dplyr::select(-contains("ES58")) |> -->
<!--   # dplyr::select(-contains("ES84")) |> -->
<!--   # dplyr::select(-contains("ES53")) |> -->
<!--   as.matrix() -->

<!-- rownames(gene_expression_matrix_day0) <- entrez_names_gene_expression_matrix_ixn -->

<!-- Diet_1 <- as.factor(metadata_phenotypes_day0$Diet) -->
<!-- # Day_tolerance <- (metadata_phenotypes_day0$Day) -->


<!-- # diet_tolerance <- interaction(Diet_1, metadata_phenotypes_day0$tolerance_5_cat) #%>% as.factor() -->
<!-- # design0 <- model.matrix(~ 0 + diet_tolerance) -->

<!-- ES_minMax0 <- metadata_phenotypes_day0$r_ES_day3 -->

<!-- design0 <- model.matrix(~ ES_minMax0) -->


<!-- y_ES0 <- DGEList( -->
<!--   counts = gene_expression_matrix_day0, -->
<!--   # names= entrez_names_gene_expression_matrix_day0, -->
<!--   # group=Active_1, -->
<!--   remove.zeros = TRUE -->
<!-- ) -->

<!-- #determine cpm cutoff -->
<!-- cpm_cutoff0 <- as.numeric(round(cpm(10, mean(y_ES0$samples$lib.size)),1)) -->
<!-- # cpm_cutoff <- 1 -->
<!-- # apply cutoff -->
<!-- keep0 <- rowSums(cpm(y_ES0) > cpm_cutoff0) >= 2 -->
<!-- y_ES0 <- y_ES0[keep0,] -->
<!-- summary(keep0) -->

<!-- y_ES0 <- calcNormFactors(y_ES0) -->
<!-- y_ES0 <- estimateDisp(y_ES0, design0)#, robust=TRUE) -->

<!-- # y_ES0$samples$group <- #grouping_var -->
<!-- #   interaction(Diet_1, metadata_phenotypes_day0$ES_5_cat) %>% as.factor() -->

<!-- # y_ES$samples$group <- #grouping_var -->
<!-- #   interaction(Diet_1, metadata_phenotypes_day0$ES_cat) %>% as.factor() -->

<!-- # contrasts_ES0 <- makeContrasts(#design, #metadata,  -->
<!-- #   contrasts = c("(diet_ESprotein.3 - diet_ESprotein.1) - (diet_ESlipid.3 - diet_ESlipid.1)", -->
<!-- #                 # "Diet_1protein", -->
<!-- #                 "(diet_ESprotein.3 + diet_ESprotein.2 + diet_ESprotein.1)/3-(diet_ESlipid.3 + diet_ESlipid.2 + diet_ESlipid.1)/3", -->
<!-- #                 "(diet_ESprotein.3 + diet_ESlipid.3)/2 - (diet_ESprotein.1 + diet_ESlipid.1)/2" -->
<!-- #                 #"diet_ESlipid.3 - diet_ESlipid.1", -->
<!-- #                 #"diet_ESprotein.3 - diet_ESlipid.3" -->
<!-- #                 ), levels=design0) -->

<!-- fit0 <- glmQLFit(y_ES0, design0)#, robust=TRUE) -->

<!-- # res0_ixn <-  glmQLFTest(fit0, contrast=contrasts_ES0[,1]) -->
<!-- res0_inf <-  glmQLFTest(fit0, coef="ES_minMax0") -->

<!-- # exactTest0 <- exactTest(fit0) -->
<!-- # res1 <-  glmQLFTest(fit, coef = "Diet_1protein:ES_1") -->
<!-- #res2 <-  glmQLFTest(fit, coef = "Diet_1lipid:ES_1") -->
<!-- # #  -->
<!-- # topTags_ES0_ixn <- topTags(res0_ixn, n=Inf) %>% data.frame() %>% -->
<!-- #   arrange(FDR) %>% -->
<!-- #   #mutate(logFC=round(logFC,2)) %>% -->
<!-- #   # mutate(across(where(is.numeric), signif, 3)) %>% -->
<!-- #   # mutate_if(is.numeric, signif, 3) %>% -->
<!-- #   rownames_to_column("gene") %>% -->
<!-- #   left_join(result_BM_merged, -->
<!-- #     #gene_GO_tbl |> -->
<!-- #               #mutate(gene = as.character(entrezgene_id)), -->
<!-- #             by = "gene") %>% -->
<!-- #   dplyr::select( -->
<!-- #     c( -->
<!-- #       "gene", -->
<!-- #       "gene_name", -->
<!-- #       "logFC", -->
<!-- #       "logCPM", -->
<!-- #       "F", -->
<!-- #       "PValue", -->
<!-- #       "FDR", -->
<!-- #       "description", -->
<!-- #       "entrezgene_id" -->
<!-- #     ) -->
<!-- #   ) %>% -->
<!-- #   distinct()# %>% View() -->


<!-- topTags_ES0_inf <- topTags(res0_inf, n=Inf) %>% data.frame() %>% -->
<!--   arrange(FDR) %>% -->
<!--   #mutate(logFC=round(logFC,2)) %>% -->
<!--   # mutate(across(where(is.numeric), signif, 3)) %>% -->
<!--   # mutate_if(is.numeric, signif, 3) %>% -->
<!--   rownames_to_column("gene") %>% -->
<!--   left_join(result_BM_merged, -->
<!--     #gene_GO_tbl |> -->
<!--               #mutate(gene = as.character(entrezgene_id)), -->
<!--             by = "gene") %>% -->
<!--   dplyr::select( -->
<!--     c( -->
<!--       "gene", -->
<!--       "gene_name", -->
<!--       "logFC", -->
<!--       "logCPM", -->
<!--       "F", -->
<!--       "PValue", -->
<!--       "FDR", -->
<!--       "description", -->
<!--       "entrezgene_id" -->
<!--     ) -->
<!--   ) %>% -->
<!--   distinct() -->
<!-- ``` -->

<!-- END SIDEBAR.... -->


# Figure 7: Corr Phenotype measures correlation ES vs logload vs tolerance
```{r}
# Corr Phenotype measures correlation ES vs logload vs tolerance
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  dplyr::select(logFC.tolerance, logFC.ES, logFC.logLoad, logFC) %>%
  ggpairs(mapping = aes(alpha=0.5)) +
  theme_classic() +
  ggtitle("ALL Genes: Correlation of Phenotypic LogFC estimates")

ggsave("figures/AllGenesCorrPhenotypeLogFC.png", width=16, height=16, units = "cm", dpi=1200)

# Corr Phenotype measures correlation ES vs logload vs tolerance
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  filter(FDR.tolerance < 0.05 | FDR < 0.05 | FDR.ES < 0.05 | FDR.logLoad < 0.05 
) %>%
  # filter(logCPM>1) %>%
  # filter(logFC.tolerance*logFC<0) %>% #View()
  dplyr::select(gene,logFC.tolerance, logFC.ES, logFC.logLoad, logFC) %>%
  column_to_rownames("gene") %>%
  ggpairs(mapping = aes(alpha=0.1),progress = FALSE) +
  theme_classic() +
  ggtitle("DE Genes: Correlation of Phenotypic LogFC estimates")

ggsave("figures/DEGenesCorrPhenotypeLogFC.png", width=16, height=16, units = "cm", dpi=1200)
```

```{r}
enrichGO(gene         = 
  topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  filter(FDR.tolerance < 0.05 & FDR < 0.05 #| FDR.ES < 0.05 | FDR.logLoad < 0.05 
) %>%
  # filter(logCPM>1) %>%
  # filter(logFC.tolerance*logFC>0) %>%
  # filter(logFC.tolerance < 0) %>%
  pull(gene) 
                                  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                # universe      = unfiltered_genes_with_GO,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame()

## kegg enrichment of genes opposite direction between tolerance and infection
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  # mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  filter(FDR.tolerance < 0.05 | FDR < 0.05 #| FDR.ES < 0.05 | FDR.logLoad < 0.05 
) %>%
  # filter(logCPM>1) %>%
  filter(logFC.tolerance*logFC<0) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_tolerance$gene
             )

topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  # mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  filter(FDR.tolerance < 0.05 | FDR < 0.05 #| FDR.ES < 0.05 | FDR.logLoad < 0.05 
) %>%
  # filter(logCPM>1) %>%
  filter(logFC.tolerance*logFC>0) %>%
  pull(gene) %>%
  enrichKEGG(gene = .,
               organism = 'scan',
               universe = topTags_tolerance$gene
             )
```

```{r}
topTags_tolerance %>%
left_join(topTags_ES_inf, by=c("gene","gene_name", "description", "entrezgene_id"),suffix = c(".tolerance", ".ES")) %>%
  left_join(topTags_logLoad_inf, by=c("gene","gene_name", "description", "entrezgene_id")) %>%
  # mutate(logFC.logLoad=logFC) %>%
  left_join(topTags_inf_edgeR, by=c("gene","gene_name", "description"), suffix=c(".logLoad","")) %>%
  # filter(FDR.tolerance < 0.05 | FDR < 0.05 #| FDR.ES < 0.05 | FDR.logLoad < 0.05 
# ) %>%
  # filter(logCPM>1) %>%
  filter(logFC.tolerance*logFC>0.5) %>% #View()
  pull(gene) %>%
  enrichKEGG(gene = .,
             universe = topTags_tolerance$gene,
               organism = 'scan',
             pvalueCutoff = 1
            
               
             
             )
```


# Correlation of gse results
```{r}
# Correlation of gse results
# gseGO_infection %>% data.frame() %>%
#   left_join(gseGO_diet %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c(".infection",".diet")) %>%
#   left_join(gseGO_ixn %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c("",".ixn")) %>%
#   ggplot(aes(x=enrichmentScore.infection, y=enrichmentScore.diet, color=enrichmentScore)) +
#   geom_density2d_filled()
# 
# # Correlation of gse results
# gseGO_infection %>% data.frame() %>%
#   left_join(gseGO_diet %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c(".infection",".diet")) %>%
#   left_join(gseGO_ixn %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c("",".ixn")) %>%
#   ggplot(aes(x=enrichmentScore.infection, y=enrichmentScore, color=enrichmentScore.diet)) +
#   geom_density2d_filled()
# 
# # Correlation of gse results
# gseGO_infection %>% data.frame() %>%
#   left_join(gseGO_diet %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c(".infection",".diet")) %>%
#   left_join(gseGO_ixn %>% data.frame(), by=c("ID", "Description", "setSize"), suffix=c("",".ixn")) %>%
#   ggplot(aes(x=enrichmentScore, y=enrichmentScore.diet, color=enrichmentScore.infection)) +
#   geom_density2d_filled()
```


# Figure 8: PCA plot of gene expression for all samples

```{r}
# PCA plot of gene expression for all samples
PCA <- plotMDS(y_ixn, gene.selection="common",
               top = 300,pch = 2, cex = 0.5, main = "PCA plot of gene expression for all samples", dim.plot=c(1,2))
```

```{r}

ggPCA <- tibble(
  SampID = colnames(PCA$distance.matrix.squared),
  x = PCA$x,
  y =PCA$y
) %>%
  mutate(SampID = str_replace_all(SampID, "\\.","-")) %>%
  left_join(metadata_phenotypes, by="SampID")

ggplot(ggPCA, aes(x=x,y=y, group=Bird.ID, shape = Day, color = Diet)) +
  geom_line(aes(linetype = MG.Status), alpha = 0.5) +
  geom_point() +
  theme_classic() +
  # theme(legend.position = "none") +
  labs(title="PCA plot of gene expression for all samples") +
  xlab(paste0("Principal Component 1 (",100*round(PCA$var.explained[1],2),"%)")) +
  ylab(paste0("Principal Component 2 (",100*round(PCA$var.explained[2],2),"%)")) #+
  # facet_wrap(~MG.Status)

ggplot(ggPCA, aes(x=x,y=y, group=interaction(Diet,MG.Status), #shape = Day,
                  color = interaction(MG.Status,Diet), linetype = Diet)) +
  #geom_line(aes(alpha = MG.Status)) +
  geom_point() +
  theme_classic() +
  # theme(legend.position = "none") +
  labs(title="PCA plot of gene expression") +
  xlab(paste0("Principal Component 1 (",100*round(PCA$var.explained[1],2),"%)")) +
  ylab(paste0("Principal Component 2 (",100*round(PCA$var.explained[2],2),"%)")) +
  scale_color_discrete("MG.Diet") +
  stat_ellipse(type = "norm",level=0.8)

#ggsave("figures/PCA_plot_all_samples.png", width = 6, height = 5, units = "in", dpi = 600)

```


## ordinal regression for eyescores

```{r}
set.seed(123)

  
dummy <- tpm_data %>% cpm(log = TRUE) %>% t() %>%
  data.frame() %>% rownames_to_column("sample") %>%
  left_join(metadata_phenotypes, by=c("sample"="SampID")) %>%
  # filter(MG.Status == "MG") %>%
  # filter(logLoad>0) %>%
  # dplyr::select(where(~ any(. != 0))) %>%
  mutate(ES = as.ordered(as.factor(ES)),
         tolerance_ord = as.ordered(as.factor(tolerance))#,
         # logload = as.ordered(as.factor(logload))
         )

# dummy <- tibble(ID = 1:50,
#                 A = sample(x = 1:200, size = 50, replace = T),
#                 B = as.factor(sample(x = c("day", "night"), size = 50, 
#                                      replace = T)),
#                 C = as.factor(sample(x = c("blue", "red", "green"), size = 50, replace = T)))
library(tidyverse)
library(VGAM)

mod_f <- function(x) {
  vglm(
    formula = as.formula(x),
    family = cumulative(parallel = TRUE,
                        reverse = FALSE),
    data = dummy
  )
}

## get genes that are included in edgeR analysis:
edgeR_analyzed_genes <- topTags_inf_edgeR %>% 
  # topTags_ES_inf %>%
  left_join(result_BM_merged, by=c("gene" = "entrezgene_id")) %>% #View()
  filter(FDR < 1.05) %>%
  mutate(match_gene_tmp = case_when(
    str_detect(gene,"^1") ~ str_replace(gene, "^1", "LOC1"),
    TRUE~gene
  )) %>%
  mutate(match_gene = case_when(
    match_gene_tmp %in% rownames(tpm_data) ~ match_gene_tmp,
    gene_tag %in% rownames(tpm_data) ~ gene_tag,
    TRUE~NA
  )) %>%
  arrange(desc(logFC)) %>%
  # filter(logFC>0.5) %>%
  pull(match_gene, name = gene)# %>% unique()

n <- length(unique(edgeR_analyzed_genes))
      # 2000

formulas <- c(paste0("ES~", rownames(tpm_data)[rownames(tpm_data) %in% edgeR_analyzed_genes
                                               ]))[1:n]

tbl <- tibble(forms = formulas)

tbl_2 <-
  tbl %>% mutate(
    all = map(formulas, mod_f) %>% 
      map(summary) %>% 
      map(coef) %>% 
      map(as_tibble, rownames = "gene")
  )


tbl_ES_results <- tbl_2$all %>% set_names(rownames(tpm_data)[1:n]) %>% 
  dplyr::bind_rows() %>%
  filter(!str_detect(gene, "Intercept")) %>%  
  left_join(result_BM_merged, by=c("gene" = "gene_tag")) %>% 
  mutate(FDR= p.adjust(`Pr(>|z|)`, method="fdr")) %>% #View()
  left_join(topTags_inf_edgeR, by=c("gene.y" = "gene"),suffix = c("_ES", "_inf")
              ) #%>%  
  # filter(FDR_inf < 0.05 #&
  #          # FDR_ES < 0.05
  #          ) %>%
  # dplyr::select(logFC, Estimate) #%>%
  # cor(use = "pairwise.complete.obs", method = "spearman")
  # View()

de_ES <- tbl_2$all %>% set_names(rownames(tpm_data)[1:n]) %>% 
  dplyr::bind_rows() %>%
  filter(!str_detect(gene, "Intercept")) %>%  
  left_join(result_BM_merged, by=c("gene" = "gene_tag")) %>% 
  mutate(FDR= p.adjust(`Pr(>|z|)`, method="fdr")) %>%
  filter(FDR < 0.05) %>%
  # filter(Estimate < 0) %>%
  pull(gene.y) #%>% #length()
  # intersect(topTags_inf_edgeR %>% filter(FDR <0.05) %>% pull(gene)) #%>% length()

topTags_inf_edgeR %>%
  filter(gene %in% de_ES) #%>% View()

de_ES[de_ES %ni% de_ES_and_inf]
```

```{r}
edgeR_analyzed_genes_DE <- topTags_inf_edgeR %>% 
  # topTags_ES_inf %>%
  left_join(result_BM_merged, by=c("gene" = "entrezgene_id")) %>% #View()
  filter(FDR < 0.05) %>%
  mutate(match_gene_tmp = case_when(
    str_detect(gene,"^1") ~ str_replace(gene, "^1", "LOC1"),
    TRUE~gene
  )) %>%
  mutate(match_gene = case_when(
    match_gene_tmp %in% rownames(tpm_data) ~ match_gene_tmp,
    gene_tag %in% rownames(tpm_data) ~ gene_tag,
    TRUE~NA
  )) %>% #View()
  arrange(desc(logFC)) %>%
  # filter(logFC>0.5) %>%
  pull(match_gene, name = gene)# %>% unique()


m = length(unique(edgeR_analyzed_genes_DE))

formulas_tolerance <- c(paste0("ES~Diet+logLoad+", rownames(tpm_data)[rownames(tpm_data) %in% edgeR_analyzed_genes
                                               ]))[1:n]#n

tbl_tolerance <- tibble(forms = formulas_tolerance)

tbl_2_tol <-
  tbl_tolerance %>% mutate(
    all = map(formulas_tolerance, mod_f) %>% 
      map(summary) %>% 
      map(coef) %>% 
      map(as_tibble, rownames = "gene")
  )

tbl_tol_results <- tbl_2_tol$all %>% set_names(rownames(tpm_data)[1:n]) %>% 
  dplyr::bind_rows() %>%
  filter(!str_detect(gene, "Intercept")) %>% #View()
  left_join(result_BM_merged, by=c("gene" = "gene_tag")) %>% 
  mutate(FDR= p.adjust(`Pr(>|z|)`, method="fdr")) %>%
  # mutate(id = c(rep(1:m, each = 3),885))# %>%
  filter(gene != "Dietprotein" & gene != "logLoad") %>% #View()
  filter(entrezgene_id %in% topTags_inf_edgeR$entrezgene_id) %>%
  mutate(gene.z = case_when(
    gene.y %in% topTags_inf_edgeR$gene ~ gene.y,
    gene %in% topTags_diet_edgeR$gene ~ gene,
    gene %in% paste0("LOC",topTags_diet_edgeR$gene) & 
      is.na(entrezgene_id) ~ str_remove(gene, "LOC"),
    TRUE ~ NA
  )) %>%
    filter(entrezgene_id %in% topTags_inf_edgeR$entrezgene_id & gene.z %in% topTags_inf_edgeR$gene) %>%
    dplyr::select(-c(wikigene_name,ensembl_gene_id))%>%
  right_join(topTags_inf_edgeR, by=c("entrezgene_id" = "entrezgene_id", "description" = "description", "gene.z" = "gene", "gene_name" = "gene_name"
                                    ),
            suffix = c("_tol", "_inf"))
    # group_by(`gene_name`) %>%
    # filter(n()>2) %>% View()
  #filter(gene.z != gene_inf)
  #filter(!is.na(entrezgene_id)) %>%
```


functional enrichments of "tolerance" aka ES significant genes
```{r}
enrichGO_tol_ord_pos <- tbl_tol_results %>%
  filter(FDR_tol < 0.05 & Estimate > 0) %>%
  pull(entrezgene_id) %>%
  enrichGO( gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() #%>% View()

enrichGO_tol_ord_neg <- tbl_tol_results %>%
  filter(FDR_tol < 0.05 & Estimate < 0) %>%
  pull(entrezgene_id) %>%
  enrichGO( gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() #%>% View()

tbl_tol_results %>%
  filter(FDR_tol < 0.05 & Estimate > 0) %>%
  pull(entrezgene_id) %>%
  enrichGO( gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) %>% data.frame() %>%
  View()

enrichGO_tol_ord <- tbl_tol_results %>%
  filter(FDR_tol < 0.05) %>%
  pull(entrezgene_id) %>%
  enrichGO( gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) #%>% data.frame() 

gseGO_tol_ord <- tbl_tol_results %>% 
  mutate(est= case_when(
    # Estimate > 5 ~ 5,
    # Estimate < -5 ~ -5,
    TRUE ~ Estimate
  )) %>%
  arrange(desc(est)) %>%
  filter(!is.na(est)) %>%
  pull(est, entrezgene_id) %>%
  gseGO(geneList     = .,
         OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1
)

enrichKEGG_tol_ord <- tbl_tol_results %>%
  filter(FDR_tol < 0.05) %>%
  pull(entrezgene_id) %>%
  enrichKEGG( gene = .,
             #universe = topTags_tolerance$gene,
               organism = 'scan',
             pvalueCutoff = 1,
             qvalueCutoff = 1
             )

gseKEGG_tol_ord <- tbl_tol_results %>% 
  mutate(est= case_when(
    # Estimate > 5 ~ 5,
    # Estimate < -5 ~ -5,
    TRUE ~ Estimate
  )) %>%
  arrange(desc(est)) %>%
  filter(!is.na(est)) %>%
  pull(est, entrezgene_id) %>%
  gseKEGG(geneList     = .,
               organism = 'scan',
               #universe = topTags_ixn$gene,
               #minGSSize    = 120,
               pvalueCutoff = 1,
               verbose      = FALSE)



cnetplot(setReadable(simplify(enrichGO_tol_ord), orgdb), 
           # foldChange = ., 
           showCategory = 10,
           species = "scan",#node_label="gene",
           cex.params = list(gene_label = 0.4, category_label = 0.4),
           color.params = list(foldChange = tbl_tol_results %>%
                                 mutate(est= case_when(
    Estimate > 5 ~ 5,
    Estimate < -5 ~ -5,
    TRUE ~ Estimate
  )) %>%
  arrange(desc(est)) %>%
  pull(est, entrezgene_id))
  ) 
```

make custom cnetplot:
tranlsation GO:0006412
ubiquitin dependent protein catabolism GO:0006511
excluded: macromolecule catabolism GO:GO:0009057
mitochondrial membrane GO:0031966
immune response GO:0006955
defense response GO:0006952
lipid metabolic process GO:0006629

```{r}
enrichGO_tol_ord2 <- tbl_tol_results %>%
  filter(FDR_tol < 0.2) %>%
  pull(entrezgene_id) %>%
  enrichGO( gene = .,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                #universe      = unfiltered_genes_with_GO,
                ont           = "ALL",
                pAdjustMethod = "BH",
                pvalueCutoff  = 1.1,
                qvalueCutoff  = 1.1) |>
  # filter(ONTOLOGY %in% c("MF","BP")) |>
  mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))

enrichGO_tol_ord2 %>%
  #simplify() %>%
  setReadable(orgdb) %>%
  filter(ID %in% c("GO:0006412","GO:0006511",
                   "GO:0031966","GO:0006955",# "GO:0009057",
                   "GO:0006952","GO:0006629")) %>%
cnetplot(., layout = "graphopt", 
           # foldChange = ., 
           showCategory = 6,
           species = "scan",#node_label="gene",
           cex.params = list(gene_label = 0.2, category_label = 0.5),
           color.params = list(foldChange = tbl_tol_results %>%
                                 mutate(est= -case_when(
    Estimate > 5 ~ 5,
    Estimate < -5 ~ -5,
    TRUE ~ Estimate
  )) %>%
  arrange(desc(est)) %>%
  pull(est, entrezgene_id))
  ) 

ggsave("figures/GO_term_cnetplot_ES.png", width = 8, height = 8, units = "in", dpi = 1200)
```




```{r}
# vglm(formula = "ES~LOC103823426", family = cumulative(parallel = TRUE, #jchain
#                         reverse = FALSE),
#     data = dummy) %>% summary()

# ES_day0_analysis
formulas_day0ES <- c(paste0("ES_day0_analysis~", rownames(tpm_data)[rownames(tpm_data) %in% edgeR_analyzed_genes
                                               ]))[1:n]

tbl_day0ES <- tibble(forms = formulas_day0ES)

tbl_2_day0ES <- tbl_day0ES %>% mutate(all = map(formulas_day0ES, mod_f) %>% map(summary) %>% map(coef) %>% map(as_tibble, rownames = "gene")
                        # gla = all %>% map(summary),
                        # coef_tmp = gla %>% map(coef),
                        # coef = coef_tmp %>% map(as_tibble, rownames = "gene")
                        # aug = all %>% map(augment),
                        # AIC = all%>% map_dbl("AIC")
                        )# %>% dplyr::select(-all, -gla, -coef_tmp)

# tbl_2$coef %>% purrr::reduce(c) %>% as.data.frame(names(.)) %>% rownames_to_column("gene") %>%
#   filter(!str_detect(gene, "Intercept")) %>%
#   arrange(desc(estimate)) %>%
#   head(10)


tbl_2_day0ES$all %>% set_names(rownames(tpm_data)[1:n]) %>% dplyr::bind_rows() %>%
  filter(!str_detect(gene, "Intercept")) %>%  left_join(result_BM_merged, by=c("gene" = "gene_tag")) %>% mutate(FDR= p.adjust(`Pr(>|z|)`, method="fdr")) %>% View()
```


# Final Fig 4 comparing GO term enrichments.
```{r, fig.height=8, fig.width=12}
fig4_plot_list <- list(
simplify(gseGO_infection)@result %>%
  mutate(comparison = "infection (all)"),
simplify(gseGO_infection_protein)@result %>%
  mutate(comparison = "infection (protein)"),
simplify(gseGO_infection_lipid)@result %>%
  mutate(comparison = "infection (lipid)"),
gseKEGG_infection@result %>%
  mutate(comparison = "infection (all) KEGG"),
gseKEGG_infection_protein@result %>%
  mutate(comparison = "infection (protein) KEGG"),
gseKEGG_infection_lipid@result %>%
  mutate(comparison = "infection (lipid) KEGG"),
simplify(gseGO_diet_uninf)@result %>%
  mutate(comparison = "diet (uninfected)")
)

DE_GO_terms_any_inf_simple <- bind_rows(fig4_plot_list[[1]], 
          fig4_plot_list[[2]],
          fig4_plot_list[[3]]
          ) %>%
  filter(p.adjust < 0.05) %>%
  pull(ID) %>% unique()

fig4_plot_list <- list(
(gseGO_infection)@result %>%
  mutate(comparison = "infection (all)"),
(gseGO_infection_protein)@result %>%
  mutate(comparison = "infection (protein)"),
(gseGO_infection_lipid)@result %>%
  mutate(comparison = "infection (lipid)"),
gseKEGG_infection@result %>%
  mutate(comparison = "infection (all) KEGG"),
gseKEGG_infection_protein@result %>%
  mutate(comparison = "infection (protein) KEGG"),
gseKEGG_infection_lipid@result %>%
  mutate(comparison = "infection (lipid) KEGG"),
(gseGO_diet_uninf)@result %>%
  mutate(comparison = "diet (Protein-Lipid)")
)

bind_rows(fig4_plot_list[[1]], 
          fig4_plot_list[[2]],
          fig4_plot_list[[3]],
          fig4_plot_list[[7]]) %>%
  filter(ID %in% DE_GO_terms_any_inf_simple) %>%
  mutate(Description = str_replace_all(Description, "biological process involved in ", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " (", ID, ") ")) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
    Description == "monoatomic ion transport" & comparison == "infection (all)" ~ Term,
    Description == "monoatomic ion transport" ~ NA,
    comparison == "infection (all)" ~ Term,
    TRUE ~ NA
  ),
  y_labels = case_when(
    Description == "monoatomic ion transmembrane transport" & comparison == "infection (all)" ~ 0.42,
    Description == "monoatomic ion transport" & comparison == "infection (all)" ~ 0.36,
    TRUE ~ 0
  )) %>%
  group_by(Term) %>%
  mutate(Term_labels2 = case_when(
    n() == 4 & comparison == "infection (all)" ~ Term,
    n() == 3 & !is.na(Term_labels) & comparison == "infection (all)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (protein)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (lipid)" ~ Term,
    n() == 1 ~ Term,
    TRUE ~ NA) 
    )%>%
  ungroup() %>% 
  arrange(desc(enrichmentScore)) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`#-log(p.adjust)
                )) +
  geom_col(#position = "dodge",
           position = position_dodge(preserve = 'single')) +
  geom_text(aes(y = y_labels, label = Term_labels2, hjust = as.numeric(enrichmentScore) > 0), alpha=1) + 
  xlab("GO Term") +
  scale_alpha_discrete(range=c(0.2, 1)) +
  scale_fill_manual(values = c("yellow","purple", "blue","red")) +
  coord_flip() + theme_classic() +
  theme(axis.title.y=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank(),axis.text = element_text(size = 14), axis.title=element_text(size=16))

ggsave("figures/GO_term_enrichment_comparison.png", width = 12, height = 10, units = "in", dpi = 600)
```
how about we scale the logFC values for each of those so they are on 0,1 scale? maybe, but that could be messy b/c of outliers.. z score fine but goes too far out.. maybe just leave it out? or rescale? we'll see.

```{r, fig.height=8}

DE_KEGG_terms_any_inf <- bind_rows(fig4_plot_list[[4]], 
          fig4_plot_list[[5]],
          fig4_plot_list[[6]]) %>%
  filter(p.adjust < 0.05) %>%
  pull(ID) %>% unique()

bind_rows(fig4_plot_list[[4]], 
          fig4_plot_list[[5]],
          fig4_plot_list[[6]]) %>%
  filter(ID %in% DE_KEGG_terms_any_inf) %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " (", ID, ") ")) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
    # Description == "monoatomic ion transport" & comparison == "infection (all)" ~ Term,
    # Description == "monoatomic ion transport" ~ NA,
    comparison == "infection (all) KEGG" ~ Term,
    TRUE ~ NA
  ),
  y_labels = case_when(
    Description == "Spliceosome" & comparison == "infection (protein) KEGG	" ~ 0.1,
    TRUE ~ 0
  )) %>%
  arrange(desc(enrichmentScore)) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`#-log(p.adjust)
                )) +
  geom_col(position = "dodge") +
  geom_text(aes(y = y_labels, label = Term_labels, hjust = as.numeric(enrichmentScore) > 0), alpha=1) + 
  xlab("GO Term") +
  scale_alpha_discrete(range=c(0.6, 1)) +
  scale_fill_manual(values = c("purple", "blue","red")) +
  coord_flip() + theme_classic() +
  theme(axis.title.y=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank())
```







# Supplemental Code

Plot individual genes
```{r}
genes_to_plot <- c(
  "103820477", #bpgm
  "103822775", #cytokine CXCR5
  "103814802",
  "103823620", # same again, huge diff but not in others
  "103818584", # same group as below
  "103816392", # cullin DE for ordinal but not edgeR
  "103812987", # chemokine c-c ixn diff bigger in protein tolerance
                   "103820477", #bpgm, goes opposite between diets
                   "103817495", # LARP1, important for translation gene expression (lysosomes)
  "127059202", # feather beta keratin, DE in tolerance ixn
  "103813205", # BPI fold-containing family C protein
  "103823136", # ADAM metallopeptidase with thrombospondin type 1
  "127060733", # germ cell nuclear acidic protein-like
                   "103813247", # Galanin receptor 3 GAL3 (gprotein coupled receptor, up in tol inf and ixn)
                   "127061143", # Cathepsin S-like, DE in tolerance and diet
                   "108961920", 
                   "103823426", #JCHAIN
                   "103822244", # very different in diet, (mas-related G-protein coupled receptor member )
                        # it seems like the one above just driven by a few birds.
                   "103812721", # insig1, diff in diet(very pretty difference, higher in prot)
                   "103821804", # SLC4A1, important for pH of blood maintenance. not DE but higher in lipid inf
                   "103826733", #CPT1A, top DE gene for diet
  "103813189", #aldh1l2
  "103820599",
"127059388") # gene that maybe differs tolerance from infection.
i <- 1

metadata_phenotypes2 <- metadata_phenotypes_ixn %>% 
  mutate(tolerance_1 = ES_AUC_7/(1+logLoad_AUC_14), 
         tolerance_2 = ES_AUC_7/(1+logLoad_day14),
         tolerance_3 = ES_day7/(1+logLoad_day7),
         tolerance_4 = r_ES_day3/(1+r_logLoad_day3), # maybe best?
         tolerance_5 = (1+r_logLoad_day3)/(1+r_ES_day3)-1, # this format works better for "tolerance"
         #tolerance_6 = (1+logLoad_day14)/(1+r_ES_AUC_14)-1,
         tolerance_0 = 1+logLoad/(1+ES)-1,
         tolerance_01 = scale(tolerance,center=min(tolerance),scale=diff(range(tolerance)))
         ) %>%
  mutate(tolerance_5_cat = dplyr::ntile(tolerance_5,3),
         #tolerance_5_cat = dplyr::ntile(tolerance_5,3),
         # tolerance_5_cat = dplyr::ntile(tolerance_1,3),
         tolerance_cat = dplyr::ntile(tolerance,3),
         tolerance_cat_num = as.numeric(tolerance_cat)) %>%
  mutate(diet.tolerance = paste0(Diet, ".",tolerance_cat)) #%>%
  # group_by(Bird.ID) %>%
  # mutate(tolerance_mean = mean(tolerance))

# plot individual genes
y_ixn %>%
  cpm(log=T) %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  filter(gene == genes_to_plot[i]) %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  # filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene=="103822886") %>% #NPR3
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata_phenotypes2, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet,ES_minMax, ES_day0_analysis,ES, tolerance, logLoad)) %>% #View()
  group_by(BirdID) %>% #View()
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, 0),
                day14 = replace_na(`14`, 0),
                day21 = replace_na(`21`, 0)) %>%
  mutate(diff_21 = (day21) - (day0),
         diff_14 = (day14) - (day0)) %>%
  pivot_longer(c("diff_14", diff_21), names_to = "diff_day", values_to = "counts_diff") %>% #View()
  #filter(day==0) %>%
  #drop_na() %>% View()
  ggplot(aes(x = Diet, y = counts_diff, color = diet, group = "BirdID", size=ES_day7)) +
    geom_point() +
    #ggbeeswarm::geom_quasirandom(alpha=1) +
    # geom_line() +
    facet_wrap(~BirdID) +
    theme_bw()


# plot individual genes (alternate)
y_ixn %>%
  cpm(log=T) %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>% #View()
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  # filter(gene == "103819802") %>% #pctp, top DE gene for diet uninf
  # filter(gene == genes_to_plot[i]) %>%
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  # filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  filter(gene=="103820730") %>% 
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata_phenotypes2, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  #mutate(diff_21 = (day21) - (day0),
         # diff_14 = (day14) - (day0)) %>%  
  pivot_longer(c("day14", "day21", "day0"), names_to = "day", values_to = "counts") %>%  #View()
  # filter(day!="day21") %>%
  arrange(desc(counts)) %>% 
  ggplot(aes(x = fct_reorder(as.factor(BirdID),(as.numeric(as.factor(condition)))), y = counts, fill = MG.Diet, group=day, alpha=ES#, group = "BirdID"#, size=tolerance_cat_num
             )) +
    # geom_point(position = "jitter") +
    # ggbeeswarm::geom_quasirandom(alpha=1) +
    geom_col(position = "dodge") +
    # geom_line() +
    facet_wrap(~Diet, scales = "free_x") +
    # scale_y_log10() +
    theme_classic()
```

```{r}
# plot individual genes (alternate)
y_tolerance %>%
  cpm(log=T) %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  filter(gene == genes_to_plot[i]) %>%
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  #filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata_phenotypes2, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  # mutate(diff_21 = (day21) - (day0),
  # diff_14 = (day14) - (day0)) %>% View()
  pivot_longer(c("day14", "day21", "day0"), names_to = "day", values_to = "counts") %>% #View()
  #filter(day==0) %>%
  ggplot(aes(x = day, y = (counts), color = ES, size=ES, shape=diet, group = "BirdID"#, size=tolerance_cat_num
             )) +
    geom_point() +
    geom_line() +
    ylim(-5,1) +
    # ggbeeswarm::geom_quasirandom(alpha=1) +
    # geom_line() +
    geom_smooth(method="lm") +
    # facet_wrap(~day) +
    # scale_y_log10() +
    theme_classic() + facet_wrap(~BirdID)


# plot individual genes (alternate2)
y_tolerance$counts %>%
  cpm() %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  filter(gene == genes_to_plot[i]) %>%
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  #filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata_phenotypes2, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  #mutate(diff_21 = (day21) - (day0),
         # diff_14 = (day14) - (day0)) %>%
  pivot_longer(c("day14", "day21", "day0"), names_to = "day", values_to = "counts") %>% #View()
  # filter(#day=="day14",
         # MG.Status == "MG") %>% #View()
  ggplot(aes(x = ES, y = log2(counts+1), #color = condition,# shape=day#, 
             group = "BirdID"#
             # , size=tolerance_cat_num
             )) +
    geom_point() +
    geom_line() +
    # ggbeeswarm::geom_quasirandom(alpha=1) +
    # geom_line() +
    geom_smooth(method="lm") +
    # facet_wrap(~day) +
    # scale_y_log10() +
    theme_classic() + 
    facet_wrap(~Diet)
```

temporary
```{r}

# plot individual genes (alternate2)
y_tolerance$counts %>%
  cpm(.,log=TRUE) %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  pivot_longer(-gene, names_to = "sample", values_to = "counts") %>%
  separate_wider_delim(sample, names = c("sampleID","BirdID", "condition-diet", "day"), delim = "_", too_many = "merge") %>%
  separate_wider_delim(`condition-diet`, names = c("condition","diet"), delim = ".", too_many = "merge") %>%
  # filter(gene == "103826733") %>% #CPT1A, top DE gene for diet
  filter(gene == genes_to_plot[i]) %>%
  # filter(gene == "103824949") %>% #bone morph protein (tolerance relevant)
  # filter(gene == "103822409") %>% #FGD2, gene that is expressed in macrophages
  # filter(gene == "103823426") %>%
  # filter(gene == "103813189") %>% # ALDH1L2, biggest FC in DE genes for lipid infection
  # filter(gene == "103823426") %>% #JCHAIN
  # filter(gene == "103827129") %>% #TENT5C
  # filter(gene == "103820477") %>%
  # filter(gene=="103815985") %>%
  # filter(gene=="108962424") %>%
  # filter(gene == "127061207") %>% #Evi5-like (oncogene)
  # filter(gene=="103824071") %>% #BCAN, biggest expression FC for diet, higher in lipid
  # filter(gene == "103817526") %>%  # Ablim3
  left_join(metadata_phenotypes2, by = c("sampleID" = "Sample.ID")) %>%
  dplyr::select(-c(sampleID,Day,MG.Diet.Day, MG.Day,Diet.Day, Exposure.Diet.Day,MG.active,Exposure.MG,Exposure.Diet.Day, Exposure.Diet)) %>%
  group_by(BirdID) %>%
  pivot_wider(names_from = "day", values_from = "counts") %>%
  # mutate(day0 = replace_na(`0`, 0),
  #               day14 = replace_na(`14`, 0),
  #               day21 = replace_na(`21`, 0)) %>%
  mutate(day0 = replace_na(`0`, NA),
                day14 = replace_na(`14`, NA),
                day21 = replace_na(`21`, NA)) %>%
  #mutate(diff_21 = (day21) - (day0),
         # diff_14 = (day14) - (day0)) %>%
  pivot_longer(c("day14", "day21", "day0"), names_to = "day", values_to = "counts") %>% #View()
  # filter(day=="day14") %>%
  filter(day!="day0") %>%
  # filter(MG.Status == "MG") %>% #View()
  ggplot(aes(x = MG.Status,
             y = counts, #color = condition#, 
             shape=day#, group = "BirdID"#, size=tolerance_cat_num
             )) +
    geom_point() +
    # geom_line() +
    # ggbeeswarm::geom_quasirandom(alpha=1) +
    # geom_line() +
    geom_smooth(method="lm") +
    # facet_wrap(~day) +
    # scale_y_log10() +
    theme_classic() + 
    facet_wrap(~Diet)
```


I want to make a bar chart of ribosomal gene's logFC across all of the conditions, let's try it:
```{r}
topTags_inf_edgeR %>%
  left_join(topTags_diet_edgeR,by = c("gene" = "gene", "gene_name" = "gene_name", "logCPM"="logCPM", "description" = "description"), suffix = c("_inf","_diet")) %>%
  left_join(topTags_ixn_edgeR,by = c("gene" = "gene", "gene_name" = "gene_name", "logCPM"="logCPM", "description" = "description")) %>%
  left_join(topTags_tolerance, by = c("gene" = "gene", "gene_name" = "gene_name", #"logCPM"="logCPM", 
                                      "description" = "description"#, "entrezgene_id" = "entrezgene_id"
                                      ), suffix = c("_interaction", "_tolerance")) %>%
  left_join(topTags_ES_inf, by = c("gene" = "gene", "gene_name" = "gene_name", #"logCPM"="logCPM", 
                                   "description" = "description")) %>%
  left_join(topTags_lipid_inf_edgeR,by = c("gene" = "gene", "gene_name" = "gene_name",  "description" = "description"), suffix = c("_ES","_inf_lipid")) %>%
  left_join(topTags_protein_inf_edgeR,by = c("gene" = "gene", "gene_name" = "gene_name",  "description" = "description")) %>%
  mutate(logFC_inf_prot = logFC,
         FDR_inf_prot = FDR) %>%# View()
  left_join(tbl_tol_results, by = c("gene" = "gene.y", "gene_name" = "gene_name_inf", "description" = "description_inf")) %>%
  mutate(logFC_ES_cat = Estimate ) %>%
  filter(str_detect(gene_name, '^RPL|^RPS')) %>%
  dplyr::select(c(gene_name,contains("logFC_"))) %>%
  pivot_longer(-gene_name, names_to = "condition", values_to = "logFC") %>%
  ggplot(aes(x = reorder(gene_name,logFC), y = logFC#, fill = condition
             )) +
  geom_col(position = "dodge") +
  theme_classic() +
  facet_wrap(~condition) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  ggtitle("logFC of Ribosomal genes across comparisons")


ggsave(here("figures/ribosome_genes_fig.png"), width = 50, height = 20, units = "cm", dpi=1200)
```




# Final Work

## Text table files for supplement

Table results for DE genes (all comparisons)
```{r}
topTags_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_inf_all.csv"))
topTags_diet_edgeR %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_diet_all.csv"))
topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_diet_uninf.csv"))
topTags_ixn_edgeR %>% 
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_infxdiet.csv"))
topTags_protein_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_inf_protein.csv"))
topTags_lipid_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_inf_lipid.csv"))
topTags_logLoad_inf %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_pathogen_load.csv"))


tbl_tol_results %>%
  filter(!is.na(logCPM)) %>%
  dplyr::select(entrezgene_id, gene, Estimate, `Std. Error`, `z value`, `Pr(>|z|)`, FDR_tol) %>%
  rename("FDR_tol" = "FDR") %>%
  filter(FDR < 0.05) %>%
  write_excel_csv(., here("supplement/DE_genes/DE_genes_EyeScore.csv"))
```

Tables with all genes included
```{r}
tbl_tol_results %>%
  full_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM"="logCPM", "description" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  left_join(topTags_lipid_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description")) %>%
  left_join(topTags_protein_inf_edgeR, by = c("gene.z" = "gene", "gene_name" = "gene_name", "logCPM_inf"="logCPM", "description" = "description"),suffix = c("_inf_lipid", "_inf_prot"))%>%
  filter(!is.na(logCPM_inf)) %>%
  dplyr::select(
    gene,
    entrezgene_id_inf,
    description,
    Estimate,
    `Std. Error`,
    `z value`,
    `Pr(>|z|)`,
    FDR_tol,
    logCPM_inf,
    logFC_inf,
    FDR_inf,
    logFC_diet,
    FDR_diet,
    logFC_logLoad,
    FDR_logLoad,
    FDR_ixn,
    logFC_inf_lipid,
    FDR_inf_lipid,
    logFC_inf_prot,
    FDR_inf_prot
  ) %>%
  rename("entrezgene_id_inf" = "entrezgene_id" , "Estimate" = "EyeScore_Estimate" ,
         "FDR_tol" = "FDR_EyeScore", "logCPM_inf" = "logCPM") %>%
  unique() %>%
  write_excel_csv(., here("supplement/all_genes_combined_estimates.csv"))
```


## write tables of GO and KEGG outputs

Just one big file.
```{r}
# Top 10 terms for each by p.val
gseGO_inf_df <- gseGO_infection %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_all")

gseGO_diet_df <- gseGO_diet_uninf %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "diet_uninf")

gseGO_ixn_df <- gseGO_ixn %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "interaction_diet:infection")

gseGO_infection_protein_df <- gseGO_infection_protein %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_protein")

gseGO_infection_lipid_df <- gseGO_infection_lipid %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_lipid")

gseGO_logLoad_df <- gseGO_logLoad %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "pathogen_load")

gseGO_ES_df <- gseGO_tol_ord %>%
  data.frame() %>%
  group_by(ONTOLOGY) %>%
  head(n=30) %>%
  mutate(comparison = "eye_score")

gseGO_all_df <- rbind(gseGO_inf_df, gseGO_diet_df, gseGO_ixn_df, gseGO_infection_protein_df, gseGO_infection_lipid_df, gseGO_logLoad_df, gseGO_ES_df)

gseGO_all_df %>%
  write_excel_csv(., here("supplement/functional_enrichments/GO_enrichments.csv"))
  

gseKEGG_inf_df <- gseKEGG_infection %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "infection_all")

gseKEGG_diet_df <- gseKEGG_diet_uninf %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "diet_uninf")

gseKEGG_ixn_df <- gseKEGG_ixn %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "interaction_diet:infection")

gseKEGG_infection_protein_df <- gseKEGG_infection_protein %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "infection_protein")

gseKEGG_infection_lipid_df <- gseKEGG_infection_lipid %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "infection_lipid")

gseKEGG_logLoad_df <- gseKEGG_logLoad %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "pathogen_load")

gseKEGG_ES_df <- gseKEGG_tol_ord %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "eye_score")

gseKEGG_all_df <- rbind(gseKEGG_inf_df, gseKEGG_diet_df, gseKEGG_ixn_df, gseKEGG_infection_protein_df, gseKEGG_infection_lipid_df, gseKEGG_logLoad_df, gseKEGG_ES_df)

gseKEGG_all_df %>%
  write_excel_csv(., here("supplement/functional_enrichments/KEGG_enrichments.csv"))


enrichGO_ES_pos_corr_redness_df <- enrichGO_tol_ord_pos %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "eye_score_pos")


enrichGO_ES_neg_corr_redness_df <- enrichGO_tol_ord_neg %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "eye_score_neg")

enrichGO_all_ES_df <- rbind(enrichGO_ES_pos_corr_redness_df, enrichGO_ES_neg_corr_redness_df)

enrichGO_all_ES_df %>%
  write_excel_csv(., here("supplement/functional_enrichments/GO_ES_enrichments_DE.csv"))
```



Correlation between Inf logFC and ES Estimates
```{r}
tbl_tol_results %>%
  filter(FDR_inf < 0.05) %>%
  # filter(abs(Estimate) > 5 & abs(logFC)>0.05) %>%
  ggplot(aes(x = logFC, y = Estimate)) +
  geom_point() +
  ggrepel::geom_label_repel(aes(label = gene_name_tol), max.overlaps = 20) +
  ggpubr::stat_cor(method = "spearman", label.x = 1, label.y=30) +
  ggpubr::stat_regline_equation(label.x = 1, label.y = 25)+
  # xlim(-8,8) +
  # ylim(-8,8) +
  #geom_label() +
  theme_minimal()
```


```{r}
gseKEGG_logLoad <- topTags_logLoad_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(geneList     = .,
               organism = 'scan',
               #universe = topTags_ixn$gene,
               #minGSSize    = 120,
               pvalueCutoff = 1,
               verbose      = FALSE)
```


# Figure 1

## table to replace the upsetplot
```{r}
tbl_tol_results %>%
  left_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM"="logCPM", "description_tol" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description_tol" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  left_join(topTags_lipid_inf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description")) %>%
  left_join(topTags_protein_inf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description"),suffix = c("_inf_lipid", "_inf_prot")) %>%
  filter(#FDR_tol < 0.05 & 
           FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixn < 0.05 #& 
           & FDR_inf_lipid < 0.05
           # FDR_logLoad < 0.05 
           #|
           # gene_name_tol %in% c("JCHAIN", "EXOG",
           #                    "LOC103823677", "MZB1", "LOC103818060",
           #                    "MYD88", "LOC108961394",
           #                    "PTGDR", "PTGER2", "NEFM", "NEFL")
           ) %>%
  # filter(gene_name_tol %in% c("JCHAIN", "EXOG",
  #                             "LOC103823677", "MZB1", "LOC103818060", 
  #                             "MYD88", "LOC108961394",
  #                             "PTGDR", "PTGER2", "NEFM", "NEFL")) %>%
  dplyr::select(gene_name_tol, description_tol, 
                logFC_inf, 
                logFC_inf_lipid,
                logFC_inf_prot,
                # logFC_logLoad, 
                logFC_diet, 
                logFC_ixn, 
                FDR_inf,
                #Estimate, #logCPM_inf,
                # FDR_tol, 
                FDR_diet,
                FDR_ixn,
                FDR_inf_lipid,
                FDR_inf_prot
                ) %>%
  arrange(desc(logFC_inf)) %>%
  # mutate(across(is.numeric, round, 3)) %>% 
  gt::gt() %>%
  gt::cols_hide(c(FDR_inf, FDR_ixn, FDR_diet, FDR_inf_prot, FDR_inf_lipid)) %>%
  gt::tab_header(title = "Top Shared DE Genes logFC estimates") %>%
  gt::fmt_number(decimals = 2) %>%
  gt::cols_label(gene_name_tol = "Gene Name",
                 description_tol = "Description",
                 logFC_inf = "logFC Infection (all)",
                 logFC_inf_lipid = "logFC Infection (lipid)",
                 logFC_inf_prot = "logFC Infection (protein)",
                 logFC_diet = "logFC Diet (protein-lipid)",
                 logFC_ixn = "logFC Interaction"
                 #Estimate = "Estimate (ES)",
                 #logCPM_inf = "logCPM"
                 ) %>%
  gt::cols_width(gene_name_tol ~ "5em",
                 description_tol ~ "15em",
                 logFC_inf ~ "5em",
                 logFC_inf_lipid ~ "5em",
                 logFC_inf_prot ~ "5em",
                 logFC_diet ~ "5em",
                 logFC_ixn ~ "5em",
                 # Estimate ~ "5em",
                 # logCPM_inf ~ "5em"
                 ) %>%
  gt::data_color(columns = c(logFC_inf, logFC_inf_lipid, logFC_inf_prot, logFC_diet, logFC_ixn),
             fn = scales::col_numeric(palette = c("blue","white", "red"), domain = c(-5,5))) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf,
                                   rows = FDR_inf < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_diet,
                                   rows = FDR_diet < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_ixn,
                                   rows = FDR_ixn < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
                gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf_lipid,
                                   rows = FDR_inf_lipid < 0.05)
  ) %>%
  gt::tab_style(style = 
              list(
            gt::cell_text(weight = "bold")
              ),
            locations = gt::cells_body(columns = logFC_inf_prot,
                                   rows = FDR_inf_prot < 0.05)
  ) %>%
  gtsave(file = here::here("tables/top_genes_ixndietinf_colors.png"))
```


# Figure 4

## upsetplot
upset plot of Inf, Diet, ixn, and ES genes
```{r, fig.width=12, fig.height=16}

lab_size = 2.5
pt_size = 0.9

tbl_tol_results %>%
  left_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM"="logCPM", "description_tol" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description_tol" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  left_join(topTags_lipid_inf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description")) %>%
  left_join(topTags_protein_inf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description"),suffix = c("_inf_lipid", "_inf_prot")) %>% #View()
  mutate(
    Estimate = case_when(
      Estimate > 20 ~ 20,
      Estimate < -20 ~ -20,
      TRUE ~ Estimate
    ),
    est = as.numeric(scale(Estimate)), 
  DE_inf = case_when(
    FDR_inf < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_prot = case_when(
    FDR_inf_prot < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_diet = case_when(
    FDR_diet < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ixn = case_when(
    FDR_ixn < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_logLoad = case_when(
    FDR_logLoad < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ES = case_when(
    FDR_tol < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  goi = case_when(
    gene_name_tol %in% c("JCHAIN","INSRR","NEFM", "ALDH1L2"
                         #"NEFL", "USP46"#,
                         #"LOC103823677", "MZB1", "LOC103818060", 
                         #"MYD88", "LOC108961394",
                         # "INSRR"
                         #"PTGDR", "PTGER2", "NEFM", "NEFL"
                         ) ~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_inf = case_when(
    FDR_inf < 0.05 & abs(logFC_inf)>1~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_inf_prot = case_when(
    FDR_inf_prot < 0.05 & abs(logFC_inf_prot)>1.5~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 & abs(logFC_inf_lipid)>2.5~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_diet = case_when(
    FDR_diet < 0.05 & abs(logFC_diet)>2.5~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_ixn = case_when(
    FDR_ixn < 0.05 & abs(logFC_ixn)>2.5~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_logLoad = case_when(
    FDR_logLoad < 0.05 & abs(logFC_logLoad)>1~ gene_name_tol,
    TRUE ~ NA
  ),
  goi_ES = case_when(
    FDR_tol < 0.05 & FDR_inf < 0.05 & abs(est)>4~ gene_name_tol,
    abs(est) > 2 & FDR_tol < 0.05 & FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixn < 0.05 ~ gene_name_tol,
    TRUE ~ NA
  )
  ) %>% #View()
  upset(., c("DE_inf",#"DE_inf_prot", "DE_inf_lipid", 
             "DE_diet", "DE_ixn", "DE_ES"#, "DE_logLoad"
             ), name="Comparison",
        base_annotations=list(
        'Intersection size'=intersection_size(text = list(size=5, vjust=-0.15),
            text_colors=c(
                on_background='black', on_bar='white'
            )
        )
        + annotate(
            geom='text', x=Inf, y=Inf,
            label=paste('Total:', nrow(topTags_inf_edgeR)),
            vjust=1.2, hjust=1.2, size = 5
        )
        + ylab('# of genes')
    ),
        annotations = list(
        'Infection\n(all)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf,color=-log10(FDR_inf), label=goi#_inf
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 10) + scale_color_continuous(name="-log10(FDR)")
        ),
        'Infection\n(lipid)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf_lipid,color=-log10(FDR_inf_lipid), label=goi#_inf_lipid
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE)+ ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR) ")
        ),
        'Infection\n(protein)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_inf_prot,color=-log10(FDR_inf_prot), label=goi#_inf_lipid
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)  ")
        ),
        'Diet'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_diet,color=-log10(FDR_diet), label=goi#_diet
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)   ")
        ),
        'Interaction\n(Protein-Lipid)'=(
            # note that aes(x=intersection) is supplied by default and can be skipped
            ggplot(mapping=aes(y=logFC_ixn,color=-log10(FDR_ixn), label=goi#_ixn
                               ))
            # checkout ggbeeswarm::geom_quasirandom for better results!
            + ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)    ")
        ),
        'Eye Score'=(
            ggplot(mapping=aes(y=est,color=-log10(FDR_tol),label=goi#_ES
                               ))
            + ggbeeswarm::geom_quasirandom(size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(#min.segment.length = 0.1, 
                                                       direction = "y",size=lab_size, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR)     ")
        )#,
        # 'logLoad'=(
        #     ggplot(mapping=aes(y=logFC_logLoad))
        #     + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.5,
        #         na.rm=TRUE)
        # )
        ),
        keep_empty_groups=TRUE, themes=upset_default_themes(text=element_text(size=18),axis.title=element_text(size=18,face="bold")),
        min_degree=2, min_size = 8,
    # moves legends over the set sizes
    guides='over') & theme(legend.margin=margin(t=0, r=0, b=1, l=0, 'cm'))





ggsave("figures/DE_genes_multinode_ES_combined_upsetplot.png", width = 12, height = 16, units = "in", dpi = 1200)
```


## table of top genes between ES and logFC infection
```{r}
tbl_tol_results %>%
  filter(FDR_tol < 0.05 & FDR_inf < 0.05) %>%
  filter(gene_name_tol %in% c("JCHAIN", "EXOG",
                              "LOC103823677", "MZB1", "LOC103818060", 
                              "MYD88", "LOC108961394",
                              "PTGDR", "PTGER2", "NEFM", "NEFL")) %>%
  dplyr::select(gene_name_tol, description_tol, logFC, #FDR_inf,
                Estimate, logCPM#,FDR_tol, 
                ) %>%
  arrange(desc(logFC)) %>%
  mutate(across(is.numeric, round, 2)) %>%
  gt::gt() %>%
  gt::tab_header(title = "Top Genes between ES and Infection") %>%
  gt::cols_label(gene_name_tol = "Gene Name",
                 description_tol = "Description",
                 logFC = "logFC (Infection)",
                 Estimate = "Estimate (ES)",
                 logCPM = "logCPM") %>%
  gt::cols_width(gene_name_tol ~ "5em",
                 description_tol ~ "15em",
                  logFC ~ "5em",
                 Estimate ~ "5em",
                 logCPM ~ "5em") %>%
  gt::gtsave(file = here("tables/top_genes_ES_inf.png"))
  
```

alternate version that is wider
```{r}
tbl_tol_results %>%
  left_join(topTags_diet_uninf_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM"="logCPM", "description_tol" = "description"), suffix = c("_inf", "_diet")) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name",
                                        #"logCPM"="logCPM", 
                                        "description_tol" = "description"
                                        ),suffix = c("_inf", "_logLoad")
            ) %>%# View()
  left_join(topTags_ixn_edgeR, by = c("gene.z" = "gene", "gene_name_inf" = "gene_name", "logCPM_inf"="logCPM", "description_tol" = "description"),suffix = c("_logLoad", "_ixn")) %>%
  filter(#FDR_tol < 0.05 & 
           FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixn < 0.05 #& 
           # FDR_logLoad < 0.05 
           #|
           # gene_name_tol %in% c("JCHAIN", "EXOG",
           #                    "LOC103823677", "MZB1", "LOC103818060",
           #                    "MYD88", "LOC108961394",
           #                    "PTGDR", "PTGER2", "NEFM", "NEFL")
           ) %>%
  # filter(gene_name_tol %in% c("JCHAIN", "EXOG",
  #                             "LOC103823677", "MZB1", "LOC103818060", 
  #                             "MYD88", "LOC108961394",
  #                             "PTGDR", "PTGER2", "NEFM", "NEFL")) %>%
  dplyr::select(gene_name_tol, description_tol, 
                logFC_inf, 
                # logFC_logLoad, 
                logFC_diet, 
                # logFC_ixn, #FDR_inf,
                Estimate, logCPM_inf#,FDR_tol, 
                ) %>%
  arrange(desc(logFC_inf)) %>%
  mutate(across(is.numeric, round, 2)) %>%
  gt::gt() %>%
  gt::tab_header(title = "Top Genes between ES and Infection") %>%
  gt::cols_label(gene_name_tol = "Gene Name",
                 description_tol = "Description",
                 logFC_inf = "logFC (Infection)",
                 Estimate = "Estimate (ES)",
                 logCPM_inf = "logCPM") %>%
  gt::cols_width(gene_name_tol ~ "5em",
                 description_tol ~ "15em",
                  logFC_inf ~ "5em",
                 Estimate ~ "5em",
                 logCPM_inf ~ "5em")# %>%
  gt::gtsave(file = here("tables/top_genes_all.png"))
```