---
title: "biomaRt canaryGO prep"
author: "Carson Stacy"
date: "2022-09-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(topGO)
library(biomaRt)
```

```{r}
#TODO: CHECK HOW MANY lncRNA show up in our edgeR outputs.
# if they aren't interesting, let's get rid of them and redo our edgeR output.
# also probably throw out the tRNAs anyway.
# I'll check if BiomaRt works for this.

#load in edgeR output to get gene names
# load in edgeR output
MGLpost0 <- readr::read_csv("~/Desktop/canary/contrasts/MGLpost0.csv")
MGpost0 <- readr::read_csv("~/Desktop/canary/contrasts/MGpost0.csv")

# load in InterProScan annotations file

InterProScan_annotations <- readr::read_delim("~/Documents/GitHub/CanarySeq/geneAnnotations/canaryGOviaIPS.tsv", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, na = "NA", trim_ws = TRUE)
#give useful column names
colnames(InterProScan_annotations) <- 
  c(
    'accessionID',
    'seqMD5',
    'seq_len',
    'analysis_type',
    'sign_accession',
    'sign_description',
    'start',
    'stop',
    'e_val',
    'match_status',
    'date_run',
    'IPannot_accession',
    'IPannot_description',
    'GOterms'
  )

#Total of 472k rows, how many unique genes? First let's get the gene names from column 1 transcript IDs

# I commend this out when rerunning due to the call it makes
mart <- useMart(biomart = "ensembl", dataset = "scanaria_gene_ensembl")

transcript_ids <- #InterProScan_annotations$accessionID 
  #note these weren't all of the genes that we have. I'll use edgeR input instead to get them all.
  MGpost0$...1

res <- getBM(attributes = c('ensembl_transcript_id',
                            'ensembl_peptide_id',
                            'ensembl_gene_id', 
                            'uniprot_gn_symbol',
                            'wikigene_name',
                            'entrezgene_id',
                            'external_gene_name',
                            'go_id',
                            'description'
                            ),
             # filters = 'ensembl_transcript_id', 
             values = transcript_ids,
             mart = mart)


# Great, I got GO terms for each gene with this. Need to format them properly.

#first let me get GO terms grouped to gene names
res_min <- res %>% 
  mutate(wikigene_name = ifelse(wikigene_name %in% "", external_gene_name, wikigene_name))%>%
  group_by(ensembl_gene_id) %>%
  mutate(GOterms = paste0(go_id, collapse = ",")) %>%
  slice_head(n=1) %>% # get only one row per gene
  ungroup()
  #mutate(GOterms = gsub("^;+|;+$|(;)+", "\\1", GOterms, perl=TRUE)) # %>%  # tidy up the extra ;;; in Go terms
# res_min <- res2 %>%  slice_head(n=1) %>% 

res_min <- res_min %>%
  mutate(GOterms = gsub("^,+|,+$|(,)+", "\\1", GOterms, perl=TRUE)) %>%  # tidy up the extra ;;; in Go terms
  mutate_all(na_if,"")

# check to see how gene names have GO terms
## external_gene_names
res_min %>% 
  drop_na(GOterms, external_gene_name) %>% 
  pull(external_gene_name) %>% 
  unique() %>% 
  length() #7179

#wikigene names
res_min %>% 
  drop_na(GOterms, wikigene_name) %>% 
  pull(wikigene_name) %>% 
  unique() %>%
  length() #8252
```

Add in new GO terms identified by Blast2GO

```{r}

###NOTE: looks like I NEEDed TO HAVE all gene names, not just the ones from res_min. I'm losing genes from the lack of gene names. I need to map them all to edgeR output instead here!

geneID2GO_combine_names <- res_min[,c(5,7,10)] %>%
  mutate(...1 = ifelse(is.na(res_min$wikigene_name), res_min$external_gene_name, res_min$wikigene_name)) %>%
  dplyr::select(-c(wikigene_name, external_gene_name))

geneID_names <- left_join(MGpost0[1], geneID2GO_combine_names, by = "...1")
colnames(geneID_names)[1] <- "seq_name"

# build the table of original GO terms with gene IDs
geneID2GO_build_BM <- tibble(
                          # canary_annotations$geneID_ncbi,
                          geneID_names$seq_name,
                          "\t",
                          geneID_names$GOterms
                          )

colnames(geneID2GO_build_BM) <- c("seq_name", "\t", "go_i_ds")

# load in Blast2GO results
Blast2GO <- read_delim("~/Desktop/canary/newGOterms_nomatchDEgenes.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE) %>%
  janitor::clean_names() %>%
  dplyr::select(seq_name, go_i_ds) %>%
#change formatting of the GO term seperators to ,
  dplyr::mutate(go_i_ds = str_replace_all(go_i_ds, "; ", ",")) %>%
  dplyr::mutate(go_i_ds = str_replace_all(go_i_ds, "[PFC]:", ""))

combinedGOterms <- left_join(geneID2GO_build_BM, Blast2GO, by = "seq_name")

# now condense the two go_i_ds columns into single column of GO terms.
combinedGOterms <- 
  tidyr::unite(combinedGOterms,
               GOterms,
               go_i_ds.x:go_i_ds.y, 
                 na.rm = TRUE, remove = FALSE,
                 sep = ",")
```

Now let's create files needed to load these in as annotationDBs into topGO. First I am trying to just use the GO terms pulled from BioMart DBs.
```{r}
# canary_annotations_BMwiki <- res_min %>% drop_na(GOterms, wikigene_name)

geneID2GO_build_BM <- tibble(
                          # canary_annotations$geneID_ncbi,
                          seq_name = combinedGOterms$seq_name,
                          "\t",
                          GOterms = combinedGOterms$GOterms
                          )

# write out the geneID2GO_BM.map file
write.table(geneID2GO_build_BM, file="~/Desktop/geneID2GO_BM.map",
            quote = F, row.names = F, col.names = F)
save(geneID2GO_build_BM, file = "~/Documents/GitHub/CanarySeq/geneAnnotations/geneID2GO.RData")

# read in with readMappings
canary_geneID2GO_BM <- topGO::readMappings(file = "~/Desktop/geneid2GO_BM.map")

# save these mappings
save(canary_geneID2GO_BM, file = "~/Desktop/canary/canaryGeneID2GOMappings_BM.RData")
```

Now get Gene expression data. Loaded in from EdgeR
```{r}

#load in canary geneID2GO map
load(file = "~/Desktop/canary/canaryGeneID2GOMappings_BM.RData")

# Numeric ID values (in edgeR output) of genes whose names match Gene Ontology Estimates with edgeR output gene names
matches <- which(toupper(names(canary_geneID2GO_BM)) %in% MGpost0$...1)
          # these names came from `edgeR diff expression.R` file

# Numeric ID values (in eggNOG output) of genes whose names match Gene Ontology Estimates with edgeR output gene names
matches_rnaIndex <- which(MGpost0$...1 %in% toupper(names(canary_geneID2GO_BM)))

```

```{r}
# I don't think i need these.
noMatch_BMwiki <- toupper(names(canary_geneID2GO_BM))[-matches]

noMatch_edgeR.BMwiki <- MGpost0$...1[-matches_rnaIndex]

head(noMatch_edgeR.BMwiki)
```

# Choose DEgenes filter here:

```{r}
#genes that are in both.
# used unique because there are some duplicates in eggNOG output. (per isoform)
genesMatched <- unique(toupper(names(canary_geneID2GO_BM))[matches])

# Identify genes names that are DE in edgeR with FDR < 0.05
# total of 273 first time I ran this with default settings
DEgenes <- filter(MGpost0, FDR < 0.05) %>% pull(`...1`)
DEgenes <- filter(MGpost0, FDR < 0.05 & abs(logFC) > 1) %>% pull(`...1`)
DEgenes <- filter(MGpost0, abs(logFC) > 0.4) %>% pull(`...1`)
DEgenes <- filter(MGpost0, -(logFC) > 0.4 & PValue < 0.05) %>% pull(`...1`)
# DEgenes <- filter(MGpost0, (logFC) > 0.585) %>% pull(`...1`)
# DEgenes <- filter(MGpost0, FDR < 0.05 & logFC < -0.585) %>% pull(`...1`)

# not great results on this one: no FDR sig results
# DEgenes <- filter(MGpost0, logFC > 0.585 & PValue < 0.05) %>% pull(`...1`)
# DEgenes <- filter(MGpost0, logFC < -0.585 & PValue < 0.1) %>% pull(`...1`)
#Fold change of 1.5


# this DEgene gives lot's of good results once adding in B2GO results
# DEgenes <- filter(MGpost0, abs(logFC) > 0.2 ) %>% pull(`...1`)
# surprisingly this one also looks pretty good. Next is seeing if pval effects.
DEgenes <- filter(MGpost0, logFC > 0.2 ) %>% pull(`...1`)
# this resulted in one single GO term being upregulated (inflammation)
DEgenes <- filter(MGpost0, logFC > 0.2 & PValue < 0.05) %>% pull(`...1`)
# now let's do the repressed genes as above
DEgenes <- filter(MGpost0, logFC < -0.2 ) %>% pull(`...1`)
# this resulted in one single GO term being upregulated (inflammation)
DEgenes <- filter(MGpost0, logFC < -0.2 & PValue < 0.05) %>% pull(`...1`)

# try using a different cutoff value for logFC per https://rdrr.io/bioc/edgeR/man/glmTreat.html
# for the upregulated genes, log2(1.1) gives sig results, but log2(1.2) doesn't give fdr corrected sig results.
DEgenes <- filter(MGpost0, logFC > log2(1.2)) %>% pull(`...1`)
DEgenes <- filter(MGpost0, logFC < log2(1.1)) %>% pull(`...1`)
#for downlregulated here, the log(1.1) and log(1.2) give similar results
# DEgenes <- filter(MGpost0, logFC < -log2(1.2)) %>% pull(`...1`)
# DEgenes <- filter(MGpost0, abs(logFC) > log2(1.2)) %>% pull(`...1`)

#count how many of the genes that are DE have GO terms.
# 159
length(which(DEgenes %in% genesMatched))

# genes names with GO terms and DE.
DEgeneswithGO <- which(DEgenes %in% genesMatched)

# select only genes that are DE and have GO terms
DEgenesGO <- DEgenes[DEgeneswithGO]

DEgenes[-DEgeneswithGO]
write.table(DEgenes[-DEgeneswithGO],"~/Desktop/canary/noMatchDEgenes.csv",sep=",",quote=FALSE,row.names=FALSE)
length(DEgenes[-DEgeneswithGO]) #count now that don't have GO terms
tmp <- DEgenes[-DEgeneswithGO]

tmp
res_min$wikigene_name[res_min$wikigene_name %in% tmp]
```




```{r}
#first, let's subset our canary geneIDtoGO to just these genes.
canary_geneID2GO_BM_DE <- #unique(
  canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% DEgenesGO)]
 # )
# canary_geneID2GO_BM_DE <- canary_geneID2GO_BM_DE[!duplicated(canary_geneID2GO_BM_DE)]


names(canary_geneID2GO_BM_DE)[!(DEgenesGO %in% names(canary_geneID2GO_BM_DE))]

#Let me also get my full edgeR gene list.
canary_geneID2GO_BM_edgeR <- canary_geneID2GO_BM[which(toupper(names(canary_geneID2GO_BM)) %in% genesMatched)]
# canary_geneID2GO_BM_edgeR <- canary_geneID2GO_BM_edgeR[!duplicated(canary_geneID2GO_BM_edgeR)]

# weird formatting creating the geneList for topGO. this code seemed to work for me.
all_genes <- factor(as.integer(toupper(MGpost0$...1) %in% DEgenesGO)) # create 0 1 factors.
names(all_genes) <- toupper(MGpost0$...1) # add gene names
str(all_genes)
```

```{r}
#create topGO data obj.
GOdata=new('topGOdata', 
           ontology='BP', # 'BP' = Biological Process. Also tried 'ALL' but doesn't work.
           allGenes = all_genes, # all 10188 genes that are in MGpost0 data from edgeR. factor=1 for those that are DE.
           annot = annFUN.gene2GO, # annotation function for custom annotation
           # gene2GO = canary_geneID2GO_BM # annotation db from eggNOG output.
# I am thinking about switching this for one filtered to just edgeR genes:
           gene2GO = canary_geneID2GO_BM_edgeR # annotation db from BiomaRt.
           # gene2GO = canary_geneID2GO_BM_DE#, 
           # geneSel = DE_genes
           )
```


```{r}
# define test using the weight01 algorithm (default) with fisher
weight_fisher_result=runTest(GOdata, algorithm='weight01', statistic='fisher') 
 
# generate a table of results: we can use the GenTable function to generate a summary table with the results from tests applied to the topGOdata object.
allGO=usedGO(GOdata)
all_res=GenTable(GOdata, weightFisher=weight_fisher_result, orderBy='weightFisher', topNodes=length(allGO), numChar = 200)

#performing BH correction on our p values
p.adj=round(p.adjust(all_res$weightFisher,method="BH"),digits = 4)
 
# create the file with all the statistics from GO analysis
all_res_final=cbind(all_res,p.adj)
all_res_final=all_res_final[order(all_res_final$p.adj),]
 
#get list of significant GO before multiple testing correction
results.table.p= all_res_final[which(as.numeric(all_res_final$weightFisher)<=0.1),]
 
#get list of significant GO after multiple testing correction
results.table.bh=all_res_final[which(all_res_final$p.adj<=0.1),]
 
#save first top 50 ontolgies sorted by adjusted pvalues
# write.table(all_res_final[1:50,],"~/Desktop/canary/summary_topGO_analysis.csv",sep=",",quote=FALSE,row.names=FALSE)
 
# PLOT the GO hierarchy plot: the enriched GO terms are colored in yellow/red according to significance level
 
pdf(file='~/Desktop/canary/topGOPlot_fullnames.pdf', height=12, width=12, paper='special', pointsize=18)
showSigOfNodes(GOdata, score(weight_fisher_result), useInfo = "none", sigForAll=FALSE, firstSigNodes=2,.NO.CHAR=50)
dev.off()
```


TEMPORARY:
```{r}
# save the upregulated results table for log2(1.1)
# up1.1.results.table.p <- tmp.results.table.p
# up1.1.results.table.bh <- tmp.results.table.bh
```

