---
title: "Canary_GenoxPheno"
author: "Carson Stacy"
date: "`r Sys.Date()`"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache=TRUE, cache.lazy = FALSE)
```


```{r libraries, warning=FALSE, message=FALSE}
# ensure BiocManager is installed, install if not
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.16")

#ensure all required packages are installed
# if(!"biomaRt" %in% installed.packages()) BiocManager::install("biomaRt")
if(!"topGO" %in% installed.packages()) BiocManager::install("topGO")
# if(!"edgeR" %in% installed.packages()) BiocManager::install("edgeR")
if(!"tidyverse" %in% installed.packages()) install.packages("tidyverse")
if(!"corrr" %in% installed.packages()) install.packages("corrr")
if(!"MESS" %in% installed.packages()) install.packages("MESS")

library(tidyverse)
library(edgeR)
library(corrr)
# library(topGO)
library(dplyr)
library(tidyr)
```



# Individual Genes of Interest

It looks like the CCR# receptors are enriched in the PROTEIN group but not the lipid group. 
We take a look at how the expression of these genes changed over time for prot vs lipid ctl birds

```{r getDataIn}
g.tpm.l <- read.delim("~/Documents/GitHub/CanarySeq/globinModifiedGTFResults/rsem.merged.gene_tpm.tsv", 
                                          header=TRUE)
row.names(g.tpm.l) <- (g.tpm.l$gene_id)
# View(g.tpm.l)
g.tpm.l$ES89_Blue.049_Control.protein_0 <- NULL #remove the bad sample

############ make the metadata table ######################
meta <- data.frame(SampID = colnames(g.tpm.l[,c(3:106)]))
meta <- meta %>% separate(SampID, 
                          c("Sample.ID", "Bird.ID",
                            "MG.Diet", "Day"), "_", 
                          extra = "warn", fill = "warn", 
                          remove = FALSE) 
# View(meta)

meta <- meta %>% separate(MG.Diet, 
                          c("MG.Status", "Diet"), 
                          sep = "[.]", extra = "warn", 
                          fill = "warn", 
                          remove = FALSE) 

meta$MG.Diet.Day <- paste0(meta$MG.Diet, ".", meta$Day)
meta$MG.Day <- paste0(meta$MG.Status, ".", meta$Day)
meta$Diet.Day <- paste0(meta$Diet, ".", meta$Day)

meta <- meta %>% mutate(MG.active = if_else(Day < 1, "NO", 
                                            if_else(MG.Status == "Control",
                                                    "NO", "YES")))
meta$Exposure.MG <- paste0(meta$MG.active, ".", meta$MG.Status)
meta$Exposure.Diet <- paste0(meta$MG.active, ".", meta$Diet)

# View(meta)
row.names(meta) <- (meta$SampID)
############ diff exp #################################
My.dge<-DGEList(counts=g.tpm.l[,c(3:106)], group=meta$MG.Diet.Day)
summary(My.dge)
#barplot(My.dge$samples$lib.size*1e-6, names=1:ncol(My.dge), ylab="Library size (millions)")

## creates object from table of counts, 
## with a data frame counting information for each sample
# View(My.dge$samples)
#keep <- filterByExpr(My.dge)
keep <- rowSums(cpm(My.dge)>1) >= 6
My.dge <- My.dge[keep, , keep.lib.sizes=FALSE]
My.dge <- calcNormFactors(My.dge)
geneExpr <- My.dge$counts
## uses TMM (weighted trimmed mean of M-values) 
## to calculate normalization factors (i.e. effective library sizes) 
## for each sample.

## rule of thumb, require that genes have at least 10-15 reads 
## in at least one condition to be considered expressed
## removing unexpressed genes increases our statistical power 
## to detect differential expression (because fewer tests)
```
```{r tidy data}
# tidy the data for graphing
tidygeneExpr <- geneExpr %>%
  as_tibble(rownames = "gene") %>%
  pivot_longer(!gene, 
               names_to = c("Lane", "Bird", "MG.Diet", "Day"),
               names_sep = "_",
               values_to = "tpm") %>%
  separate(col = MG.Diet, c("MG", "Diet"))

remove(geneExpr, g.tpm.l, My.dge)
```


See all of the CCR gene expression in control birds
```{r}


# let's look at the CCR# genes now in all control birds.
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  filter(MG == "Control") %>%
  filter(str_detect(gene, 'CCR')) %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day_jitter, 
             y = #tpm,
             log10tpmadd1,
             color = gene)) +
  geom_line(aes(group=interaction(Bird,gene))) +
  geom_point() +
  geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  facet_wrap(~Diet+ MG) +
  theme_classic() 

```



See all of the adaptive immune response gene expression in control birds
```{r warning=FALSE, eval=FALSE}
# load in .Rdata of GOdata_up and GOdata_don in Diet0 contrast
load(file = "~/Documents/GitHub/CanarySeq/GOanalysis/GOdataUPDN_Diet0.RData",
     envir = knitr::knit_global())

head(GOdata_up)

# find out how many contain GOTERM.
GOTERM <- "GO:0007186" #G Protein Coupled Receptor signaling pathway
GOTERM <- "GO:0002250" # adaptive immune response
GOTERM_ <- "GO:0030890" # positive regulation of B cell proliferation

# # select rows with GOTERM
ID2GO_up %>%
  filter(str_detect(name, GOTERM))

GcoupledUP <- genesInTerm(GOdata_up)[GOTERM][[1]][genesInTerm(GOdata_up)[GOTERM][[1]] %in% DEgenes_up]
GcoupledDN <- genesInTerm(GOdata_down)[GOTERM][[1]][genesInTerm(GOdata_down)[GOTERM][[1]] %in% DEgenes_down]

AdImUP <- genesInTerm(GOdata_up)[GOTERM][[1]][genesInTerm(GOdata_up)[GOTERM][[1]] %in% DEgenes_up]
AdImDN <- genesInTerm(GOdata_down)[GOTERM][[1]][genesInTerm(GOdata_down)[GOTERM][[1]] %in% DEgenes_down]

# let's look at the adaptive immune genes now in all control birds.
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  filter(MG == "Control") %>%
  filter(gene %in% AdImUP) %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day_jitter, 
             y = #tpm,
             log10tpmadd1,
             color = gene)) +
  geom_line(aes(group=interaction(Bird,gene))) +
  geom_point() +
  geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet) +
  theme_classic() +
  ggtitle("Adapt immune resp upregulated in Diet0")

# let's look at the Positive B cell reg genes now in all control birds.
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  filter(MG == "Control") %>%
  filter(gene %in% PosBUP) %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day_jitter, 
             y = tpm,
             # log10tpmadd1,
             color = gene)) +
  geom_line(aes(group=interaction(Bird,gene))) +
  geom_point() +
  geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet) +
  theme_classic() +
  ggtitle("pos b cell upregulated in Diet0")
```

It looks like the gene RSAD2 is acting differently between the two diet groups. Let's explore further
```{r}
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  # filter(MG == "Control") %>%
  filter(gene == "RSAD2") %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day, 
             y = tpm,
             # log10tpmadd1,
             fill = Bird)) +
  geom_col(position = "dodge", width = 0.5) +#aes(group=interaction(Bird,gene))) +
  # geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet+MG) +
  theme_classic() +
  guides(fill = FALSE) +
  ggtitle("RSAD2 gene expression lowest in high lipid diet uninfected birds")


```
Look at CD40 of B Cell pos reg GO term
```{r}
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  # filter(MG == "Control") %>%
  filter(gene == "CD40") %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day, 
             y = tpm,
             # log10tpmadd1,
             fill = Bird)) +
  geom_col(position = "dodge", width = 0.5) +#aes(group=interaction(Bird,gene))) +
  # geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet+MG) +
  theme_classic() +
  guides(fill = FALSE) +
  ggtitle("CD40 gene expression")
```

Alright now let us do the same thing but this time for downregulated in this scenario. First GO term I consider relevant would be translation (GO:0006412)
```{r eval=FALSE}
translationGO <- "GO:0006412"# b cell pos reg (Diet post0)

transDN <- topGO::genesInTerm(GOdata_up)[translationGO][[1]][genesInTerm(GOdata_up)[translationGO][[1]] %in% DEgenes_down]

# let's look at the translation genes now in all control birds.
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  # filter(MG == "Control") %>%
  filter(gene %in% transDN) %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day_jitter, 
             # y = tpm,
             y = log10tpmadd1,
             color = gene)) +
  geom_line(aes(group=interaction(Bird,gene))) +
  geom_point() +
  # geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet+MG, nrow = 1) +
  theme_classic() +
  ggtitle("translation genes up in LIPID in Diet0")  +
  guides(color = FALSE) 


tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  # filter(MG == "Control") %>%
  filter(gene == "RPS29") %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day, 
             y = #tpm,
             log10tpmadd1,
             fill = Bird)) +
  geom_col(position = "dodge", width = 0.5) +#aes(group=interaction(Bird,gene))) +
  facet_wrap(~Diet+MG) +
  theme_classic() +
  guides(fill = FALSE) +
  ggtitle("RPS29 gene expression")
```


11/17/22 weekly meeting canary notes
Set the thresholds. Most interesting part is going to be the discussion. What is the (potential) biological meaning of these results. How do top enriched genes relate to the bird outcomes (phenotypic data).

Getting close to outlining and writing up the results. (we can think about where we want it to go and writing it accordingly). Jeff suggests a broad journal to start. Molecular Ecology an option. Also ProcB fxnl ecology. Other similar ones are out there.

Authorship...

Intro and discussion Erin feels good about. Erin make first outline, esp what went into it prior and how we want to frame.
I can handle (results and conclusion??)
Make a figure flow chart. (lot's in the supplement)
Make a Google slides of figures we can go through. I will make this and share with you.
...will also want to put into that the phenotypic data.

I want to have the phenotype data.
Take all the data for each and do pearson corr for each. Use permutation test to get p-value.
This has been done in yeast, a good idea. 

Quantitative measures. We could use as stat the given days values. also the time to recovery or AUC as sum stats.
For path load make sure to log transformation. Also consider log transform and median center.
Note log2 transformation for path load instead of log10.

Think about using slope of decay of path load as variable to fit.

Other variables that they have include: fat scores & mass stuff, but probably not worth looking into. antibody levels didn't look different between the trt groups but maybe worth looking at. Probably do other stuff first. white blood cell stuff don't want to go there.

What makes most sense for our story would be the eye score data. 

```{r}
# let me test a gene with biggest edgeR output diff in diet0
tidygeneExpr %>% 
  # filter(Day == "0" | MG == "Control") %>%
  # filter(MG == "Control") %>%
  filter(gene == "RIMBP2") %>%
  # filter(str_detect(gene, 'LOC108963957')) %>%
  mutate(Day_jitter = (as.numeric(Day) + rnorm(length(Day), sd = 0.2)),
         log10tpmadd1 = log10(tpm+1)) %>%
  ggplot(aes(x=Day, 
             # y = tpm,
             log10tpmadd1,
             fill = Bird)) +
  geom_col(position = "dodge", width = 0.5) +#aes(group=interaction(Bird,gene))) +
  # geom_smooth(method = "lm", aes(group = gene, fill = gene)) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~Diet+MG) +
  theme_classic() +
  guides(fill = FALSE) +
  ggtitle("RIMBP2 gene expression lowest in high lipid diet uninfected birds")
```

# Genotype x Phenotype

## loading in phenotype data
```{r}
eyescores_raw <- readr::read_csv("~/Documents/GitHub/CanarySeq/Geno.x.Pheno/canary_eyescores.csv", 
    col_types = cols(Day = col_integer(), 
        Date = col_character()))


```

```{r}
eyescores_days <- eyescores_raw %>%
  mutate(Bird = stringr::str_replace(ID, " ", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD.19 239", "LD19.239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue.045", "Blue.45")) %>%
  dplyr::select(Bird, Day, ES) #%>%
  # filter(Day %in% c(0,14,21)) # get days we are interested in

eyescores_days14 <- eyescores_days %>%
  filter(Day == 14)

unique(eyescores_days14$Bird) %>% length()
unique(tidygeneExpr$Bird) %>% length()
# we have RNA-seq data for 6 of the 8 birds with phenotype scores.
unique(tidygeneExpr$Bird)[!(unique(tidygeneExpr$Bird) %in% unique(eyescores_days14$Bird))] %>% length()

eyescores_tidy <- eyescores_days %>%
  filter(Bird %in% tidygeneExpr$Bird) %>% # only birds with both phenotype and genotype data
  mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  dplyr::select(-ES) 

eyescores_tidy14 <- eyescores_days14 %>%
  filter(Bird %in% tidygeneExpr$Bird) %>% # only birds with both phenotype and genotype data
  mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  dplyr::select(-ES) #%>% # remove ES column, replaced by 'value'
  # mutate(Lane = NA, 
  #        gene = NA, 
  #        MG=)

geneExpr_wide14 <- tidygeneExpr %>% 
  filter(Bird %in% eyescores_tidy14$Bird) %>% # just correlate with birds with both data types
  # filter(Day %in% c(0,7,14)) %>% # only days with pheno & geno measures
  filter(Day == 14) %>% # alternative, just use this xday for analysis. alt to above line
  # group_by(MG) %>%
  pivot_wider(id_cols = gene, 
              values_from = tpm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df14 <- eyescores_tidy14 %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14)) %>%
  pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

eyescore_row_vec14 <- c(NA, as.numeric(as.vector(eyescore_row_df14[1,-1])))
colnames(eyescore_row_df14) <-  colnames(geneExpr_wide14)

genPhen_dat14 <- rbind(geneExpr_wide14, eyescore_row_df14)

genPhen_dat_t.tmp14 <- as_tibble(cbind(bird = names(genPhen_dat14), t(genPhen_dat14)))

colnames(genPhen_dat_t.tmp14) <- genPhen_dat_t.tmp14[1,]

geno.x.pheno14 <- genPhen_dat_t.tmp14 %>% 
  dplyr::slice(-1) %>%
  hablar::retype()
  
# 
# eyescore_wide <- eyescore_wide_tmp[rep(1, nrow(geneExpr_wide)),]
# 
# 
# 
# 
# geno.x.pheno <- bind_cols(
#   geneExpr_wide,
#   eyescore_wide
# )

```

```{r}
# subset only genes that show some possiblility of DE via edgeR:
# load in edgeR output
MGLpost0 <- readr::read_csv("~/Documents/GitHub/CanarySeq/contrasts/MGLpost0.csv")
```


```{r}
library(corrr)

cor.geno.x.pheno14 <- geno.x.pheno14 %>%
  correlate()

cor.geno.x.pheno.tidy14 <- cor.geno.x.pheno14 %>%
  stretch()

cor.geno.x.pheno.tidy14 %>%
  filter(x == "eye_score") %>%
  arrange(desc(abs(r)))
  
```

It appears there exists a positive correlation between phenotypic eye scores and tpm of LOC103818289 and JCHAIN.

```{r viz key genes Day 14}
# our gene of interest
goi <- "LOC103818289"
goi <- "JCHAIN"
goi <- "LOC108963957"

tidygeneExpr %>%
  filter(gene == goi) %>%
  ggplot(aes(x=Day, y =tpm, color = MG, group = Bird, shape = Diet)) +
  geom_line() +
  geom_point() +
  theme_classic() +
  facet_wrap(~MG)

eyescores_tidy %>%
  ggplot(aes(x=Bird, y =10*value)) +
  geom_bar(data = filter(tidygeneExpr, gene==goi & Day == 14),
           stat= "identity",
           aes(x=Bird, y = tpm), alpha = 0.5, fill = "red"
           ) +
  geom_bar(stat = "identity", fill = "blue", alpha = 0.5) +
  theme_classic() + 
  scale_x_discrete(guide = guide_axis(angle = 70)) #+
  # facet_wrap(~MG)
```

```{r}
tidygeneExpr %>%
  filter(gene %in% c("LOC103818289", "LOC108963957")) %>%
  filter(Day == 14) %>%
  ggplot(aes(x=Bird, y =tpm, fill = gene, group = Bird, shape = Diet)) +
  geom_col(position = "jitter") +
  theme_classic() +
  facet_wrap(~MG) + 
  scale_x_discrete(guide = guide_axis(angle = 70))
```


# Day 21

```{r}

eyescores_days21 <- eyescores_raw %>%
  dplyr::mutate(Bird = str_replace(ID, " ", ".")) %>%
  dplyr::mutate(Bird = str_replace(Bird, "Orange", "orange")) %>%
  dplyr::mutate(Bird = str_replace(Bird, "- ", "")) %>%
  dplyr::mutate(Bird = str_replace(Bird, "LD.19 239", "LD19.239")) %>%
  dplyr::mutate(Bird = str_replace(Bird, "Blue.045", "Blue.45")) %>%
  dplyr::select(Bird, Day, ES) %>%
  # filter(Day %in% c(0,21,21)) # get days we are interested in
  dplyr::filter(Day == 21)

unique(eyescores_days21$Bird) %>% length()
unique(tidygeneExpr$Bird) %>% length()
# we have RNA-seq data for 6 of the 8 birds with phenotype scores.
unique(tidygeneExpr$Bird)[!(unique(tidygeneExpr$Bird) %in% unique(eyescores_days21$Bird))] %>% length()

eyescores_tidy21 <- eyescores_days21 %>%
  dplyr::filter(Bird %in% tidygeneExpr$Bird) %>% # only birds with both phenotype and genotype data
  dplyr::mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  dplyr::select(-ES) 

geneExpr_wide21 <- tidygeneExpr %>% 
  dplyr::filter(Bird %in% eyescores_tidy21$Bird) %>% # just correlate with birds with both data types
  # filter(Day %in% c(0,7,21)) %>% # only days with pheno & geno measures
  dplyr::filter(Day == 21) %>% # alternative, just use this xday for analysis. alt to above line
  # group_by(MG) %>%
  tidyr::pivot_wider(id_cols = gene, 
              values_from = tpm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df21 <- eyescores_tidy21 %>%
  dplyr::distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide21)) %>%
  tidyr::pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  dplyr::mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

eyescore_row_vec21 <- c(NA, as.numeric(as.vector(eyescore_row_df21[1,-1])))
colnames(eyescore_row_df21) <-  colnames(geneExpr_wide21)

genPhen_dat21 <- rbind(geneExpr_wide21, eyescore_row_df21)

genPhen_dat_t.tmp21 <- as_tibble(cbind(bird = names(genPhen_dat21), t(genPhen_dat21)))

colnames(genPhen_dat_t.tmp21) <- genPhen_dat_t.tmp21[1,]

geno.x.pheno21 <- genPhen_dat_t.tmp21 %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

remove(geneExpr, genPhen_dat_t.tmp14, genPhen_dat21)
#corr generation.

cor.geno.x.pheno.tidy21 <- geno.x.pheno21 %>%
  dplyr::select(-gene) %>%
  corrr::correlate(use = 'pairwise.complete.obs') %>%
  # corrr::rearrange(method = "Heatmap") %>%
  corrr::shave() %>%
  corrr::stretch()

remove(cor.geno.x.pheno21)

cor.geno.x.pheno.tidy21 %>%
  dplyr::filter(x == "eye_score") %>%
  dplyr::arrange(desc(abs(r)))
```

```{r viz key genes Day 21}
# some gene of interest for day 14
goi <- "LOC103818289"
goi <- "JCHAIN"
goi <- "LOC108963957"

# these are for day 21:
goi <- "LOC103826451"
goi <- "LOC115485505"
goi <- "YPEL1"
goi <- "LOC103818289"

# DE but don't see:
goi <- "RUBCN"

tidygeneExpr %>%
  filter(gene == goi) %>%
  ggplot(aes(x=Day, y =tpm, color = MG, group = Bird, shape = Diet)) +
  geom_line() +
  geom_point() +
  theme_classic() +
  facet_wrap(~MG)

print(goi)
eyescores_tidy21 %>%
  # make sure that only birds with both are included
  filter(Bird %in% filter(tidygeneExpr, gene==goi & Day == 21)$Bird) %>%
  ggplot(aes(x=Bird, y =10*value)) +
  geom_bar(data = filter(tidygeneExpr, gene==goi & Day == 21),
  # geom_bar(data = filter(tidygeneExpr, gene %in% c("LOC103826451", "LOC115485505","YPEL1") & Day == 21),
           stat= "identity",
           aes(x=Bird, y = tpm), alpha = 0.5, fill = "red"
           ) +
  geom_bar(stat = "identity", fill = "blue", alpha = 0.5) +
  theme_classic() + 
  scale_x_discrete(guide = guide_axis(angle = 70)) +
  ggtitle("red = tpm, blue = eyescore*10") + 
  facet_wrap(~MG)
```

```{r}
# look at all of the interesting relationships between genes before unloading that data
cor.geno.x.pheno.tidy21 %>%
  filter(r <= -0.8) %>%
  arrange(r)
```

```{r}
tidygeneExpr %>%
  filter(gene %in% c("RPS3A", "WHAMM")) %>%
  ggplot(aes(x=Day, y =log10(tpm), color = gene, group = interaction(gene, Bird), shape = Bird)) +
  geom_line() +
  geom_point() +
  theme_classic() +
  facet_wrap(~MG)
```

```{r}
tidygeneExpr %>%
  filter(gene %in% c("OCIAD2", "P2RY1")) %>%
  ggplot(aes(x=Day, y =tpm, color = gene, group = interaction(gene, Bird), shape = Diet)) +
  geom_line() +
  geom_point() +
  theme_classic() +
  facet_wrap(~MG)
```

# Checking correlatioons only for genes that were identified as DE via edgeR in post0 contrast
```{r}
eyescores_tidy14 <- eyescores_days14 %>%
  dplyr::filter(Bird %in% tidygeneExpr$Bird) %>% # only birds with both phenotype and genotype data
  dplyr::mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  dplyr::select(-ES) #%>% # remove ES column, replaced by 'value'
  # mutate(Lane = NA, 
  #        gene = NA, 
  #        MG=)

# # tidy the data for graphing
# tidygeneExpr <- geneExpr %>%
#   as_tibble(rownames = "gene") %>%
#   pivot_longer(!gene, 
#                names_to = c("Lane", "Bird", "MG.Diet", "Day"),
#                names_sep = "_",
#                values_to = "tpm") %>%
#   separate(col = MG.Diet, c("MG", "Diet"))

geneExpr_wide14_DE <- tidygeneExpr %>% 
  dplyr::filter(Bird %in% eyescores_tidy14$Bird) %>% # just correlate with birds with both data types
  # filter(Day %in% c(0,7,14)) %>% # only days with pheno & geno measures
  dplyr::filter(Day == 14) %>% # alternative, just use this xday for analysis. alt to above line
  dplyr::filter(gene %in% filter(MGLpost0, FDR < 0.2)$...1) %>%
  pivot_wider(id_cols = gene, 
              values_from = tpm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df14 <- eyescores_tidy14 %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide14_DE)) %>%
  pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  dplyr::mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

eyescore_row_vec14 <- c(NA, as.numeric(as.vector(eyescore_row_df14[1,-1])))
colnames(eyescore_row_df14) <-  colnames(geneExpr_wide14_DE)

genPhen_dat14_DE <- rbind(geneExpr_wide14_DE, eyescore_row_df14)

genPhen_dat_t.tmp14_DE <- as_tibble(cbind(bird = names(genPhen_dat14_DE), t(genPhen_dat14_DE)))

colnames(genPhen_dat_t.tmp14_DE) <- genPhen_dat_t.tmp14_DE[1,]

geno.x.pheno14_DE <- genPhen_dat_t.tmp14_DE %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_DE <- geno.x.pheno14_DE %>%
  correlate(method = "spearman") %>%
  rearrange() %>%
  shave() 

# cor.geno.x.pheno_DE #%>%
  # rplot() + 
  # scale_x_discrete(guide = guide_axis(angle = 45))
```

```{r}
cor.geno.x.pheno_DE %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))
  
```

```{r}
# genes with the highest correlation (regardless of DE output)
tidygeneExpr %>% # genes that correlate strongly with phenotypic eyescores
  filter(gene %in% c("LOC103818289", 
                     "JCHAIN", 
                     "LOC108963957", 
                     "TENT5C", 
                     "LOC103818060", 
                     "LOC103823243", 
                     "CCR10", 
                     "LOC108963113")) %>%
  mutate(Day = as.numeric(Day)) %>%
  left_join(., eyescores_tidy, by = c("Bird", "Day")) %>% # adding column of eyescore
  ggplot(aes(x=Day, y =log10(tpm), color = MG, group = interaction(gene, Bird))) +
  geom_line() +
  geom_point(aes(size = as.numeric(value))) +
  theme_classic() +
  facet_wrap(~gene)


tidygeneExpr %>% # genes that correlate strongly with phenotypic eyescores
  filter(gene %in% c("LOC103818289", "JCHAIN", "LOC108963957", "TENT5C", "LOC103818060", "LOC103823243", "CCR10")) %>%
  mutate(Day = as.numeric(Day)) %>%
  left_join(., eyescores_tidy, by = c("Bird", "Day")) %>% # adding column of eyescore
  ggplot(aes(x=Day, y =log10(tpm), color = MG, group = interaction(gene, Bird)), size = value) +
  geom_line() +
  geom_point(aes(size = value)) +
  theme_classic() +
  facet_wrap(~gene)

tidygeneExpr %>%
  filter(gene %in% c("PSMA7", "LOC115484857", "LECT2")) %>%
  ggplot(aes(x=Day, y =log10(tpm), color = MG, group = interaction(gene, Bird))) +
  geom_line() +
  geom_point() +
  theme_classic() +
  facet_wrap(~gene)


goi <- "LOC103818289"
eyescores_tidy14 %>%
  # make sure that only birds with both are included
  filter(Bird %in% filter(tidygeneExpr, gene==goi & Day == 14)$Bird) %>%
  ggplot(aes(x=Bird, y =10*value)) +
  geom_bar(data = filter(tidygeneExpr, gene==goi & Day == 14),
  # geom_bar(data = filter(tidygeneExpr, gene %in% c("LOC103826451", "LOC115485505","YPEL1") & Day == 21),
           stat= "identity",
           aes(x=Bird, y = tpm), alpha = 0.5, fill = "red"
           ) +
  geom_bar(stat = "identity", fill = "blue", alpha = 0.5) +
  theme_classic() + 
  scale_x_discrete(guide = guide_axis(angle = 60)) +
  ggtitle("red = tpm, blue = eyescore*10") #+ 
  # facet_wrap(~MG)
```

Other thing I need to try. Look at the correlation of MG vs control. (FC at those time points)
only at MG birds. MGpost0 use the FC to see the correlation. 
Use vs time 0 of the same bird.

The actual response (FC) might be more informative than what I've done so far...


Are certain genes important for clearing pathogens.

Consider the PCR data here as well, I should use the phenotype (qPCR)

See if infection phenotype correlates with some genes but load doesn't. That's an interesting story!

Consider using day14 given high 0 inflation we would expect at day21

# other phen data we have is the antibody data. 

Think about these in terms of different tolerance, not differences in susceptible.


# comparing with log2FC instead of raw data
```{r}
tidygeneFC <- tidygeneExpr %>%
  group_by(Bird, gene) %>%
  summarise(gene = gene, Bird = Bird,
            MG = MG, Diet = Diet,
            FC21 = round((tpm[3]+1)/(tpm[1]+1), 5),
            FC14 = round((tpm[2]+1)/(tpm[1]+1), 5),
            log2FC21 = round(log2(FC21),5),
            log2FC14 = round(log2(FC14),5)
            ) %>%
  distinct()

save(tidygeneFC, file = "~/Documents/GitHub/CanarySeq/Geno.x.Pheno/tidygeneFC.RData")

remove(tidygeneExpr)

eyescores_tidy <- eyescores_days %>%
  filter(Bird %in% tidygeneFC$Bird) %>% # only birds with both phenotype and genotype data
  mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  dplyr::select(-ES) 
```


```{r}

load("~/Documents/GitHub/CanarySeq/Geno.x.Pheno/tidygeneFC.RData",
     envir = knitr::knit_global())


MGLpost0 <- readr::read_csv("~/Documents/GitHub/CanarySeq/contrasts/MGLpost0.csv")
MGPpost0 <- readr::read_csv("~/Documents/GitHub/CanarySeq/contrasts/MGPpost0.csv")
MGd2114 <- readr::read_csv("~/Documents/GitHub/CanarySeq/contrasts/MGd2114.csv")
MGpost0 <- readr::read_csv("~/Documents/GitHub/CanarySeq/contrasts/MGpost0.csv")
```

# the MGpost0 contrast
```{r}
# differentially expressed genes
DE.genes <- MGpost0 %>%
  filter(FDR < 0.2) %>%
  pull(`...1`)

geneExpr_wide14_DE_FC <- tidygeneFC %>% 
  filter(Bird %in% eyescores_tidy$Bird) %>% # just correlate with birds with both data types
  filter(MG == "MG") %>%
  # filter(Diet == "lipid") %>% # when lookat at only birds of a given diet
  drop_na(log2FC14) %>%
  filter(gene %in% DE.genes) %>%
  pivot_wider(id_cols = gene, 
              values_from = log2FC14, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df14 <- eyescores_tidy %>%
  filter(Day == 14) %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

# eyescore_row_vec14 <- c(NA, as.numeric(as.vector(eyescore_row_df14[1,-1])))
colnames(eyescore_row_df14) <-  colnames(geneExpr_wide14_DE_FC)

genPhen_dat14_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_df14)

genPhen_dat_t.tmp14_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_DE_FC), t(genPhen_dat14_DE_FC)))

colnames(genPhen_dat_t.tmp14_DE_FC) <- genPhen_dat_t.tmp14_DE_FC[1,]

geno.x.pheno14_DE_FC <- genPhen_dat_t.tmp14_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_DE_FC <- geno.x.pheno14_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))


# Use to compare top genes for diferrent measures.
day14vsday14.cor <- cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (day 14 eyescore vs day14 FC vs day0)
```{r}

# main pos corr
goi <- "MEF2D" # 0.79, myocyte enhancer factor 2D
# goi <- "THUMPD2" # 0.79, 
# goi <- "LOC103818060" # 0.78, Ig lambda chain C region-like

genPhen_dat_t.tmp14_DE_FC %>%
  dplyr::slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  # add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>% 
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") & eyescore d14 MGpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
# main neg corr
goi <- "LOC108963561" # -0.72, LOC108963561 1-phosphatidylinositol phosphodiesterase-like
# goi <- "RPS17"
goi <- "ABLIM3" # -0.71, LOC103817526 – actin binding LIM protein family member 3

genPhen_dat_t.tmp14_DE_FC %>%
  dplyr::slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  # add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  dplyr::mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") & eyescore D14 MGpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
```

## eye score: alternative measurements

```{r}
eyescores_tidy_summary <- eyescores_tidy %>%
  group_by(Bird) %>%
  summarise(max = max(value),
            auc = MESS::auc(Day,value, type = "spline"),
            auc_daynorm = auc/(max(Day)-min(Day)),
            recovery_time = min(Day[Day > 7 & value == min(value[Day > 7])]) # find day when eyescore is lowest post infection
            )



recovery_trajectory <-
  eyescores_tidy %>%
    group_by(Bird) %>%
    # filter(Day >=2) %>%
    filter(Day >= min(Day[value == max(value)])) %>% # start with point of max infection
    filter(Day <= min(Day[value == 0])) %>% # get rid of trailing 0's
    group_modify(~ broom::tidy(lm(value ~ Day, data = .x, na.action=na.omit), conf.int = TRUE)) %>%
    filter(term == "Day") 

eyescores_tidy_summary$recovery_trajectory <-
  recovery_trajectory$estimate

recovery_trajectory %>%
    left_join(tidygeneFC %>% group_by(Bird) %>% summarise(MG = MG, Diet = Diet) %>% distinct()) %>%
    ggplot(aes(Bird, estimate, 
               # ymin = conf.low, 
               # ymax = conf.high, 
               height = 0, color = MG)) +
      geom_point() +
      theme_classic() + 
      scale_x_discrete(guide = guide_axis(angle = 60)) +
      # geom_vline(xintercept = 0, lty = 4) +
      # geom_errorbar() +
  facet_wrap(~Diet, scales = "free_x")
  


```

### visualize eyescore data

```{r viz eyescore data}
eyescores_tidy %>%
  arrange(Bird) %>%
  left_join(tidygeneFC %>% group_by(Bird) %>% summarise(MG = MG, Diet = Diet) %>% distinct()) %>%
  group_by(Bird) %>%
  filter(MG == "MG") %>%
ggpubr::ggline(x="Day", y="value", color = "Bird", shape = "Diet")
```



## auc norm by day (avg eye score across days)
```{r}
eyescore_row_aucDayNorm <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(auc,max, recovery_trajectory, recovery_time),#id_cols = measure, 
              values_from = auc_daynorm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_aucDayNorm)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC[,!duplicated(colnames(genPhen_dat_t.tmp14_auc_DE_FC))] %>%
  dplyr::slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_aucNorm_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  corrr::correlate(method = "pearson") %>%
  corrr::rearrange() %>%
  corrr::shave() 

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

day14vsAUCnorm.cor <- cor.geno.x.pheno_aucNorm_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```

### viz cool genes (avg score)
```{r}
goi <- "ERP44" # 0.80, LOC103826109 - endoplasmic reticulum protein 44
# goi <- "LOC103818060" # 0.78, Ig lambda chain C region-like
# goi <- "MEF2D" # 0.77, myocyte enhancer factor 2D

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "'average' eye score"
                  )

## same as results from day 14 only comparison
goi <- "LOC108963561" # -0.74, LOC108963561 1-phosphatidylinositol phosphodiesterase-like
# goi <- "ABLIM3" # -0.71, LOC103817526 – actin binding LIM protein family member 3
# goi <- "PHACTR1" # -0.69, LOC103814391 - phosphatase and actin regulator 1

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "'average' eye score"
                  )


```



## auc raw
```{r}
eyescore_row_auc <- eyescores_tidy_summary %>% 
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc_daynorm, recovery_time, recovery_trajectory),#id_cols = measure, 
              values_from = auc, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_auc)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_auc_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

day14vsAUC.cor <- cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (AUC)

imo this measure appears flawed, might not want to use it. depends greatly on the total number of days measured for the bird.
```{r}
goi <- "THUMPD2" # 0.72
# goi <- "CTBP1" # 0.66

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  mutate(gene = as.factor(gene), bird = gene,
         eye_score = as.numeric(eye_score)) %>%
  hablar::retype() %>%
  ggpubr::ggscatterhist(x = 'eye_score',
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(",
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "AUC of eye score"
                  )


goi <- "LOC108963561" # -0.68
# goi <- "LOC103813357" # -0.59 

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "AUC of eye score"
                  )


```



## Trajectory of Recovery
```{r}
eyescore_row_traj <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc_daynorm, auc, recovery_time),#id_cols = measure, 
              values_from = recovery_trajectory, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_traj_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_traj)

genPhen_dat_t.tmp14_traj_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_traj_DE_FC), t(genPhen_dat14_traj_DE_FC)))

colnames(genPhen_dat_t.tmp14_traj_DE_FC) <- genPhen_dat_t.tmp14_traj_DE_FC[1,]


geno.x.pheno14_traj_DE_FC <- genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_traj_DE_FC <- geno.x.pheno14_traj_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 


day14vsTraj.cor <- cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  mutate(gene = case_when(x == "eye_score" ~ y,
         TRUE ~ x)
  ) %>%
  arrange((r))


# positive correlation
cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)


## for negative correlated genes
cor.traj.genes <- cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r)) %>%
  mutate(gene = case_when(x == "eye_score" ~ y,
         TRUE ~ x)
  ) #%>%
  # filter(r < -0.5) %>%
  # pull(gene)
write_delim(cor.traj.genes, file = "~/Documents/GitHub/CanarySeq/GOanalysis/cor.traj.genes.csv")

```


## max eyescore
```{r}
eyescore_row_max <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(auc, auc_daynorm, recovery_time, recovery_trajectory),#id_cols = measure, 
              values_from = max, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_max_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_max)

genPhen_dat_t.tmp14_max_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_max_DE_FC), t(genPhen_dat14_max_DE_FC)))

colnames(genPhen_dat_t.tmp14_max_DE_FC) <- genPhen_dat_t.tmp14_max_DE_FC[1,]


geno.x.pheno14_max_DE_FC <- genPhen_dat_t.tmp14_max_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_max_DE_FC <- geno.x.pheno14_max_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

day14vsMax.cor <- cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# positive correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (max eyescore)
```{r}
goi <- "CNOT8" # 0.70
# goi <- "CUNH5orf51" # 0.69

genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "max eye score"
                  )

goi <- "DES"
# goi <- "LOC115484764"
goi <- "PEAR1"

genPhen_dat_t.tmp14_max_DE_FC %>% 
  dplyr::slice(-1) %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGpost0", sep = ""),
                  xlab = "max eye score"
                  )


```








## Combine top gene lists for the different phenotypes
```{r}
day14vsALL <-
  dplyr::bind_rows(lst(day14vsAUC.cor, 
                       day14vsAUCnorm.cor,
                       day14vsday14.cor,
                       day14vsMax.cor,
                       day14vsTraj.cor),
                   .id = 'id') %>%
  mutate(gene = 
           case_when(
             x == "eye_score" ~ y,
             y == "eye_score" ~ x
             )#,
         # r_health = 
         #   case_when(
         #     id == "day14vsTraj.cor" ~ -r,
         #     TRUE ~ r
         #     )
         )

neg_all_genes <- day14vsALL %>%
  group_by(id) %>%
  filter(id != "day14vsMax.cor", id != "day14vsAUC.cor") %>%
  filter(r < -0.6)

pos_all_genes <- day14vsALL %>%
  group_by(id) %>%
  filter(id != "day14vsMax.cor", id != "day14vsAUC.cor") %>%
  filter(r > 0.6) #%>% View()

split(neg_all_genes$gene, neg_all_genes$id) %>%
   ggVennDiagram::ggVennDiagram() + ggtitle("genes with r < -0.6")

split(pos_all_genes$gene, pos_all_genes$id) %>%
   ggVennDiagram::ggVennDiagram() + ggtitle("genes with r > 0.6")


neg_all_genes7 <- day14vsALL %>%
  group_by(id) %>%
  filter(id != "day14vsMax.cor", id != "day14vsAUC.cor") %>%
  filter(r < -0.7)

pos_all_genes7 <- day14vsALL %>%
  group_by(id) %>%
  filter(id != "day14vsMax.cor", id != "day14vsAUC.cor") %>%
  filter(r > 0.7) #%>% View()

split(neg_all_genes7$gene, neg_all_genes7$id) %>%
   ggVennDiagram::ggVennDiagram() + ggtitle("genes with r < -0.7")

split(pos_all_genes7$gene, pos_all_genes7$id) %>%
   ggVennDiagram::ggVennDiagram() + ggtitle("genes with r > 0.7")

# counts of phenotype per gene passing threshold
neg_all_genes7 %>% group_by(gene) %>% count() %>% arrange(desc(n)) %>% head()
pos_all_genes7 %>% group_by(gene) %>% count() %>% arrange(desc(n)) %>% head()

neg_all_genes %>% group_by(gene) %>% count() %>% arrange(desc(n)) %>% head()
pos_all_genes %>% group_by(gene) %>% count() %>% arrange(desc(n)) %>% head()

# penalized average correlation coefficient per gene.
neg_all_genes7 %>% group_by(gene) %>% count(wt = r/3) %>% arrange(n) %>% head()
pos_all_genes7 %>% group_by(gene) %>% count(wt = r/3) %>% arrange(desc(n)) %>% head()

neg_all_genes %>% group_by(gene) %>% count(wt = r/3) %>% arrange(n) %>% head()
pos_all_genes %>% group_by(gene) %>% count(wt = r/3) %>% arrange(desc(n)) %>% head()



day14vsALL %>%
  group_by(id) %>%
  filter(id != "day14vsMax.cor", id != "day14vsAUC.cor") %>%
  group_by(gene) %>% count(wt = r/3) %>% arrange(desc(n)) %>% head(10)
```

Neg Corr genes: ABLIM phenotype by diet in infected birds
```{r}

goi <- "ABLIM3" #actin binding LIM protein family member 3
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )

goi <- "LOC108963561" #1-phosphatidylinositol phosphodiesterase-like
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )



```
pos corr genes of interest
```{r}


goi <- "LOC103815182" # fatty-acid amide hydrolase 1-like
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )



goi <- "LOC103818269" # Ig lambda chain V-1 region-like
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )


goi <- "LOC103818060" # Ig lambda C region like
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )


goi <- "MEF2D" # myocyte enhancer factor 2D
# goi <- "THUMPD2" #tRNA methylation
genPhen_dat_t.tmp14_traj_DE_FC %>%
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(",
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )


goi <- "JCHAIN" #joining chain of multimeric IgA and IgM
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )
```





## corr of different phenotypes
```{r}
eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  corrr::correlate() %>%
  # corrplot::corrplot() %>%
  corrr::rplot(print_cor = TRUE) +
  ggtitle("Correlation of different phenotype measures of eyescore")
  # corrr::network_plot()
```






# MGLpost0 correlations

Interesting for MGpost0, but it looks like some of the genes correlate more strongly in the lipid vs protein
```{r}
# I want to use biggest net possible, so I don't subset by this contrast.
# might want to reconsider and subset for 'DEish' genes for each contrast
# # differentially expressed genes
# DE.genes <- MGLpost0 %>%
#   filter(FDR < 0.2) %>%
#   pull(`...1`)

geneExpr_wide14_DE_FC <- tidygeneFC %>% 
  filter(Bird %in% eyescores_tidy$Bird) %>% # just correlate with birds with both data types
  filter(MG == "MG") %>%
  filter(Diet == "lipid") %>% # when lookat at only birds of a given diet
  drop_na(log2FC14) %>%
  filter(gene %in% DE.genes) %>%
  pivot_wider(id_cols = gene, 
              values_from = log2FC14, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df14 <- eyescores_tidy %>%
  filter(Day == 14) %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

# eyescore_row_vec14 <- c(NA, as.numeric(as.vector(eyescore_row_df14[1,-1])))
colnames(eyescore_row_df14) <-  colnames(geneExpr_wide14_DE_FC)

genPhen_dat14_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_df14)

genPhen_dat_t.tmp14_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_DE_FC), t(genPhen_dat14_DE_FC)))

colnames(genPhen_dat_t.tmp14_DE_FC) <- genPhen_dat_t.tmp14_DE_FC[1,]

geno.x.pheno14_DE_FC <- genPhen_dat_t.tmp14_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_DE_FC <- geno.x.pheno14_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (day 14 eyescore vs day14 FC vs day0)
```{r}

# main pos corr
goi <- "C2CD2" # 0.93, C2 calcium dependent domain containing 2
# goi <- "TRAPPC10" # 0.88, trafficking protein particle complex subunit 10
# goi <- "LOC115485288" # 0.88, protein-tyrosine kinase 2-beta-like
# goi <- "MEF2D" # 0.85, myocyte enhancer factor 2D

genPhen_dat_t.tmp14_DE_FC %>%
  slice(-1) %>%
  add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
# main neg corr
goi <- "LOC108963561" # -0.96, LOC108963561 1-phosphatidylinositol phosphodiesterase-like
# goi <- "LOC103813357" # -0.82, LOC103813357 - histone H1

goi <- "LOC108963957" # most DE gene mgpost0

genPhen_dat_t.tmp14_DE_FC %>%
  dplyr::slice(-1) %>%
  add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
```

## eye score: alternative measurements

```{r}
eyescores_tidy_summary <- eyescores_tidy %>%
  group_by(Bird) %>%
  summarise( max = max(value),
            auc = MESS::auc(Day,value, type = "spline"),
            auc_daynorm = auc/(max(Day)-min(Day)))
```

## auc norm by day (avg eye score across days)
```{r}
eyescore_row_aucDayNorm <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc),#id_cols = measure, 
              values_from = auc_daynorm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_aucDayNorm)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_auc_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```

### viz cool genes (avg score)
```{r}
goi <- "MEF2D" # 0.88, myocyte enhancer factor 2D
# goi <- "LOC115485288" # 0.87, LOC115485288 - Now named: LOC108964492 - protein tyrosine kinase 2 beta


genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "'average' eye score"
                  )

## same as results from day 14 only comparison
goi <- "LOC108963561" # -0.94, LOC108963561 1-phosphatidylinositol phosphodiesterase-like
goi <- "ABLIM3" # -0.86, LOC103817526 – actin binding LIM protein family member 3
# goi <- "PHACTR1" # -0.80, LOC103814391 - phosphatase and actin regulator 1

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "'average' eye score"
                  )


```



## auc raw
```{r}
eyescore_row_auc <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc_daynorm),#id_cols = measure, 
              values_from = auc, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_auc)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_auc_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


## max eyescore
```{r}
eyescore_row_max <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(auc, auc_daynorm),#id_cols = measure, 
              values_from = max, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_max_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_max)

genPhen_dat_t.tmp14_max_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_max_DE_FC), t(genPhen_dat14_max_DE_FC)))

colnames(genPhen_dat_t.tmp14_max_DE_FC) <- genPhen_dat_t.tmp14_max_DE_FC[1,]


geno.x.pheno14_max_DE_FC <- genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_max_DE_FC <- geno.x.pheno14_max_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (max eyescore)
```{r}
goi <- "RHOC" # 0.92, LOC103823262 - ras homolog family member C
# goi <- "LOC103821046" # 0.84, mesoderm induction early response protein 3-like

genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "max eye score"
                  )

goi <- "LOC108963561" # -0.78, 1-phosphatidylinositol phosphodiesterase-like
# goi <- "LOC115485035" # -0.78, zinc finger protein 239-like

genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "lipid") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGLpost0", sep = ""),
                  xlab = "max eye score"
                  )


```



# MGPpost0 correlations

```{r}
# I want to use biggest net possible, so I don't subset by this contrast.
# might want to reconsider and subset for 'DEish' genes for each contrast
# # differentially expressed genes
# DE.genes <- MGPpost0 %>%
#   filter(FDR < 0.2) %>%
#   pull(`...1`)

geneExpr_wide14_DE_FC <- tidygeneFC %>% 
  filter(Bird %in% eyescores_tidy$Bird) %>% # just correlate with birds with both data types
  filter(MG == "MG") %>%
  filter(Diet == "protein") %>% # when lookat at only birds of a given diet
  drop_na(log2FC14) %>%
  filter(gene %in% DE.genes) %>%
  pivot_wider(id_cols = gene, 
              values_from = log2FC14, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             MG,
                             Bird)
              ) %>%
  dplyr::select(gene, order(names(.))) %>%
  rename_all(~stringr::str_replace(.,"^.*_","")) %>%
  dplyr::select(gene, order(names(.)))

eyescore_row_df14 <- eyescores_tidy %>%
  filter(Day == 14) %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(id_cols = measure, 
              values_from = value, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = measure, .keep="unused") %>%
  dplyr::select(gene, order(names(.)))# %>%
  # select(Bird %in% tidygeneExpr$Bird)

# eyescore_row_vec14 <- c(NA, as.numeric(as.vector(eyescore_row_df14[1,-1])))
colnames(eyescore_row_df14) <-  colnames(geneExpr_wide14_DE_FC)

genPhen_dat14_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_df14)

genPhen_dat_t.tmp14_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_DE_FC), t(genPhen_dat14_DE_FC)))

colnames(genPhen_dat_t.tmp14_DE_FC) <- genPhen_dat_t.tmp14_DE_FC[1,]

geno.x.pheno14_DE_FC <- genPhen_dat_t.tmp14_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_DE_FC <- geno.x.pheno14_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (day 14 eyescore vs day14 FC vs day0)
```{r}

# main pos corr
goi <- "LOC103822417" # 0.96, potassium voltage-gated channel subfamily A member 3
# goi <- "SLC23A1" # 0.96, solute carrier family 23 member 1
# goi <- "TNFRSF1B" # 0.96, TNF receptor superfamily member 1B

genPhen_dat_t.tmp14_DE_FC %>%
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
# main neg corr
goi <- "RASD1" # -0.90, ras related dexamethasone induced 1
# goi <- "GPCPD1" # -0.89, 

genPhen_dat_t.tmp14_DE_FC %>%
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "Day 14 eye score"
                  )
```

## eye score: alternative measurements

```{r}
eyescores_tidy_summary <- eyescores_tidy %>%
  group_by(Bird) %>%
  summarise( max = max(value),
            auc = MESS::auc(Day,value, type = "spline"),
            auc_daynorm = auc/(max(Day)-min(Day)))
```

## auc norm by day (avg eye score across days)
```{r}
eyescore_row_aucDayNorm <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc),#id_cols = measure, 
              values_from = auc_daynorm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_aucDayNorm)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_auc_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```

### viz cool genes (avg score)
```{r}
goi <- "SLC23A1" # 0.95, solute carrier family 23 member 1
# goi <- "TENT5C" # 0.93, terminal nucleotidyltransferase 5C
# goi <- "TNFRSF1B" # 0.92, TNF receptor superfamily member 1B
goi <- "JCHAIN" # 0.90, joining chain of multimeric IgA and IgM


genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "'average' eye score"
                  )

## same as results from day 14 only comparison


goi <- "YPEL5" # -0.92, yippee like 5
goi <- "CASP6" # -0.87, caspase 6
# goi <- "MOB2" # -0.87, 

genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "'average' eye score"
                  )


```



## auc raw
```{r}
eyescore_row_auc <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc_daynorm),#id_cols = measure, 
              values_from = auc, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_auc_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_auc)

genPhen_dat_t.tmp14_auc_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_auc_DE_FC), t(genPhen_dat14_auc_DE_FC)))

colnames(genPhen_dat_t.tmp14_auc_DE_FC) <- genPhen_dat_t.tmp14_auc_DE_FC[1,]


geno.x.pheno14_auc_DE_FC <- genPhen_dat_t.tmp14_auc_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_auc_DE_FC <- geno.x.pheno14_auc_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_auc_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


## max eyescore
```{r}
eyescore_row_max <- eyescores_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(auc, auc_daynorm),#id_cols = measure, 
              values_from = max, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_max_DE_FC <- rbind(geneExpr_wide14_DE_FC, eyescore_row_max)

genPhen_dat_t.tmp14_max_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_max_DE_FC), t(genPhen_dat14_max_DE_FC)))

colnames(genPhen_dat_t.tmp14_max_DE_FC) <- genPhen_dat_t.tmp14_max_DE_FC[1,]


geno.x.pheno14_max_DE_FC <- genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  hablar::retype()

cor.geno.x.pheno_max_DE_FC <- geno.x.pheno14_max_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 

# positive correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange(desc(r))

# negative correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))

# PLOTTING all correlation
cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  ggpubr::gghistogram(., x = "r", fill = "lightgray",
   add = "mean", rug = TRUE, bins = 50)
```


### viz cool genes (max eyescore)
```{r}
goi <- "LOC115484324" # 0.93, importin-4-like (withdrawn)
# goi <- "PPIG" # 0.87, peptidylprolyl isomerase G

genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  # color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "max eye score"
                  )

goi <- "SYNDIG1L" # -0.88, synapse differentiation inducing 1 like
# goi <- "FGF8" # -0.77, fibroblast growth factor 8

genPhen_dat_t.tmp14_max_DE_FC %>% 
  slice(-1) %>%
  add_column(Diet = "protein") %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  # color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and eye_score Day 14 MGPpost0", sep = ""),
                  xlab = "max eye score"
                  )


```



# Assumptions... What is the distribution of logFC for these data. approx normal or??
```{r}
tidygeneFC %>%
  ggplot(aes(x=log2FC14)) +
  geom_histogram()

ggpubr::ggqqplot(tidygeneFC$log2FC14)

eyescores_tidy %>%
  ggplot(aes(x=value)) +
  geom_histogram()
```


# notes from meeting dec 14

fat scores is another phenotype to look at, differs greatly.
DEFINITELY need to use time to recovery as phenotype.
beta crystallin upregulation update: cornea damage 
For my GO_VIZ figs make sure to update titles to be easier to read.


# Storyboarding the paper

Goal: protein bird tolerance (prolonged pathology, but not difference in path loads.. Prev paper)... Why?
Phenotypically, what is the difference?
Pathogen load, clearance not very different but phenotypic pathology very different.
This matters b/c they're blind while v infected, sig potential fit effects. need to find out why.

Look at molecular basis by looking for differences in gene expression.
Way to look at signatures in genomics that best correlate with pathology.

Key bullet point messages: (each should have at least a figure)

- Genomic differences drive expression responses over diet and and infection. We expected diet and inf to drive, but it appears the ?innate? genomic differences do.
- Large number of Diet:Infection interactions.
  How to see the role of diet vs infection vs their interaction.
- How infection, independent of diet, drives expression differences,
  also in this part would be GO term enrichment analyses.
- phenotype correlated effects.
  quantiative phenotypes: AUC, 'average', max
  which phenotype quantiative measures works best? and which genes correlate with those?
    what is that gene list enriched for? (or the top hits)
    
- genes removed from latest annotation??
  
What drives loss in lipid birds might be different than in protein birds(who lose all their fat)

## I think I want to make a PCA plot that is nicely organized.


# pathogen loads PCR data
```{r loads}
canary_loads <- readr::read_csv("~/Documents/GitHub/CanarySeq/Geno.x.Pheno/canary_loads.csv")
```
```{r}
loads_tidy <- canary_loads %>% # only birds with both phenotype and genotype data
  # mutate(measure = "eye_score", value = ES) %>% # allow data to be merged with tidygeneExpr
  mutate(Bird = stringr::str_replace(ID, " ", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Orange", "orange")) %>%
  mutate(Bird = stringr::str_replace(Bird, "- ", "")) %>%
  mutate(Bird = stringr::str_replace(Bird, "-", ".")) %>%
  mutate(Bird = stringr::str_replace(Bird, "LD.19 239", "LD19.239")) %>%
  mutate(Bird = stringr::str_replace(Bird, "Blue.045", "Blue.45")) %>%
  dplyr::filter(Bird %in% unique(tidygeneExpr$Bird)) %>%
  dplyr::select(-ID) 

loads_tidy_summary <- loads_tidy %>%
  group_by(Bird) %>%
  summarise(max = max(logLoad),
            auc = MESS::auc(day,logLoad, type = "spline"),
            auc_daynorm = auc/(max(day)-min(day)),
            recovery_time = min(day[day > 3 & logLoad == min(logLoad[day > 3])]) 
            )


loads_recovery_trajectory <-
  loads_tidy %>%
    group_by(Bird) %>%
    # filter(Day >=2) %>%
    filter(day >= min(day[logLoad == max(logLoad)])) %>% # start with point of max infection
    filter(day <= min(day[logLoad == 0])) %>% # get rid of trailing 0's
    group_modify(~ broom::tidy(lm(logLoad ~ day, data = .x, na.action=na.omit), conf.int = TRUE)) %>%
    filter(term == "day") 

loads_tidy_summary$recovery_trajectory <-
  loads_recovery_trajectory$estimate

pheno_summary <- left_join(
  eyescores_tidy_summary,
  loads_tidy_summary,
  by = "Bird"
)
```


## corrplots
```{r}
pheno_summary %>%
  corrr::correlate(method = "pearson") %>%
  corrr::rplot(print_cor = T, ) + 
  theme(axis.text.x=element_text(angle = 45, hjust = 1)) +
  ggtitle("x = eyescore, y = logLoad of Pathogen")

pheno_summary %>%
  corrr::correlate(method = "pearson") %>%
  corrr::network_plot() +
  ggtitle("  x = eyescore, y = logLoad of Pathogen")


pheno_summary %>%
  corrr::correlate(method = "kendall") %>%
  corrr::rplot(print_cor = T, ) + 
  theme(axis.text.x=element_text(angle = 45, hjust = 1)) +
  ggtitle("x = eyescore, y = logLoad of Pathogen")

pheno_summary %>%
  corrr::correlate(method = "kendall") %>%
  corrr::network_plot() +
  ggtitle("  x = eyescore, y = logLoad of Pathogen")
```

## trajectory of pathogen load correlation to genes
```{r}
loads_row_traj <- loads_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(max, auc_daynorm, auc, recovery_time),#id_cols = measure, 
              values_from = recovery_trajectory, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_traj_DE_FC <- rbind(geneExpr_wide14_DE_FC, loads_row_traj)

genPhen_dat_t.tmp14_traj_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_traj_DE_FC), t(genPhen_dat14_traj_DE_FC)))

colnames(genPhen_dat_t.tmp14_traj_DE_FC) <- genPhen_dat_t.tmp14_traj_DE_FC[1,]


load.geno.x.pheno14_traj_DE_FC <- genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

load.cor.geno.x.pheno_traj_DE_FC <- load.geno.x.pheno14_traj_DE_FC %>%
  correlate(method = "pearson") %>%
  rearrange() %>%
  shave() 


day14vsTraj_loads.cor <- load.cor.geno.x.pheno_traj_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))
```

## visualize day14vsTraj_loads top genes
```{r}

goi <- "LOC103815742" #  interferon-induced GTP-binding protein Mx-like
# goi <- 
genPhen_dat_t.tmp14_traj_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene#,
         # eye_score = replace_na(eye_score, 0)
         ) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and traj path load MGpost0", sep = ""),
                  xlab = "traj logLoad"
                  )

```


### see all translation genes expression
```{r, warning=FALSE}

genPhen_dat_t.tmp14_traj_DE_FC %>%
  dplyr::slice(-1) %>%
  # dplyr::na_if(., abs(.) < 0.5 )
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>% 
  dplyr::mutate(bird = as.factor(gene), 
                Diet = as.factor(Diet),
                .keep="unused",
         # eye_score = replace_na(eye_score, 0)
         ) %>%
  pivot_longer(.,-c(bird, Diet, eye_score), names_to = "gene", values_to = "l2fc") %>%
  # dplyr::filter(gene %in% transDN) %>%# summary()
  dplyr::filter(gene %in% day14vsTraj.cor$gene[(nrow(day14vsTraj.cor)-25):nrow(day14vsTraj.cor)]) %>%
  ggplot(aes(y=eye_score, x=l2fc, color = Diet, group = gene)) +
  geom_point() +
  # stat_smooth(method="lm")
  geom_smooth(method = "lm", #aes(group = gene),
              se = F) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~gene, scales = "free_x") +
  theme_classic() +
  ggtitle("translation genes") # +
  # guides(color = FALSE) 



genPhen_dat_t.tmp14_DE_FC %>%
  dplyr::slice(-1) %>%
  # dplyr::na_if(., abs(.) < 0.5 )
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>% 
  dplyr::mutate(bird = as.factor(gene), 
                Diet = as.factor(Diet),
                .keep="unused",
         # eye_score = replace_na(eye_score, 0)
         ) %>%
  pivot_longer(.,-c(bird, Diet, eye_score), names_to = "gene", values_to = "l2fc") %>%
  # dplyr::filter(gene %in% transDN) %>%# summary()
  dplyr::filter(gene %in% day14vsday14.cor$y[(nrow(day14vsday14.cor)-8):nrow(day14vsday14.cor)] |
                  gene %in% day14vsday14.cor$x[(nrow(day14vsday14.cor)-8):nrow(day14vsday14.cor)]) %>%
  ggplot(aes(y=eye_score, x=l2fc, color = Diet, group = gene)) +
  geom_point() +
  # stat_smooth(method="lm")
  geom_smooth(method = "lm", #aes(group = gene),
              se = F) +
  # geom_smooth(method = "lm", aes(group = Diet, fill = Diet)) +
  facet_wrap(~gene, scales = "free_x") +
  theme_classic() +
  ggtitle("translation genes")
```



# viral loads max corr
```{r}
loads_row_max <- loads_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(recovery_trajectory, auc_daynorm, auc, recovery_time),#id_cols = measure, 
              values_from = max, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_max_DE_FC <- rbind(geneExpr_wide14_DE_FC, loads_row_max)

genPhen_dat_t.tmp14_max_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_max_DE_FC), t(genPhen_dat14_max_DE_FC)))

colnames(genPhen_dat_t.tmp14_max_DE_FC) <- genPhen_dat_t.tmp14_max_DE_FC[1,]


load.geno.x.pheno14_max_DE_FC <- genPhen_dat_t.tmp14_max_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

load.cor.geno.x.pheno_max_DE_FC <- load.geno.x.pheno14_max_DE_FC %>%
  # mutate(across(is.numeric, function(x){replace(x, which(abs(x)<0.1), NA)})) %>%
  correlate(method = "kendall", use = "pairwise.complete.obs") %>%
  rearrange() %>%
  shave() 

# load.cor.geno.x.pheno_max_DE_FC <- load.geno.x.pheno14_max_DE_FC %>%
#   correlate(method = "pearson", use = "pairwise.complete.obs") %>%
#   rearrange() %>%
#   shave() 
  


day14vsMax_loads.cor <- load.cor.geno.x.pheno_max_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  mutate(gene = case_when(x == "eye_score" ~ y,
         TRUE ~ x)
         ) %>%
  dplyr::select(c(gene, r)) %>%
  drop_na() %>%
  arrange((r))
```

```{r}

goi <- "PEAR1" # platelet endothelial aggregation receptor 1
# goi <- "LOC103824010"
genPhen_dat_t.tmp14_max_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene#,
         # eye_score = replace_na(eye_score, 0)
         ) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and max path load MGpost0", sep = ""),
                  xlab = "max path logLoad"
                  )

goi <- "LOC103822941"
goi <- "HSPA5"
goi <- "LOC103824732"
goi <- "JCHAIN"

genPhen_dat_t.tmp14_max_DE_FC %>%
  dplyr::slice(-1) %>%
  # dplyr::na_if(., abs(.) < 0.5 )
  
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  # mutate(across(is.numeric, function(x){replace(x, which(abs(x)<0.1), NA)})) %>%# trying to only correlate highly expressed genes
  mutate(bird = gene#,
         # eye_score = replace_na(eye_score, 0)
         ) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and max path load MGpost0", sep = ""),
                  xlab = "max path logLoad"
                  )
```




# viral loads auc_norm corr
```{r}
loads_row_aucNorm <- loads_tidy_summary %>%
  distinct() %>% # remove any duplicates (currently only orange.24)
  dplyr::filter(Bird %in% colnames(geneExpr_wide14_DE_FC)) %>%
  pivot_wider(!c(recovery_trajectory, max, auc, recovery_time),#id_cols = measure, 
              values_from = auc_daynorm, 
              names_from = c(#Day, #only use this if looking at mulitple days
                             Bird)
              ) %>%
  mutate(gene = "eye_score", .keep="unused") %>%
  dplyr::select(gene, order(names(.)))

genPhen_dat14_aucNorm_DE_FC <- rbind(geneExpr_wide14_DE_FC, loads_row_aucNorm)

genPhen_dat_t.tmp14_aucNorm_DE_FC <- as_tibble(cbind(bird = names(genPhen_dat14_aucNorm_DE_FC), t(genPhen_dat14_aucNorm_DE_FC)))

colnames(genPhen_dat_t.tmp14_aucNorm_DE_FC) <- genPhen_dat_t.tmp14_aucNorm_DE_FC[1,]


load.geno.x.pheno14_aucNorm_DE_FC <- genPhen_dat_t.tmp14_aucNorm_DE_FC %>% 
  dplyr::slice(-1) %>%
  hablar::retype()

load.cor.geno.x.pheno_aucNorm_DE_FC <- load.geno.x.pheno14_aucNorm_DE_FC %>%
  correlate(method = "kendall") %>%
  rearrange() %>%
  shave() 


day14vsaucNorm_loads.cor <- load.cor.geno.x.pheno_aucNorm_DE_FC %>%
  arrange(desc(eye_score)) %>%
  stretch() %>%
  filter(x=="eye_score" | y == "eye_score") %>%
  drop_na() %>%
  arrange((r))
```

```{r}

goi <- "LOC103824839" # E3 ubiquitin-protein ligase KCMF1-like

genPhen_dat_t.tmp14_aucNorm_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene#,
         # eye_score = replace_na(eye_score, 0)
         ) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and avg path load MGpost0", sep = ""),
                  xlab = "aucNorm path logLoad"
                  )

goi <- "RRP9" # 

genPhen_dat_t.tmp14_aucNorm_DE_FC %>% 
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene#,
         # eye_score = replace_na(eye_score, 0)
         ) %>%
ggpubr::ggscatterhist(x = "eye_score",
                  y = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(", 
                                goi,
                                ") and avg path load MGpost0", sep = ""),
                  xlab = "aucNorm path logLoad"
                  )
```



# Curious logFC of E3 ubiquitin-protein ligase KCMF1 (LOC103824839) uninfected birds
```{r}
tidygeneFC %>% filter(gene == "LOC103824839") %>%
  ggplot(aes(x = Bird, y = log2FC14, fill = Diet)) +
  geom_col() +
  facet_wrap(~MG, scales = "free_x") +
  theme_minimal() +
  ggtitle("log2FC of LOC103824839") +
  scale_x_discrete(guide = guide_axis(angle = 45),
                   limits = levels(tidygeneFC$Diet))
```



# make a pretty figure of the eyescore data
```{r}

# trajectory data
eyescore_trajectory_data <- eyescores_tidy %>%
    group_by(Bird) %>%
    # filter(Day >=2) %>%
    filter(Day >= min(Day[value == max(value)])) %>% # start with point of max infection
    filter(Day <= min(Day[value == 0])) %>%
  left_join(tidygeneFC %>% group_by(Bird) %>% summarise(MG = MG, Diet = Diet) %>% distinct()) 

# just visualizing the birds
eyescores_tidy %>%
  arrange(Bird) %>%
  left_join(tidygeneFC %>% group_by(Bird) %>% summarise(MG = MG, Diet = Diet) %>% distinct()) %>%
  left_join(., recovery_trajectory, by = "Bird") %>%
  group_by(Bird) %>%
  dplyr::filter(MG == "MG") %>%
  ggplot(aes(x=Day, y=value, color = Bird, group = Bird, shape = Diet)) +
  geom_point(alpha = 0.7) +
  geom_line(alpha = 0.7) +
  theme_classic() +
  ylim(0,6) +
  ylab("eye_score") +
  ggtitle("Eyescore Data") +
  guides(color="none") +
  facet_wrap(~Diet)


# visualizing all of them
eyescores_tidy %>%
  arrange(Bird) %>%
  left_join(tidygeneFC %>% group_by(Bird) %>% summarise(MG = MG, Diet = Diet) %>% distinct()) %>%
  left_join(., recovery_trajectory, by = "Bird") %>%
  group_by(Bird) %>%
  ggplot(aes(x=Day, y=value, color = MG, group = Bird, shape = Diet)) +
  geom_smooth(data = eyescore_trajectory_data,
    method = "lm", se=F) +
  geom_point(alpha = 0.7) +
  geom_line(alpha = 0.7) + 
  theme_classic() +
  ylim(0,6) +
  ylab("eye_score") +
  ggtitle("Recovery trajectory slope regressions") +
  facet_wrap(~Bird)
  



```

```{r}

goi <- "MEF2D" # myocyte enhancer factor 2D

genPhen_dat_t.tmp14_DE_FC %>%
  dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(y = "eye_score",
                  x = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(",
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "slope of eyescore recovery"
                  )
geno.x.pheno14_DE_FC %>%
  # dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(y = "eye_score (Day 14)",
                  x = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(",
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "Log2 FC Day14 vs Day0",
                  ylab = "Eyescore (Day 14)"
                  )

# now for the loads
load.geno.x.pheno14_max_DE_FC %>%
  # dplyr::slice(-1) %>%
  # add_column(Diet = "lipid") %>%
  add_column(Diet = c("lipid","lipid","protein","lipid",
  "lipid","lipid","protein","protein","protein",
  "protein","lipid","lipid","protein","lipid","protein",
  "lipid","protein","lipid")) %>%
  hablar::retype() %>%
  mutate(bird = gene) %>%
ggpubr::ggscatterhist(y = "eye_score",
                  x = goi,
                  alpha = 0.7,
                  color = "Diet",
                  shape = "Diet",
                  margin.plot = "density",
                  bins = 7,
                  title = paste("Correlation between log2(",
                                goi,
                                ") and eye_score recovery MGpost0", sep = ""),
                  xlab = "Log2 FC Day14 vs Day0",
                  ylab = "MAX pathogen load (log10)"
                  ) +
  geom_smooth(method="lm")

```

