---
title: "Code Supplement Canary-Seq"
author: "Carson Stacy"
date: "2024-07-15"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#pak("eulerr")
library(eulerr)
library(cowplot)
library(ggplot2)
library(tidyverse)
library(ggpubr)
library(here)
library(kableExtra)
library(AnnotationHub)
library(gridtext)
library(graphics)
library(data.table)
library(edgeR)
library(ComplexUpset)
library(ComplexHeatmap)
library(tidyverse)
library(clusterProfiler)
library(flextable)
library(GGally)
library(biomartr)
library(gt)

options(ggrepel.max.overlaps = Inf)
set.seed(123)
# Set hyper-parameter for functional GSEA 
exponent=0.8
```

# Data Preparation
```{r load annotations}
# get annotation hub (orgdb)
ah <- AnnotationHub()
orgs <- subset(ah, ah$rdataclass == "OrgDb")
orgdb <- query(orgs, "Serinus canaria")[[1]]
```

## load custom functions
```{r custom functions}
source(here::here("scripts/canary_data_processing_functions.R"))
```

```{r read-data}
# assign file path location of gene count file
filepath_rsem_gene_counts <- here::here("data/processed", "rsem.merged.gene_counts.tsv")
filepath_rsem_gene_tpm <- here::here("data/processed", "rsem.merged.gene_tpm.tsv")

# read in rsem.merged.gene_counts.tsv
g.counts.l <- data.table::fread(filepath_rsem_gene_counts, sep="\t") %>% as_tibble()
```


## Remove problematic samples

The Sample ES89 was removed due to failed sequencing.
Samples ES31,ES32, and ES65 were removed due to incomplete sample metadata.
Samples from bird SEM19-1481 had their metadata corrected due to an error in recording of diet received.
```{r remove-samples}
g.counts.l <- g.counts.l %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>%
  # Correct diet metadata of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       )
```


## read metadata
```{r}
# see generate_metadata.R script for metadata generation code.
metadata_phenotypes <-
  read_tsv(here("data/raw/sample_metadata/metadata_phenotypes.tsv"))

# annotation compiled in process_gene_annotation.R script.
result_BM_merged <- read_tsv(here::here("data/processed/annotation/results_BM_merged.tsv"))
```

## Filter Hemoglobulin genes
```{r}
# Generate count matrix
gene_expression_matrix <- g.counts.l %>%
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  # convert to matrix for edgeR
  as.matrix()
```


# Differential Expression Analysis with edgeR

```{r}
entrez_names_gene_expression_matrix <-
  gene_expression_matrix |>
  data.frame() |>
  rownames_to_column("external_gene_name") |> 
  left_join(#entrez_IDs |> dplyr::select(-go_id),
            result_BM_merged,
            by = "external_gene_name", multiple = "first") |> #View()
  mutate(entrezID = case_when(
    str_detect(external_gene_name, "^LOC") ~ str_replace(external_gene_name, "^LOC", ""),
    !is.na(entrezgene_id) ~ as.character(entrezgene_id),
    TRUE ~ external_gene_name
  )) |> #View()
  pull(entrezID)

rownames(gene_expression_matrix) <- entrez_names_gene_expression_matrix
```

```{r}
Diet <- as.factor(metadata_phenotypes$Diet)
MG.Active <- as.factor(metadata_phenotypes$MG.Status)


diet_inf <- interaction(Diet, MG.Active) %>% as.factor()
design <- model.matrix(~ 0 + diet_inf)
```

```{r}
y <- DGEList(
  counts = #round(
    gene_expression_matrix
  #               ,0),remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff <- as.numeric(round(cpm(10, mean(y$samples$lib.size)),1))

# apply cutoff
keep <- rowSums(cpm(y) > cpm_cutoff) >= 10
y <- y[keep,]
summary(keep)

y <- calcNormFactors(y)
y <- estimateDisp(y, design, robust=T)

y$samples$group <- #grouping_var
  diet_inf %>% as.factor()

contrast <- makeContrasts(
  #design, #metadata,
  contrasts = c(
    # Interaction contrast
    # infection response in protein - infection response in lipid
    "(diet_infprotein.MG - diet_infprotein.Control) - (diet_inflipid.MG - diet_inflipid.Control)",
    # Diet Contrast
    # Protein - lipid
    "(diet_infprotein.MG + diet_infprotein.Control)/2-(diet_inflipid.MG + diet_inflipid.Control)/2",
    # Infection contrast
    # Infected - Uninfected
    "(diet_infprotein.MG + diet_inflipid.MG)/2-(diet_infprotein.Control + diet_inflipid.Control)/2",
    # Infection in Protein Birds
    # infected protein - uninfected protein
    "diet_infprotein.MG - diet_infprotein.Control",
    # Infection in Lipid Birds
    # infected lipid - uninfected lipid
    "diet_inflipid.MG - diet_inflipid.Control",
    # Uninfected Diet Contrast
    # Protein - lipid
    "(diet_infprotein.Control)-(diet_inflipid.Control)"
  ),
  levels = design
)

fit <- glmQLFit(y, design)
```

## Infection x Diet interaction
```{r}
res_ixd_edgeR <- glmQLFTest(fit, contrast=contrast[,1])

topTags_ixd_edgeR <- topTags(res_ixd_edgeR, n=Inf) %>% 
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```

## Diet (Protein vs Lipid) (only uninfected samples)
```{r}
res_diet_edgeR <- glmQLFTest(fit, contrast=contrast[,2])

topTags_diet_edgeR <- topTags(res_diet_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)

res_diet_uninf_edgeR <- glmQLFTest(fit, contrast=contrast[,6])

topTags_diet_uninf_edgeR <- topTags(res_diet_uninf_edgeR, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```


## Changes in expression following MG exposure
```{r}
res_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,3])

topTags_inf_edgeR <- topTags(res_inf_edgeR, n=Inf) %>% 
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description",
      "entrezgene_id"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)

# protein bird response to infection
res_protein_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,4])

topTags_protein_inf_edgeR <- topTags(res_protein_inf_edgeR, n=Inf) %>%
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)


# Lipid bird infection response
res_lipid_inf_edgeR <- glmQLFTest(fit, contrast=contrast[,5])

topTags_lipid_inf_edgeR <- topTags(res_lipid_inf_edgeR, n=Inf) %>%
  data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select(
    c(
      "gene",
      "gene_name",
      "logFC",
      "logCPM",
      "F",
      "PValue",
      "FDR",
      "description"
    )
  ) %>%
  distinct(gene, .keep_all = TRUE)
```

## Pathogen Load Modeling

Using Pathogen load levels as regression covariate for gene expression.

```{r}
logLoad <- metadata_phenotypes$logLoad_alt
Diet <- metadata_phenotypes$Diet

design_logLoad <- model.matrix(~ 0 + logLoad * Diet)

y_logLoad <- DGEList(
  counts = #round(
    gene_expression_matrix#),
  #remove.zeros = TRUE
)

#determine cpm cutoff
cpm_cutoff_logLoad <- as.numeric(round(cpm(10, mean(y_logLoad$samples$lib.size)),1))

# apply cutoff
keep_logLoad <- rowSums(cpm(y_logLoad) > cpm_cutoff_logLoad) >= 10
y_logLoad <- y_logLoad[keep_logLoad,]

y_logLoad <- calcNormFactors(y_logLoad)
y_logLoad <- estimateDisp(y_logLoad, design_logLoad)
fit_logLoad <- glmQLFit(y_logLoad, design_logLoad)
res_logLoad_inf <-  glmQLFTest(fit_logLoad, coef="logLoad")


# log pathogen Load
topTags_logLoad_inf <- topTags(res_logLoad_inf, n=Inf) %>% data.frame() %>%
  arrange(FDR) %>%
  rownames_to_column("gene") %>%
  left_join(result_BM_merged,
            by = "gene") %>%
  dplyr::select( c("gene", "gene_name", "logFC", "logCPM", "F",
      "PValue", "FDR", "description", "entrezgene_id") ) %>%
  distinct(gene,.keep_all = TRUE)
```

## Merge Results
```{r}
genes_higher_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_lower_in_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_protein_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_lipid_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>% 
  deframe() 

genes_bigger_protein_infection_difference <- topTags_ixd_edgeR %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_bigger_lipid_infection_difference <- topTags_ixd_edgeR %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe() 

genes_higher_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC > 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

genes_lower_in_logLoad <- topTags_logLoad_inf %>%
  filter(FDR < 0.05 & logFC < 0) %>%
  dplyr::select(gene, logFC) %>%
  deframe()

fdr_genes_infection <- topTags_inf_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

fdr_genes_ixd <- topTags_ixd_edgeR %>% 
  filter(FDR < 0.05) %>%
  dplyr::select(gene, FDR) %>%
  deframe()

DE_genes_edgeR_upset <- topTags_inf_edgeR %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixd_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixd_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),

         higher_in_logLoad = gene %in% names(genes_higher_in_logLoad),

         lower_in_logLoad = gene %in% names(genes_lower_in_logLoad),
        logFC_infection = 
          case_when(
            gene %in% names(genes_lower_in_infection) ~ genes_lower_in_infection[gene],
            gene %in% names(genes_higher_in_infection) ~ genes_higher_in_infection[gene],
            TRUE ~ NA_real_
          ),
        logFC_diet = case_when(
          gene %in% names(genes_higher_protein_diet) ~ genes_higher_protein_diet[gene],
          gene %in% names(genes_higher_lipid_diet) ~ genes_higher_lipid_diet[gene],
          TRUE ~ NA_real_
        ),
        logFC_interaction = case_when(
          gene %in% names(genes_bigger_protein_infection_difference) ~ genes_bigger_protein_infection_difference[gene],
          gene %in% names(genes_bigger_lipid_infection_difference) ~ genes_bigger_lipid_infection_difference[gene],
          TRUE ~ NA_real_
        ),
        FDR_infection = case_when(
          gene %in% names(fdr_genes_infection) ~ fdr_genes_infection[gene],
          TRUE ~ NA_real_
        ),
        FDR_diet = case_when(
          gene %in% names(fdr_genes_diet) ~ fdr_genes_diet[gene],
          TRUE ~ NA_real_
        ),
        FDR_interaction = case_when(
          gene %in% names(fdr_genes_ixd) ~ fdr_genes_ixd[gene],
          TRUE ~ NA_real_
        ),
        logFC_logLoad = case_when(
          gene %in% names(genes_higher_in_logLoad) ~ genes_higher_in_logLoad[gene],
          gene %in% names(genes_lower_in_logLoad) ~ genes_lower_in_logLoad[gene],
          TRUE ~ NA_real_
        )
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixd = Protein_ixd_bigger_Inf | Lipid_ixd_bigger_Inf,
         DE_in_logLoad = higher_in_logLoad | lower_in_logLoad) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ))


compile results.
```{r}
merged_results <- tbl_tol_results %>% unique() %>%
  dplyr::select(-c(logFC,logCPM,F,PValue, FDR_inf)) %>%
  left_join(.,topTags_inf_edgeR, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#, "description" = "description"
                                      )
            #, suffix=c("","")
            ) %>%
  left_join(topTags_diet_uninf_edgeR, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#, #"logCPM"="logCPM", 
                                             #"description" = "description"
                                             ), suffix = c("_inf", "_diet")) %>%
  #mutate(FDR_diet = FDR) %>%
  #dplyr::select(-FDR) %>%
  left_join(topTags_logLoad_inf, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#,
                                        #"logCPM"="logCPM", 
                                       # "description" = "description"
                                        )#, suffix = c("", "")
            ) %>%
  left_join(topTags_ixd_edgeR, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#, #"logCPM_inf"="logCPM",
                                     # "description" = "description"
                                      ),suffix = c("_logLoad", "_ixd")) %>%
  left_join(topTags_lipid_inf_edgeR, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#, #"logCPM_inf"="logCPM",
                                            #"description" = "description"
                                            )#, suffix = c("", "_inf_lipid")
            ) %>%
  left_join(topTags_protein_inf_edgeR, multiple="first", by = c("gene.z" = "gene", "gene_name" = "gene_name"#, #"logCPM_inf"="logCPM", 
                                             # "description" = "description"
                                             ),suffix = c("_inf_lipid", "_inf_prot")) %>%
  mutate(
    Estimate = case_when(
      Estimate > 20 ~ 20,
      Estimate < -20 ~ -20,
      TRUE ~ Estimate
    ),
    est = as.numeric(scale(Estimate)), 
  DE_inf = case_when(
    FDR_inf < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_prot = case_when(
    FDR_inf_prot < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_diet = case_when(
    FDR_diet < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ixd = case_when(
    FDR_ixd < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_logLoad = case_when(
    FDR_logLoad < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  DE_ES = case_when(
    FDR_tol < 0.05 ~ TRUE,
    TRUE ~ FALSE
  ),
  goi = case_when(
    gene_name %in% c("JCHAIN","INSRR","NEFM", "ALDH1L2"
                         ) ~ gene_name,
    TRUE ~ NA
  ),
  goi_inf = case_when(
    FDR_inf < 0.05 & abs(logFC_inf)>1~ gene_name,
    TRUE ~ NA
  ),
  goi_inf_prot = case_when(
    FDR_inf_prot < 0.05 & abs(logFC_inf_prot)>1.5~ gene_name,
    TRUE ~ NA
  ),
  goi_inf_lipid = case_when(
    FDR_inf_lipid < 0.05 & abs(logFC_inf_lipid)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_diet = case_when(
    FDR_diet < 0.05 & abs(logFC_diet)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_ixd = case_when(
    FDR_ixd < 0.05 & abs(logFC_ixd)>2.5~ gene_name,
    TRUE ~ NA
  ),
  goi_logLoad = case_when(
    FDR_logLoad < 0.05 & abs(logFC_logLoad)>1~ gene_name,
    TRUE ~ NA
  ),
  goi_ES = case_when(
    FDR_tol < 0.05 & FDR_inf < 0.05 & abs(est)>4~ gene_name,
    abs(est) > 2 & FDR_tol < 0.05 & FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixd < 0.05 ~ gene_name,
    TRUE ~ NA
  )
  )  %>% unique()
```



# Gene Set Enrichment Analysis of DE genes

Grouped by comparison
```{r}
#gseGO
gseGO_ixd <- topTags_ixd_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    exponent     = exponent,
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

gseGO_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    exponent     = exponent,
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

gseGO_diet <- topTags_diet_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    exponent      = exponent,
    minGSSize     = 11,
    seed          = 123,
    ont           = "BP",
    pAdjustMethod = "BH",
    pvalueCutoff  = 1
  )

gseGO_diet_uninf <- topTags_diet_uninf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList      = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    exponent      = exponent,
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )



gseGO_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

gseGO_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

gseGO_logLoad <- topTags_logLoad_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "BP",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

# gseKEGG

gseKEGG_ixd <- topTags_ixd_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    exponent = exponent,
    minGSSize     = 11,
    seed         = 123,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

gseKEGG_infection <- topTags_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism     = 'scan',
    exponent     = exponent,
    seed         = 123,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

gseKEGG_infection_lipid <- topTags_lipid_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    exponent = exponent,
    pvalueCutoff = 1,
    seed         = 123,
    verbose      = FALSE
  )

gseKEGG_infection_protein <- topTags_protein_inf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    exponent = exponent,
    seed          = 123,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

gseKEGG_diet_uninf <- topTags_diet_uninf_edgeR %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    exponent = exponent,
    seed          = 123,
    pvalueCutoff = 1,
    verbose      = FALSE
  )

gseKEGG_logLoad <- topTags_logLoad_inf %>%
  arrange(desc(logFC)) %>%
  pull(logFC, gene) %>%
  gseKEGG(
    geneList     = .,
    organism = 'scan',
    exponent = exponent,
    pvalueCutoff = 1,
    seed         = 123,
    verbose      = FALSE
  )
```

# Ordinal Logistic Regression of Eye Score
```{r}
tpm_data <- data.table::fread(filepath_rsem_gene_tpm, sep="\t") %>% as_tibble() %>%
  dplyr::select(-c(
      "ES89_Blue-049_Control-protein_0",
      "ES31_Blue-055_NA_0",
      "ES32_Blue-055_NA_14",
      "ES65_106_NA_0" )) %>% 
  # dplyr::select(-contains("Blue-050")) %>%
  # change diet group of bird SEM19-1481 from MG-lipid to MG-protein
  dplyr::rename(all_of(
       c(`ES64_SEM19-1481_MG-protein_21`="ES64_SEM19-1481_MG-lipid_21",
         `ES52_SEM19-1481_MG-protein_14`="ES52_SEM19-1481_MG-lipid_14",
         `ES83_SEM19-1481_MG-protein_0`="ES83_SEM19-1481_MG-lipid_0"))
       ) %>%
  # remove globin genes
  dplyr::filter(gene_id != "LOC103824466" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  dplyr::filter(gene_id != "LOC103817749" ) %>%
  dplyr::filter(gene_id != "LOC115485052" ) %>%
  dplyr::filter(gene_id != "LOC103817748" ) %>%
  # Add rownames
  column_to_rownames("gene_id") %>%
  # remove unneeded column
  dplyr::select(-`transcript_id(s)`) %>% 
  # convert to matrix for edgeR
  as.matrix()
```

This analysis can take a significant amount of time to complete.
```{r}
dummy <- tpm_data %>% cpm(log = TRUE) %>% 
  t() %>%
  data.frame() %>% rownames_to_column("sample") %>%
  left_join(metadata_phenotypes, by=c("sample"="SampID")) %>%
  mutate(ES = as.ordered(as.factor(ES)),
         tolerance_ord = as.ordered(as.factor(tolerance))
         )

library(tidyverse)
library(VGAM)

mod_f <- function(x) {
  vglm(
    formula = as.formula(x),
    family = cumulative(parallel = TRUE,
                        reverse = FALSE),
    data = dummy
  )
}

## get genes that are included in edgeR analysis:
edgeR_analyzed_genes <- topTags_inf_edgeR %>% 
  left_join(mutate(result_BM_merged,entrezgene_id= as.character(entrezgene_id)), by=c("gene" = "entrezgene_id"),
            multiple="first"
            ) %>% 
  mutate(match_gene_tmp = case_when(
    str_detect(gene,"^1") ~ str_replace(gene, "^1", "LOC1"),
    TRUE~gene
  )) %>%
  mutate(match_gene = case_when(
    match_gene_tmp %in% rownames(tpm_data) ~ match_gene_tmp,
    gene_tag %in% rownames(tpm_data) ~ gene_tag,
    TRUE~NA
  )) %>%
  arrange(desc(logFC)) %>%
  pull(match_gene, name = gene)

n <- length(unique(edgeR_analyzed_genes))

formulas_tolerance <-
  c(paste0("ES~Diet+logLoad+", rownames(tpm_data)[rownames(tpm_data) %in% edgeR_analyzed_genes]))[1:n]

tbl_tolerance <- tibble(forms = formulas_tolerance)

## These take about an hour to run on m1 macbook air. Can load output below.
# tbl_2_tol <-
#   tbl_tolerance %>% mutate(
#     all = map(formulas_tolerance, mod_f) %>%
#       map(summary) %>%
#       map(coef) %>%
#       map(as_tibble, rownames = "gene")
#   )

# tbl_tol_results <- tbl_2_tol$all %>% set_names(rownames(tpm_data)[1:n]) %>%
#   dplyr::bind_rows() %>%
#   filter(!str_detect(gene, "Intercept")) %>%
#   left_join(result_BM_merged, by=c("gene" = "gene_tag"), multiple="first") %>%
#   mutate(FDR = p.adjust(`Pr(>|z|)`, method="fdr")) %>%
#   filter(entrezgene_id %in% topTags_inf_edgeR$entrezgene_id) %>%
#   filter(gene != "Dietprotein" & gene != "logLoad") %>%
#   mutate(gene.z = case_when(
#     gene.y %in% topTags_inf_edgeR$gene ~ gene.y,
#     gene %in% topTags_diet_uninf_edgeR$gene ~ gene,
#     gene %in% paste0("LOC",topTags_diet_edgeR$gene) &
#       is.na(entrezgene_id) ~ str_remove(gene, "LOC"),
#     TRUE ~ NA
#   )) %>%
#     filter(entrezgene_id %in% topTags_inf_edgeR$entrezgene_id & gene.z %in% topTags_inf_edgeR$gene) %>%
#   right_join(topTags_inf_edgeR, by=c("entrezgene_id" = "entrezgene_id", "description" = "description", "gene.z" = "gene", "gene_name" = "gene_name"
#                                     ),
#              multiple="first",
#             suffix = c("_tol", "_inf")) %>% unique()

# write_tsv(tbl_tol_results, here::here("results/tables/all_genes/tbl_ordinal_regression_results.tsv"))

tbl_tol_results <- read_tsv(here::here("results/tables/all_genes/tbl_ordinal_regression_results.tsv"))
```

Functional enrichments on eye score analysis results

```{r}
# enrichKEGG_tol_ord2 <- enrichKEGG(gene = tbl_tol_results %>%
#   filter(FDR_tol < 0.05) %>%
#   #filter(`Pr(>|z|)` < 0.0005) %>%
#   pull(entrezgene_id)  ,
#                organism = 'scan', 
#   keyType = "kegg",
#                universe = topTags_diet_uninf_edgeR$gene,
#                pvalueCutoff = 0.9,
#                 qvalueCutoff  = 0.9)
```

```{r}
# split this into up vs down
# enrichGO_tol_ord2_up <- tbl_tol_results %>%
#   filter(FDR_tol < 0.05 & Estimate > 0) %>%
#   #filter(`Pr(>|z|)` < 0.0005) %>%
#   pull(entrezgene_id) %>%
#   enrichGO(     gene          = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "ALL",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 qvalueCutoff  = 1) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))
# 
# enrichGO_tol_ord2_down <- tbl_tol_results %>%
#   filter(FDR_tol < 0.05 & Estimate < 0) %>%
#   #filter(`Pr(>|z|)` < 0.0005) %>%
#   pull(entrezgene_id) %>%
#   enrichGO(     gene          = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "ALL",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 qvalueCutoff  = 1) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))
# 
# enrichGO_tol_ord2 <- tbl_tol_results %>%
#   filter(FDR_tol < 0.05) %>%
#   pull(entrezgene_id) %>%
#   enrichGO(     gene          = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "ALL",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 qvalueCutoff  = 1) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))
# 
# enrichGO_ES_not_inf <- merged_results %>%
#   filter(FDR_inf > 0.05 & FDR_tol < 0.05) %>% 
#   pull(entrezgene_id) %>%
#   enrichGO(     gene          = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "BP",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 qvalueCutoff  = 1) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))
# 
# 
# 
# enrichGO_inf_not_ES <- merged_results %>%
#   filter(FDR_inf < 0.05 & FDR_tol > 0.05) %>% 
#   pull(entrezgene_id) %>%
#   enrichGO(     gene          = .,
#                 OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "BP",
#                 pAdjustMethod = "BH",
#                 pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 qvalueCutoff  = 1) |>
#   mutate(richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))

gseGO_tol_ord2 <- tbl_tol_results %>% 
  arrange(desc(`z value`)) %>%
  filter(!is.na(`z value`)) %>%
  pull(`z value`, entrezgene_id) %>%
  gseGO(
    geneList     = .,
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    exponent      = exponent,
    ont           = "ALL",
    pAdjustMethod = "BH",
    minGSSize     = 11,
    seed          = 123,
    pvalueCutoff  = 1
  )

gseKEGG_tol_ord2 <- tbl_tol_results %>%
  arrange(desc(`z value`)) %>%
  filter(!is.na(`z value`)) %>%
  pull(`z value`, entrezgene_id) %>%
  gseKEGG(
    geneList     = .,
    organism     = 'scan',
    exponent      = exponent,
    pvalueCutoff = 1,
    seed         = 123,
    verbose      = FALSE
  )
```


## Joint Functional Enrichment in All Comparisons
```{r}
all_de_genes_ordered = list(
  "Infection (all)" = merged_results %>% arrange(desc(logFC_inf)) %>%
    dplyr::select(logFC_inf, entrezgene_id.x) %>% 
    drop_na() %>% pull(logFC_inf, name = entrezgene_id.x),
  "Diet" = merged_results %>% arrange(desc(logFC_diet)) %>% 
    dplyr::select(logFC_diet, entrezgene_id.x) %>% 
    drop_na() %>% pull(logFC_diet, name = entrezgene_id.x),
  "Infection (protein)" = merged_results %>% arrange(desc(logFC_inf_prot)) %>%
    dplyr::select(logFC_inf_prot, entrezgene_id.x) %>% 
    drop_na() %>% pull(logFC_inf_prot, name = entrezgene_id.x) ,
  "Infection (lipid)" = merged_results %>% arrange(desc(logFC_inf_lipid)) %>%
    dplyr::select(logFC_inf_lipid, entrezgene_id.x) %>%
    drop_na() %>% pull(logFC_inf_lipid, name = entrezgene_id.x) ,
  "Interaction" = merged_results %>% arrange(desc(logFC_ixd)) %>% 
    dplyr::select(logFC_ixd, entrezgene_id.x) %>% 
    drop_na() %>% pull(logFC_ixd, name = entrezgene_id.x),
  "Eye Score" = merged_results %>%
    arrange(desc(`z value`)) %>%
    dplyr::select(`z value`, entrezgene_id.x) %>% 
    drop_na() %>% pull(`z value`, name = entrezgene_id.x),
  "Pathogen Load" = merged_results %>% arrange(desc(logFC_logLoad)) %>%
    dplyr::select(logFC_logLoad, entrezgene_id.x) %>% 
    drop_na() %>% pull(logFC_logLoad, name =  entrezgene_id.x)
)

gseGO_all_clusters <- clusterProfiler::compareCluster(all_de_genes_ordered, 
                                      fun="gseGO", 
    OrgDb         = orgdb,
    keyType       = 'ENTREZID',
    ont           = "ALL",
    pAdjustMethod = "BH",
    exponent     = exponent,
    minGSSize     = 11,
    maxGSSize    = 130,
    seed          = 123,
    pvalueCutoff  = 0.05
    )

gseGO_all_clusters <- enrichplot::pairwise_termsim(gseGO_all_clusters)


gseGO_all_clusters@compareClusterResult %>%
  write_excel_csv(., here("results/tables/functional_enrichments/GO_enrichments_include_phenotype.csv"))

```

# Main Text Figures

```{r}
# color.infection = "#984EA3" # was: #C77CFF
# color.diet = "#FFFF33" # was: #7CAE00
# color.interaction = "#4DAF4A" # was: NA
# color.inf.lipid = "#F8766D" # was: #F8766D
# color.inf.protein = "#00BFC4" # was: #00BFC4
# color.EyeScore = "#E41A1C" # was: darkblue
# color.logLoad = "#FF7F00" # was: orange
# #"#e6550d", "#2ca25f", "#ffcc66", "#3366cc", "#c25cb8", "#808080", "#f2a2cc"
# 
# 
# #   "#984EA3" "#FF7F00" "#FFFF33" "#A65628"
# RColorBrewer::brewer.pal(8, "Set2")
# "#E41A1C" "#377EB8" "#4DAF4A" "#984EA3" "#FF7F00" "#FFFF33" "#A65628" "#F781BF"
# "#F11A1C"
# "#66C2A5" "#FC8D62" "#8DA0CB" "#E78AC3" "#A6D854" "#FFD92F" "#E5C494" "#B3B3B3"
# "#8DD3C7" "#FFFFB3" "#BEBADA" "#FB8072" "#80B1D3" "#FDB462" "#B3DE69" "#FCCDE5"

color.infection =  "#C77CFF"
color.diet = "#7CAE00"
color.interaction = "#FFFF33" # was: NA
color.inf.lipid = "#F8766D"
color.inf.protein =  "#00BFC4"
color.EyeScore = "darkblue"
color.logLoad = "orange"
```


## Figure 1

#TODO Add Erin's code for this figure here.
```{r}

```


## Figure 2

### Fig 2A
```{r}

source(here::here("scripts/canary_data_processing_functions.R"))

fig2A <- list("Infection" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_infection],
     "Diet" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet],
     "   Infection × Diet" = DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixd]
     ) %>%
  ggeulerr(shape="circle", text_size = 2.4, alpha=0.7) +
    theme_void(base_size = 2)+
  scale_fill_manual(values = c(color.interaction, color.diet, color.infection)) +
    theme(legend.position = "none", 
          legend.title= element_blank(),
            text = element_text(size=6),
          plot.margin = margin(t = 5,  # Top margin
                             r = 0,  # Right margin
                             b = 5,  # Bottom margin
                             l = 5)) # Left margin

fig2A
  
# ggsave(plot = fig2A, filename= here("results/figures/venn_plot_interaction_fig.png"), width = 16, height = 16, units = "cm", dpi=1200)

```

Hypergeometric test of significance in overlap between diet and interaction between diet and infeciton:
```{r}
q <- length(intersect(DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixd],
                      DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet]))
m <- length(DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_diet])
n_genes <- length(unique(edgeR_analyzed_genes)) - m
k <- length(DE_genes_edgeR_upset$gene[DE_genes_edgeR_upset$DE_in_ixd])

phyper(q, m, n_genes, k, lower.tail=FALSE)

q
m
n_genes
k
```


### Fig 2B
```{r}
fig2b.ma.inf.L <- topTags_lipid_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Lipid)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  labs(x=expression("log"["2"]~"(counts per million)"),
       y=expression("log"["2"]~"(fold change)")) +
  ylim(-4,6) +
     guides(color = guide_legend(override.aes = list(size = 2) ) ) + 
      theme(
            text = element_text(size=8),
            axis.text = element_text(size=8),title = element_text(size=8))
  

fig2b.ma.inf.P <- topTags_protein_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Protein)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) + theme(legend.position = "right") + guides(color = guide_legend(reverse=TRUE)) +
  labs(x=expression("log"["2"]~"(counts per million)"),
       y=expression("log"["2"]~"(fold change)")) +
  ylim(-4,6) +
     guides(color = guide_legend(override.aes = list(size = 2) ) ) + 
      theme(
            text = element_text(size=8),
            axis.text = element_text(size=8),title = element_text(size=8))

fig2B <- ggpubr::ggarrange(fig2b.ma.inf.L, fig2b.ma.inf.P, ncol=1,common.legend = T, legend = "bottom") 

fig2B

# ggsave(plot = fig2B, filename = here("results/figures/logFC_density_DE_in_diet_fig.png"), width = 3, height = 4, units = "cm", dpi=1200)
```




### Fig 2C
```{r}
font_size <- 22

tbl_tol_results %>%
  left_join(
    topTags_diet_uninf_edgeR,
    by = c(
      "gene.z" = "gene",
      "gene_name" = "gene_name",
      #"logCPM"="logCPM",
      "description" = "description"
    ),
    suffix = c("_inf", "_diet")
  ) %>%
  mutate(FDR_diet = FDR) %>%
  dplyr::select(-FDR) %>%
  left_join(
    topTags_logLoad_inf,
    by = c(
      "gene.z" = "gene",
      "gene_name" = "gene_name",
      #"logCPM"="logCPM",
      "description" = "description"
    ),
    suffix = c("_inf", "_logLoad")
  ) %>% # View()
  left_join(
    topTags_ixd_edgeR,
    by = c(
      "gene.z" = "gene",
      "gene_name" = "gene_name",
      #"logCPM_inf"="logCPM",
      "description" = "description"
    ),
    suffix = c("_logLoad", "_ixd")
  ) %>%
  left_join(
    topTags_lipid_inf_edgeR,
    by = c(
      "gene.z" = "gene",
      "gene_name" = "gene_name",
      #"logCPM_inf"="logCPM",
      "description" = "description"
    )
  ) %>%
  left_join(
    topTags_protein_inf_edgeR,
    by = c(
      "gene.z" = "gene",
      "gene_name" = "gene_name",
      #"logCPM_inf"="logCPM",
      "description" = "description"
    ),
    suffix = c("_inf_lipid", "_inf_prot")
  ) %>%
  filter(FDR_inf < 0.05 & FDR_diet < 0.05 & FDR_ixd < 0.05
         & FDR_inf_lipid < 0.05) %>%
  dplyr::select(
    gene_name,
    description,
    logFC_inf,
    logFC_inf_lipid,
    logFC_inf_prot,
    logFC_diet,
    logFC_ixd,
    FDR_inf,
    FDR_diet,
    FDR_ixd,
    FDR_inf_lipid,
    FDR_inf_prot
  ) %>%
  arrange(desc(logFC_inf)) %>%
  mutate(
    gene_name = replace_na(gene_name, "TENT5C"),
    description = replace_na(description, "terminal nucleotidyltranferase 5C"),
    description = ifelse(description == "uncharacterized protein DDB_G0290301-like", "uncharacterized LOC127061056", description)
  ) %>% 
  gt::gt() %>%
  gt::tab_options(table.font.size = font_size,table.width = pct(100)) %>%
  gt::cols_hide(c(FDR_inf, FDR_ixd, FDR_diet, FDR_inf_prot, FDR_inf_lipid)) %>%
  gt::tab_header(title = "31 Genes Differentially Expressed by Diet, Infection, and Diet×Infection") %>%
  gt::fmt_number(decimals = 2) %>%
  gt::cols_label(
    gene_name = "Gene Name",
    description = "Description",
    logFC_inf = "logFC Infection (all)",
    logFC_inf_lipid = "logFC Infection (lipid)",
    logFC_inf_prot = "logFC Infection (protein)",
    logFC_diet = "logFC Diet (protein vs lipid)",
    logFC_ixd = "logFC Interaction"
  ) %>%
  gt::cols_width(
    gene_name ~ "8em",
    description ~ "20em",
    logFC_inf ~ "5em",
    logFC_inf_lipid ~ "5em",
    logFC_inf_prot ~ "5em",
    logFC_diet ~ "5em",
    logFC_ixd ~ "5.2em"
  ) %>%
  gt::data_color(
    columns = c(
      logFC_inf,
      logFC_inf_lipid,
      logFC_inf_prot,
      logFC_diet,
      logFC_ixd
    ),
    fn = scales::col_numeric(
      palette = c("blue", "white", "red"),
      domain = c(-5, 5)
    )
  ) %>%
  gt::cols_align(
    columns = c(
      logFC_inf,
      logFC_inf_lipid,
      logFC_inf_prot,
      logFC_diet,
      logFC_ixd
    ),
    align = "center"
  ) %>%
  gt::tab_style(
    style =
      list(gt::cell_text(weight = "bold")),
    locations = gt::cells_body(columns = logFC_inf,
                               rows = FDR_inf < 0.05)
  ) %>%
  gt::tab_style(
    style =
      list(gt::cell_text(weight = "bold")),
    locations = gt::cells_body(columns = logFC_diet,
                               rows = FDR_diet < 0.05)
  ) %>%
  gt::tab_style(
    style =
      list(gt::cell_text(weight = "bold")),
    locations = gt::cells_body(columns = logFC_ixd,
                               rows = FDR_ixd < 0.05)
  ) %>%
  gt::tab_style(
    style =
      list(gt::cell_text(weight = "bold")),
    locations = gt::cells_body(columns = logFC_inf_lipid,
                               rows = FDR_inf_lipid < 0.05)
  ) %>%
  gt::tab_style(
    style =
      list(gt::cell_text(weight = "bold")),
    locations = gt::cells_body(columns = logFC_inf_prot,
                               rows = FDR_inf_prot < 0.05)
  ) %>%
  gt::tab_options(data_row.padding = gt::px(15)) %>%
  gt::gtsave(
    file = here::here("results/figures/top_genes_ixndietinf_colors.png"),
    zoom = 2,
    vwidth = 3900,
    vheight = 4000,
    expand = 20
  )
```


### Figure 2 combined
Figure 2 Panels
```{r, fig.width=8, fig.height=6}
fig2C <- ggdraw() + draw_image(here("results/figures/top_genes_ixndietinf_colors.png")) 

fig2AB <- cowplot::plot_grid(fig2A,NULL, fig2B, ncol=1, labels = c("A","B",""), rel_heights = c(1.1,0.12,2))
fig2 <- cowplot::plot_grid(fig2AB, NULL, fig2C, rel_widths = c(0.9, 0.0025, 1.3), ncol=3,labels = c(" ","  C", " "))

fig2

ggsave(plot = fig2, filename = here("results/figures/fig2.png"), width = 8, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure 3

### Fig 3A
```{r}
cor_dietvInfL_DE <- left_join(topTags_lipid_inf_edgeR, topTags_diet_uninf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
    left_join(topTags_inf_edgeR,
              by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% 
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, color = "black", size=0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2.6, label.y = 6,  size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  geom_rug(col=color.diet, alpha = 0.2, sides="l",
           length = unit(0.05, "npc")) +
  geom_rug(col=color.inf.lipid, alpha = 0.2, sides = "b",
           length = unit(0.05, "npc")) +
  theme_classic() +
  ylab("") +
  xlab("log fold change\n infection (lipid)") +
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8))

ggsave(here::here("results/figures/DEGenesBothCorrlipInfDietLogFC.png"), width=16, height=16, units = "cm", dpi=1200)

cor_dietvInfP_DE <- left_join(topTags_protein_inf_edgeR, topTags_diet_uninf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  left_join(topTags_inf_edgeR,by=c("gene","gene_name", "description")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% 
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, size=0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 6,  size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  geom_rug(col=color.diet, alpha = 0.2, sides="l",length = unit(0.05, "npc")) +
  geom_rug(col=color.inf.protein, alpha=0.2, sides = "b",length = unit(0.05, "npc")) +
  theme_classic() +
  ylab("") +
  xlab("log fold change\n infection (protein)") +
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8))

ggsave(here::here("results/figures/DEGenesBothCorrprotInfDietLogFC.png"), width=8, height=8, units = "cm", dpi=1200)

cor_dietvInf_DE <- left_join(topTags_inf_edgeR, topTags_diet_uninf_edgeR,
          by=c("gene","gene_name", "description"),suffix = c(".inf", ".diet")) %>%
  filter(FDR.inf < 0.05 | FDR.diet < 0.05) %>% 
  ggplot(aes(x=logFC.inf, y=logFC.diet)) +
  geom_point(alpha = 0.5, size=0.5) +
  geom_smooth(method = "lm") +
  ggpubr::stat_cor(label.x = -2, label.y = 6, size = 2.5, digits = 2, label.sep="\n",
                   aes(label = after_stat(r.label))) +
  geom_rug(col=color.diet, alpha = 0.2, sides="l",length = unit(0.05, "npc")) +
  geom_rug(col=color.infection, alpha = 0.2, sides = "b",length = unit(0.05, "npc")) +
  theme_classic() +
  xlab("log fold change\n infection (all)") +
  ylab("log fold change\n diet (Protein vs Lipid)") +
  theme(panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.text = element_text(size = 8), axis.title=element_text(size=8)
        ) 

ggsave(here::here("results/figures/DEGenesBothCorrInfDietLogFC.png"), width=8, height=8, units = "cm", dpi=1200)


cor_dietvInf_DE
cor_dietvInfP_DE
cor_dietvInfL_DE
```

### Fig 3B

```{r}
fig3_plot_list <- list(
(gseGO_infection)@result %>%
  mutate(comparison = "infection (all)"),
(gseGO_infection_protein)@result %>%
  mutate(comparison = "infection (protein)"),
(gseGO_infection_lipid)@result %>%
  mutate(comparison = "infection (lipid)"),
gseKEGG_infection@result %>%
  mutate(comparison = "infection (all) KEGG"),
gseKEGG_infection_protein@result %>%
  mutate(comparison = "infection (protein) KEGG"),
gseKEGG_infection_lipid@result %>%
  mutate(comparison = "infection (lipid) KEGG"),
(gseGO_diet_uninf)@result %>%
  mutate(comparison = "diet (protein vs lipid)")
)
```

```{r, fig.height=8}

DE_GO_terms_any_inf_simple <- bind_rows(fig3_plot_list[[1]], 
          fig3_plot_list[[2]],
          fig3_plot_list[[3]]
          ) %>%
  filter(p.adjust < 0.05) %>%
  pull(ID) %>% unique()


DE_KEGG_terms_any_inf <- bind_rows(fig3_plot_list[[4]], 
          fig3_plot_list[[5]],
          fig3_plot_list[[6]]) %>%
  filter(p.adjust < 0.05) %>%
  pull(ID) %>% unique()


bind_rows(fig3_plot_list[[4]], 
          fig3_plot_list[[5]],
          fig3_plot_list[[6]]) %>%
  filter(ID %in% DE_KEGG_terms_any_inf) %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " (", ID, ") ")) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
    comparison == "infection (all) KEGG" ~ Term,
    TRUE ~ NA
  ),
  y_labels = case_when(
    Description == "Spliceosome" & comparison == "infection (protein) KEGG	" ~ 0.1,
    TRUE ~ 0
  )) %>%
  arrange(desc(enrichmentScore)) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`
                )) +
  geom_col(position = "dodge") +
  geom_text(aes(y = y_labels, label = Term_labels, hjust = as.numeric(enrichmentScore) > 0), alpha=1) + 
  xlab("GO Term") +
  scale_alpha_discrete(range=c(0.6, 1)) +
  scale_fill_manual(values = c(color.infection, color.inf.lipid, 
                               color.inf.protein)) +
  coord_flip() + theme_classic() +
  theme(axis.title.y=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank())


barplot_enrichments <- 
  bind_rows(fig3_plot_list[[1]], 
          fig3_plot_list[[2]],
          fig3_plot_list[[3]],
          fig3_plot_list[[7]]) %>%
  filter(ID %in% DE_GO_terms_any_inf_simple) %>%
  mutate(Description = str_replace_all(Description, "biological process involved in ", ""),
         `p.adj<0.05` = case_when(p.adjust < 0.05 ~ "YES",
                            TRUE ~ 'NO')) %>%
  mutate(Term = paste0(" ",Description, " "#, "(", ID, ") "
                       )
         ) %>%
  mutate(Term = fct_reorder(as.factor(Term), -enrichmentScore)) %>%
  mutate(Term_labels = case_when(
     TRUE ~ NA
  ),
   y_labels = case_when(
     TRUE ~ 0
   )) %>%
  group_by(Term) %>%
  mutate(Term_labels2 = case_when(
    n() == 4 & comparison == "infection (all)" ~ Term,
    n() == 3 & !is.na(Term_labels) & comparison == "infection (all)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (protein)" ~ Term,
    n() == 2 & is.na(Term_labels) & comparison == "infection (lipid)" ~ Term,
    n() == 1 ~ Term,
    TRUE ~ NA) 
    )%>%
  ungroup() %>% 
  arrange(desc(enrichmentScore)) %>% 
  filter(Description %in% c("translation", "RNA processing", "gene expression", 
                     "monoatomic ion transport", "cell communcation",
                     "signaling", "lipid metabolic process",
                     "cellular response to stimulus", "response to stimulus",
                     "immune system process", "defense response", 
                     "immune response","response to other organism",
                     "defense response to other organism"
                     )) %>%
  ggplot(., aes(x= reorder(Term, -enrichmentScore), y=enrichmentScore, group = comparison, fill = comparison, alpha=`p.adj<0.05`, color=`p.adj<0.05`
                )) +
    geom_col(
             position = position_dodge(preserve = 'single'), 
             linewidth = 0.2) +
    geom_text(aes(y = y_labels, label = Term_labels2, hjust = as.numeric(enrichmentScore) > 0), alpha=1,
              size = 2.5, color = "black") + 
    xlab("GO Term") +
    scale_alpha_discrete(range=c(0.4, 1), guide="none") +
    scale_fill_manual(values = c(color.diet,color.infection, 
                                 color.inf.lipid,color.inf.protein) ) +
    scale_color_manual(values = c("#00000000","black"), guide="none") +
    coord_flip(clip="off") + theme_classic() +
    geom_hline(yintercept = 0) +
    ylim(-0.9,0.9) +
    theme(axis.title.y=element_blank(),
            axis.text.y=element_blank(),
            axis.ticks.y=element_blank(),
          axis.text = element_text(size = 8), axis.title=element_text(size=8)) +
      theme(legend.position = "top", legend.text = element_text(size=8,hjust=0.5),
            legend.title = element_text(size=8),
            legend.key.size = unit(0.25, 'cm'),legend.justification="center",
            panel.border = element_rect(color="black", fill = NA, linewidth = 0.5)) + 
    guides(fill = guide_legend(reverse=TRUE,nrow=2)) +
  #geom_vline(xintercept = 13.5) +
  geom_vline(xintercept = c(1:12)+0.5, linetype = "dotted", color="grey")
    # annotate("text", y=0.75, x = 15, label = "upregulated", size=8) +
    # annotate("text", y=-0.65, x = 15, label = "downregulated", size=8)

barplot_enrichments 

ggsave(plot = barplot_enrichments, here::here("results/figures/GO_term_enrichment_comparison.png"),
       width = 10, height = 10, units = "in", dpi = 1200)
```




### Fig 3C

Heatmap of DE genes in all contrasts
```{r, fig.height=10, fig.width=8}
logFC_all_contrasts <- topTags_inf_edgeR %>%
  arrange(gene) %>%
  mutate(repressed_in_infection = gene %in% names(genes_lower_in_infection),
         induced_in_infection = gene %in% names(genes_higher_in_infection),
         higher_in_Protein = gene %in% names(genes_higher_protein_diet),
         higher_in_Lipid = gene %in% names(genes_higher_lipid_diet),
         Protein_ixd_bigger_Inf = gene %in% names(genes_bigger_protein_infection_difference),
         Lipid_ixd_bigger_Inf = gene %in% names(genes_bigger_lipid_infection_difference),
         # Here we use ifelse to assign logFC values from the named vector based on the gene presence.
        logFC_infection = topTags_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_interaction = topTags_ixd_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        FDR_infection = FDR, 
        FDR_diet = topTags_diet_uninf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_interaction = topTags_ixd_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        logFC_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        logFC_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(logFC),
        FDR_inf_lipid = topTags_lipid_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_inf_protein = topTags_protein_inf_edgeR |> arrange(gene) |> filter(gene == gene) |> pull(FDR),
        FDR_logLoad = topTags_logLoad_inf |> arrange(gene) |> filter(gene == gene) |> pull(FDR)#,
   ) %>%
  mutate(DE_in_infection = repressed_in_infection | induced_in_infection,
         DE_in_diet = higher_in_Protein | higher_in_Lipid,
         DE_in_ixd = Protein_ixd_bigger_Inf | Lipid_ixd_bigger_Inf) %>%
  mutate(gene_label = case_when(
    abs(logFC)>1 ~ gene_name,
    TRUE ~ NA_character_
  ),
  nice_gene_name = case_when(
    !is.na(gene_name) ~ gene_name,
    TRUE ~ paste0("LOC",gene)
  ))

# Define significance level
sig.cutoff <- 0.05

df_heatmap <- logFC_all_contrasts %>%
  # filter(gene %in% genes_de_all_3) %>%
  filter(DE_in_infection | DE_in_diet | DE_in_ixd) %>%
  dplyr::select(
    gene,
    gene_name,
    nice_gene_name,
    description,
    logFC_infection,
    logFC_diet,
    logFC_interaction,
    FDR_infection,
    FDR_diet,
    FDR_interaction,
    logFC_inf_lipid,
    logFC_inf_protein,
    FDR_inf_lipid,
    FDR_inf_protein
  )

# Split into logFC and FDR matrices
logFC_matrix <- as.matrix(df_heatmap[, c("logFC_infection", "logFC_inf_lipid", "logFC_inf_protein", "logFC_diet", "logFC_interaction"
                                         )])
rownames(logFC_matrix) <-  paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)

FDR_matrix <- as.matrix(df_heatmap[, c("FDR_infection", "FDR_inf_lipid", "FDR_inf_protein", "FDR_diet", "FDR_interaction"
                                       )])
rownames(FDR_matrix) <- paste0(df_heatmap$nice_gene_name, " - ", df_heatmap$description)



significance_matrix <- ifelse(FDR_matrix < sig.cutoff, "*", " ") %>%
  replace_na(" ")

colnames(logFC_matrix) <- c("Inf (all)", "Inf (L)", "Inf (P)", "Diet", "Interaction"
                            )

df <- scale(logFC_matrix)

factoextra::fviz_nbclust(k.max = 54, df, factoextra::hcut, method = "wss", nboot = 500) +
geom_vline(xintercept = 8, linetype = 2)+
labs(subtitle = "Elbow method")



heatmap_enrichment_terms <- list(
        # old list
    # text1 = list("Translation", "Translation Regulation", 
    #              "Ribonucleoprotein Complex"),
    # text2 = list("Nuclear Protein Complex", "Homologous Recombination"),
    # text3 = list(expression(Transferase~Activity^"\u2020")),
    # text4 = list("Lysosome", "Protein Kinase Activity", "Phosphorylation"),
    # text5 = list("Innate Immune Response", "Defense Response", "Apoptosis"),
    # text6 = list(expression(Lysosome^"\u2020"),
    #              expression(Lytic~Vacuole^"\u2020")),
    # text7 = list(expression(Apoptosis^"\u2020"),
    #              expression(Channel~Activity^"\u2020")),
    # text8 = list(expression(ECM-receptor~interaction^"\u2020"),
    #              expression(Nucleosome^"\u2020"))
    
    text1 = list("Ribonucleoprotein Complex","Ribosome"),
    text2 = list("Ribosome"),
    text3 = list(expression(Lysosome^"\u2020")),
    text4 = list("ECM-receptor interaction"),
    text5 = list(expression(Lysosome^"\u2020"), "Phosphoric Diester Hydrolase"),
    text6 = list(expression(Lytic~Vacuole^"\u2020"), 
      expression(Cytokine-Cytokine~Receptor^"\u2020"),
      expression(Lysosome^"\u2020") ),
    text7 = list(#expression(Influenza~A^"\u2020"),
                 # expression(Apoptosis^"\u2020"),
                 "Innate Immune Response", "Defense Response" ),
    text8 = list("Nuclear Lumen", "Pyrophosphatase Activity")
)

# note how we set the width of this empty annotation
ha = rowAnnotation(foo = anno_empty(border = FALSE, 
    width = max_text_width(unlist(heatmap_enrichment_terms)) + unit(4, "mm")))

# change padding 
ht_opt$DIMNAME_PADDING = unit(0.25, "cm")
#ht_opt$COLUMN_ANNO_PADDING = unit(0.25, "cm")

heatmap2 <- Heatmap(logFC_matrix, name = "logFC", 
                   cluster_rows = TRUE,
                   show_row_dend = TRUE,
                   width = unit(8, "cm"),
                   heatmap_legend_param = list(direction = "horizontal",
                                               legend_width = unit(3, "in"), 
                                               title = expression(
                                                 "log"["2"]~"(fold change)"
                                                 ),
                                               legend_position = "topleft",
                                               title_gp = gpar(fontsize = 12),
                                               labels_gp = gpar(fontsize=12)
                                               ),
                   cluster_columns = FALSE, 
                   split = 8, #8,
                   show_row_names = FALSE,
                   show_column_names = TRUE,
                   column_names_side = "top",
                   clustering_distance_rows = "canberra",
                   clustering_method_rows = "ward.D2",
                   row_names_gp = gpar(fontsize = 12),
                   column_names_gp = gpar(fontsize = 12),
                   column_names_rot = 0,
                   column_names_centered = TRUE,
                   right_annotation = ha,
                   row_dend_reorder = TRUE,
                   cell_fun = function(j, i, x, y, width, height, fill) {
                       grid.rect(x, y, width, height, gp = gpar(fill = fill, col = NA))
                       if (significance_matrix[i, j] == "*") {
                           grid.text("*", x, y, gp = gpar(col = "black", alpha=0.5, fontsize = 6))
                       }
                   }
                   ) 
ht <- draw(heatmap2, heatmap_legend_side = "bottom")
```

NOTE: seed setting doesn't keep cluster orders. Have to manually adjust the enrichments for each cluster to line up with the corresponding cluster. 
```{r, fig.height=9, fig.width=8, message=FALSE, warning=FALSE}
png(file=here::here("results/figures/logFC_heatmap.png"), width=8, height=16, res=1200,units = "in",bg = "transparent")

draw(heatmap2, heatmap_legend_side = "bot")

for (i in 1:8) {
    decorate_annotation("foo", slice = i, {
        grid.rect(x = 0.01, width = unit(0.75, "mm"), gp = gpar(fill = "black", col = NA), just = "left")
        text_lines <- heatmap_enrichment_terms[[i]]
        n_lines <- length(text_lines)
        # Calculate the center point: Half the number of lines from middle
        start_y <- unit(0.5, "npc")  # middle of the slice
        # Adjust starting y position up by half the number of lines
        adjust_y <- unit(n_lines / 2 - 0.5, "lines") * 1.5 # adjustment from center

        for (j in seq_along(text_lines)) {
            text_line <- text_lines[[j]]
            # Adjust position: start from center and move up or down
            line_y <- start_y - adjust_y + 1.5*unit(j - 1, "lines")
            grid.text(text_line, x = unit(4, "mm"), y = line_y, just = "left")
        }
    })
}

decorate_annotation("foo",  { 
    grid.text("           Functional Enrichments", 
              x = unit(0.4,"npc") - unit(0.2,"cm"),
              y = unit(1, "npc") + unit(0.55, "cm"),
                just = "top", 
              gp = gpar(fontface="bold")) 
})

dev.off()
```

enrichment of each cluster
```{r}
list_GO_results <- list()
j=0
for (i in row_order(ht)) {
  # print(i)
  genes_i <- rownames(heatmap2@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_GO_results[[j]] <- enrichGO(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
                OrgDb         = orgdb,
                keyType       = 'ENTREZID',
                ont           = "ALL",
                pAdjustMethod = "BH",
                minGSSize     = 11,
                pvalueCutoff  = 0.8,
                qvalueCutoff  = 0.8) %>%
    data.frame() %>%
    arrange(p.adjust)
  
}

list_kegg_results <- list()
j=0
for (i in row_order(ht)) {
  genes_i <- rownames(heatmap2@matrix)[i] %>%
    word(., 1)
  j = j+1
  list_kegg_results[[j]] <- enrichKEGG(gene = logFC_all_contrasts %>%
    filter(gene_name %in% genes_i) %>%
    pull(gene)  ,
               organism = 'scan', keyType = "kegg",
               universe = topTags_diet_uninf_edgeR$gene,
               pvalueCutoff = 0.9,
                qvalueCutoff  = 0.9) %>%
    data.frame() %>%
    arrange(p.adjust)
}

list_GO_results
list_kegg_results
```


### Figure 3 combined

```{r, fig.width=7, fig.height=8}
fig3C <- ggdraw() + draw_image(here("results/figures/logFC_heatmap.png"), scale = 1.25, halign=0.4) +
  theme_void()

fig3A <- cowplot::plot_grid(
  cor_dietvInf_DE + theme(
    rect = element_rect(fill = "transparent"),
    plot.background = element_rect(fill = "transparent",
                                   colour = NA_character_)
  ),
  NULL,
  
  cor_dietvInfL_DE + theme(
    rect = element_rect(fill = "transparent"),
    plot.background = element_rect(fill = "transparent",
                                   colour = NA_character_)
  ),
  NULL,
  cor_dietvInfP_DE + theme(
    rect = element_rect(fill = "transparent"),
    plot.background = element_rect(fill = "transparent",
                                   colour = NA_character_)
  ),
  ncol = 5,
  rel_widths = c(1, -0.15, 1, -0.15, 1)
)


fig3AB <-
  cowplot::plot_grid(
    NULL,
    fig3A,
    NULL,
    barplot_enrichments ,
    ncol = 1,
    rel_heights = c(0.05, 0.5,-0.01, 1.75),
    labels = c("A", " ", "", "B"),
    vjust = c(1.5, 0, 0, 4)
  )
fig3AB


fig3 <- cowplot::plot_grid(fig3AB, NULL, fig3C, ncol=3, labels = c("", "", "C"),
                           # rel_widths = c(2.2,0.1,1.65)
                           rel_widths = c(2.3, 0, 1.8)
                           )
fig3

ggsave(plot = fig3, filename = here::here("results/figures/fig3.png"), width = 7, height = 8, units = "in", dpi=1200, bg="white")
```


## Figure 4

### Fig 4A 

Venn Diagram of DE gene overlap between phenotype measures and infection



```{r}
fig4a.1 <- list("Infection" = merged_results$gene.z[merged_results$DE_inf],
     "Pathogen\n Load" = merged_results$gene.z[merged_results$DE_logLoad],
     "Eye Score" = merged_results$gene.z[merged_results$DE_ES]
     ) %>%
  ggeulerr(shape="circle", text_size = 3, alpha=0.75) +
  scale_fill_manual(values = #c("darkblue","#C77CFF","orange"
                               c(color.EyeScore, color.infection, color.logLoad
                               ) , name = " ") +
    theme_void(base_size = 18) +
    theme(#legend.position = "bottom", 
          legend.title= element_blank(),
          plot.margin = margin(t = 5,  # Top margin
                             r = 10,  # Right margin
                             b = 5,  # Bottom margin
                             l = 10)) %>% # Left margin
theme(legend.position = "none"
      ) 


fig4a.1

ggsave(plot = fig4a.1, filename= here("results/figures/fig4a_venn_plot_phenotype_fig.png"), width = 16, height = 16, units = "cm", dpi=1200, bg="white")
```


part II: table Table of genes DE in infection, eye score, and pathogen load analyses
```{r}
merged_results %>%
  filter(FDR_inf < 0.05 & FDR_tol < 0.05) %>%
  filter(gene_name %in% c("JCHAIN", "EXOG",
                              "LOC103823677", "MZB1", "LOC103818060",
                              "MYD88", "LOC108961394",
                              "PTGDR", "PTGER2", "NEFL")) %>%
  dplyr::select(gene_name, gene, description, 
                logFC_inf, Estimate
                ) %>%
  arrange(desc(logFC_inf)) %>%
  mutate(across(where(is.numeric), round, 2)) %>%
  gt::gt() %>%
 gt::tab_header(title = gt::html("Notable Genes among <span style='color:#A37DE2;'>189</span> shared in Eye Score and Infection")) %>%
  gt::cols_hide(gene) %>%
  # fmt_markdown() |>
  gt::cols_label(gene_name = "Gene Name",
                 description = "Description",
                 logFC_inf = "logFC (Infection)",
                 Estimate =  html("
      <span style='display:inline-block; position:relative;'>
        β
        <span style='position:absolute; top:-0.3em; left:0.05em;'>ˆ</span>
      </span><sub>gene</sub><br>(Eye Score)"),
                 ) %>%
  #md("$\\hat\\beta$   (Eye Score)")
  gt::cols_width(gene_name ~ "5em", description ~ "15em",
                 logFC_inf ~ "8em", Estimate ~ "8em") %>%
  gt::gtsave(file = here("results/figures/top_genes_all.png"))
```

```{r}
fig4a.2 <- ggdraw() + draw_image(here("results/figures/top_genes_all.png")) + theme_void()

fig4A <- cowplot::plot_grid(fig4a.1, fig4a.2, ncol=2, labels = c("A"," "),align = "h", rel_widths = c(1.4,2))
fig4A
```




### Fig 4B 

change this one...
```{r}
# plot of JCHAIN expression across levels of eye score
tidy_tpm <- tpm_data %>%
  cpm(log = TRUE) %>% 
  data.frame() %>%
  rownames_to_column("gene_name") %>%
  pivot_longer(cols = -gene_name, names_to = "SampID", 
               values_to = "tpm") %>%
  mutate(SampID = gsub(SampID, pattern = "\\.", replacement = "-")) %>%
  left_join(metadata_phenotypes, by = "SampID") %>%
  left_join(result_BM_merged, by = c("gene_name"="gene_tag"),
            multiple="first") 

tidy_tpm %>%
  filter(gene_name %in% c("LOC103813055", "LOC103815619", 
                     "LOC103816082", "LOC103816190", 
                     "LOC103816359", "LOC103818060",
                     "LOC103818306", "LOC103819798",
                     "LOC103820497", "LOC103821174",
                     "LOC103822938", "LOC103823426", 
                     "LOC103823677", "LOC103823791",
                     "LOC103824735", "LOC108961920",
                     "LOC115484495", "LOC127060478")
         ) %>%#View()
  filter(MG.active == "YES") %>%
  ggplot(aes(x=ES, y = tpm, color = Diet)) +
    geom_point(alpha=0.5) + 
  geom_smooth(method="lm", se=FALSE) +
    facet_wrap(~gene_name.y, scales="free_y")

tidy_tpm %>%
  filter(str_detect(description, "ribosomal protein") & !str_detect(description, "mitochondrial")
         ) %>%#View()
  #filter(MG.active == "YES") %>%
  ggplot(aes(x=ES, y = tpm, color = Diet,group = interaction(Bird.ID
, gene_name.y))) +
  geom_line(alpha=0.1) +
    geom_point(alpha=0.5) +
  geom_smooth(method="lm", se=FALSE,alpha=0.3,aes(group=Diet)) #+
    #facet_wrap(~gene_name.y, scales="free_y")
  
```


network plot of genes DE accoridng to Eye Scores.

```{r, fig.height=4, fig.width=8}
# options(ggrepel.max.overlaps = Inf)

# enrichKEGG_tol_ord2@result$Description <- gsub(enrichKEGG_tol_ord2@result$Description, pattern = " \\- Serinus canaria \\(common canary\\)" , replacement = "")

# network_ES_plot <- enrichGO_tol_ord2 
# (enrichKEGG_tol_ord2+ enrichGO_tol_ord2) #%>%
# setReadable(orgdb) %>%
# filter(ID %in% c("GO:0006412","GO:0006511",
#                 "GO:0031966","GO:0006955",
#                 "GO:0006952","GO:0006629")) %>%# data.frame() %>% View()
  
# clusterProfiler::cnetplot(., layout = "graphopt", 
#            #showCategory = 3,
#            nodel_label ="category",
#            max.overlaps=Inf,
#            species = "scan",
#            cex.params = list(gene_label = 0, category_label = 0.55),
#            color.params = list(foldChange = tbl_tol_results %>%
#                                  mutate(est= -case_when(
#     Estimate > 5 ~ 5,
#     Estimate < -5 ~ -5,
#     TRUE ~ Estimate
#   )) %>%
#   arrange(desc(est)) %>%
#   pull(est, entrezgene_id))
#   ) + 
#   theme(legend.position = "right") + 
#   scale_color_continuous( expression("-",hat(beta)~"Eye Score"),
#                           type = "viridis")
# 
# network_ES_plot
```

```{r}
# all_de_genes = list(
#   "Eye Score" = merged_results %>% filter(FDR_tol < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Pathogen Load" = merged_results %>% filter(FDR_logLoad < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Infection (all)" = merged_results %>% filter(FDR_inf < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Diet" = merged_results %>% filter(FDR_diet < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Infection (protein)" = merged_results %>% filter(FDR_inf_prot < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Infection (lipid)" = merged_results %>% filter(FDR_inf_lipid < 0.05)%>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character(),
#   "Interaction" = merged_results %>% filter(FDR_ixd < 0.05) %>% dplyr::select(entrezgene_id.x) %>% drop_na() %>% pull(entrezgene_id.x) %>% as.character()
# )

# xx <- clusterProfiler::compareCluster(all_de_genes, 
#                                       fun="enrichGO", 
#                OrgDb         = orgdb,
#                 keyType       = 'ENTREZID',
#                 ont           = "ALL",
#                 pAdjustMethod = "BH",
#                 #pvalueCutoff  = 1,
#                 minGSSize     = 11,
#                 maxGSSize     = 100,
#                 qvalueCutoff  = 0.05,
#                 pvalueCutoff=0.5)
# 
# xx <- enrichplot::pairwise_termsim(xx)
# 
# p1 <- clusterProfiler::emapplot(xx,pie.params = list(legend_n = 7), showCategory=5, layout.params	=list(layout="kk")) + ggsci::scale_fill_bmj()
# 
# p1
```


```{r}
# all_de_genes_ordered = list(
#   "Infection (all)" = merged_results %>% arrange(desc(logFC_inf)) %>%
#     dplyr::select(logFC_inf, entrezgene_id.x) %>% 
#     drop_na() %>% pull(logFC_inf, name = entrezgene_id.x),
#   "Diet" = merged_results %>% arrange(desc(logFC_diet)) %>% 
#     dplyr::select(logFC_diet, entrezgene_id.x) %>% 
#     drop_na() %>% pull(logFC_diet, name = entrezgene_id.x),
#   "Infection (protein)" = merged_results %>% arrange(desc(logFC_inf_prot)) %>%
#     dplyr::select(logFC_inf_prot, entrezgene_id.x) %>% 
#     drop_na() %>% pull(logFC_inf_prot, name = entrezgene_id.x) ,
#   "Infection (lipid)" = merged_results %>% arrange(desc(logFC_inf_lipid)) %>%
#     dplyr::select(logFC_inf_lipid, entrezgene_id.x) %>%
#     drop_na() %>% pull(logFC_inf_lipid, name = entrezgene_id.x) ,
#   "Interaction" = merged_results %>% arrange(desc(logFC_ixd)) %>% 
#     dplyr::select(logFC_ixd, entrezgene_id.x) %>% 
#     drop_na() %>% pull(logFC_ixd, name = entrezgene_id.x),
#   "Eye Score" = merged_results %>%
#     arrange(desc(`z value`)) %>%
#     dplyr::select(`z value`, entrezgene_id.x) %>% 
#     drop_na() %>% pull(`z value`, name = entrezgene_id.x),
#   "Pathogen Load" = merged_results %>% arrange(desc(logFC_logLoad)) %>%
#     dplyr::select(logFC_logLoad, entrezgene_id.x) %>% 
#     drop_na() %>% pull(logFC_logLoad, name =  entrezgene_id.x)
# )
# 
# gseGO_all_clusters <- clusterProfiler::compareCluster(all_de_genes_ordered, 
#                                       fun="gseGO", 
#     OrgDb         = orgdb,
#     keyType       = 'ENTREZID',
#     ont           = "ALL",
#     pAdjustMethod = "BH",
#     exponent     = exponent,
#     minGSSize     = 11,
#     maxGSSize    = 130,
#     seed          = 123,
#     pvalueCutoff  = 0.05
#     )
# 
# gseGO_all_clusters <- enrichplot::pairwise_termsim(gseGO_all_clusters)

# p2 <- clusterProfiler::emapplot(xx2, #pie.params = list(legend_n = 7, pie = "count"), 
#                                 #layout = "kk",
#                                 showCategory=12, 
#                                 legend_n = 3,
#                                 layout.params	=list(layout="star"), #group_category = T, group_legend = F,
#                                 color="p.adjust"#,
#    #cex_label_group = 1.5, node_label = "group", 
#     #circular = TRUE, colorEdge = TRUE
#    ) + ggsci::scale_fill_bmj()
# 
# p2

```

```{r}
# Choose subset of categories to show:
categories <- c("immune response",
                "immune system process",
                "response to other organism",
                "ribosome",
                "translation",
                "ubiquitin-dependent protein catabolic process",
                "protein ubiquitination",
                "proteolysis",
                "mitochondrion"
)

g <- gseGO_all_clusters %>%
  DOSE::setReadable(OrgDb = orgdb) %>%
  clusterProfiler::cnetplot(.,
                            set.seed=123,
                            showCategory=categories,
                            # see all enriched terms by using below instead:
                            #showCategory=12,
                            cex_gene=2,
                            cex_category=2,
                            node_label = "category",
                            layout.params	=list(layout="kk"),
                            pie.params = list(legend_n = 3, pie = "count", alpha=0.8)) + 
  theme(legend.position = "right") +
  scale_fill_manual(values = c(color.infection, color.diet, color.inf.protein,
                               color.inf.lipid,'#4844ac' #paste0("#00008B"#color.EyeScore
                                                       #,"BF")
                    , color.logLoad))

g

ggsave(plot = g, filename= here("results/figures/fig4b_network_annotated.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")

g_unannotated <- gseGO_all_clusters %>%
  DOSE::setReadable(OrgDb = orgdb) %>%
  clusterProfiler::cnetplot(.,
                            set.seed=123,
                            showCategory=categories,
                            layout = "kk", 
                            cex_gene=2,
                            cex_category=2,
                            node_label = "none",
                            pie.params = list(legend_n = 3, pie = "count")) + 
  theme(legend.position = "none") +
  scale_fill_manual(values = c(color.infection, color.diet, color.inf.protein,
                               color.inf.lipid, '#4844ac',#color.EyeScore, 
                               color.logLoad))

ggsave(plot = g_unannotated, filename= here("results/figures/fig4b_network.png"), 
       width = 7, height = 7, units = "in", dpi=1200, bg="white")
```


```{r}
enrichplot::dotplot(gseGO_all_clusters,showCategory=100) + 
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25))
```

```{r}
# Choose the categories to show:
categories <- c(#"GO:0006955", # 
                "immune response",
                # "GO:0002376", # 
                "immune system process",
                # "GO:0006952", # 
                 #"defense response",
                "response to other organism",
                
                # "GO:0005840", # 
                "ribosome",
                 #"peptide metabolic process",
                 #"structural constituent of ribosome",
                 #"amide biosynthetic process",
                # "GO:0006412", # 
                "translation",
                # "GO:0006511", # 
                "ubiquitin-dependent protein catabolic process",
                # "GO:0016567", # 
                "protein ubiquitination",
                # "GO:0043161", # 
                 #"proteasome-mediated ubiquitin-dependent protein catabolic process",
                # "GO:0006508" # 
                "proteolysis",
                "mitochondrion"
)

p3 <- clusterProfiler::emapplot(xx2, pie.params = list(legend_n = 7, pie = "count"), 
                                #layout = "kk",
                                showCategory=categories, 
                                #showCategory=19,
                                layout.params	=list(layout="kk"), #group_category = T, group_legend = F,
                                color="p.adjust"#,
   #cex_label_group = 1.5, node_label = "group", 
    #circular = TRUE, colorEdge = TRUE
   ) + ggsci::scale_fill_bmj()

p3

clusterProfiler::cnetplot(xx2, showCategory = categories, layout="fr")


dotplot(xx2, showCategory = 8, by = "pvalue") +
  geom_path(aes(GeneRatio, Description), group=1,linewidth=0.3,color='steelblue') #+
  coord_polar(theta='y')

```


try with kegg
```{r}
xx.kegg <- clusterProfiler::compareCluster(all_de_genes_ordered, 
                                      fun="gseKEGG", 
    #OrgDb         = orgdb,
    organism     = 'scan',
    #keyType       = 'ENTREZID',
    #ont           = "ALL",
    pAdjustMethod = "BH",
    exponent=exponent,
    #maxGSSize     = 115,
    seed          = 123,
    pvalueCutoff  = 0.05
    )

xx.kegg <- enrichplot::pairwise_termsim(xx.kegg)

xx.kegg <- mutate(xx.kegg, Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " ") )

network_plot_kegg <- clusterProfiler::emapplot(xx.kegg,# pie.params = list(legend_n = 7, pie = "count"), 
                                layout = "kk",
                                #showCategory=categories, 
                                #showCategory=19,
                                layout.params	=list(layout="kk")#, #group_category = T, group_legend = F,
                                #color="p.adjust"#,
   #cex_label_group = 1.5, node_label = "group", 
    #circular = TRUE, colorEdge = TRUE
   ) + ggsci::scale_fill_bmj()

cnetplot(xx.kegg)
dotplot(xx.kegg)
network_plot_kegg
```


```{r}
fig4B_new <- ggdraw() + draw_image(here("results/figures/fig4a_new_network_plot.png")) + theme_void()
```

### Figure 4 combined
```{r, fig.height=7, fig.width=7}
fig4 <- cowplot::plot_grid(fig4A, fig4B_new, ncol=1, labels = c("", "B"), rel_heights = c(3,2.5))
fig4

ggsave(plot = fig4, filename= here("results/figures/fig4_phenotype.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")
```



# Supplemental Figures

helper functions.
```{r plot-DE-cleanerCode}
sig.cutoff <- 0.05
logFC.cutoff <- 0

source(here::here("scripts/canary_data_processing_functions.R"))
```

## Figure S1: MA plots
```{r}
# Density plot of estimated gene expression changes due to infection, according to diet.
s1.density <- topTags_lipid_inf_edgeR %>%
  left_join(., topTags_protein_inf_edgeR, by = c("gene","gene_name","description"),suffix = c("_lipid", "_protein")) %>%
  left_join(., topTags_inf_edgeR, by = c("gene","gene_name","description")) %>%
  pivot_longer(c(logFC_lipid, logFC_protein), names_to = "diet", values_to = "logFC_plot") %>%
  mutate(diet =
           case_when(diet == "logFC_protein" ~ "protein",
                     diet == "logFC_lipid" ~ "lipid")) %>%
mutate(fdr = case_when(
  diet == "lipid" ~ FDR_lipid,
  diet == "protein" ~ FDR_protein
)) %>%
  filter(FDR_lipid < 0.05| FDR_protein < 0.05 | FDR < 0.05) %>%
  ggplot(aes(x=logFC_plot,  color = diet, fill = diet
             )) +
  geom_density(alpha=0.5) +
  xlim(-3, 3) +
  xlab("logFC") +
  theme_classic(base_size = 18) +
    theme(legend.position = c(0.8,0.6))


# ggsave(plot = s1.density, filename = here("results/figures/logFC_density_DE_in_diet_fig.png"), width = 16, height = 16, units = "cm", dpi=1200)

s1.ma.inf <- topTags_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (all)",show.names = F, 
                   logFC.cutoff = logFC.cutoff)

s1.ma.diet <- topTags_diet_uninf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Diet",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  geom_text(inherit.aes = F,data= data.frame(label = c("higher on lipid diet", "higher on protein diet"),
                             x = c(10,9.5),
                             y= c(-4,5), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red"))


s1.ma.interaction <- topTags_ixd_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG-Diet Interaction",show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  geom_text(inherit.aes = F,data= data.frame(label = c("Greater MG effect \non lipid diet", "Greater MG effect \non protein diet"),
                             x = c(10,9.5),
                             y= c(-4.25,4.25), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red"))

s1.ma.inf.L <- topTags_lipid_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Lipid)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) 

s1.ma.inf.P <- topTags_protein_inf_edgeR %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("MG Infection (Protein)",show.names = F, 
                   logFC.cutoff = logFC.cutoff) + theme(legend.position = "right") + guides(color = guide_legend(reverse=TRUE)) 

s1.legend <- get_legend(s1.ma.inf.P)

fig.s1.ma <- cowplot::plot_grid(s1.ma.inf, #s1.ma.inf.L, s1.ma.inf.P+ theme(legend.position = "none"), 
                             s1.ma.diet, s1.ma.interaction, 
                              s1.legend, labels = c(LETTERS[seq(1,3)], "") )

fig.s1.ma

fig.s1 <- cowplot::plot_grid(fig.s1.ma, s1.density, ncol=1, labels = c("", "D"))

fig.s1

# ggsave(plot = fig.s1, filename = here("results/figures/fig.s1.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")
```


## Figure S2: KEGG plots
```{r}
gseGO_inf_df <- gseGO_infection %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_all")

gseGO_diet_df <- gseGO_diet_uninf %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "diet_uninf")

gseGO_ixn_df <- gseGO_ixd %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "interaction_diet:infection")

gseGO_infection_protein_df <- gseGO_infection_protein %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_protein")

gseGO_infection_lipid_df <- gseGO_infection_lipid %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "infection_lipid")

gseGO_logLoad_df <- gseGO_logLoad %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(ONTOLOGY = "BP") %>%
  mutate(comparison = "pathogen_load")

gseGO_ES_df <- gseGO_tol_ord2 %>%
  data.frame() %>%
  group_by(ONTOLOGY) %>%
  head(n=30) %>%
  mutate(comparison = "eye_score")

gseGO_all_df <- rbind(gseGO_inf_df, gseGO_diet_df, gseGO_ixn_df, gseGO_infection_protein_df, gseGO_infection_lipid_df, gseGO_logLoad_df, gseGO_ES_df)

gseGO_all_df %>%
  write_excel_csv(., here("results/tables/functional_enrichments/GO_enrichments.csv"))
  

gseKEGG_inf_df <- gseKEGG_infection %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "infection_all")

gseKEGG_diet_df <- gseKEGG_diet_uninf %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "diet_uninf")

gseKEGG_ixd_df <- gseKEGG_ixd %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "interaction_diet:infection")

gseKEGG_infection_protein_df <- gseKEGG_infection_protein %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "infection_protein")

gseKEGG_infection_lipid_df <- gseKEGG_infection_lipid %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "infection_lipid")

gseKEGG_logLoad_df <- gseKEGG_logLoad %>%
  data.frame() %>%
    head(n=30) %>%
  mutate(comparison = "pathogen_load")

gseKEGG_ES_df <- gseKEGG_tol_ord2 %>%
  data.frame() %>%
  head(n=30) %>%
  mutate(comparison = "eye_score")

gseKEGG_all_df <- rbind(gseKEGG_inf_df, gseKEGG_diet_df, gseKEGG_ixd_df, gseKEGG_infection_protein_df, gseKEGG_infection_lipid_df, gseKEGG_logLoad_df, gseKEGG_ES_df)

gseKEGG_all_df %>%
  write_excel_csv(., here("results/tables/functional_enrichments/KEGG_enrichments.csv"))
```

```{r}
fig.s2a <- gseGO_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %in% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>% 
  arrange(pvalue) %>%
  slice_head(n=10) %>%
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","no")) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 45, side="left")) + 
  # cut off long names
  ylab(NULL) +
  ggtitle("GO Enrichments") +
  theme_bw() +
  facet_wrap(~comparison, nrow=5, scales = "free_y") +
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))


fig.s2b <- gseKEGG_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %in% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>%
  arrange(pvalue) %>%
  slice_head(n=10) %>% 
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","yes")) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 42)) + # cut off long names
  ylab(NULL) +
  ggtitle("KEGG Enrichments") +
  theme_bw() +
  facet_wrap(~comparison, nrow=5, scales = "free_y") + 
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))

fig.s2 <- ggpubr::ggarrange(fig.s2a, fig.s2b, common.legend = T,legend = "right")

fig.s2

# ggsave(plot = fig.s2, filename = here("results/figures/fig.s2.png"), width = 7, height = 8, units = "in", dpi=1200, bg="white")
```


## Figure S3: Gene Expression Correlations by Contrast

```{r}
fig.s3 <- merged_results %>%
  dplyr::select(DE_inf, gene.z,entrezgene_id_inf, Estimate, logFC_inf, logFC_inf_lipid, logFC_inf_prot,
                logFC_diet, logFC_ixd, logFC_logLoad) %>%
  rename(all_of(c("logFC_inf" = "Infection \n(all)" ,
         "logFC_inf_lipid" = "Infection \n(lipid)",
         "logFC_inf_prot" = "Infection \n(protein)",
         "logFC_diet"="Diet" ,
         "logFC_ixd"="MG-Diet \nInteraction",
         "Estimate" = "Eye \nScore",
         "logFC_logLoad"="Pathogen\n Load" 
         ))) %>%
  mutate(`Differentially\nExpressed\nInfection` = ifelse(DE_inf,"yes","no")) %>%
  relocate(`Differentially\nExpressed\nInfection`, .before = DE_inf) %>%
  dplyr::select(-DE_inf) %>%
  mutate(rowid = paste0(gene.z, "_", entrezgene_id_inf)) %>%
  group_by(rowid) %>%
  slice_head(n=1) %>%
  column_to_rownames("gene.z") %>%
  dplyr::select(-c(rowid, entrezgene_id_inf)) %>%
  GGally::ggpairs(., aes(color=fct_rev(`Differentially\nExpressed\nInfection`), alpha=0.3),
    upper = list(continuous = GGally::wrap("cor", size = 2))) + 
  theme_classic() +
  theme(axis.text = element_text(size = 7), 
        axis.title=element_text(size=7), axis.text.x=element_text(angle=90, hjust=1))

fig.s3

# ggsave(plot = fig.s3, filename = here("results/figures/fig.s3.png"), width = 7, height = 7, units = "in", dpi=1200, bg="white")
```

## Figure S4: Upset All Contrasts

```{r}
lab_size = 1.5
pt_size = 0.9

fig.s4 <- merged_results %>%
  mutate(MG = DE_inf, Diet = DE_diet, MGxDiet = DE_ixd, EyeScore = DE_ES) %>%
  upset(., c(#"DE_inf", "DE_diet", "DE_ixd", "DE_ES"
    "MG", "Diet", "MGxDiet", "EyeScore"
             ), name="Comparison",
        base_annotations=list(
        'Intersection size'=intersection_size(text = list(size=3, vjust=-0.15),
            text_colors=c(
                on_background='black', on_bar='white'
            )
        )
        + annotate(
            geom='text', x=Inf, y=Inf,
            label=paste('Total:', nrow(topTags_inf_edgeR)),
            vjust=1.2, hjust=1.2, size = 3
        )
        + ylab('# of genes')
    ),
        annotations = list(
        'Infection\n(all)'=(
            ggplot(mapping=aes(y=logFC_inf,color=-log10(FDR_inf), label=goi#_inf
                               )) + 
              ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + ggrepel::geom_text_repel(
                                                       direction = "y",
                                                       size=lab_size,
nudge_x = -0.4,force=0.1,
                                                       max.overlaps = 10) + scale_color_continuous(name="-log10(adj.p.value)", limits = c(0,5)) + theme(legend.direction = "vertical")
        ),
        'Infection\n(lipid)'=(
            ggplot(mapping=aes(y=logFC_inf_lipid,color=-log10(FDR_inf_lipid), label=goi)) +
              ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE)+ ggrepel::geom_text_repel(
                                                       direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + scale_color_continuous(name="-log10(FDR) ") + theme(legend.position='none')
        ),
        'Infection\n(protein)'=(
            ggplot(mapping=aes(y=logFC_inf_prot,color=-log10(FDR_inf_prot), label=goi)) +
              ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + 
              ggrepel::geom_text_repel(direction = "y",size=lab_size,nudge_x = -0.4, force=0.1, max.overlaps = 5) + 
              scale_color_continuous(name="-log10(FDR)  ") +
              theme(legend.position='none')
        ),
        'Diet\n(Protein-Lipid)'=(
            ggplot(mapping=aes(y=logFC_diet,color=-log10(FDR_diet), label=goi
                               )) + 
              ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) +
              ggrepel::geom_text_repel(direction = "y",
                                       size=lab_size,nudge_x = -0.4, 
                                       force=0.1, max.overlaps = 5) + 
              scale_color_continuous(name="-log10(FDR)   ") + 
              theme(legend.position='none')
        ),
        'Interaction\n(Protein-Lipid)'=(
            ggplot(mapping=aes(y=logFC_ixd,color=-log10(FDR_ixd), label=goi)) + 
              ggbeeswarm::geom_quasirandom(alpha=0.5, size=pt_size,
                na.rm=TRUE) + 
              ggrepel::geom_text_repel(direction = "y",
                                       size=lab_size,nudge_x = -0.4, 
                                       force=0.1, max.overlaps = 5) + 
              scale_color_continuous(name="-log10(FDR)    ") + 
              theme(legend.position='none')
        ),
        'Eye Score'=(
            ggplot(mapping=aes(y=est,color=-log10(FDR_tol),label=goi)) +
              ggbeeswarm::geom_quasirandom(size=pt_size, na.rm=TRUE) + 
              ggrepel::geom_text_repel(direction = "y",
                                       size=lab_size,
                                       nudge_x = -0.4,
                                       force=0.1, max.overlaps = 5) +
              scale_color_continuous(name="-log10(FDR)     ") + 
              guides(fill = guide_legend(override.aes = list(size = 0.5))) +
              theme(legend.position='none')
        )#,
        # 'logLoad'=(
        #     ggplot(mapping=aes(y=logFC_logLoad))
        #     + ggbeeswarm::geom_quasirandom(alpha=0.5, size=0.5,
        #         na.rm=TRUE)
        # )
        ),
        keep_empty_groups=TRUE, 
        themes=upset_default_themes(text=element_text(size=8),
                          axis.title=element_text(size=8,face="bold")),
        min_degree=2, min_size = 8,
        # moves legends over the set sizes
        guides='over') & theme(legend.margin=margin(t=0, r=0, b=1, l=0, 'cm'))

# ggsave(plot = fig.s4,        filename = "results/figures/fig.s4.DE_genes_multinode_ES_combined_upsetplot.png",        width = 6, height = 9, units = "in", dpi = 1200)
```


## Figure S5: MA plot for phenotype analysis
```{r}
s5.ma.logLoad <- topTags_logLoad_inf %>%
  clean_and_mutate_data() %>% 
  create_gene_plot("Pathogen Load Effect on Gene Expression",
                   show.names = F, 
                   logFC.cutoff = logFC.cutoff) +
  scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  theme(plot.title = element_text(size=10), legend.position = "bottom")

s5.ma.eyeScore <- merged_results %>%
  filter(!is.na(Estimate)) %>%
    mutate(delabel = case_when(
      !is.na(gene_name) ~ gene_name,
      TRUE ~ paste0("LOC",gene)
    )) %>%
    mutate(delabel = case_when(
      FDR_tol < sig.cutoff ~ delabel,
      abs(Estimate) > logFC.cutoff ~ delabel,
      TRUE ~ NA
    )) %>%
    mutate(diffexpressed = case_when(
      FDR_tol > sig.cutoff ~ "Neither",
      Estimate < 0 ~ "Down",
      Estimate > 0 ~ "Up"
    )) %>%
    mutate(diffexpressed = factor(diffexpressed, levels = c("Neither", "Down", "Up"))) %>%
    arrange(diffexpressed) %>%
  mutate(logCPM = logCPM_inf) %>%
  ggplot(., aes(x = logCPM, 
                   y = Estimate, 
                   col = diffexpressed, 
                   label = delabel
                   )) +
    geom_point(size=0.5) +
    theme_classic() +
    scale_color_manual("Differentially\n Expressed",values = c("black", "blue", "red")) +
      scale_size_area() +
      ggtitle("Eye Score Effect on Gene Expression") + 
      theme(legend.position="none",
            text = element_text(size=10),
            axis.text = element_text(size=10)) +
    # add custom text to plot
  geom_text(inherit.aes = F,data= data.frame(label = c("Expression + correlates\n   with Eye Score", "Expression - correlates\n  with Eye Score"),
                             x = c(10,10),
                             y= c(-15,15), delabel = c(T,T), 
                             diffexpressed=c("Down", "Up")
                             ), 
            aes(x=x, y=y, label=label), size=2.5, color = c("blue", "red")) +
    guides(color = guide_legend(reverse=TRUE)) +
  theme(plot.title = element_text(size=10))

fig.s5 <- cowplot::plot_grid(s5.ma.logLoad, s5.ma.eyeScore, nrow=2, labels = c("AUTO"),
                             rel_heights = c(1.3,1))

fig.s5

# ggsave(plot = fig.s5, filename = here("results/figures/fig.s5.png"),        width = 4, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure S6: GO and KEGG (Phenotype)

```{r}
fig.s6a <- gseGO_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE = ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %ni% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>% 
  arrange(pvalue) %>%
  slice_head(n=10) %>%
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_size_continuous(range = c(2, 10)) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","no")) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 45, side="right")) + 
  ylab(NULL) +
  ggtitle("GO Enrichments") +
  theme_bw() +
  facet_wrap(~fct_rev(comparison), nrow=5, scales = "free_y") +
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))


fig.s6b <- gseKEGG_all_df %>%
  mutate(Description = str_replace(Description, " - Serinus canaria \\(common canary\\)", " "),
         DE =ifelse(p.adjust < 0.05, "YES", "NO"),
         comparison = factor(comparison, levels = c("infection_all","infection_lipid", "infection_protein","diet_uninf","interaction_diet:infection", "eye_score", "pathogen_load"),
                             labels = c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction", "Eye Score", "Pathogen Load"))
         ) %>%
  filter(comparison %ni% c("Infection (all)", "Infection (lipid)", "Infection (protein)", "Diet", "MG-Diet Interaction")) %>%
  group_by(comparison) %>%
  arrange(pvalue) %>%
  slice_head(n=10) %>% 
  ggplot(.,
       showCategory = 10,
       aes(enrichmentScore, fct_reorder(Description, enrichmentScore))) +
  geom_segment(aes(xend = 0, yend = Description, linetype = fct_rev(DE))) +
  geom_point(aes(color = p.adjust, size = setSize)) +
  scale_color_gradientn(
    colours = c("#f7ca64", "#46bac2", "#7e62a3"),
    trans = "log10",
    guide = guide_colorbar(reverse = TRUE, order = 1)
  ) +
  scale_linetype_discrete("Significant\nenrichment", labels =  c("yes","yes")) +
  scale_size_continuous(range = c(2, 10)) +
  scale_y_discrete(label = function(x) stringr::str_trunc(x, 42)) +
  ylab(NULL) +
  ggtitle("KEGG Enrichments") +
  theme_bw() +
  facet_wrap(~fct_rev(comparison), nrow=5, scales = "free_y") + 
      theme(legend.position="none",
            text = element_text(size=6),
            axis.text = element_text(size=6))

fig.s6 <- ggpubr::ggarrange(fig.s6a, fig.s6b, common.legend = T,legend = "right")
fig.s6

#ggsave(plot = fig.s6, filename = here("results/figures/fig.s6.png"), width = 7, height = 6, units = "in", dpi=1200, bg="white")
```


## Figure S7: PCA plot
```{r}
PCA <- plotMDS(y, gene.selection="common",
               top = 300, pch = 2, cex = 0.5, main = "PCA plot of gene expression for all samples", dim.plot=c(1,2))

ggPCA <- tibble(
  SampID = colnames(PCA$distance.matrix.squared),
  x = PCA$x,
  y =PCA$y
) %>%
  mutate(SampID = str_replace_all(SampID, "\\.","-")) %>%
  left_join(metadata_phenotypes, by="SampID")

fig.s7 <- ggplot(ggPCA, aes(x=x,y=y, group=interaction(Diet,MG.active),
                  color = interaction(MG.active,Diet), linetype = Diet)) +
  geom_point() +
  theme_classic() +
  labs(title="PCA plot of gene expression") +
  xlab(paste0("Principal Component 1 (",100*round(PCA$var.explained[1],2),"%)")) +
  ylab(paste0("Principal Component 2 (",100*round(PCA$var.explained[2],2),"%)")) +
  scale_color_discrete("MG.Diet") +
  stat_ellipse(type = "norm",level=0.8)

# ggsave(plot = fig.s7, filename = here("results/figures/fig.s7.png"), width = 7, height = 6, units = "in", dpi=1200, bg="white")
```



# Suppplemental Tables

## Table S2: DE genes in the interaction between diet and infection response

The 20 largest magnitude DE (10 up and 10 down) genes for each contrast.
```{r}
table.s3.ixd <- topTags_ixd_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

table.s3.inf <- topTags_inf_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG Infection") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

table.s3.diet <- topTags_diet_uninf_edgeR %>%
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=10) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "Diet (Protein vs Lipid)") %>%
  dplyr::select(comparison, gene, gene_name, logFC, FDR, description)

bind_rows(table.s3.ixd, table.s3.inf, table.s3.diet) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  dplyr::select(-comparison) |>
  rename(gene = "Top DE Genes",
         gene_name = "Gene Name") |>
  replace_na(list(rep("", 5))) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  pack_rows("MG-Diet Interaction", 1, 20) |>
  pack_rows("MG Infection", 21, 40) |>
  pack_rows("Diet (Protein-Lipid)", 41, 60) #|>
  # save_kable(here("results/figures/table.s2.png"))
```



## Table S3: Significant GO terms

```{r}
gseGO_all_df |>
    rownames_to_column("rowname") |>
  dplyr::select(comparison, ID, Description, setSize, enrichmentScore, NES, pvalue, p.adjust, qvalue, rank, ONTOLOGY) |>
  mutate(comparison = case_when(
    comparison == "infection_all" ~ "Infection (all)",
    comparison == "diet_uninf" ~ "Diet (Protein-Lipid)",
    comparison == "interaction_diet:infection" ~ "MG-Diet Interaction",
    comparison == "infection_protein" ~ "Infection (protein)",
    comparison == "infection_lipid" ~ "Infection (lipid)",
    comparison == "pathogen_load" ~ "Pathogen Load",
    comparison == "eye_score" ~ "Eye Score"
  )) |>
  mutate(across(is.numeric, signif, 3)) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  # pack_rows("MG-Diet Interaction", 1, 20) |>
  # pack_rows("MG Infection", 21, 40) |>
  # pack_rows("Diet (Protein-Lipid)", 41, 60) 
  collapse_rows(columns = 1, valign="top", latex_hline = "major") #|>
  #save_kable(here("results/figures/table.s3.png"))
```

## Table S4: Heatmap Cluster GO Enrichments
```{r}
table.s4.go <- list_GO_results |>
  bind_rows(.id= "cluster") |>
  rownames_to_column("rowname") |>
  group_by(cluster) |>
  arrange(pvalue) |>
  filter(Count > 3) |>
  slice_head(n=5) |>
  dplyr::select(cluster, ID, Description, GeneRatio, BgRatio,p.adjust) |>
  mutate(across(is.numeric, signif, 3)) 

table.s4.kegg <- list_kegg_results |>
  bind_rows(.id= "cluster") |>
  rownames_to_column("rowname") |>
  group_by(cluster) |>
  arrange(pvalue) |>
  filter(Count > 3) |>
  slice_head(n=5) |>
  dplyr::select(cluster, ID, Description, GeneRatio, BgRatio, p.adjust) |>
  mutate(Description = 
           str_replace(Description,
                       " - Serinus canaria \\(common canary\\)", "")) |>
  mutate(across(is.numeric, signif, 3))
  
  
bind_rows(table.s4.go, table.s4.kegg) |>
  group_by(cluster) |>
  arrange(cluster) |>
  mutate(across(is.numeric, as.character, 3)) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header")) |>
  collapse_rows(columns = 1, valign="top", latex_hline = "major")


```

## Table S5: DE genes for pathogen load
```{r}
 topTags_logLoad_inf |>
  filter(FDR<0.05) %>%
  group_by(sign(logFC)) %>%
  arrange(desc(abs(logFC))) %>%
  slice_head(n=20) %>%
  arrange(logFC) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(gene, gene_name, logFC, FDR, description) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  rename(gene = "Top DE Genes",
         gene_name = "Gene Name") |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header"))
```

## Table S6: DE genes for Eye Score
```{r}
tbl_tol_results |>
  filter(FDR_tol<0.05) %>%
  group_by(sign(Estimate)) %>%
  arrange(`Pr(>|z|)`) %>%
  slice_head(n=20) %>%
  arrange(Estimate) %>%
  ungroup() %>%
  mutate(comparison = "MG-Diet Interaction") %>%
  dplyr::select(gene.z, gene_name, Estimate,`Pr(>|z|)`, description) |>
  mutate(across(is.numeric, signif, 3)) |>
  mutate(across(is.numeric, as.character, 3)) |>
  rename(gene.z = "Top DE Genes",
         gene_name = "Gene Name"
         ) |>
  kbl(booktabs = T, longtable = T) |>
    kable_styling("striped", latex_options = c("repeat_header"))
```

